---
title: "Inbred Strain Expression"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

The purpose of this workflow is look closely at inbred strain expression
in hepatocytes. We look at similarities and differences across strains.

This workflow uses the vst normalized data generated by 1.1_Expression.Rmd


```{r load_libraries, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
is.interactive = FALSE
#is.interactive = TRUE
needed.packages <- c("here", "pheatmap", "hexbin", "RColorBrewer", "gprofiler2",
"knitr", "DESeq2", "sva", "qtl2", "igraph", "wordcloud")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}
```

```{r source_code}
all.code.dir <- list.files(here("code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r read_expression}
transformed.data.file <- here("Data", "RNASeq", "StrainsEffCts9_vst.RDS")
expr <- readRDS(transformed.data.file)
```

```{r info}
transcript.info.file <- here("Data", "RNASeq", "RNASeq_gene_info.RData")
transcript.info <- readRDS(transcript.info.file)
col.table <- as.matrix(read.table(here("Data", "support_files", "strain.color.table.txt"), 
    sep = "\t", comment.char = "%", stringsAsFactors = FALSE))
```

```{r id_table}
ind.name <- colnames(expr)
ind.label <- sapply(strsplit(ind.name, ""), function(x) paste(tail(x, 2), collapse = ""))
treat.label <- substr(ind.label, 1, 1)
ind <- substr(ind.label, 2,2)
strain <- mapply(function(x,y) gsub(x, "", y), ind.label, ind.name)
u_strain <- unique(strain)
id.table <- cbind(strain, treat.label, ind)
full.col <- col.table[match(id.table[,1], col.table[,4]),3]
u_strain.col <- col.table[match(u_strain, col.table[,4]),3]
```

```{r sep_strain_expr}
#separate control expression by strain
strain_expr <- lapply(u_strain, function(x) expr[,intersect(which(strain == x), which(treat.label == "C"))])
names(strain_expr) <- u_strain
```

The following plot shows the correlation among individuals.

```{r ind_cor}
ind.cor <- cor(expr)
if(is.interactive){quartz()}
ind.order <- hclust(dist(ind.cor))$order
ordered.strain.clust <- unique(substr(rownames(ind.cor)[ind.order], 1, 2))
final.ind.order <- unlist(lapply(ordered.strain.clust, function(x) which(substr(rownames(ind.cor), 1, 2) == x)))

pheatmap(ind.cor[final.ind.order,final.ind.order], cluster_rows = FALSE, cluster_cols = FALSE)

```

The following plot shows the correlation of expression between
pairs of strains. Overall, the correlation is very high. 

```{r expr_cor}
strain_sd <- sapply(strain_expr, function(x) apply(x, 1, sd))
strain_mean <- sapply(strain_expr, function(x) apply(x, 1, mean))
all_sd <- apply(strain_mean, 1, sd)
sd_col <- colors.from.values(all_sd, use.pheatmap.colors = TRUE)
expr.cor <- cor(strain_mean)
```

That is, genes that are highly expressed in the hepatocytes
of one strain also tend to be highly expressed in the hepatocytes
of other strains. The minimum expression correlation between strains
is R = `r signif(min(expr.cor), 2)`

```{r cor_heat}
pheatmap(expr.cor, display_numbers = TRUE)
```

The following correlation plots show a more detailed view of the 
expression correlation. Transcripts are colored by their standard
deviation across all strains.

```{r within_strain_sd, fig.width = 10, fig.height = 10}
#pdf("~/Desktop/pair_cor.pdf", height = 10, width = 10)
pairs(strain_mean, col = sd_col, pch = 16)
#dev.off()
```

We looked more closely at the most variable genes across strains.
It turns out that many of the most highly variable transcripts are
labeled as predicted genes or pseudogenes.


```{r gene_description}
ordered.description <- transcript.info[match(rownames(strain_mean), transcript.info[,1]),"description"]
terms.remove <- c("predicted", "pseudogene")
remove.list <- lapply(terms.remove, function(x) grep(x, ordered.description))
remove.idx <- sort(unique(unlist(remove.list)))
all.others <- setdiff(1:length(ordered.description), remove.idx)
sd.list <- lapply(remove.list, function(x) all_sd[x])
names(sd.list) <- terms.remove
sd.list$"All-Others" <- all_sd[all.others]
boxplot(sd.list, ylab = "Gene Expression SD", 
main = "Variance of pseudogenes is high")
strain_mean <- strain_mean[all.others,]
expr.cor <- cor(strain_mean)
```

When we remove these genes from the analysis, the minimum correlation
in expression across strain pairs is `r signif(min(expr.cor), 2)`.

```{r new_cor}
pheatmap(expr.cor, display_numbers = TRUE)
```

The correlation plots are as follows:

```{r cor_pairs, fig.height = 10, fig.width = 10}
#pdf("~/Desktop/pair_cor_new.pdf", height = 10, width = 10)
pairs(strain_mean, col = sd_col[all.others], pch = 16)
#dev.off()
```

These high correlations indicate that across all the strains, gene
expression in hepatocytes is largely the same. However, there are 
some differences across the strains, that we may be able to explain 
based on genotype, or epigenetic modifications.

There are some very high-variance genes still. The following plots show
the counts of words from the gene descriptions and counts of the first
three letters of each gene name. This counts large families of Mup-, 
Cyt-, and Rbs-/Rbl- genes. (These are Mups (major urinary proteins), 
cytochrome oxidases/reductases, and ribosomal genes.

```{r high_var}
all_sd <- all_sd[all.others]
#hist(all_sd, breaks = 100, xlab = "SD", main = "Expression SD")
high.sd <- 1
high.var.genes <- which(all_sd > high.sd)
var.high.idx <- match(names(high.var.genes), transcript.info[,1])
var.high.idx <- var.high.idx[which(!is.na(var.high.idx))]
high.var.info <- transcript.info[var.high.idx,c(2,3,4,14)]
info.words <- unlist(strsplit(high.var.info[,4], " "))
#barplot(table(info.words))
remove.words <- c("[Source:MGI", "1", "2", "2,", "3", "a,", "protein")
remove.idx <- sapply(remove.words, function(x) which(info.words == x))
pared.words <- info.words[-unique(unlist(remove.idx))]
#barplot(table(pared.words))
word.count <- table(pared.words)
common.words <- sort(word.count[which(word.count > 10)])

par(mar = c(4,8,4,4))
barplot(common.words, las = 2, horiz = TRUE, main = "common gene descriptions")

gene.names <- high.var.info[,1]
sub.names <- substr(gene.names, 1, 3)
#gene.names[order(sub.names)]
name.count <- table(sub.names)
high.count <- which(name.count > 2)
if(is.interactive){quartz(width = 11, height = 5)}
barplot(sort(name.count[high.count]), las = 2, horiz = TRUE, main = "common substrings")

high_var_enrich <- gost(names(high.var.genes), organism = "mmusculus")
plot.enrichment(high_var_enrich, num.terms = 30)
```

## Expression clusters by strain {.tabset .tabset-fade .tabset-pills}

Although the hepatocyte transcriptomes are highly correlated across 
strains, the variance that is there is highly related to genotype.

The following plot shows the first five principle components of
the transcriptomes. The transcriptomes cluster based on strain,
indicating that genotype is a primary factor determining gene expression. 

The purpose of this project is to idensify factors in local genotype,
and local epigenotype that lead to variation in expression across
these strains.

```{r var_pc, fig.height = 7, fig.width = 7}
all_expr <- Reduce("cbind", lapply(strain_expr, function(x) x[all.others,]))
if(is.interactive){quartz()}
expr.decomp <- plot.decomp(t(all_expr), cols = full.col, pc = 5)

expr.mean <- rowMeans(all_expr)
expr.sd <- apply(all_expr, 1, sd)
#plot(expr.mean, expr.sd)
```

## Strain separation by individual PCs {.tabset .tabset-fade .tabset-pills}

The following plots show how the strains are separated by each PC individually.

```{r pc_sep, results = "asis"}
for(i in 1:ncol(expr.decomp$u)){
    cat("### PC", i, "\n")
    if(is.interactive){quartz()}
    strain.vals <- lapply(u_strain, function(x) expr.decomp$u[which(strain == x),i])
    names(strain.vals) <- u_strain
    strain.means <- sapply(strain.vals, mean)
    strain.order <- order(strain.means)
    boxplot(strain.vals[strain.order], main = paste("PC", i), col = u_strain.col[strain.order])
    cat("\n\n")
}
```

## Expression variance and strain separation {.tabset .tabset-fade .tabset-pills}

We performed principle components analysis on transcripts of increasing 
variance across strains. The progressive PCA plots are shown below.
There is no strain separation for the low variance genes, and the strain
separation increases as the variance of the blocks increases.

```{r low_to_high_var, results = "asis"}
inbred_F_file <- here("Results", "eQTL", "Inbred_F_Stats.RDS")
if(!file.exists(inbred_F_file)){
    strain_F <- apply(all_expr, 1, function(x) anova(aov(x~as.factor(strain)))$"F value"[1])
    saveRDS(strain_F, inbred_F_file)
}else{
    strain_F <- readRDS(inbred_F_file)
}

num.chunks <- 15
chunked.sd <- chunkV(sort(all_sd), num.chunks)

#boxplot(chunked.sd)
sd_enrich <- vector(mode = "list", length = num.chunks)
sd_sil_coef <- matrix(NA, nrow = num.chunks, ncol = length(u_strain))
colnames(sd_sil_coef) <- u_strain
rownames(sd_sil_coef) <- apply(cbind(sapply(chunked.sd, min), sapply(chunked.sd, max)), 1, function(x) paste0(signif(x[1], 2), "-", signif(x[2], 2)))
for(i in 1:length(chunked.sd)){
    cat("###", round(min(chunked.sd[[i]]), 2), "to", round(max(chunked.sd[[i]]), 2), "\n")
    if(is.interactive){report.progress(i, num.chunks)}
    gene.locale <- match(names(chunked.sd[[i]]), rownames(all_expr))
    if(is.interactive){quartz()}
    expr.decomp <- plot.decomp(t(all_expr[gene.locale,]), cols = full.col)    
    sd_sil_coef[i,] <- silhouette_coef(expr.decomp$u, strain)
    sd_enrich[[i]] <- gost(names(chunked.sd[[i]]), organism = "mmusculus", 
        sources = c("GO", "KEGG", "REACTOME"))
    #pheatmap(cor(t(all_expr[gene.locale,])), show_rownames = FALSE, show_colnames = FALSE)
    #pheatmap(all_expr[gene.locale,], show_rownames = FALSE, show_colnames = FALSE)
    #plot.decomp(all_expr[gene.locale,], pc = 5)
    cat("\n\n")
}
```

## Strain separation by silhouette

The following heatmap shows how each strain diverges from the rest with
genes of increasing variance.

NZO has a higher than expected divergence score because it has very low
within-group variation.

PWK has a lower than expeted divergence score because it has very high
within-group variation.

```{r divergence, fig.height = 7, fig.width = 7}
div.order <- order(colSums(sd_sil_coef))
pheatmap(sd_sil_coef[,div.order], cluster_rows = FALSE, cluster_cols = FALSE)

par(mfrow = c(1,2))
boxplot(t(sd_sil_coef), horizontal = TRUE, las = 2)
abline(v = 0)

chunked.F <- lapply(chunked.sd, function(x) log10(strain_F[match(names(x), names(strain_F))]))
boxplot(chunked.F, horizontal = TRUE)
```


## Strain separation by variance

For each of these bins of genes, we used the F statistic from a strain-based
ANOVA to quantify how well the strains were separated in mean expression.
The distribution of the log of the F statistics in each bin is shown below.

As the SD gets higher, so does the average F statistic, indicating that the
strains become more separable.

```{r silhouette_dist, fig.height = 5, fig.width = 10}
par(mfrow = c(1,2))
plot.with.model(all_sd, log10(strain_F))
plot_in_bins(all_sd, log10(strain_F), bin.type = "percentile", n.bins = num.chunks)
```

## Functional enrichment by increasing SD {.tabset .tabset-fade .tabset-pills}

We looked at the functional enrichment of the genes in bins from
very low varation across strains to very high variation across 
strains.

The enrichments are shown below, first a detailed view of 
each bin, and then as a full gruop. At first glance, it looks as if
core cellular processes are represented in the low-variance bins,
whereas hepatocyte-specific processes are represented in the 
high-variance bins. 

For all these cells, their first function is to be cells. These
genes have low variance across the strains. Their next function is
to be a hepatocyte

```{r detail_enrich, results = "asis", fig.height = 10, fig.width = 10}
num.terms = 10; order.by = "p_value";max.term.size = 2000
for(i in 1:length(sd_enrich)){
    cat("### SD:", rownames(sd_sil_coef)[i], "\n")
    if(is.interactive){quartz(width = 8, height = 6)}
    layout(matrix(c(1,2,3,2), nrow = 2, byrow = TRUE))
    plot.enrichment(enrichment = sd_enrich[[i]], num.terms = num.terms, 
        order.by = order.by, max.term.size = max.term.size) 
    plot.enrichment.wordcloud(sd_enrich[[i]], num.terms = 10, 
        order.by = order.by, max.term.size = max.term.size, max.edge.lwd = 3,
        max.vertex.cex = 20)
    if(is.interactive){dev.off()}
    cat("\n\n")
}
```

## Group Enrichment

```{r bin_enrich, fig.width = 7, fig.height = 14}

dev.off()
if(is.interactive){quartz()}
plot.enrichment.group(sd_enrich, max.term.size = 2000, cluster_cols = FALSE, 
    cluster_rows = FALSE, transformation = sqrt)
```


## Expression variance and LOD score

How much of the variation in gene expression across strains is related 
to local genotype? We examined this question by looking at LOD scores in 
the DO mice vs. 

The figue below on the left 

We looked in the DO at the same genes to see whether genes with higher
standard deviation in the inbred mice had higher LOD scores in the DO.

The following plot shows the LOD score distributions for each set of 
genes in the bins with increasing SD. Although there are some very high
LOD scores in the low-SD bins, there is a clear positive relationship 
between SD and LOD score. 

```{r do_comp, fig.width = 10, fig.height = 10}
do.data <- load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData"))

counts <- dataset.mrna$data$raw
covar <- dataset.mrna$covar.matrix

mean.count <- colMeans(counts)
rna.seq <- counts[,which(mean.count >= 5)]
rna.seq <- floor(rna.seq)
norm.rnaseq <- vst(t(rna.seq))
adj.do <- adjust(t(norm.rnaseq), covar, retain.intercept = TRUE)

do.cor <- cor(t(adj.do))
#quartz();pheatmap(do.cor)

do.sd <- apply(adj.do, 2, sd)
do.mean <- apply(adj.do, 2, mean)

eqtl.r2 <- readRDS(here("Results", "eQTL", "eQTL.R2.RDS"))

#LOD score by chunk
#quartz()
par(mfrow = c(2,2))
chunked.r2 <- lapply(chunked.sd, function(x) eqtl.r2[match(names(x), rownames(eqtl.r2)),1])
boxplot(chunked.r2, ylab = "LOD score", main = "Variance explained with\nincreasing inbred SD",
notch = TRUE)

#quartz(width = 10, height = 5)
#stripchart(lapply(chunked.r2, function(x) round(x, 2)), method = "stack", 
#pch = 16, col = "darkgray", offset = 0.08, vertical = TRUE, cex = 0.5, add = TRUE)


mean.r2 <- sapply(chunked.r2, function(x) mean(x, na.rm = TRUE))
mean.sd <- sapply(chunked.sd, function(x) mean(x, na.rm = TRUE))
plot(mean.sd, mean.r2, main = "Mean R2 score vs. mean inbred SD",
    xlab = "mean SD", ylab = "mean R2", pch = 16)

do.chunked.expr <- sapply(chunked.sd, function(x) adj.do[,match(names(x), colnames(adj.do))])
do.chunked.sd <- lapply(do.chunked.expr, function(x) apply(x, 2, sd))
boxplot(do.chunked.sd, main = "SD in DO Expression with\nincreasing Inbred SD")

#quartz(width = 10, height = 5)
#stripchart(lapply(do.chunked.sd, function(x) round(x, 2)), method = "stack", 
#pch = 16, col = "darkgray", offset = 0.08, vertical = TRUE, cex = 0.5)


mean.do.sd <- sapply(do.chunked.sd, function(x) mean(x, na.rm = TRUE))
plot(mean.do.sd, mean.r2, main = "Mean R2 vs. mean DO SD",
    xlab = "mean SD", ylab = "mean LOD", pch = 16)
```

The figures below show the mean and sd of genes in the inbred and outbred
populations against each other. There is an incredibly strong relationship 
between mean expression in the DO and mean expression in the inbred mice. 
It is stronger than some of the pairwise correlations between inbred strains.

The variance across genes is not quite as well correlated as the mean. 

```{r sd_mean, fig.width = 10, fig.height = 5}
#sd in inbreds vs. sd in DO
mean.do.expr <- sapply(do.chunked.expr, function(x) apply(x, 2, function(y) mean(y, na.rm = TRUE)))
mean.inbred.expr <- sapply(chunked.sd, function(x) rowMeans(all_expr[match(names(x), rownames(all_expr)),]))

par(mfrow = c(1,2))
plot.with.model(unlist(mean.inbred.expr), unlist(mean.do.expr), 
    xlab = "Mean expression across inbred strains",
    ylab = "Mean expression across DO", report = "cor.test")

plot.with.model(unlist(chunked.sd), unlist(do.chunked.sd), 
    xlab = "SD across inbred strains",
    ylab = "SD across DO", report = "cor.test")

```


```{r summ_table}
#the following code checks out some examples of low-variance 
#genes with high LOD scores
#the variance is low in inbred and DO, but there is a clear
#cis effect in the DO. This does not always match the pattern
#of expression seen in the inbreds - epistasis, trans effects?

comp.table <- cbind(unlist(mean.inbred.expr), unlist(chunked.sd), 
    unlist(mean.do.expr), unlist(do.chunked.sd), unlist(chunked.r2))
colnames(comp.table) <- c("Mean_Inbred_Expr", "Inbred_SD", "Mean_DO_Expr", "DO_SD", "LOD")
```



```{r, eval = FALSE}
plot_in_bins(comp.table[,"DO_SD"], comp.table[,"LOD"], n.bins = num.chunks, bin.type = "even")
plot_in_bins(comp.table[,"DO_SD"], comp.table[,"LOD"], n.bins = num.chunks, bin.type = "percentile")

plot_in_bins(comp.table[,"Inbred_SD"], comp.table[,"DO_SD"], n.bins = 25, bin.type = "even")
plot_in_bins(comp.table[,"Inbred_SD"], comp.table[,"DO_SD"], n.bins = 25, bin.type = "percentile")
```


```{r checks, eval = FALSE}
gene.id <- "ENSMUSG00000029507"
#gene.id <- "ENSMUSG00000006019"

#gene.id <- "ENSMUSG00000078674" #high variance
gene.id <- sample(names(chunked.sd[[length(chunked.sd)]]), 1)
gene.name <- transcript.info[which(transcript.info[,"ensembl_gene_id"] == gene.id)[1],"external_gene_name"]
quartz(width = 7, height = 4)
strip.expr <- lapply(u_strain, function(x) all_expr[gene.id,which(strain == x)])
strain.mean <- sapply(strip.expr, mean)
mean.order <- order(strain.mean, decreasing = FALSE)
stripchart(strip.expr[mean.order], vertical = TRUE, pch = 16, 
    col = col.table[match(u_strain, col.table[,4]),3][mean.order],
    group.names = u_strain[mean.order], main = gene.name, cex = 1.5)
abline(h = mean(unlist(strip.expr)))

test.scan <- scan1(genoprobs, dataset.mrna$data$rz[,gene.id])
gene.chr <- transcript.info[which(transcript.info[,1] == gene.id)[1],"chromosome_name"]
chr.scan <- scan1coef(genoprobs[,gene.chr], dataset.mrna$data$rz[,gene.id])

#quartz(width = 5, height = 7)
par(mfrow = c(3,1))
boxplot(strip.expr, col = u_strain.col, pch = 16, vertical = TRUE, main = "Inbred Expression")
plot(test.scan, map = map, main = "eQTL scan in DO")
plot_coefCC(chr.scan, map = map, main = "coef scan in DO")

max.marker <- rownames(test.scan)[which.max(test.scan)]
max.chr <- strsplit(max.marker, "_")[[1]][1]
marker.geno <- genoprobs[[max.chr]][,,max.marker]

ind.locale <- match(rownames(adj.do), rownames(genoprobs[[1]]))
#quartz(width = 8, height = 5)
par(mfrow = c(2,4))
for(i in 1:8){
    plot(adj.do[,gene.id]~marker.geno[ind.locale,i], xlab = "", ylab = "", main = colnames(marker.geno)[i])
    #plot(dataset.mrna$data$rz[,gene.id]~marker.geno[ind.locale,i], xlab = "", ylab = "", main = colnames(marker.geno)[i])
}

part.cor.file <- here("Results", "eQTL", "DO_expr_part_cor.RDS")
if(!file.exists(part.cor.file)){
    part.cor <- pcor.shrink(adj.do)
    saveRDS(part.cor, part.cor.file)
}else{
    part.cor <- readRDS(part.cor.file)
}

reg.cor <- cor(t(adj.do))

library(igraph)
thresh.cor <- round(part.cor, 2)
diag(thresh.cor) <- 0
deg <- rowSums(abs(thresh.cor))
hist(deg, breaks = 100)

gene.idx <- 23
test.genes <- colnames(thresh.cor)[which(thresh.cor[gene.idx,] > 0)]
full.list <- c(colnames(thresh.cor)[gene.idx], test.genes)
transcript.info[match(full.list, transcript.info[,1]),2]
test <- gost(full.list, organism = "mmusculus")
quartz()
plot.enrichment(test, num.terms = 30)

net <- graph_from_adjacency_matrix(thresh.cor[full.list, full.list], weighted = TRUE)

V(net)$name <- transcript.info[match(full.list, transcript.info[,1]),2]
quartz()
plot(net)


deg <- degree(net)

hist(deg, breaks = 100)

common.ind <- intersect(names(deg), names(eqtl))
deg.locale <- match(common.ind, names(deg))
lod.locale <- match(common.ind, names(eqtl))
#head(cbind(names(deg)[deg.locale], names(eqtl)[lod.locale]))
plot.with.model(deg[deg.locale], eqtl.lod[lod.locale], report = "cor.test")
plot_in_bins(deg[deg.locale], eqtl.lod[lod.locale], n.bins = num.chunks, bin.type = "percentile")
plot_in_bins(eqtl.lod[lod.locale], deg[deg.locale], n.bins = num.chunks, bin.type = "percentile")
```