---
title: "Analyzing ChromHMM Results"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

## Introduction
The purpose of this workflow is to run all eQTL scans in the DO data.

```{r load_libraries}
library(here); library(qtl2)
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE, pattern = ".R")
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_data}
#A file containing the cis eQTL coefficients for each haplotype and each transcript
do.qtl.file <- here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData")
do.data <- load(do.qtl.file)
```

Do local scans only for each transcript.

```{r run_eQTL}
mrna <- dataset.mrna$data$rz
mrna.info <- dataset.mrna$annot.mrna
covar <- dataset.mrna$covar.matrix

write.csv(mrna.info, here("Data", "DOQTL", "mRNA_Info.csv"), 
quote = FALSE, row.names = FALSE)

#the following function runs qtl2 for the chromosome
#on which the specified transcript is encoded, to 
#get a LOD score and the haplotype coefficients.
#It returns the LOD score and haplotype coefficients
#for the marker nearest the transcript.

scan.one.transcript <- function(idx){
    transcript.result <- list("LOD" = NA, "coef" = NA)
    mrna.chr <- unlist(mrna.info[idx,"chr"])
    mrna.nearest <- mrna.info[idx,"nearest.marker.id"]
    chr.locale <- which(names(genoprobs) == mrna.chr)
    if(length(chr.locale) > 0){
        transcript.vals <- mrna[,idx]
        if(mrna.chr != "X"){
            #get LOD scores for the chromosome
            cis.lod <- scan1(genoprobs[,chr.locale], transcript.vals, 
            kinship = K[[chr.locale]], addcovar = covar)
        
            #and coefficients
            cis.coef <- scan1coef(genoprobs[,chr.locale], transcript.vals, 
            kinship = K[[chr.locale]], addcovar = covar)


            transcript.result <- list("LOD" = cis.lod[mrna.nearest,1], 
            "coef" = cis.coef[mrna.nearest,])
        }
    }
return(transcript.result)
}

eqtl.file <- here("Data", "DOQTL", "eQTL_statistics.RDS")
save.periodically(scan.one.transcript, max.iter = nrow(mrna.info), 
filename = eqtl.file, save.every = 100)
```

```{r test_divergence, eval = FALSE}
library(DESeq2)
library(sva)
counts <- dataset.mrna$data$raw
mean.count <- colMeans(counts)
rna.seq <- counts[,which(mean.count >= 5)]
cts <- floor(rna.seq)
norm.counts <- vst(t(cts))
adj.count <- t(adjust(t(norm.counts), covar))
all_sd <- apply(adj.count, 1, sd)

#plot.decomp(t(norm.counts))
#plot.decomp(t(adj.count))

num.chunks <- 15
chunked.sd <- chunkV(sort(all_sd), num.chunks)
sd_enrich <- vector(mode = "list", length = num.chunks)
for(i in 1:length(chunked.sd)){
    if(is.interactive){report.progress(i, num.chunks)}
    gene.locale <- match(names(chunked.sd[[i]]), rownames(norm.counts))
    if(is.interactive){quartz()}
    sd_enrich[[i]] <- gost(names(chunked.sd[[i]]), organism = "mmusculus", sources = c("GO", "KEGG", "REACTOME"))
    #quartz();plot.enrichment(sd_enrich[[i]], num.terms = 30)   
}
boxplot(chunked.sd)

pdf(paste0("~/Desktop/DO_var_enrich.",num.chunks, ".pdf"), width = 7, height = 14)
plot.enrichment.group(sd_enrich, max.term.size = 2000, cluster_cols = FALSE, 
    cluster_rows = FALSE, transformation = sqrt)
dev.off()

```