---
title: "Expression Data Wrangling"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

The purpose of this workflow is to normalize the expression data from the
epigenetics experiment for downstream workflows.

This workflow applies a variance stabilizing transformation (vst) (from DESeq2 package).
The covariate in this data set is treatment: five of the strains were treated with
dexamethasone. We therefore include treatment in the vst model. 


```{r source_code}
library(here)
all.code.dir <- list.files(here("code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], pattern = ".R", full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```


```{r load_libraries,  warning = FALSE, message = FALSE, error = FALSE}
needed.packages <- c("here", "pheatmap", "DESeq2", "sva", "knitr", "pdist", "dendextend",
"biomaRt", "gProfileR")
load_libraries(needed.packages)
is.interactive = FALSE #set to FALSE when running from the command
#is.interactive = TRUE
```

```{r read_expr}
all.var <- ls()
rna.seq.file <- here("Data", "RNASeq", "StrainsEffCts9.txt")
rna.seq.loaded <- as.logical(length(which(all.var == "rna.seq")))

if(!rna.seq.loaded){
    rna.seq <- as.matrix(read.table(rna.seq.file, header = TRUE, stringsAsFactors = FALSE, sep = "\t"))
    min.mean = 5
    #filter to transcripts with at least a mean count of 5

    mean.counts <- apply(rna.seq, 1, mean)
    pass.filter <- which(mean.counts >= min.mean)
    rna.seq <- rna.seq[pass.filter,]
}
strains <- substr(colnames(rna.seq), 1, 2)
```

Identify the covariates

```{r covar}
ind.name <- colnames(rna.seq)
ind.label <- sapply(strsplit(ind.name, ""), function(x) paste(tail(x, 2), collapse = ""))
treat.label <- substr(ind.label, 1, 1)
```

## Remove Treatment Group

We are only using the controls in this analysis. Here we 
remove the treatment animals.

```{r remove_treat}
c.locale <- which(treat.label == "C")
rna.seq <- rna.seq[,c.locale]

#make an ID table
ind.name <- colnames(rna.seq)
ind.label <- sapply(strsplit(ind.name, ""), function(x) paste(tail(x, 2), collapse = ""))
treat.label <- substr(ind.label, 1, 1)
ind <- substr(ind.label, 2,2)
strain <- mapply(function(x,y) gsub(x, "", y), ind.label, ind.name)
u_strain <- unique(strain)
id.table <- cbind(strain, treat.label, ind)

#make strain color vector
col.table <- read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", comment.char = "!", stringsAsFactors = FALSE)
full.col <- col.table[match(id.table[,1], col.table[,4]),3]
u_strain.col <- col.table[match(u_strain, col.table[,4]),3]
```

## Floor counts

There are many counts that are between 0 and 1. Round all count values
to the next lower integer. The count data should have individuals in *columns*.

```{r raw_expr}
cts <- floor(rna.seq)
```


Apply the VST using treatment as a covariate.

```{r vst, fig.height = 10, fig.width = 10}
strain.df <- data.frame(as.factor(strain))
rownames(strain.df) <- colnames(cts) 
colnames(strain.df) <- "strain"
dds <- DESeqDataSetFromMatrix(cts, colData = DataFrame(strain.df), design = ~ strain)

vsd <- vst(dds, blind = FALSE)
vsd.blind <- vst(cts)

lgd <- log2(cts+1)

par(mfrow = c(2,2))
plot.decomp(t(cts), cols = full.col, main = "Raw Counts")
plot.decomp(t(assay(vsd)), cols = full.col, main = "VST")
plot.decomp(t(vsd.blind), cols = full.col, main = "Blind VST")
plot.decomp(t(lgd), cols = full.col, main = "Log2")
```

## Within-strain correlations {.tabset .tabset-fade .tabset-pills}

The following figures show the correlations in log10 expression between 
individuals of a single strain.

These look reasonable overall, although some genes had quite a bit of 
variability from individual to individual. 

Since we are interested in genetic regulation of gene expression, we
should select genes whose inter-individual variance is small compared
to the inter-strain variance.

```{r individual_correlations, results = "asis"}
for(i in 1:length(u_strain)){
  cat("###", u_strain[i], "\n")
  strain.locale <- which(strain == u_strain[i])
  if(is.interactive){quartz()}
  pairs(assay(vsd)[,strain.locale])
  cat("\n\n")
}
```

```{r save_data}
transformed.data.file <- here("Data", "RNASeq", "StrainsEffCts9_vst.RDS")
saveRDS(assay(vsd), transformed.data.file)
```

We also create a rank Z normalized form of the data for mapping.

```{r rankz, fig.height = 5, fig.width = 5}
z.expr <- apply(t(assay(vsd)), 1, rankZ)
boxplot(z.expr[,1:10])
plot.decomp(t(z.expr), cols = full.col, main = "RankZ Normalized Data Colored by Strain")
norm.data.file <- here("Data", "RNASeq", "StrainsEffCts9_norm.RDS")
saveRDS(z.expr, norm.data.file)
```
