---
title: "Compare ChromHMM Models"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


The purpose of this workflow is to compare different numbers of states specified 
for ChromHMM. We want to select the model that is best correlated with expression.
This workflow can be run after 2.Chromatin_and_Expression.Rmd has been run for multiple 
state numbers. It uses the "Prop\_Expression\_Cor" files to find the maximum and minimum 
associations between proportion of each state and the expression of each transcript. 
It plots these maxima and minima across the states analyzed.

## Source code

```{r source_code}
library("here")
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE, pattern = ".R")
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```


## Collect results
Collect all results for each state. The results are stored in the Prop\_Expression\_Cor
files in each analysis directory. These files hold the r values describing the correlation
between the proportion of each state and expression level in inbred mice.


```{r collect_results}
state.num <- list.files(here("Results", "ChromHMM"), full.names = TRUE)
state.names <- basename(state.num)
state.as.num <- as.numeric(sapply(strsplit(state.names, "_"), function(x) x[1]))
state_order <- order(state.as.num)
all.prop <- vector(mode = "list", length = length(state.num))
names(all.prop) <- basename(state.num)[state_order]
for(i in 1:length(state.num)){
	prop.file <- list.files(path = state.num[state_order[i]], pattern = "Prop_Expression_Cor", full.names = TRUE)
	if(length(prop.file) > 0){all.prop[[i]] <- readRDS(prop.file)}
}
```

## Compare results
Calculate the median correlations across all states for all models. Plot the maximum
and minimum for each model. 

```{r process, fig.width = 10, fig.height = 4}
all.prop.r <- lapply(all.prop, function(x) x$all.r)
all.prop.p <- lapply(all.prop, function(x) x$all.p)
med.prop.r <- lapply(all.prop.r, function(x) if(length(x) > 0){apply(x, 2, function(y) median(y, na.rm = TRUE))})

max.med.prop <- lapply(med.prop.r, function(x) if(length(x) > 0){c(min(x), max(x))})
min.prop <- unlist(lapply(max.med.prop, function(x) x[1]))
max.prop <- unlist(lapply(max.med.prop, function(x) x[2]))

min.max.table <- rbind(min.prop, max.prop)

xvals <- as.numeric(sapply(strsplit(names(min.prop), "_"), function(x) x[1]))

plot.two.y.axes(xvals, min.max.table[2,], xvals, min.max.table[1,], lwd = c(3,3), legend.pos = "topright",
xlab = "N States", ylab1 = "Maximum Median Correlation", ylab2 = "Minimum Median Correlation", 
plot.type = c("b", "b"), pch = c(16, 16))
```

```{r test, eval = FALSE}
best.pair <- apply(min.max.table, 2, function(x) sum(abs(x)))
barplot(best.pair, las = 2)


#get the number of states with correlations to each gene.
num.state.cor <- t(sapply(1:nrow(all.prop.r[[1]]), function(y) sapply(all.prop.r, function(x) length(which(!is.na(x[y,])))))))
n.genes.with.state.cor <- sapply(4:16, function(x) length(unique(which(num.state.cor == x, arr.ind = TRUE)[,1])))
names(n.genes.with.state.cor) <- 4:16
barplot(n.genes.with.state.cor)

#get the best correlation between state and gene across all states and genes
min.state.cor <- t(sapply(1:nrow(all.prop.r[[1]]), function(y) sapply(all.prop.r, function(x) min(x[y,], na.rm = TRUE))))
max.state.cor <- t(sapply(1:nrow(all.prop.r[[1]]), function(y) sapply(all.prop.r, function(x) max(x[y,], na.rm = TRUE))))
min.state.cor[which(!is.finite(min.state.cor))] <- NA
max.state.cor[which(!is.finite(max.state.cor))] <- NA

boxplot(max.state.cor, las = 2)
boxplot(min.state.cor, las = 2)

plot(rowMeans(max.state.cor, na.rm = TRUE), rowMeans(min.state.cor, na.rm = TRUE))
abline(h = 0, v = 0)
plot(colMeans(max.state.cor, na.rm = TRUE), colMeans(min.state.cor, na.rm = TRUE))


what <- which(max.state.cor < -0.9, arr.ind = TRUE)

idx = 1
barplot(rbind(abs(min.state.cor[idx,]), max.state.cor[idx,]), beside =TRUE, las = 2)

best.state.by.gene <- unlist(sapply(1:nrow(all.prop.r[[1]]), function(x) which.max(abs(min.state.cor[x,])+max.state.cor[x,])))
```

Although the seven-state model has the best positive correlation with a state, the
six-state model is almost as good and has a much better negative correlation.

```{r compare_further}
boxplot.r <- function(r.mats, p.mats, state.num){
    form.states <- paste0(state.num, "_states_C")
    state.locale <- which(names(r.mats) == form.states)
    
    r.mat <- r.mats[[state.locale]]
    p.mat <- p.mats[[state.locale]]

    plot.new()
    plot.window(xlim = c(0,(ncol(r.mat)+1)), ylim = c(-1, 1))
    for(i in 1:ncol(r.mat)){
        p.col <- colors.from.values(-log10(p.mat[,i]), use.pheatmap.colors = TRUE)
        not.na <- which(!is.na(r.mat[,i]))
        points(x = jitter(rep(i, nrow(r.mat))), y = r.mat[,i], col = p.col, 
        pch = 16, cex = 0.5)
        boxplot(r.mat[not.na,i], at = i, col = "lightgray", add = TRUE)
        }
    axis(2);mtext(side = 2, "Correlation Coefficient", line = 2.5)
    abline(h = 0)
    par(xpd = TRUE)
    text(x = 1:ncol(r.mat), y = -1.1, labels = paste0("State", 1:ncol(r.mat)), 
    srt = 90, adj = 1)
    mtext(side = 3, text = paste0(state.num, "-state model"), line = 1)
    par(xpd = FALSE)
}

show.emissions <- function(state.num){
    form.states <- paste0(state.num, "_states_C")
    emissions.file <- here("Data", "ChromHMM", form.states, paste0("emissions_", state.num, ".png"))
    cat("![](",emissions.file,")\n")
}
```

## Correlation Distributions {.tabset .tabset-fade .tabset-pills}

The following plots show the distributions of correlations between states and 
gene expression for the states tested.


```{r cor_plots, results = "asis"}
for(i in 1:length(state.as.num)){
    cat("###", state.names[i], "\n")
    boxplot.r(all.prop.r, all.prop.p,  state.as.num[i])
    show.emissions(state.as.num[i])
    cat("\n\n")
}
```
