---
title: "Map chromatin to genotype"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


The purpose of this workflow is to create a "chromaprobs" object from a genoprobs
DO object. We used the 8-state chromatin output from ChromHMM.

Here we used those states and their locations to create the chromaprobs object. 
The genoprobs object consists of individuals in rows, locations in columns, and 
haplotypes in the third dimension. The chromaprobs object with have chromatin state
in the third dimension. 

The basic pipeline for mapping chromatin to genotype is as follows: 

1. For each genetic locus, identify the genotype of each individual. 
    * For example, individual 1 is heterozygous A/J and B6 at locus 1.
2. Use the genotype to select the chromatin states from the 8-state model at that position.
    * For example, the A/J chromatin state at locus 1 is 6, and the B6 chromatin state 
  at locus 1 is 2
3. Replace the genotype with the appropriate states.
    * Now the chromaprobs for individual 1 at locus one are 0.5 state 6 and 0.5 state 2.


In an update of this workflow, it can now be used to make a chromaprobs file for
a single mark to better test the importance of each chromatin mark.

Use the code below to specify which mark should be tested.

```{r set_mark}
mark <- NULL #use NULL to make a chromaprobs file for all states
#mark <- "H3K4me1"
```
  
```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
needed.packages <- c("here")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```


Load the DO data. We will be using the genoprobs object for our mapping.

```{r load_do_data}
all.var <- ls()
data.loaded <- as.logical(length(which(all.var == "dataset.expr")))
if(!data.loaded){
  load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_veiwer_v0.Rdata"))
}
```

In addition, we load the bed files from the 8-state ChromHMM model. 
These tell us the chromatin state at each position for each inbred 
strain.

```{r load_bed}
num.states = 8
form.states <- paste0(formatC(num.states, digits = 2, flag = 0), "_states_C")
bed.files <- list.files(here("Data", "ChromHMM", form.states), full.names = TRUE, pattern = ".bed")
strains <- sapply(strsplit(basename(bed.files), "_"), function(x) x[1])

#read in all bed files
all.bed <- vector(mode = "list", length = length(strains))
names(all.bed) <- strains
for(i in 1:length(strains)){
    all.bed[[i]] <- read.table(bed.files[i], sep = "\t", skip = 1, 
    stringsAsFactors = FALSE)
  }
```

Load the strain key. This helps us keep track of the different ways strains are named.

```{r strain.key}
strain.key <- read.table(here("Data", "support_files", "strain.color.table.txt"), sep = "\t",
stringsAsFactors = FALSE, comment.char = "*")
```

Load the emissions probabilities so we can assign marks to states.

```{r emissions}
emissions <- read.table(here("Data", "ChromHMM", form.states, "emissions_8.txt"), header = TRUE,
sep = "\t", row.names = 1)
mark.cols <- c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3")
names(mark.cols) <- colnames(emissions)
barplot(t(as.matrix(round(emissions, 2))), col = mark.cols, horiz = TRUE, las = 2)
legend("bottomright", fill = mark.cols, legend = colnames(emissions))


if(!is.null(mark)){
  mark.locale <- which(colnames(emissions) == mark)
  mark.states <- emissions[,mark.locale]
}
```

Map the chromatin states onto the genotypes. 
Occasionally we hit a marker that is exactly on the border between two ChromHMM regions.
One ChromHMM region begins at the exact location that the last one ended, so there is
no way to resolve which state

```{r chromaprobs}
chromprob.file <- here("Results", "ChromHMM", form.states, paste0("chromprobs", mark, ".RData"))
if(!file.exists(chromprob.file)){
  chromaprobs <- genoprobs

  #put the bed files in the same order as the haplotypes in genoprobs
  bed.order <- order.strains(colnames(genoprobs[[1]]), names(all.bed), strain.key)
  all.bed <- all.bed[bed.order]

  #This function converts genotype to chromatin state for a single marker
  #for a single individual
  geno.to.chrom <- function(ind.geno, strain.state, total.states){
    chrom.states <- sapply(1:total.states, function(x) sum(ind.geno[which(strain.state == x)]))
    return(chrom.states)
  }

  for(i in 1:length(chromaprobs)){ #for each chromosome
    print(i)
    chr = names(chromaprobs)[i]
    chr.idx <- lapply(all.bed, function(x) which(x[,1] == paste0("chr", chr)))
    chr.bed <- all.bed
    for(ch in 1:length(chr.bed)){
      chr.bed[[ch]] <- chr.bed[[ch]][chr.idx[[ch]],]
    }
    
    marker.names <- dimnames(chromaprobs[[i]])[[3]]
    marker.pos <- as.numeric(sapply(strsplit(marker.names, "_"), function(x) x[2]))
    
    for(m in 1:length(marker.pos)){ #for each marker on the chromosome
      #find the overlapping region in the bed files
      report.progress(m, length(marker.pos))
      
      #find the chromatin bins in which the marker resides
      overlapping.region <- sapply(chr.bed, function(x) 
      intersect(which(x[,2] <= marker.pos[m]), which(x[,3] >= marker.pos[m])[1]))
      
      #get the state for each inbred strain at this location
      strain.state <- mapply(function(x,y) x[y,4], chr.bed, overlapping.region)
      marker.geno <- genoprobs[[i]][,,m]

      #replace the genotype probabilities with state probabilities
      #cbind(round(marker.chrom[1,], 2), strain.state)
      #chromsum <- sapply(1:num.states, function(x) sum(marker.chrom[1,which(strain.state == x)]))
      #names(chromsum) <- 1:num.states
      #round(marker.geno[1,], 2)
      #strain.state
      #round(chromsum, 2)
      
      #replace genotypes with states for all individuals for this marker
      marker.chrom <- t(apply(marker.geno, 1, 
      function(x) geno.to.chrom(ind.geno = x, strain.state = strain.state, total.states = num.states)))
      colnames(marker.chrom) <- 1:ncol(marker.chrom)

      chromaprobs[[i]][,,m] <- marker.chrom
    } #end looping over markers on the chromosome
  } #end looping over chromosomes

  saveRDS(chromaprobs, chromprob.file)
}
```