---
title: "Map chromatin to genotype"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


The purpose of this workflow is to impute chromatin states onto the DO mice
for each transcript using their genotypes at the marker nearest the TSS
of the transcript.

We then scan these chromatin probabilities in qtl2 to infer the effects of 
chromatin state on each transcript.

The basic pipeline for mapping chromatin to genotype is as follows: 

1. For each transcript identify the DO haplotypes at the marker nearest the TSS.
2. Use the haplotypes to calculate the chromatin states for each DO individual.
3. Calculate chromatin effects on transcript abundance using qtl2.

```{r source_code}
library("here")
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
args <- commandArgs(trailingOnly=T)
num.states <- as.numeric(args[1])
delete.previous <- as.logical(args[2])

if(is.na(num.states)){num.states = 14; delete.previous = FALSE}

needed.packages <- c("qtl2", "gtools", "parallel", "doParallel", "abind",
  "Matrix")
load_libraries(needed.packages, personal.library = TRUE)

results.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))
#barplot(1:length(chrom.colors), col = chrom.colors)
is.interactive = FALSE
#is.interactive = TRUE
n.cores = detectCores() - 1
#n.cores = 19

data(CCcolors)
chrom.colors <- colors.from.values(1:num.states, use.pheatmap.colors = TRUE)
```


Load data.

```{r load_do_data}
transcript.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
transcript.haplotypes <- readRDS(here("Data", "DOQTL", "transcript.haplotypes.RDS"))

#rankZ normalized expression from Svenson_DO850_for_eQTL_viewer_v9.RData
#generated by 2.0_chromaprobs_setup.Rmd
expr.data <- readRDS(here("Data", "DOQTL", "DO_Expression.RDS")) 
expr <- expr.data$expr
covar <- expr.data$covar
#pull out diet as an interactive covariate
int.covar <- covar[,2,drop=FALSE]
chrom.state.file <- file.path(results.dir, "Expanded_Chromatin.RDS") #generated by 2.0_chromaprobs_setup.Rmd
chrom.states <- readRDS(chrom.state.file)
chrom.mats <- readRDS(file.path(results.dir, paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

common.transcripts <- intersect(colnames(expr), names(chrom.states)) #transcripts that are shared between chromatin and DO expression data
hf.locale <- which(covar[,"diethf"] == 1)
chow.locale <- which(covar[,"diethf"] == 0)
```

Load the strain key. This helps us keep track of the different ways strains are named.

```{r strain.key}
strain.key <- read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", stringsAsFactors = FALSE, comment.char = "*")
```

For transcripts we perform the following steps. 

1. Impute local chromatin state in DO from inbred chromatin state matrices.
2. For each chromatin state, calculate the variance explained
    at each position in the gene. Retain the maximum coefficient
    for each state across all positions.
3. For each chromatin state, calculate beta coefficient each position 
    in the gene. Retain all coefficients to align with the inbred data.
4. For haplotypes, calculated variance explained for each individual
    state (not the full haplotype).
    This makes sure we have one degree of freedom across all our 
    models so we can directly compare them. This also allows a 
    chromatin state to potentially explain more variance in gene
    expression than an individual haplotype. If both A/J and B6 have
    a repressor in a given position, and that causes reduced expression,
    the repressor (chromatin state) will explain more variance in 
    gene expression than either individual haplotype.
    Retain the maximum variance explained for each state across all positions
5. For haplotypes calculate the beta coefficient for each haplotype.

We do not calculate LOD scores here. We used to, but we are not 
using them, and we will save time by not calculating them.

```{r scan_chQTL_and_eQTL}
chrom.coef.file <- file.path(results.dir, "chQTL.state.coef.RDS")
chrom.r2.file <- file.path(results.dir, "chQTL.state.R2.RDS")

geno.coef.file <- file.path(results.dir, "eQTL.hap.coef.RDS")
geno.r2.file <- file.path(results.dir, "eQTL.hap.R2.RDS")

if(delete.previous){
  unlink(chrom.coef.file)
  unlink(chrom.r2.file)
  unlink(geno.coef.file)
  unlink(geno.r2.file)
}

save.every = 100

if(file.exists(chrom.coef.file)){
  chrom.coef <- readRDS(chrom.coef.file)
  chrom.r2 <- readRDS(chrom.r2.file)
  geno.coef <- readRDS(geno.coef.file)
  geno.r2 <- readRDS(geno.r2.file)

  #figure out where we left off.
  not.na <- which(!is.na(geno.r2[,1]))  
  left.off <- max(not.na)
  start.at <- left.off + 1
}else{
    start.at <- 1
    #if we are at the beginning, intialize all the objects for holding
    #results. 
    geno.coef <- vector(mode = "list", length = length(common.transcripts))
    names(geno.coef) <- common.transcripts

    #make space for the full set of chromaprob coefficients for
    #each transcript.
    chrom.coef <- chrom.r2 <- vector(mode = "list", length = length(common.transcripts))
    names(chrom.coef) <- names(chrom.r2) <- common.transcripts
  
    #keep variance explained for each haplotype in each transcript
    geno.r2 <- matrix(NA, ncol = 8, nrow = length(common.transcripts))
    rownames(geno.r2) <- common.transcripts 
    colnames(geno.r2) <- LETTERS[1:8]

}

```


```{r stat_fun}

#Scans effects of each state individually
scan_chrom_transcript_by_state <- function(gene.id, 
  geno.type = c("chromatin", "haplotype"), perm.order = NULL){

  geno.type = geno.type[1]

  gene.name <- transcript.info[which(transcript.info[,"ensembl_gene_id"] == gene.id),"external_gene_name"][1]
  if(is.na(gene.name)){return(NA)}

  expr.locale <- which(colnames(expr) == gene.id)
  chrom.locale <- which(names(chrom.mats) == gene.id)

  if(length(expr.locale) == 0 || length(chrom.locale) == 0){
    return(NA)
  }

    geno <- get_one_geno(gene.name, transcript.info, transcript.haplotypes, 
        chrom.states, strain.key, geno_type = geno.type, perm.order = perm.order)

    if(is.na(geno)){return(NA)}

    #limit to only females because only females are in the inbred data
    geno[[1]] <- geno[[1]][grep("F", rownames(geno[[1]])),,,drop=FALSE]

    #check for variation
    test.var <- round(apply(geno[[1]], 3, function(x) apply(x, 2, function(y) var(y))), 2)
    
    if(geno.type == "chromatin"){
      no.var <- length(which(apply(test.var, 1, function(x) length(unique(x))) > 1)) <= 1
    }else{
      no.var <- all(test.var[,1] == 0)
    }
    
    if(no.var){return(NA)}
   
    #calculate R2
    #adjust expression for covariates first, so we don't include 
    #the covariates in the variance explained
    adj.expr <- adjust(expr[grep("F", rownames(expr)),expr.locale,drop=FALSE], covar)

    all.models <- apply(geno[[1]], 3, function(x) apply(x, 2, function(y) lm(adj.expr~y)))
    all.r2 <- sapply(all.models, function(x) sapply(x, function(y) summary(y)$adj.r.squared))
    #r2.max <- apply(all.r2, 2, max) #maximum R2 by position
    #r2.max <- apply(all.r2, 1, max) #maximum R2 by state

    all.coef <- sapply(all.models, function(x) sapply(x, function(y) coef(y)[2]))

    if(geno.type == "chromatin"){
      #Make sparse matrices to save space. Don't add dimnames. 
      #these take up a lot of space, we can use dimnames from 
      #chrom.mats later.

      non.zero.idx <- which(all.coef != 0, arr.ind = TRUE)
      non.zero.val <- all.coef[which(all.coef != 0)]
      sparse.coef <- sparseMatrix(non.zero.idx[,1], non.zero.idx[,2], x = non.zero.val,
        dims = c(nrow(all.coef), ncol(all.coef))) 

      non.zero.idx <- which(all.r2 != 0, arr.ind = TRUE)
      non.zero.val <- all.r2[which(all.r2 != 0)]
      sparse.r2 <- sparseMatrix(non.zero.idx[,1], non.zero.idx[,2], x = non.zero.val,
        dims = c(nrow(all.r2), ncol(all.r2))) 

    }else{
      sparse.coef <- all.coef
      sparse.r2 <- all.r2
    }  
  results <- list("coef" = sparse.coef, "r2" = sparse.r2)
  return(results)
}

```

```{r run_remaining}
#new tests to constrain degrees of freedom
#tests each state individually

if(start.at < length(common.transcripts)){ #if we haven't scanned all the transcripts yet
  sink(file.path(results.dir, "progress.txt"))
  #for each remaining transcript
  for(i in start.at:length(common.transcripts)){
    print(i)

    if(is.interactive){report.progress(i, length(common.transcripts))}
    
    chrom.results <- scan_chrom_transcript_by_state(gene.id = common.transcripts[i], 
      geno.type = "chromatin")

    if(length(chrom.results) > 1){
      chrom.coef[[i]] <- chrom.results$coef
      chrom.r2[[i]] <- chrom.results$r2
    }else{
      chrom.coef[[i]] <- NA
      chrom.r2[[i]] <- NA
    }

    gene.results <- scan_chrom_transcript_by_state(gene.id = common.transcripts[i], 
      geno.type = "haplotype")

    if(length(gene.results) > 1){
      geno.coef[[i]] <- gene.results$coef
      geno.r2[i,] <- gene.results$r2
    }else{
      geno.coef[[i]] <- NA
    }
    
    if(is.interactive){
      gene.name <- transcript.info[which(transcript.info[,"ensembl_gene_id"] == gene.id), "external_gene_name"][1]
      plot.chrom.effects_by_state(gene.name, expr, covar, chrom.mats, transcript.info, 
        transcript.haplotypes, chrom.states, strain.key)
    }

    #save every once in a while
    if(i %% save.every == 0){
      saveRDS(chrom.coef, chrom.coef.file)
      saveRDS(chrom.r2, chrom.r2.file)
      saveRDS(geno.coef, geno.coef.file)
      saveRDS(geno.r2, geno.r2.file)
    }
  }
  #also save at the end of everything.
  saveRDS(chrom.coef, chrom.coef.file)
  saveRDS(chrom.r2, chrom.r2.file)
  saveRDS(geno.coef, geno.coef.file)
  saveRDS(geno.r2, geno.r2.file)
  sink()
}
```

## Assess variance explained

Compare the variance explained by individual chromatin states
to the variance explained by individual haplotypes.

The following boxplot shows the distribution of variance
explained by each state. 

```{r var_exp_state}
chrom.r2.mat <- t(sapply(chrom.r2, function(x) if(length(x) > 1){apply(as.matrix(x), 1, max)}else{rep(NA, num.states)}))
boxplot(chrom.r2.mat, las = 2)
```

The following boxplot shows the variance explained
by each haplotype.

```{r var_exp_hap}
boxplot(geno.r2)
```

The plot below shows the variance explained by individual
haplotypes vs. the variance explained by individual chromatin
states.

There are many cases in which an individual chromatin state
explains more variance than an individual haplotype. This 
suggests that multiple haplotypes share a chromatin state
that is influencing gene expression.

There are also some cases in which an individual haplotype 
explains more variance than any individual chromatin state.


```{r var_exp}
max.chrom.r2 <- apply(chrom.r2.mat, 1, function(x) if(!all(is.na(x))){max(x, na.rm = TRUE)}else{NA})
max.geno.r2 <- apply(geno.r2, 1, function(x) if(!all(is.na(x))){max(x, na.rm = TRUE)}else{NA})

plot(max.geno.r2, max.chrom.r2, xlab = "R2 Haplotype",
    ylab = "R2 Chromatin State")
abline(0,1)
```

```{r counts}
r2.table <- cbind(round(max.chrom.r2, 3), round(max.geno.r2, 3))
chrom.better <- length(which(apply(r2.table, 1, function(x) x[1] > x[2])))
hap.better <- length(which(apply(r2.table, 1, function(x) x[1] < x[2])))
both.same <- length(which(apply(r2.table, 1, function(x) x[1] == x[2])))
counts <- c("Chromatin" = chrom.better, "Equal" = both.same, 
  "Haplotype" = hap.better)
barplot_with_num(counts, main = "Feature with Greater R2", ylab = "# Transcripts")
```


## Coefficients by position {.tabset .tabset-fade .tabset-pills}

```{r coef_by_pos, results = "asis", fig.width = 7, fig.height = 5}

gene.names <- transcript.info[match(names(chrom.coef), transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
matched.chrom.mats <- chrom.mats[match(names(chrom.coef), names(chrom.mats))]
centered.coord <- lapply(1:length(gene.names),
  function(x) if(length(matched.chrom.mats[[x]]) > 1){center.on.feature(gene.names[x], 
  transcript.info, as.numeric(rownames(matched.chrom.mats[[x]])), feature = "full")}else{NA})

ymin <- min(sapply(chrom.coef, min), na.rm = TRUE)
ymax <- max(sapply(chrom.coef, max), na.rm = TRUE)

#find indices where both coordinates and coefficients have data
coord.len <- sapply(centered.coord, length)
coef.len <- sapply(chrom.coef, function(x) if(length(x) > 1){return(ncol(x))}else{1})
use.idx <- intersect(which(coord.len > 1), which(coef.len > 1))

nbins = 25
coord.bins <- segment.region(-0.5, 1.5, nbins)
plot.coord <- coord.bins[1:(length(coord.bins)-1)]
all.coord <- unlist(lapply(use.idx, function(x) as.numeric(names(centered.coord[[x]]))))

state.coef.file <- file.path(results.dir, "State_Coef_SE_DO.RDS")

if(!file.exists(state.coef.file)){

  state.coef.se <- vector(mode = "list", length = num.states)
  names(state.coef.se) <- 1:num.states

  for(st in 1:num.states){
    state.coef <- unlist(lapply(use.idx, function(x) chrom.coef[[x]][st,]))

    binned.coef <- lapply(1:(length(coord.bins)-1), 
      function(x) state.coef[intersect(which(all.coord >= coord.bins[x]), 
      which(all.coord < coord.bins[(x+1)]))])
    #boxplot(binned.coef)

    coef.mean <- sapply(binned.coef, function(x) mean(x, na.rm = TRUE))
    #coef.se <- sapply(binned.coef, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))
    coef.test <- lapply(binned.coef, function(x) t.test(x, conf.level = 0.99))
    coef.conf <- sapply(coef.test, function(x) x$conf.int)

    coef.info <- rbind(coef.mean, coef.conf)
    colnames(coef.info) <- plot.coord
    state.coef.se[[st]] <- coef.info
  }
  saveRDS(state.coef.se, state.coef.file)
}else{
  state.coef.se <- readRDS(state.coef.file)
}

#plot results
#pdf("~/Desktop/state.effects.in.DO.pdf")
for(st in 1:num.states){
  cat("### State", st, "\n")

  coef.mean <- state.coef.se[[st]]["coef.mean",]
  #coef.se <- state.coef.se[[st]]["coef.se",]
  coef.conf <- state.coef.se[[st]][2:3,]

  plot.new()
  #plot.window(xlim = c(-0.5, 1.5), ylim = c(min(coef.mean-coef.se, na.rm = TRUE), max(coef.mean+coef.se, na.rm = TRUE)))
  plot.window(xlim = c(-0.5, 1.5), ylim = c(min(coef.conf, na.rm = TRUE), max(coef.conf, na.rm = TRUE)))
  #plot.poly.xy(poly.top.x = plot.coord, poly.top.y = coef.mean+coef.se, 
  #  poly.bottom.x = plot.coord, poly.bottom.y = coef.mean-coef.se,
  #  border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
  plot.poly.xy(poly.top.x = plot.coord, poly.top.y = coef.conf[1,], 
    poly.bottom.x = plot.coord, poly.bottom.y = coef.conf[2,],
    border = NULL, col = "gray", lwd = 1, new.plot = FALSE)

  abline(h = 0)
  axis(1)
  axis(2)
  abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
  mtext(paste("State", st, "Effects in DO"), side = 3, line = 1.5)
  cat("\n\n")
}
#dev.off()
```


## Variance Explained by Position {.tabset .tabset-fade .tabset-pills}

```{r var_by_pos, results = "asis", fig.width = 7, fig.height = 5}

ymin <- min(sapply(chrom.r2, min), na.rm = TRUE)
ymax <- max(sapply(chrom.r2, max), na.rm = TRUE)

state.var.file <- file.path(results.dir, "State_Var_SE_DO.RDS")
if(!file.exists(state.var.file)){

  state.var.se <- vector(mode = "list", length = num.states)
  names(state.var.se) <- 1:num.states

  for(st in 1:num.states){
    state.r2 <- unlist(lapply(use.idx, function(x) chrom.r2[[x]][st,]))

    binned.r2 <- lapply(1:(length(coord.bins)-1), 
      function(x) state.r2[intersect(which(all.coord >= coord.bins[x]), 
      which(all.coord < coord.bins[(x+1)]))])
    #boxplot(binned.r2)

    r2.mean <- sapply(binned.r2, function(x) mean(x, na.rm = TRUE))
    r2.se <- sapply(binned.r2, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))

    var.info <- rbind(r2.mean, r2.se)
    colnames(var.info) <- plot.coord
    state.var.se[[st]] <- var.info
}
  saveRDS(state.var.se, state.var.file)
}else{
  state.var.se <- readRDS(state.var.file)
}


for(st in 1:num.states){
  cat("### State", st, "\n")

  r2.mean <- state.var.se[[st]]["r2.mean",]
  r2.se <- state.var.se[[st]]["r2.se",]

  plot.new()
  plot.window(xlim = c(-0.5, 1.5), ylim = c(min(r2.mean-r2.se, na.rm = TRUE), 
    max(r2.mean+r2.se, na.rm = TRUE)))
  plot.poly.xy(poly.top.x = plot.coord, poly.top.y = r2.mean+r2.se, 
    poly.bottom.x = plot.coord, poly.bottom.y = r2.mean-r2.se,
    border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
  abline(h = 0)
  axis(1)
  axis(2)
  abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
  mtext(paste("State", st, "Variance Explained in DO"), side = 3, line = 1.5)
  cat("\n\n")
}
#dev.off()
```

```{r examples, eval = FALSE}
#example of haplotype explaining much more than chromatin
test <- which(apply(r2.table, 1, function(x) x[1] <= 0.4 && x[2] >= 0.2))

#example of chromatin explaining more than haplotype
test <- which(apply(r2.table, 1, function(x) x[1] >= 0.2 && x[2] <= 0.4))

test <- which(apply(r2.table, 1, function(x) x[1] == x[2]))

#boxplot(r2.table[test,])
#gene.name = "Pkd2"

pdf("~/Desktop/test.pdf", width = 9, height = 9)
for(g in 1:length(test)){
  gene.id <- names(test)[g]
  gene.name <- transcript.info[which(transcript.info[,"ensembl_gene_id"] == gene.id)[1], "external_gene_name"]
  plot.chrom.effects_by_state(gene.name, expr, covar, chrom.mats, 
      transcript.info, transcript.haplotypes, chrom.states, strain.key)
}
dev.off()
```
