---
title: "Map chromatin to genotype"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


The purpose of this workflow is to impute chromatin states onto the DO mice
for each transcript using their genotypes at the marker nearest the TSS
of the transcript.

We then scan these chromatin probabilities in qtl2 to infer the effects of 
chromatin state on each transcript.

The basic pipeline for mapping chromatin to genotype is as follows: 

1. For each transcript identify the DO haplotypes at the marker nearest the TSS.
2. Use the haplotypes to calculate the chromatin states for each DO individual.
3. Calculate chromatin effects on transcript abundance using qtl2.

```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
args <- commandArgs(trailingOnly=T)
num.states <- as.numeric(args[1])
delete.previous <- as.logical(args[2])

if(is.na(num.states)){num.states = 10;delete.previous = FALSE}

needed.packages <- c("here", "qtl2")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}
results.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))
#barplot(1:length(chrom.colors), col = chrom.colors)
is.interactive = FALSE
#is.interactive = TRUE
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
data(CCcolors)
chrom.colors <- colors.from.values(1:num.states, use.pheatmap.colors = TRUE)
```


Load data.

```{r load_do_data}
transcript.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
transcript.haplotypes <- readRDS(here("Data", "DOQTL", "transcript.haplotypes.RDS"))

#rankZ normalized expression from Svenson_DO850_for_eQTL_viewer_v9.RData
#generated by 2.0_chromaprobs_setup.Rmd
expr.data <- readRDS(here("Data", "DOQTL", "DO_Expression.RDS")) 
expr <- expr.data$expr
covar <- expr.data$covar
#pull out diet as an interactive covariate
int.covar <- covar[,2,drop=FALSE]
chrom.state.file <- file.path(results.dir, "Expanded_Chromatin.RDS") #generated by 2.0_chromaprobs_setup.Rmd
chrom.states <- readRDS(chrom.state.file)
chrom.mats <- readRDS(file.path(results.dir, paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

#not all of the transcripts are in the chromatin state
#object. We should go back to the beginning and re-run
#everything with the new expression object as a reference
common.transcripts <- intersect(colnames(expr), names(chrom.states)) #transcripts that are shared between chromatin and DO expression data
hf.locale <- which(covar[,"diethf"] == 1)
chow.locale <- which(covar[,"diethf"] == 0)
```

Load the strain key. This helps us keep track of the different ways strains are named.

```{r strain.key}
strain.key <- read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", stringsAsFactors = FALSE, comment.char = "*")
```

For transcript we perform the following steps. To reduce memory
requirements, we do all steps for each transcript and keep only the
LOD score and coefficient scores for each transcript.

1. Impute local chromatin state in DO from inbred chromatin state matrices.

2. Use qtl2 to calculate LOD scores and state coefficients for all states
and transcripts.


Run R/qtl2 on the chromatin probabilities for each transcript.

Also scan all the transcripts with the actual genoprobs at the 
cis location to calculate eQTL LOD scores and coefficients.


```{r scan_chQTL_and_eQTL}
chrom.lod.file <- file.path(results.dir, "chQTL.lod.RDS")
chrom.coef.file <- file.path(results.dir, "chQTL.coef.RDS")
chrom.r2.file <- file.path(results.dir, "chQTL.R2.RDS")
chrom.effect.file <- file.path(results.dir, "chromatin_state_effect.RDS")

geno.lod.file <- file.path(results.dir, "eQTL.lod.RDS")
geno.coef.file <- file.path(results.dir, "eQTL.coef.RDS")
geno.r2.file <- file.path(results.dir, "eQTL.R2.RDS")

if(delete.previous){
  unlink(chrom.lod.file)
  unlink(chrom.coef.file)
  unlink(chrom.r2.file)
  unlink(chrom.effect.file)
  unlink(geno.lod.file)
  unlink(geno.coef.file)
  unlink(geno.r2.file)
}

save.every = 100

model.names <- c("Additive", "Diet-Interactive", "Chow", "HF")

if(file.exists(chrom.coef.file)){
  chrom.lod <- readRDS(chrom.lod.file)
  chrom.coef <- readRDS(chrom.coef.file)
  chrom.r2 <- readRDS(chrom.r2.file)
  geno.lod <- readRDS(geno.lod.file)
  geno.coef <- readRDS(geno.coef.file)
  geno.r2 <- readRDS(geno.r2.file)
  chrom.effect <- readRDS(chrom.effect.file)

  #figure out where we left off.
  not.na <- which(!is.na(geno.lod[,1]))  
  left.off <- max(not.na)
  start.at <- left.off + 1
}else{
  start.at <- 1
  #if we are at the beginning, intialize all the objects for holding
  #results. Make space for one set of genoprob coefficients for each 
  #transcript
  geno.coef <- vector(mode = "list", length = length(common.transcripts))
  names(geno.coef) <- common.transcripts

  #make space for the full set of chromaprob coefficients for
  #each transcript.
  chrom.coef <- chrom.lod <- chrom.r2 <- chrom.effect <- vector(mode = "list", length = length(common.transcripts))
  names(chrom.coef) <- names(chrom.coef) <- names(chrom.r2) <- names(chrom.effect) <- common.transcripts
  
  #there is only one lod score per transcript for the haplotypes
  geno.lod <- geno.r2 <- matrix(NA, ncol = length(model.names), nrow = length(common.transcripts))
  rownames(geno.lod) <- rownames(geno.r2) <- common.transcripts 
  colnames(geno.lod) <- colnames(geno.r2) <- model.names
}
```


```{r run_remaining}
if(start.at < length(common.transcripts)){ #if we haven't scanned all the transcripts yet
  sink(file.path(results.dir, "progress.txt"))
  #for each remaining transcript
  #for(i in 201:300){
  for(i in start.at:length(common.transcripts)){
    print(i)
    if(is.interactive){report.progress(i, length(common.transcripts))}
    #gene.id <- transcript.info[which(transcript.info[,"external_gene_name"] == "Irf5"),"ensembl_gene_id"][1]
    #i <- which(common.transcripts == gene.id)
    gene.id <- common.transcripts[i]
    gene.name <- transcript.info[which(transcript.info[,"ensembl_gene_id"] == gene.id),"external_gene_name"][1]
    expr.locale <- which(colnames(expr) == gene.id)
    chrom.locale <- which(names(chrom.mats) == gene.id)

    #chromatin scan
    chrom.geno <- get_one_geno(gene.name, transcript.info, transcript.haplotypes, 
        chrom.states, strain.key, geno_type = "chromatin")
    
    #chrom.locale <- which(names(chrom.mats) == gene.id)
    #plot.chrom.mat(t(chrom.mats[[chrom.locale]]), num.states = 9)
    #test <- Reduce("+", lapply(1:dim(chrom.geno[[1]])[1], function(x) chrom.geno[[1]][x,,]))
    #quartz();pheatmap(test, cluster_rows = FALSE, cluster_cols = FALSE)
    if(is.na(chrom.geno)){next()}

    #check for variation
    test.var <- round(apply(chrom.geno[[1]], 3, function(x) apply(x, 2, function(y) var(y))), 2)
    no.var <- length(which(apply(test.var, 1, function(x) length(unique(x))) > 1)) <= 1
    
    if(no.var){next()}
   
    #calculate proportion of each state
    prop.state <- lapply(1:num.states, function(x) rowMeans(chrom.geno[[1]][,x,]))
    prop.effect <- sapply(1:num.states, function(x) if(var(prop.state[[x]]) > 0){plot.with.model(prop.state[[x]], 
    expr[,expr.locale], confidence = 0.95, plot.results = FALSE)}else{rep(NA, 5)})
    chrom.effect[[i]] <- prop.effect
    #barplot(t(prop.effect["slope.fit",]), names = 1:num.states)
    #par(mfrow = c(3,3))
    #for(i in 1:9){if(var(prop.state[[i]]) > 0){plot.with.model(prop.state[[i]], expr[,expr.locale], xlab = i, ylab = "Expression")}}
    #test.mat <- cbind(Reduce("cbind", prop.state), expr[,expr.locale])
    #colnames(test.mat) <- c(1:9, "expr")
    #expr.order <- order(expr[,expr.locale])
    #has.var <- apply(test.mat, 2, var)
    #pheatmap(test.mat[expr.order,which(has.var > 0)], scale = "column", show_rownames = FALSE, cluster_rows = FALSE)
    #state.col <- colors.from.values(1:9, use.pheatmap.colors = TRUE)
    #par(mfrow = c(3,3))
    #for(i in 1:9){
    #  plot(expr[expr.order,expr.locale], test.mat[expr.order,i], col = state.col[i], type = "p", lwd = 2)
    #  }

    #calculate chromatin LOD scores
    chrom.lod.add <- scan1(chrom.geno, expr[,expr.locale], addcovar = covar)
    chrom.lod.int <- scan1(chrom.geno, expr[,expr.locale], addcovar = covar, intcovar = covar[,2,drop=FALSE])
    chrom.chow.lod <- scan1(chrom.geno, expr[,expr.locale], addcovar = covar[chow.locale,])
    chrom.hf.lod <- scan1(chrom.geno, expr[,expr.locale], addcovar = covar[hf.locale,])
    chrom.lod.table <- cbind(chrom.lod.add, chrom.lod.int, chrom.chow.lod, chrom.hf.lod)
    colnames(chrom.lod.table) <- model.names
    #boxplot(chrom.lod.table)
    chrom.lod[[i]] <- chrom.lod.table

    #calculate chromatin coefficients
    chrom.add.coef <- scan1coef(genoprobs = chrom.geno, pheno = expr[,expr.locale], 
    addcovar = covar)
    chrom.int.coef <- scan1coef(genoprobs = chrom.geno, pheno = expr[,expr.locale], 
    addcovar = covar, intcovar = int.covar)
    chrom.chow.coef <- scan1coef(genoprobs = chrom.geno, pheno = expr[,expr.locale], 
    addcovar = covar[chow.locale,])
    chrom.hf.coef <- scan1coef(genoprobs = chrom.geno, pheno = expr[,expr.locale], 
    addcovar = covar[hf.locale,])
    chrom.coef.list <- list(chrom.add.coef[,1:num.states], chrom.int.coef[,1:num.states], chrom.chow.coef[,1:8], chrom.hf.coef[,1:num.states])
    names(chrom.coef.list) <- model.names
    chrom.coef[[i]] <- chrom.coef.list

    #calculate chromatin R2
    #adjust expression for covariates first, so we don't include 
    #the covariates in the variance explained
    adj.expr <- adjust(expr[,expr.locale,drop=FALSE], covar)
    #chrom.models.add <- apply(chrom.geno[[1]], 3, function(x) lm(expr[,expr.locale]~covar+x))
    chrom.models.add <- apply(chrom.geno[[1]], 3, function(x) lm(adj.expr~x))
    ch.add.r2 <- sapply(chrom.models.add, function(x) summary(x)$adj.r.squared)
    #plot(ch.add.r2, ylim = c(0, max(ch.add.r2)))
    chrom.models.int <- apply(chrom.geno[[1]], 3, function(x) lm(adj.expr~x*int.covar))
    ch.int.r2 <- sapply(chrom.models.int, function(x) summary(x)$adj.r.squared)
    #plot(ch.int.r2, ylim = c(0, max(ch.int.r2)))
    chrom.chow.models <- apply(chrom.geno[[1]][chow.locale,,], 3, function(x) lm(adj.expr[chow.locale,]~x))
    ch.chow.r2 <- sapply(chrom.chow.models, function(x) summary(x)$adj.r.squared)
    #plot(ch.chow.r2)
    chrom.hf.models <- apply(chrom.geno[[1]][hf.locale,,], 3, function(x) lm(adj.expr[hf.locale,]~x))
    ch.hf.r2 <- sapply(chrom.hf.models, function(x) summary(x)$adj.r.squared)
    #plot(ch.hf.r2)
    chrom.r2.results <- cbind(ch.add.r2, ch.int.r2, ch.chow.r2, ch.hf.r2)
    colnames(chrom.r2.results) <- model.names
    chrom.r2[[i]] <- chrom.r2.results
    #barplot(apply(chrom.r2.results, 2, function(x) max(x, na.rm = TRUE)))
    #boxplot(chrom.r2.results)
    #pairs(chrom.r2.results)

    #Calculate the same statistics for the genetic model
    geno.geno <- get_one_geno(gene.name, transcript.info, transcript.haplotypes, 
        chrom.states, strain.key, geno_type = "genotype")

    #LOD scores for genetics
    geno.lod.add <- scan1(geno.geno, expr[,expr.locale], addcovar = covar)
    geno.lod.int <- scan1(geno.geno, expr[,expr.locale], addcovar = covar, intcovar = int.covar)
    geno.lod.chow <- scan1(geno.geno, expr[,expr.locale], addcovar = covar[chow.locale,])
    geno.lod.hf <- scan1(geno.geno, expr[,expr.locale], addcovar = covar[hf.locale,])
    geno.lod.results <- c(geno.lod.add, geno.lod.int, geno.lod.chow, geno.lod.hf)
    names(geno.lod.results) <- model.names
    #barplot(geno.lod.results)
    geno.lod[i,] <- geno.lod.results

    geno.coef.add <- scan1coef(genoprobs = geno.geno, pheno = expr[,expr.locale], 
    addcovar = covar)
    geno.coef.int <- scan1coef(genoprobs = geno.geno, pheno = expr[,expr.locale], 
    addcovar = covar, intcovar = int.covar)
    #barplot(geno.coef.add[,LETTERS[1:8]])
    geno.coef.chow <- scan1coef(genoprobs = geno.geno, pheno = expr[,expr.locale], 
    addcovar = covar[chow.locale,])
    geno.coef.hf <- scan1coef(genoprobs = geno.geno, pheno = expr[,expr.locale], 
    addcovar = covar[hf.locale,])
    geno.coef.results <- rbind(geno.coef.add[,c(LETTERS[1:8], "diethf")],
                              geno.coef.int[,c(LETTERS[1:8], "diethf")],
                              c(geno.coef.chow[,LETTERS[1:8]], NA),
                              c(geno.coef.hf[,LETTERS[1:8]], NA))
    rownames(geno.coef.results) <- model.names
    #quartz();pheatmap(geno.coef.results, cluster_rows = FALSE, cluster_cols = FALSE)
    geno.coef[[i]] <- geno.coef.results

    #haplotype coefficients for genetics
    geno.r2.add <- summary(lm(adj.expr~geno.geno[[1]][,,1]))$adj.r.squared
    geno.r2.int <- summary(lm(adj.expr~geno.geno[[1]][,,1]*int.covar))$adj.r.squared
    geno.r2.chow <- summary(lm(adj.expr[chow.locale,]~+geno.geno[[1]][chow.locale,,1]))$adj.r.squared
    geno.r2.hf <- summary(lm(adj.expr[hf.locale,]~geno.geno[[1]][hf.locale,,1]))$adj.r.squared
    geno.r2.results <- c(geno.r2.add, geno.r2.int, geno.r2.chow, geno.r2.hf)
    names(geno.r2.results) <- model.names
    #barplot(geno.r2.results)
    geno.r2[i,] <- geno.r2.results

    if(is.interactive){
      plot.chrom.effects(gene.name, expr, covar, chrom.mats, 
      transcript.info, transcript.haplotypes, chrom.states, strain.key)
    }

    #save every once in a while
    if(i %% save.every == 0){
      saveRDS(chrom.lod, chrom.lod.file)
      saveRDS(chrom.coef, chrom.coef.file)
      saveRDS(chrom.r2, chrom.r2.file)
      saveRDS(geno.lod, geno.lod.file)
      saveRDS(geno.coef, geno.coef.file)
      saveRDS(geno.r2, geno.r2.file)
      saveRDS(chrom.effect, chrom.effect.file)
    }
  }
  #also save at the end of everything.
  saveRDS(chrom.lod, chrom.lod.file)
  saveRDS(chrom.coef, chrom.coef.file)
  saveRDS(chrom.r2, chrom.r2.file)
  saveRDS(geno.lod, geno.lod.file)
  saveRDS(geno.coef, geno.coef.file)
  saveRDS(geno.r2, geno.r2.file)
  saveRDS(chrom.effect, chrom.effect.file)
}

sink(NULL)
```

## Bulk State Effects

We can also look at the overall effects of each state on gene expression
in the DO, as we did in the inbred strains. To do this, we aggregate 
all genes with a given state at each position, and fit a model to explain
gene expression using chromatin state. 

Actually, maybe we can't. I've been calculating genoprobs for each gene
individually because there isn't enough memory anywhere to calculate them
all at once. We also have more data here than n the inbreds so we should 
be able to get a pretty good estimate of the effects using one gene
at a time...
