---
title: "Analyze Chromatin local QTL results"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

## Introduction

The purpose of this workflow is to analyze the results from 2.1_chromaprobs.Rmd
That workflow imputed local chromatin states for each transcript, and scanned
those states for effects on transcription in the DO.

This workflow compares the eQTL mapped to chromatin and to haplotypes.
  
```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
library(here);library(qtl2);library(gprofiler2);library(stringr)
library(msigdbr);library(fgsea);library(pheatmap);library(biomaRt)
library(knitr)
```

```{r load_biomart}
all.var <- ls()
lib.loaded <- as.logical(length(which(all.var == "mus")))
if(!lib.loaded){
  mus <- useMart("ENSEMBL_MART_ENSEMBL", dataset = "mmusculus_gene_ensembl", 
  host = "may2017.archive.ensembl.org")
}

#att <- listAttributes(mus)
#fil <- listFilters(mus)
#query.term <- "island"
#att[grep(query.term, att[,1]),]
#fil[grep(query.term, fil[,1]),]
```

```{r source_code}
is.interactive = FALSE
#is.interactive = TRUE
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
data(CCcolors)
chrom.colors <- colors.from.values(1:8, use.pheatmap.colors = TRUE)
```


## Load data

```{r load_data}
ch.coef <- readRDS(here("Results", "chQTL", "chQTL.coef.RDS"))
ch.lod <- readRDS(here("Results", "chQTL", "chQTL.lod.RDS"))
ch.r2 <- readRDS(here("Results", "chQTL", "chQTL.R2.RDS"))
e.coef <- readRDS(here("Results", "eQTL", "eQTL.coef.RDS"))
e.lod <- readRDS(here("Results", "eQTL", "eQTL.lod.RDS"))
e.r2 <- readRDS(here("Results", "eQTL", "eQTL.R2.RDS"))

chrom.mats <- readRDS(here("Results", "ChromHMM", "008_states_C", "Chromatin_States_8_full_gene_1000.RData"))
chrom.states <- readRDS(here("Data", "chQTL", "Expanded_Chromatin.RDS"))

do.data <- load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData"))
expr <- dataset.mrna$data$rz
expr.norm <- dataset.mrna$data$norm
covar <- dataset.mrna$covar.matrix
diet.locale <- lapply(c(0,1), function(x) which(covar[,"diethf"] == x))
transcript.haplotypes <- readRDS(here("Data", "DOQTL", "transcript.haplotypes.RDS"))

strain.key <- read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", stringsAsFactors = FALSE, comment.char = "*")
```

```{r transcript_info}
transcript.info.file <- here("Data", "RNASeq", "RNASeq_gene_info.RData")
if(!file.exists(transcript.info.file)){
    all.genes <- unique(names(chrom.states), colnames(expr))
    transcript.info <- getBM(attributes = c("ensembl_gene_id", "external_gene_name", 
    "chromosome_name", "start_position", "end_position", "transcription_start_site", 
    "exon_chrom_start", "exon_chrom_end", "5_utr_start","5_utr_end","3_utr_start", 
    "3_utr_end", "strand"), 
    filters = "ensembl_gene_id", values = all.genes, mart = mus)
    saveRDS(transcript.info, transcript.info.file)
}else{
  transcript.info <- readRDS(transcript.info.file)
}
gene.names <- transcript.info[match(rownames(e.lod), transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
```

## Matching genetic eQTLs with chromatin eQTLs

The following plot shows the LOD scores of genetic eQTLs and chromatin eQTLs
plotted against each other. For the most part, the LOD scores agree with each
other very well.

```{r set_thresh}
lod.change.thresh = 2.5
```

However, there is also a fairly large group of genes for which
there is a mismatch (a LOD score difference of more than `r lod.change.thresh`),
between the two associations. These are genes for which there is a cis-eQTL,
but relatively lower association with the chromatin state.
The distribution of these differences is shown in the right-hand panel.

```{r e.ch.comp, fig.width = 8, fig.height = 4}
max.ch.lod <- t(sapply(ch.lod, function(x) if(length(x) > 0){apply(x, 2, function(y) max(y, na.rm = TRUE))}else{rep(NA, 4)}))
par(mfrow = c(1,2))
plot(e.lod[,1], max.ch.lod[,1], pch = 16, col = "gray", cex = 0.5,
xlab = "genetic eQTL LOD score", ylab = "chromatin eQTL LOD score",
main = "Genetic vs. Chromatin-based eQTL LOD scores")
abline(0,1, col = "salmon")
abline(-lod.change.thresh, 1, col = "salmon")

lod.diff <- e.lod[,1] - max.ch.lod[,1]
hist(lod.diff, breaks = 100, main = "LOD score differences")
ordered.lod <- sort(lod.diff, decreasing = TRUE)
abline(v = lod.change.thresh, col = "salmon")

small.change <- which(lod.diff <= lod.change.thresh)
big.change <- which(lod.diff > lod.change.thresh)
```

## Genetic and Chromatin R2 Comparison

We also calculated the R2 for each linear model using either haplotype
probabilities or chromatin probabilities plus covariates to explain 
transcript level. This way we can get a sense of how much variance
in the expression each type of data is explaining.

The chromatin variance explained and genetics variance explained are 
very highly correlated. There are many cases in which the chromatin does
not explain as much variance as the genetics.

```{r r2_comp}
max.ch.r2 <- t(sapply(ch.r2, function(x) if(length(x) == 0){return(rep(NA, 4))}else{apply(x, 2, function(y) max(y, na.rm = TRUE))}))
r2.diff <- e.r2 - max.ch.r2
large.r2.diff <- 0.01

par(mfrow = c(1,2))
plot(e.r2[,1], max.ch.r2[,1], pch = 16, col = "gray", cex = 0.7,
xlab = "R2 Genetic eQTL", ylab = "R2 Chromatin eQTL", 
main = "Comparison of Chromatin and Genetic R2")
abline(0,1)
abline(-large.r2.diff, 1, col = "salmon")

hist(r2.diff[,1], breaks = 100, main = "R2 Difference Distribution", 
xlab = "R2 Difference")
abline(v = large.r2.diff, col = "salmon")
```

We defined genes with a big mismatch between the genetic and chromatin 
eQTL scores as genes with an R2 difference greater than `r large.r2.diff`. 

```{r define_mismatch}
r2diff.order <- apply(r2.diff, 2, function(x) order(x, decreasing = TRUE))
big.diff <- apply(r2.diff, 2, function(x) rownames(r2.diff)[which(x > large.r2.diff)])
```

This set of `r length(big.diff[[1]])` genes is highly enriched for
mitochrondrial and ribosomal genes. 

```{r mismatch_enrich, fig.height = 10, fig.width = 7}
enrich <- lapply(big.diff, function(x) gost(x, organism = "mmusculus", source = "GO"))
if(is.interactive){quartz(height = 10, width = 7)}
plot.enrichment.group(enrich, sort.by = "p_value", n.terms = 30, max.term.size = 500)
```

## Mismatched Gene Characterization
There are many possibile ways to explain the genes for which there is a 
large mismatch between the genetic eQTL and the chromatin eQTL. We have 
no way to investigate most of these hypotheses, but we can look at some.

Chromatin imputation errors:
  1. Cell type
  2. Chromatin changes due to high-fat diet
  3. Age

Epistatic regulatory mechanism encoded in the genetics, but not reflected in chromatin:
  1. Hyper/Hypomethylated genes 
  2. Post-transcriptional regulation
  3. Cell cycle variation (Histone methylation is known to vary across cell cycle)

## Cell type specificity {.tabset .tabset-fade .tabset-pills}

Here we investigate whether the genes that have widely diverging eQTL and
chromatin eQTL LOD scores are primarily expressed in another cell type.

First we create gene lists for each cell type associated with the Aizarani 
liver cell type data sets.

```{r msigdb}
#get cell-type specific gene sets (category C8)
#see https://www.gsea-msigdb.org/gsea/msigdb/genesets.jsp?
cell.sets <- msigdbr(species = "Mus musculus", category = "C8", subcategory = "")
cell.genes <- set_to_list(cell.sets, set.name = "AIZARANI")
```

Then we used each of these as a gene set in gene set enrichment.

```{r gsea}
#order the genes by the difference in R2 between the genetic
#and chromatin eQTLs.
ordered.genes <- lapply(1:ncol(r2.diff), function(x) r2.diff[r2diff.order[,x],x])
names(ordered.genes) <- colnames(r2.diff)
ordered.genes <- lapply(ordered.genes, function(x) x[which(!is.na(x))])

for(i in 1:length(ordered.genes)){
  ordered.gene.names <- transcript.info[match(names(ordered.genes[[i]]), transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
  query <- ordered.genes[[i]]
  names(query) <- ordered.gene.names
  cell.enrich <- fgsea::fgseaMultilevel(pathways = cell.genes, stats = query, 
  minSize=15, maxSize=600, gseaParam = 1, scoreType = "pos")
  namecol <- which(colnames(cell.enrich) == "pathway")
  path.names <- sapply(strsplit(gsub("AIZARANI_LIVER_", "", cell.enrich[[namecol]]), "_"), function(x) paste(x[2:length(x)], collapse = "_"))
  if(is.interactive){quartz()}
  plot.gsea.results(cell.enrich, "padj", path.names, plot.label = colnames(r2.diff)[i])
  leading.genes <- cell.enrich[[8]][which(cell.enrich[[2]] < 0.05)]
  names(leading.genes) <- cell.enrich[[1]][which(cell.enrich[[2]] < 0.05)]
}

```

The numerals in this dataset refer to t-SNE clusters. We also merged 
these clusters to see if there is a more general cell-type signature.s

There is no longer an enrichment if I merge the cell type clusters.

```{r merge_aizarani}
cell.types <- sapply(strsplit(path.names, "_"), function(x) paste(x[1:(length(x)-1)], collapse = "_"))
u_cell.types <- unique(cell.types)
merged.cell.genes <- vector(mode = "list", length = length(u_cell.types))
names(merged.cell.genes) <- u_cell.types
for(i in 1:length(u_cell.types)){
  type.locale <- which(cell.types == u_cell.types[i])
  merged.genes <- unique(unlist(cell.genes[type.locale]))
  merged.cell.genes[[i]] <- merged.genes
}

for(i in 1:length(ordered.genes)){
  ordered.gene.names <- transcript.info[match(names(ordered.genes[[i]]), transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
  query <- ordered.genes[[i]]
  names(query) <- ordered.gene.names
  merged.cell.enrich <- fgsea::fgseaMultilevel(pathways = merged.cell.genes, 
    stats = query, minSize=15, maxSize=600, gseaParam = 1, scoreType = "pos")
  namecol <- which(colnames(merged.cell.enrich) == "pathway")
  if(is.interactive){quartz()}
  plot.gsea.results(merged.cell.enrich, "pval", p.thresh = 0.05, 
  plot.label = colnames(r2.diff)[i])
}
```

### Conclusion
There is no strong evidence that that the mismatches between genetic
and chromatin eQTLs are related to cell type. 

## GSEA of mismatched genes for GO terms {.tabset .tabset-fade .tabset-pills}

```{r msigdb_go, results = "asis"}
#get GO term gene sets
#see https://www.gsea-msigdb.org/gsea/msigdb/genesets.jsp?
go.mf <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "GO:MF")
go.cc <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "GO:CC")
go.bp <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "GO:BP")

go.mf.genes <- set_to_list(go.mf)
go.bp.genes <- set_to_list(go.bp)
go.cc.genes <- set_to_list(go.bp)

ordered.gene.names <- ordered.genes
for(i in 1:length(ordered.gene.names)){
  names(ordered.gene.names[[i]]) <- transcript.info[match(names(ordered.genes[[i]]), transcript.info[,"ensembl_gene_id"]), "external_gene_name"]
}

go.mf.enrich <- fgsea::fgseaMultilevel(pathways = go.mf.genes, 
  stats = ordered.gene.names[[1]], minSize=15, maxSize=600, gseaParam = 1, 
  scoreType = "pos")

go.bp.enrich <- fgsea::fgseaMultilevel(pathways = go.bp.genes, 
  stats = ordered.gene.names[[1]], minSize=15, maxSize=600, gseaParam = 1, 
  scoreType = "pos")

go.cc.enrich <- fgsea::fgseaMultilevel(pathways = go.cc.genes, 
  stats = ordered.gene.names[[1]], minSize=15, maxSize=600, gseaParam = 1, 
  scoreType = "pos")
```


### GO:MF
```{r gomf_results}
plot.gsea.results(go.mf.enrich, "padj", p.thresh = 0.05, plot.label = "GO:MF")
```

### GO:BP
```{r gobp_results}
plot.gsea.results(go.bp.enrich, "padj", p.thresh = 0.05, plot.label = "GO:BP")
```

### GO:CC
```{r gocc_results}
plot.gsea.results(go.cc.enrich, "padj", p.thresh = 0.05, plot.label = "GO:CC")
```

### Conclusion
There is some weak enrichment for ribosomal genes and genes translocated to the
ER. Does this suggest the posibility of post-transcriptional regulation not
captured by the chromatin? I don't think it does, necessarily, but that's
another possibility to discuss.

## Chromatin changes related to high-fat diet

Chromatin state is modified in response to environmental factors,
for example high-fat diet. We cannot capture such changes because
we are imputing the chromatin state from control inbred animals.
We do not directly measure chromatin state in the DO mice on different diets.

If there are chromatin changes in the HF animals that change transcriptional
regulation, we would expect that chromatin state would explain 
gene expression better in the chow animals than in the high-fat diet 
animals. This was not the case. The figures below show the correlation
between the genetic and chromatin eQTLs in the chow and HF animals
independently. 

```{r chow_hf_comp}
par(mfrow = c(1,3))
plot.with.model(e.r2[,"Chow"], max.ch.r2[,"Chow"], xlab = "genetic eQTL R2", 
ylab = "chromatin eQTL R2", main = "Chow Animals", pch = 16, col = "gray", cex = 0.7)
abline(0,1)

plot.with.model(e.r2[,"HF"], max.ch.r2[,"HF"], xlab = "genetic eQTL R2", 
ylab = "chromatin eQTL R2", main = "HF Animals", pch = 16, col = "gray", cex = 0.7)
abline(0,1)

plot.with.model(r2.diff[,"Chow"], r2.diff[,"HF"], col = "gray",
xlab = "Chow R2 Difference", ylab = "HF R2 Difference", main = "Chow vs. HF")
abline(0,1)
```

### Conclusion 
We cannot explain variation in expression through chromatin any 
better in chow animals than in HF animals. Therefore we cannot conclude
that the mismatches between the genetic eQTLs and chromatin eQTLs are
related to diet-related changes in chromatin state.

## Positional Chromatin LOD score

We normalized the transcript position to examine whether high LOD
scores occurred mostly at the TSS, or at other regions.

It actually appears that there is a dip in average LOD score right
at the TSS, and a peak in average LOD score in the initial part of 
the gene after the TSS.

```{r norm_lod}
test.idx <- 1:length(ch.lod)
#test.idx <- big.diff.locale

norm.ch <- vector(mode = "list", length = length(ch.lod[test.idx]))
names(norm.ch) <- rownames(e.lod)[test.idx]
for(x in 1:length(test.idx)){
  if(is.interactive){report.progress(x, length(test.idx))}
  if(!is.null(ch.lod[[x]])){
  norm.ch[[x]] <- center.on.feature(gene.names[x], transcript.info, ch.lod[[x]][,1], 
  feature = "full")
  }
}

lod.by.pos <- plot.centered.vals(norm.ch, seq.by = 0.01, min.representation = 10, 
ylim = c(0, 10), plot.label = "Mean LOD Score by Position", 
ylab = "Mean LOD Score", return.means = FALSE)
abline(v = c(0,1))
```

Many genes have no variation in chromatin state across strains right at the TSS.
We excluded these genes and looked at average LOD score by position only for the
genes that have variation in chromatin state at the TSS across strains.

For these genes, the highest LOD scores are immediately upstream of the TSS

```{r lod_with_var}
chrom.gene.names <- transcript.info[match(names(chrom.mats), transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
norm.states <- lapply_pb(1:length(chrom.mats), 
function(x) if(length(chrom.mats[[x]]) > 1){center.on.feature(chrom.gene.names[x], 
transcript.info, chrom.mats[[x]][,1], feature = "full")}else{NA})
tss.locale <- lapply(norm.states, function(x) get.nearest.pt(0, as.numeric(names(x))))
tss.states <- lapply(1:length(tss.locale), function(x) if(length(chrom.mats[[x]]) > 1){chrom.mats[[x]][tss.locale[[x]],]}else{NA})
num.states <- sapply(tss.states, function(x) if(length(x) > 1){length(unique(x))}else{NA})
var.locale <- which(num.states > 1)
var.genes <- names(chrom.mats)[var.locale]

var.gene.locale <- match(var.genes, names(norm.ch))

var.lod.by.pos <- plot.centered.vals(norm.ch[var.gene.locale], 
seq.by = 0.01, min.representation = 10, 
ylim = c(0, 10), plot.label = "Mean LOD Score by Position", 
ylab = "Mean LOD Score", return.means = FALSE)
abline(v = c(0,1))
```

Plotting these two scores on the same graph shows the dip in LOD score
for all genes (black) compared with the peak at the TSS for the genes 
with variation at that point (blue).

```{r var_vs_all}
mean.var.lod <- colMeans(var.lod.by.pos, na.rm = TRUE)
mean.all.lod <- colMeans(lod.by.pos, na.rm = TRUE)

plot(as.numeric(colnames(var.lod.by.pos)), mean.var.lod, type = "l", col = "blue",
ylab = "Mean LOD score", xlab = "Relative Position")
points(as.numeric(colnames(lod.by.pos)), mean.all.lod, type = "l")
abline(v = c(0,1))
```

The genes with variation in chromatin state at the TSS are enriched for 
cell motility and ECM. Could these be related to immune cell migration?

Genes without variation in chromatin state at the TSS are enriched for
metabolism and biosynthesis.

```{r var_enrich, fig.width = 9, fig.height = 8}
var.enrich <- gost(var.genes, organism = "mmusculus", 
sources = c("GO", "KEGG", "REACTOME"))
plot.enrichment(var.enrich, num.terms = 30,
plot.label = "Genes with Chromatin State Variation at TSS")

invar.locale <- which(num.states == 1)
invar.genes <- names(chrom.mats)[invar.locale]
invar.enrich <- gost(invar.genes, organism = "mmusculus", 
sources = c("GO", "KEGG", "REACTOME"))
plot.enrichment(invar.enrich, num.terms = 30, 
plot.label = "Genes with No Variation in Chromatin State at TSS")
```

### Conclusion
Most genes have no variation in chromatin at the TSS, and thus chromatin
state at this position is not used for regulating gene expression across
strains. 

However, a subset of genes do have variation in chromatin state at the TSS,
and for these genes, chromatin state at this position is used in regulation
of gene expression.


## Chromatin state coefficients by position {.tabset .tabset-fade .tabset-pills}

In the figures below, we show the average coefficients for each state 
across the gene body. We only include non-zero coefficients. This tells
us the effect of each state at each position when it actually varies
at that position. 

I had done this earlier using all coefficients, and it looked as if state
7 had a dip in effect at the TSS. It does, but only because it is invariant
at the TSS across strains for most genes. We should also have a measure
of presence and variance across the gene body.

The chromatin states have the same directionality biases that we observed
in the inbred strains. For example, states 1 and 2 are negatively associated
with expression, and state 7 is positively associated with expression.

We see *very* similar spatial pattern to that seen in the inbred mice.

State 8 coefficients dip right at the TSS.
State 7 coefficients are high everywhere, but take a dip at the TSS.
  This is a little different from the inbred mice, where the dip seems 
  to occur upstream of the TSS.
State 5 coefficients were high in the body of the gene
State 4 coefficients were pretty close to 0 everywhere
State 3 coefficients took a little dip by the TSS
State 1 and state 2 coefficients were low everywhere, especially near the TSS

```{r norm_coef, results = "asis", fig.width = 10, fig.height = 4}
norm.ch.coef <- lapply(ch.coef, function(x) x[[1]])
names(norm.ch.coef) <- rownames(e.lod)
for(x in 1:length(ch.coef)){
  if(is.interactive){report.progress(x, length(ch.coef))}
  if(!is.null(ch.coef[[x]])){
  norm.pos <- center.on.feature(gene.names[x], transcript.info, ch.coef[[x]][[1]][,1], 
  feature = "full")
  rownames(norm.ch.coef[[x]]) <- names(norm.pos)
  }
}

all.state.coef <- vector(mode = "list", length = 8)
names(all.state.coef) <- 1:8
for(s in 1:8){
  if(is.interactive){quartz(width = 10, height = 4)}
  cat("### State", s, "\n")
  norm.state <- lapply(norm.ch.coef, function(x) x[,s])
  non.missing.state <- norm.state
  for(i in 1:length(norm.state)){
    zero.locale <- which(non.missing.state[[i]] == 0)
    if(length(zero.locale) > 0){
      non.missing.state[[i]][zero.locale] <- NA
    }
  }
  all.state.coef[[s]] <- plot.centered.vals(non.missing.state, seq.by = 0.01, min.representation = 10, 
  ylim = c(-0.3, 0.3), plot.label = paste("State", s), 
  ylab = "Mean Non-Zero Coefficient", return.means = FALSE)
  abline(h = 0, v = c(0,1))
  cat("\n\n")
}
```

## Effect size of chromatin does not depend on position {.tabset .tabset-fade .tabset-pills}

Do the dips we see above represent dips toward zero or away from zero?
Below we show the absolute value of the average coefficient to get a better
handle on which states have stronger effects near the TSS, and which states
have weaker effects.

The strength of the effect of state 7 takes a dip at the TSS.
The strength of the effect of state 1 also take a bit of a dip right at
the TSS, but is high on either side.
The average effect of state 4 goes up a bit at the TSS.
States 2, 3, 5, and 8 are all fairly level across the gene.


```{r coef_magnitude, results = "asis", fig.width = 10, fig.height = 4}
all.state.mag <- vector(mode = "list", length = 8)
names(all.state.mag) <- 1:8
for(s in 1:8){
  if(is.interactive){quartz(width = 10, height = 4)}
  cat("### State", s, "\n")
  norm.state <- lapply(norm.ch.coef, function(x) x[,s])
  abs.state <- lapply(norm.state, function(x) if(length(x) > 0){abs(x)}else{NA})
  non.missing.state <- abs.state
  for(i in 1:length(norm.state)){
    zero.locale <- which(non.missing.state[[i]] == 0)
    if(length(zero.locale) > 0){
      non.missing.state[[i]][zero.locale] <- NA
    }
  }
  all.state.mag[[s]] <- plot.centered.vals(non.missing.state, seq.by = 0.01, 
  min.representation = 10, ylim = c(0, 0.5), plot.label = paste("State", s), 
  ylab = "Mean Non-Zero Coefficient Magnitude", return.means = FALSE)
  abline(h = 0, v = c(0,1))
  #abline(h = mean(all.state.mag[[s]], na.rm = TRUE))
  cat("\n\n")
}

cat("### Mean State Effect Size\n")
barplot(sapply(all.state.mag, function(x) mean(x, na.rm = TRUE)), col = chrom.colors,
main = "Mean Effect Magnitude")
cat("\n")
```

## State Variation by Position {.tabset .tabset-fade .tabset-pills}

The figures below show the proportion of the genes at any given 
gene body position that have variation in the given state across
strains. 

What these figures show is that the placement of state 7 is highly 
consistent right at the TSS because there is a dip in variation right 
at that point. Variation then increases later in the body of the gene. 
This is the lowest value seen for any state, suggesting its placement
at the TSS is important to transcription. 

On the other hand, the variation in state 8 goes up at the TSS, suggesting
that it may play a more regulatory role in that position.

Other states are much less conserved relative to the position in the gene
body. States 4, 5, 6 have high variation throught the gene body. 

State 1 is an interesting one. We know from the analysis in the 
inbred animals that its abundance is very low near the TSS and then 
increases as you move toward the TES. However the proportion of genes
with variation in that state across strains decreases as you move toward
the TSS.

```{r state_variance, results = "asis", fig.width = 10, fig.height = 4}
all.state.var <- vector(mode = "list", length = 8)
names(all.state.var) <- 1:8
for(s in 1:8){
  if(is.interactive){quartz(width = 10, height = 4)}
  cat("### State", s, "\n")
  norm.state <- lapply(norm.ch.coef, function(x) x[,s])
  non.missing.state <- norm.state
  for(i in 1:length(norm.state)){
    non.zero.locale <- which(non.missing.state[[i]] != 0)
    if(length(non.zero.locale) > 0){
      non.missing.state[[i]][non.zero.locale] <- 1
    }
  }
  all.state.var[[s]] <- plot.centered.vals(non.missing.state, seq.by = 0.01, 
  min.representation = 10, ylim = c(0, 1.1), plot.label = paste("State", s), 
  ylab = "Proportion Genes With Variation in State", return.means = FALSE)
  abline(h = 0, v = c(0,1))
  cat("\n\n")
}
```

