---
title: "Full eQTL"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

## Introduction

The purpose of this workflow is to perform the full eQTL scan in the DO
liver data. This will enable us to investigate whether transcripts with 
trans eQTL are enriched in our set of genes with a mismatch between the
genetic eQTL and the chromatin eQTL.

Load libraries.

```{r load_libraries}
library(here);library(qtl2);library(parallel)
```

Load data.

```{r load_data}
do.data <- load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData"))
expr <- dataset.mrna$data$rz
covar <- dataset.mrna$covar.matrix
```

Run the eQTL scan.

```{r run_eqtl}
n.cores <- detectCores() - 1
scan1.file <- here("Results", "eQTL", "Full.Transcriptome.RDS")
if(!file.exists(scan1.file)){
  scan1.result <- scan1(genoprobs, expr, kinship = K, addcovar = covar, cores = n.cores)
  saveRDS(scan1.result, scan1.file)
}else{
  scan1.result <- readRDS(scan1.file)
}
```

Detect transcripts with trans eQTL.

```{r trans_eQTL}
lod.thresh = 6
Mb.buffer = 10 #miminum distance of a trans effect
annots <- dataset.mrna$annot.mrna
scan1.markers <- rownames(scan1.result)
scan1.chr <- sapply(strsplit(scan1.markers, "_"), function(x) x[1])
scan1.pos <- sapply(strsplit(scan1.markers, "_"), function(x) x[2])

trans.locus.file <- here("Results", "eQTL", "Trans.eQTL.RDS")

if(!file.exists(trans.locus.file)){
  trans.loci <- vector(mode = "list", length = nrow(annots))
  names(trans.loci) <- annots[,1]

  for(i in 1:ncol(scan1.result)){
    if(max(scan1.result[,i]) < lod.thresh){next()}
    #plot(scan1.result, i, map = map)
    id <- annots[i,1]
    chr <- annots[i,3]
    gene.start <- annots[i,4]
    gene.end <- annots[i,5]
    nearest.marker <- annots[i,8]
    if(!is.na(nearest.marker)){
      cis.lod <- scan1.result[nearest.marker,i]
    }else{
      cis.lod <- "X"
    }

    #check for trans loci on other chromosomes
    not.chr.locale <- which(scan1.chr != chr)
    not.chr.lod <- scan1.result[not.chr.locale,i]
    not.chr.hits <- which(not.chr.lod > lod.thresh)

    if(length(not.chr.hits) > 0){
      not.chr.trans <- scan1.result[not.chr.locale[not.chr.hits],i]
    }else{
      not.chr.trans <- NULL
    }

    #check for trans loci on the same chromsome
    chr.locale <- which(scan1.chr == chr)
    chr.pos <- as.numeric(scan1.pos[chr.locale])/1e6
    upstream <- which(chr.pos < (gene.start - Mb.buffer))
    if(length(upstream) > 0){
      up.chr.hits <- which(scan1.result[chr.locale[upstream],i] > lod.thresh)
      up.chr.trans <- scan1.result[chr.locale[upstream[up.chr.hits]],i]
    }else{
      up.chr.trans <- NULL
    }

    downstream <- which(chr.pos > (gene.start + Mb.buffer))
    if(length(downstream) > 0){
      down.chr.hits <- which(scan1.result[chr.locale[downstream],i] > lod.thresh)
      down.chr.trans <- scan1.result[chr.locale[downstream[down.chr.hits]],i]
    }else{
      down.chr.trans <- NULL
    }

    all.trans.loci <- c(not.chr.trans, up.chr.trans, down.chr.trans)
    result <- list("cis.LOD" = cis.lod, "trans.LOD" = all.trans.loci)
    trans.loci[[i]] <- result
  }
  saveRDS(trans.loci, trans.locus.file)
}
```

Do a bit of analysis.

```{r rel_strength}
compare.lod <- function(locus.results){
  cis.lod <- NA; max.trans <- NA
  if(length(locus.results) > 1){
    cis.lod <- locus.results[[1]]
    trans.lod <- locus.results[[2]]
    if(length(trans.lod) > 0){
      max.trans <- max(trans.lod)
    }
  }
  result <- c("cis" = cis.lod, "trans" = max.trans)
}

lod.table <- t(sapply(trans.loci, compare.lod))
plot(as.numeric(lod.table[,1]), as.numeric(lod.table[,2]),
xlab = "cis LOD", ylab = "Maximum trans LOD")
abline(0,1)
```