---
title: "RRBS Analysis"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

The purpose of this workflow is to look for correlations between DNA methylation
and gene expression within-strain, and across-strain.

This workflow uses the output from the RRBS workflow.

```{r load_libraries, echo = FALSE}
is.interactive = FALSE
#is.interactive = TRUE
seq.by = 0.1 #the density at which to bin methylation across relative gene positions from -2 to 2
needed.packages <- c("here", "pheatmap", "hexbin", "RColorBrewer", "gprofiler2",
"knitr")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}
```

```{r source_code}
all.code.dir <- list.files(here("code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r read_rrbs}
rrbs.file <- here("Data", "RRBS", "All.Methylation.RDS")
all.var <- ls()
rrbs.loaded <- as.logical(length(which(all.var == "all.rrbs")))

if(!rrbs.loaded){
	all.rrbs <- readRDS(rrbs.file)
}

transcript.info.file <- here("Data", "RNASeq", "RNASeq_gene_info.RData")
transcript.info <- readRDS(transcript.info.file)

ind.name <- names(all.rrbs)
ind.label <- sapply(strsplit(ind.name, ""), function(x) paste(tail(x, 2), collapse = ""))
treat.label <- substr(ind.label, 1, 1)
ind <- substr(ind.label, 2,2)
strain <- mapply(function(x,y) gsub(x, "", y), ind.label, ind.name)
u_strain <- unique(strain)
id.table <- cbind(strain, treat.label, ind)
```

```{r load_expression_data}
strain.expr <- readRDS(here("Data", "RNASeq", "Strain_Mean_C_Expression.RData"))
scaled.expr <- lapply(strain.expr, scale)
#scaled.expr <- lapply(strain.expr, function(x) x - x[2]) #center on B6

col.table <- as.matrix(read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", comment.char = "%", stringsAsFactors = FALSE))

u_genes <- unique(transcript.info[,"external_gene_name"])
```


```{r strain}
strains <- col.table[,6] #the column that matches the names in the RRBS data

test.num <- length(u_genes)
survey.buffer = 5000 #The number of bp to look up and downstream of specified regions
plot.buffer = 2000 #The number of bp around a point to plot
mean.buffer = 1000 # The number of bp around a point for calculating mean methylation
```

Get methylation percentages for all genes and all strains given the above bp buffers.

```{r gene_methyl}
strain.gene.methylation <- vector(mode = "list", length = length(strains))
names(strain.gene.methylation) <- strains

for(st in 1:length(strains)){
  gene.methyl.file <- here("Data", "RRBS", paste0("All.Gene.Methylation.", strains[st], ".RDS"))

  if(!file.exists(gene.methyl.file)){
    cat("Getting methylation for", strains[st], "\n")
    gene.methyl <- lapply_pb(u_genes[1:test.num], 
    function(x) get.methyl.by.gene(x, gene.info.table = transcript.info, 
    methyl.data = all.rrbs, methyl.id.table = id.table, 
    upstream.buffer = survey.buffer, downstream.buffer = survey.buffer, 
    strains = strains[st], treatment = "C", strain.means = FALSE))
    names(gene.methyl)  <- u_genes[1:test.num]
    saveRDS(gene.methyl, gene.methyl.file)
  }else{
    gene.methyl <- readRDS(gene.methyl.file)
  }
  strain.gene.methylation[[st]] <- gene.methyl
}
```

Calculate average RRBS for each gene across replicates. 

```{r avg_rrbs}
avg_methyl <- function(methyl_mat, min_rep = 2){
  n_rep <- apply(methyl_mat, 1, function(x) length(which(!is.na(x))))
  to_avg <- which(n_rep >= min_rep)
  if(length(to_avg) == 0){
    return(NA)
  }else{
    avg_methyl <- rowMeans(methyl_mat[to_avg,,drop=FALSE], na.rm = TRUE)
    return(avg_methyl)
  }
}

strain.avg.file <- here("Results", "RRBS", "Strain.Avg.RRBS.RDS")
if(!file.exists(strain.avg.file)){
  strain_avg_rrbs <- vector(mode = "list", length = length(strains))
  names(strain_avg_rrbs) <- strains
  for(st in 1:length(strains)){
    if(is.interactive){
      cat(strains[st], "\n")
      strain_avg_rrbs[[st]] <- lapply_pb(strain.gene.methylation[[st]], 
      function(x) if(length(x) > 3){avg_methyl(x, min_rep = 2)}else{NA})
      names(strain_avg_rrbs[[st]]) <- names(strain.gene.methylation[[st]])
    }else{
      strain_avg_rrbs[[st]] <- lapply(strain.gene.methylation[[st]], 
      function(x) if(length(x) > 3){avg_methyl(x, min_rep = 2)}else{NA})
      names(strain_avg_rrbs[[st]]) <- names(strain.gene.methylation[[st]])
    }
  }
  saveRDS(strain_avg_rrbs, strain.avg.file)
}else{
  strain_avg_rrbs <- readRDS(strain.avg.file)
}
```

Combine methylation data for each gene across strains.

```{r comb_strains}
methyl.mat.file <- here("Results", "RRBS", "Aligned.Methyl.Mats.RDS")
if(!file.exists(methyl.mat.file)){
  methyl.mats <- vector(mode = "list", length = length(strain_avg_rrbs[[1]]))
  names(methyl.mats) <- names(strain_avg_rrbs[[1]])
  for(i in 1:length(methyl.mats)){
    if(is.interactive){report.progress(i, length(methyl.mats))}
    all.strain.rrbs <- lapply(strain_avg_rrbs, function(x) x[[i]])
    all.pos <- sort(unique(unlist(lapply(all.strain.rrbs, function(x) as.numeric(names(x))))))
    strain.mat <- matrix(NA, ncol = length(all.pos), nrow = length(all.strain.rrbs))
    rownames(strain.mat) <- names(all.strain.rrbs)
    colnames(strain.mat) <- all.pos
    for(j in 1:length(all.strain.rrbs)){
      strain.mat[j,match(names(all.strain.rrbs[[j]]), all.pos)] <- all.strain.rrbs[[j]]
    }
    methyl.mats[[i]] <- strain.mat
    #pheatmap(strain.mat, cluster_rows = FALSE, cluster_cols = FALSE)
  }
saveRDS(methyl.mats, methyl.mat.file)
}else{
  methyl.mats <- readRDS(methyl.mat.file)
}
```

Normalize gene coordinates to run from 0 to 1 from the TSS to the TES.

```{r norm_rrbs}
norm.rrbs.file <- here("Results", "RRBS", "RRBS.Normalized.Position.RDS")

if(!file.exists(norm.rrbs.file)){
  #first center all the coordinates
    norm_rrbs_coord <- lapply_pb(1:length(methyl.mats), 
        function(x) center.on.feature(gene.name = names(methyl.mats)[x], 
        gene.info = transcript.info, vals = methyl.mats[[x]][1,], 
        feature = "full"))
    #put the normalized coordinates onto the methyl.mats
    norm_rrbs <- methyl.mats
    names(norm_rrbs) <- names(methyl.mats)
    for(i in 1:length(norm_rrbs)){
      colnames(norm_rrbs[[i]]) <- names(norm_rrbs_coord[[i]])
    }
    saveRDS(norm_rrbs, norm.rrbs.file)
}else{
    norm_rrbs <- readRDS(norm.rrbs.file)
}
```

## Methylation density by gene position {.tabset .tabset-fade .tabset-pills}
We looked at where the densest DNA methylation occurred relative to the gene body.
Methylation is densest at the TSS, and very sparse within gene bodies. 

Density of methylated positions.
For each gene, we looked at the distance of each DNA methylation position 
to the next position. Around 80% of DNA methylation positions were
a single base pair from the next position in all strains.

```{r methyl_dens}
methyl_consec <- lapply(methyl.mats, function(x) if(length(x) > 1){consec.pairs(as.numeric(colnames(x)))}else{NA})
methyl_dist <- lapply(methyl_consec, function(x) if(length(x) >  1){x[,2] - x[,1]}else{NA})
for(i in 1:length(methyl_dist)){
  if(!is.null(colnames(norm_rrbs[[i]]))){
    consec.coord <- consec.pairs(as.numeric(colnames(norm_rrbs[[i]])))
    mean.coords <- rowMeans(consec.coord)
    names(methyl_dist[[i]]) <- mean.coords
  }
}
if(is.interactive){quartz()}
methyl_dens_file <- here("Results", "RRBS", "Methylation_Density.RDS")
if(!file.exists(methyl_dens_file)){
  avg_methyl_dist <- plot.centered.vals(val.list = methyl_dist, 
    seq.by = seq.by, min.representation = 10, ylim = c(0, 3500),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position", return.means = FALSE)
  abline(v = c(0,1))
  saveRDS(avg_methyl_dist, methyl_dens_file)
}else{
  avg_methyl_dist <- readRDS(methyl_dens_file)
  plot.centered.mat(avg_methyl_dist, ylim = c(0, 3500),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position")
      abline(v = c(0,1))
}
```

```{r density_fig, eval = FALSE}
pdf(here("Documents", "3.Manuscript", "Figures", "methylation_density.pdf"), 
  width = 5, height = 4)
  plot.centered.mat(avg_methyl_dist, ylim = c(0, 3500),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position", min.upstream = -1, 
    max.downstream = 2)
    abline(v = c(0,1), lty = 2, col = "darkgray", lwd = 2)
dev.off()
```


## DNA methylation percentage

The following plots show the percent methylation across the gene body. 
Methylation sites outside gene bodies have 50% 
methylation. Sites at the TSS have very low percentage methylation,
and sites within the gene body have very high methylation.

```{r within_strain, results = "asis"}
strain.methyl.file <- here("Results", "RRBS", "Strain.RRBS.Percent.RDS")
if(!file.exists(strain.methyl.file)){
  strain_rrbs_mat <- vector(mode = "list", length = length(strains))
  names(strain_rrbs_mat) <- strains
  for(st in 1:length(strains)){
    cat("###", strains[st], "\n")
    strain_rrbs <- lapply(norm_rrbs, function(x) x[st,])
    strain_rrbs_mat[[st]] <- plot.centered.vals(val.list = strain_rrbs, 
    return.means = FALSE, seq.by = seq.by, ylim = c(0,100), 
    plot.label = paste("Percent Methylation Across the Gene Body in", strains[st]))
    abline(v = c(0,1))
    cat("\n\n")
  }
  saveRDS(strain_rrbs_mat, strain.methyl.file)
}else{
  strain_rrbs_mat <- readRDS(strain.methyl.file)
}
```


```{r methyl_heat, eval = FALSE}
#methylation heatmaps ordered by methylation percent at the TSS
strain.locale <- order.strains(strains, names(strain.expr[[1]]), col.table)
for(st in 1:length(strain_rrbs_mat)){
  tss.pos <- which(colnames(strain_rrbs_mat[[st]]) == "0")
  tss.order <- order(strain_rrbs_mat[[st]][,tss.pos])
  ordered.id <- transcript.info[match(rownames(strain_rrbs_mat[[st]])[tss.order], transcript.info[,"external_gene_name"]),"ensembl_gene_id"]
  ordered.expr <- sapply(strain.expr[match(ordered.id, names(strain.expr))], function(x) x[strain.locale[st]])
  expr.df <- data.frame(ordered.expr)
  if(is.interactive){quartz()}
  pheatmap(strain_rrbs_mat[[1]][tss.order,], cluster_rows = FALSE, cluster_cols = FALSE, 
  show_rownames = FALSE)
}

```

The pattern of methylation was the same across all strains. 
The following plot shows the median percent methylation by position.

```{r avg_methyl_percent}
all_rrbs_med <- t(sapply(strain_rrbs_mat, function(x) apply(x, 2, function(y) median(y, na.rm = TRUE))))
rel.pos <- as.numeric(colnames(all_rrbs_med))

if(is.interactive){quartz()}
plot.new()
plot.window(xlim = c(min(rel.pos), max(rel.pos)), ylim = c(0, 100))
for(i in 1:nrow(all_rrbs_med)){
  points(rel.pos, all_rrbs_med[i,], col = col.table[i,3], lwd = 2, type = "l")
}
axis(1);axis(2)
abline(v = c(0,1))
mtext(side = 1, text = "Relative Genomic Position", line = 2.5)
mtext(side = 2, text = "Median Percent DNA Methylation", line = 2.5)
```

```{r percent_fig, eval = FALSE}
pdf(here("Documents", "3.Manuscript", "Figures", "methylation_percentage.pdf"), 
  width = 5, height = 4)
  plot.centered.mat(all_rrbs_med, ylim = c(0, 100), 
  ylab = "Median Percent DNA Methylation", min.upstream = -1, max.downstream = 2)
  abline(v = c(0,1), lwd = 2, lty = 2, col = "darkgray")
dev.off()
```


## Within-Strain Correlation Between Methylation and Expression

The following plots show the correlation with gene expression 
across these normalized positions calculated above within each 
strain. These plots do not look at variation across strains, 
only across genes within one strain. 

There is a weak negative correlation between DNA methylation 
and gene expression at the TSS (r = -0.2). There are also two
weak, but consistent signals of positive correlation between
DNA methylation and gene expression just upstream of the TSS.
As weak as these blips are, they are very consistent across the
strains.

Even farther upstream of the TSS, there is another negative 
correlation between DNA methylation and gene expression. This
correlation may be slightly stronger than the correlation at the
TSS. 

```{r rrbs_expr_cor, fig.width = 10, fig.height = 5}
rrbs.expr.cor <- matrix(NA, nrow = length(strains), ncol = ncol(strain_rrbs_mat[[1]]))
rownames(rrbs.expr.cor) <- strains
colnames(rrbs.expr.cor) <- colnames(strain_rrbs_mat[[1]])
strain.expr.order <- order.strains(strains,names(strain.expr[[1]]), col.table)

#align gene methylation percentages with expression in this strain.
common.genes <- Reduce("intersect", lapply(strain_rrbs_mat, rownames))
gene.id <- transcript.info[match(common.genes, 
  transcript.info[,"external_gene_name"]),"ensembl_gene_id"]
gene.idx <- match(gene.id, names(strain.expr))
strain.expr.mat <- t(sapply(gene.idx, function(x) if(length(strain.expr[[x]]) > 1){strain.expr[[x]]}else{rep(NA, 9)}))
rownames(strain.expr.mat) <- common.genes
colnames(strain.expr.mat) <- rownames(strain.expr[[1]])

for(st in 1:length(strains)){
  strain_mat <- strain_rrbs_mat[[st]][match(common.genes, rownames(strain_rrbs_mat[[st]])),]
  rrbs.expr.cor[st,] <- apply(strain_mat, 2, function(x) cor(x, strain.expr.mat[,strain.expr.order[st]], use = "pairwise.complete.obs", method = "pearson"))
}

rel.pos <- as.numeric(colnames(rrbs.expr.cor))
strain.col <- col.table[match(strains, col.table[,6]),3]
if(is.interactive){quartz(width = 9, height = 5)}
layout(matrix(c(1,2), nrow = 1), width = c(1, 0.3))
par(mar = c(4,4,4,0))
plot.new()
plot.window(xlim = c(min(rel.pos), max(rel.pos)), ylim = c(min(rrbs.expr.cor), max(rrbs.expr.cor))) 
for(st in 1:nrow(rrbs.expr.cor)){
  points(rel.pos, rrbs.expr.cor[st,], col = strain.col[st], type = "l", lwd = 2)
}
axis(1)
axis(2)
mtext(text = "Relative Gene Position", side = 1, line = 2.5)
mtext(text = "Pearson Correlation Between Expression and Methylation", side = 2, line = 2.5)
abline(v = c(0,1))
abline(h = 0, col = "darkgray", lty = 2)
plot.new()
plot.window(xlim = c(0,1), ylim = c(0,1))
par(mar = c(4,0,4,0))
legend(0,0.9, col = strain.col, lty = 1, legend = col.table[match(strains, col.table[,6]),1], lwd = 3)
#pheatmap(rrbs.expr.cor, cluster_cols = FALSE, cluster_rows = FALSE)
```

```{r cor_fig, eval = FALSE}
pdf(here("Documents", "3.Manuscript", "Figures", "methylation_within_strain_correlation.pdf"), 
  width = 5, height = 4)
rel.pos <- as.numeric(colnames(rrbs.expr.cor))
plot.centered.mat(rrbs.expr.cor, ylim = c(-0.15, 0.15), 
  plot.label = "Within-strain correlation between gene expression and DNA methylation",
  min.upstream = -1, max.downstream = 2)
abline(v = c(0,1), lty = 2, col = "darkgray", lwd = 2)
abline(h = 0)
dev.off()
```

## Across-strain variation in DNA methylation

For each gene, we also looked for inter-strain variation in expression 
associated with DNA methylation.

We used the aligned DNA methylation values generated above. For
each gene, we correlated DNA methylation percent with expression 
across strains. 

Overall, there is an ever-so-slight negative correlation between 
percent DNA methylation and gene expression across strains. 


```{r inter_strain}

rrbs.by.gene.file <- here("Results", "RRBS", "RRBS.by.Gene.RDS")

if(!file.exists(rrbs.by.gene.file)){
  gene.by.strain <- lapply_pb(common.genes, function(y) sapply(strain_rrbs_mat, function(x) x[which(rownames(x) == y),]))
  names(gene.by.strain) <- common.genes
  saveRDS(gene.by.strain, rrbs.by.gene.file)
}else{
  gene.by.strain <- readRDS(rrbs.by.gene.file)
}

gene.expr.cor.file <- here("Results", "RRBS", "RRBS.Expr.Cor.Across.Strains.RDS")
if(!file.exists(gene.expr.cor.file)){
  gene.expr.cor <- matrix(NA, nrow = length(gene.by.strain), ncol = ncol(strain_rrbs_mat[[1]]))
  colnames(gene.expr.cor) <- colnames(strain_rrbs_mat[[1]])
  rownames(gene.expr.cor) <- names(gene.by.strain)

  expr.order <- order.strains(strains, names(strain.expr[[1]]), col.table)
  for(i in 1:length(gene.by.strain)){
    if(is.interactive){report.progress(i, length(gene.by.strain))}
    gene.id <- transcript.info[match(names(gene.by.strain)[i], 
      transcript.info[,"external_gene_name"]),"ensembl_gene_id"]
    gene.idx <- match(gene.id, names(strain.expr))
    if(length(scaled.expr[[gene.idx]]) > 1){
      one.strain.expr <- scaled.expr[[gene.idx]][expr.order,1]
      methyl.mat <- apply(gene.by.strain[[i]], 2, function(x) bin.vector(x, c(0, 50, 100)))
      has.all <- which(apply(methyl.mat, 1, function(x) all(!is.na(x))))
      if(length(has.all) > 0){
        gene.expr.cor[i,has.all] <- apply(methyl.mat[has.all,,drop=FALSE], 1, function(x) cor(x, one.strain.expr))
      }
    }
  }
  saveRDS(gene.expr.cor, gene.expr.cor.file)
}else{
  gene.expr.cor <- readRDS(gene.expr.cor.file)
}
```

```{r across_strain_cor_fig, eval = FALSE}
pdf(here("Documents", "3.Manuscript", "Figures", "methylation_expr_across_strains.pdf"), 
  width = 5, height = 4)
plot.centered.mat(gene.expr.cor, ylim = c(-0.15, 0.15), 
  ylab = "Pearson Correlation r Between DNA Methylation and Gene Expression",
  plot.label = "", min.upstream = -1, max.downstream = 2)
abline(v = c(0,1), lwd = 2, col = "darkgray", lty = 2)
abline(h = 0)
dev.off()
```

The top figure below shows an example alignment for one gene.
The color scale runs from blue to red indicating hypo- to 
hypermethylation.

The bottom figure shows the correlation between each position
with methylation, and gene expression in the inbred mice. 

```{r rrbs_example, fig.width = 10, fig.height = 4}
#gene.name = "Gnai3"
gene.name <- "Irf5"

gene.locale <- which(names(gene.by.strain) == gene.name)
if(is.interactive){quartz()}
methyl.mat <- apply(t(gene.by.strain[[gene.locale]]), 2, function(x) bin.vector(x, c(0, 50, 100)))
colnames(methyl.mat) <- round(as.numeric(colnames(methyl.mat)), 2)
pheatmap(methyl.mat, cluster_rows = FALSE, cluster_cols = FALSE,
main = gene.name)
#boxplot(methyl.mat)
if(is.interactive){quartz()}
plot(as.numeric(colnames(gene.expr.cor)), gene.expr.cor[gene.name,], 
xlab = "Relative Genomic Position", pch = 16, type = "h", main = gene.name,
ylab = "Correlation Between DNA Methylation and Gene Expression")
abline(h = 0)

```

## Variation in Percent Methylation Across Strains

We looked at the variation across all genes by subtracting the maximum methylation
percent from the minimum methylation percent across strains at each position.

```{r methyl_var}

get_methyl_var <- function(methyl.by.strain.mat){
  var.v <- rep(NA, nrow(methyl.by.strain.mat))
  has.vals <- which(apply(methyl.by.strain.mat, 1, function(x) all(!is.na(x))))
  if(length(has.vals) > 0){
    #methyl.max <- apply(methyl.by.strain.mat[has.vals,,drop=FALSE], 1, function(x) max(x, na.rm = TRUE))
    #methyl.min <- apply(methyl.by.strain.mat[has.vals,,drop=FALSE], 1, function(x) min(x, na.rm = TRUE))
    #var.v[has.vals] <- methyl.max - methyl.min
    var.v[has.vals] <- apply(methyl.by.strain.mat[has.vals,,drop=FALSE], 1, function(x) sd(x, na.rm = TRUE))
  }
  return(var.v)
}

get_methyl_mean <- function(methyl.by.strain.mat){
  mean.v <- rep(NA, nrow(methyl.by.strain.mat))
  has.vals <- which(apply(methyl.by.strain.mat, 1, function(x) all(!is.na(x))))
  if(length(has.vals) > 0){
    mean.v[has.vals] <- apply(methyl.by.strain.mat[has.vals,,drop=FALSE], 1, function(x) mean(x, na.rm = TRUE))
  }
  return(mean.v)
}

methyl_var <- t(sapply(gene.by.strain, get_methyl_var))
methyl_mean <- t(sapply(gene.by.strain, get_methyl_mean))
```

The following plot shows distributions of percent methylation by position
across all genes. This plot makes it obious that there is low methylation
with low variance in methylation at the TSS, and high methylation with 
low variance toward the TES. All other positions have varying means, but
also very high variaion in percent methylation.

```{r methyl_var_box}
boxplot(methyl_mean, group.names = colnames(methyl_mean))
```


The following plot shows mean methylation plotted against the standard
deviation of methylation. This plot reiterates the observation that 
positions with very high or very low methylation are also consistent
across strains. 

```{r mean_var}
if(is.interactive){quartz()}
plot(as.vector(methyl_mean), as.vector(methyl_var), xlab = "Methylation Mean",
  ylab = "Methylation SD")
colnames(methyl_var) <- colnames(methyl_mean) <- rownames(gene.by.strain[[1]])
```

The following plot shows the same data as above, but with the methylation 
percent binned. In this plot it is easier to see that methylation varies 
vary little across strains when it is very high or very low.

```{r mean_var_box}
mean.bins <- seq(0, 100, 10)
var_by_percent <- sapply(1:ncol(methyl_mean), function(y) sapply(1:(length(mean.bins) - 1), function(x) mean(methyl_var[intersect(which(methyl_mean[,y] >= mean.bins[x]), which(methyl_mean[,y] <= mean.bins[(x+1)])),y], na.rm = TRUE)))
rownames(var_by_percent) <- apply(cbind(mean.bins[1:10], mean.bins[2:11]),1, function(x) paste0(x[1], "-", x[2]))
colnames(var_by_percent) <- round(as.numeric(colnames(methyl_mean)), 2)

#pdf("~/Desktop/Methyl_variation.pdf", width = 7, height = 5)
boxplot(t(var_by_percent), ylab = "Methylation SD", xlab = "Methylation Mean")
#dev.off()
```

The following plot again shows the same information as above, but this 
time split by relative genomic position. The $x$-axis shows the genomic
position relative to the gene body. The $y$-axis shows binned percent
methylation mean, and the individual cells show the variation in percent
methylation for the corresponding methylation mean and gene position.

This plot shows that the low variation seen in positions with high or
low mean methylation does not depend on position. All positions with 
high or low percent methylation are stable across strains. Positions 
with mid-range DNA methylation have higher variance. This variation
may be slightly higher upstream of gene bodies, but the effect is 
not very strong.


```{r mean_var_heat, eval = FALSE}
pheatmap(var_by_position, cluster_rows = FALSE, cluster_cols = FALSE, 
  main = "Methylation Mean and SD by Relative Genomic Position")
```

The following plot shows that variaion in DNA methylation is lower 
at the TSS than at other genomic positions. This is due to the 
observation that variation is low when average methylation is low,
and the TSS has the lowest variation in methylation on average.

```{r var_by_position}
#quartz();barplot(colMeans(methyl_var, na.rm = TRUE))
if(is.interactive){quartz(width = 8, height = 5)}
#pdf("~/Desktop/methylation_var.pdf", width = 5, height = 4)
plot.centered.mat(methyl_var, ylim = c(0, 10), 
ylab = "Percent Methylation SD Across Strains",
plot.label = "Methylation Variation", min.upstream = -1, max.downstream = 2)
abline(v = c(0,1), lwd = 2, col = "darkgray", lty = 2)
#dev.off()
```

The following plot shows the number of genes that have variation
in methylation percent at each position. Now I'm confused. There is
a very high number of genes with variation at the TSS. More than 
anywhere else. But this variation is of particularly low magnitude?

```{r number_var}
num.var.genes <- apply(methyl_var, 2, function(x) length(which(x > 0)))
if(is.interactive){quartz()}
plot(as.numeric(colnames(methyl_var)), num.var.genes, 
  xlab = "Relative Gene Position", ylab = "Number of Genes with Variation", type = "l",
  lwd = 3, col = "darkgray")
abline(v = c(0,1), lty = 2)

#plot(sort(methyl_var[,21])) #TSS variation
```

