---
title: "RRBS and Chromatin Modifications"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

The purpose of this workflow is combine data about DNA methylation and 
chromatin modifications. Here we look at whether the effect of DNA methylation
on expression is affected by chromatin state.


```{r load_libraries, echo = FALSE, message = FALSE}
num.states = 14
is.interactive = FALSE
#is.interactive = TRUE
needed.packages <- c("here", "pheatmap", "hexbin", "RColorBrewer", "gprofiler2",
"knitr")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}
results.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))
```

```{r source_code}
all.code.dir <- list.files(here("code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r read_data}
#DNA methylation data aligned across strains
rrbs.mats <- readRDS(here("Results", "RRBS", "Aligned.Methyl.Mats.RDS"))
norm.rrbs <- readRDS(here("Results", "RRBS", "RRBS.Normalized.Position.RDS"))
norm.chrom <- readRDS(file.path(results.dir, "Chromatin.States.Gene.Coords.RDS"))
strain.expr <- readRDS(here("Data", "RNASeq", "Strain_Mean_C_Expression.RData"))
transcript.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
```

```{r scale_expression_data}
scaled.expr <- lapply(strain.expr, scale)
#scaled.expr <- lapply(strain.expr, function(x) x - x[2]) #center on B6

col.table <- as.matrix(read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "\t", comment.char = "%", stringsAsFactors = FALSE))
```

```{r match_genes_across_data}
#match up the genes in the chromatin, expression, and methylation data for ease of use later
rrbs.gene.id <- transcript.info[match(names(rrbs.mats), transcript.info[,"external_gene_name"]),"ensembl_gene_id"]
common.id <- Reduce("intersect", list(rrbs.gene.id, names(norm.chrom), names(strain.expr)))
common.chrom.locale <- match(common.id, names(norm.chrom))
common.rrbs.locale <- match(common.id, rrbs.gene.id)
common.expr.locale <- match(common.id, names(strain.expr))

matched.chrom <- norm.chrom[common.chrom.locale]
matched.rrbs <- rrbs.mats[common.rrbs.locale]
matched.norm.rrbs <- norm.rrbs[common.rrbs.locale]
names(matched.rrbs) <- names(matched.norm.rrbs) <- common.id
matched.expr <- strain.expr[common.expr.locale]

gene.names <- transcript.info[match(common.id, transcript.info[,"ensembl_gene_id"]),"external_gene_name"]
```

## Differential methylation by state {.tabset .tabset-fade .tabset-pills}

We looked at whether CpG sites were differentially methylated based
on the chromatin state surrounding them.

To do this, we first created a set of matrices for each measurement with 
standardized coordinates. This involves binning both the chromatin and
the DNA methylation signals. We lose a little resolution here, but I think
the patterns are still clear and reliable.

```{r match_epigen}

std.coord <- seq(-0.5, 1.5, 0.05) #relative genomic coordinates for binning chromatin and methylation
coord.mids <- sapply(1:(length(std.coord)-1), function(x) mean(c(std.coord[x], std.coord[(x+1)])))

align_chrom_rrbs <- function(chrom.mat, rrbs.mat, bins){
    
    bin.mids <- sapply(1:(length(bins)-1), function(x) mean(c(bins[x], bins[(x+1)])))

    chrom.coord <- as.numeric(colnames(chrom.mat))
    rrbs.coord <- as.numeric(colnames(rrbs.mat))

    #bin both sets of coordinates into the standardized coordinates
    chrom.coord.idx <- lapply(1:(length(bins)-1), 
        function(x) intersect(which(chrom.coord >= bins[x]), which(chrom.coord < bins[(x+1)])))
    rrbs.coord.idx <- lapply(1:(length(bins)-1), 
        function(x) intersect(which(rrbs.coord >= bins[x]), which(rrbs.coord < bins[(x+1)])))

    num.chrom.bin <- sapply(chrom.coord.idx, length)
    num.rrbs.bin <- sapply(rrbs.coord.idx, length)

    #has.data <- sort(union(which(num.chrom.bin > 0), which(num.rrbs.bin > 0)))

    binned.chrom <- lapply(chrom.coord.idx, function(x) if(length(x) > 0){chrom.mat[,x,drop=FALSE]}else{NA})
    binned.rrbs <- lapply(rrbs.coord.idx, function(x) if(length(x) > 0){rrbs.mat[,x,drop=FALSE]})

    #for each state, make a presence/absence matrix identifying whether that 
    #state is present in that bin for each strain
    state.bin.mats <- vector(mode = "list", length = num.states)
    for(st in 1:num.states){
        state.mat <- matrix(NA, nrow = nrow(chrom.mat), ncol = length(binned.chrom))
        colnames(state.mat) <- bin.mids
        rownames(state.mat) <- rownames(chrom.mat)
        state.present <- lapply(binned.chrom, function(x) if(length(x) > 1){apply(x, 1, function(y) length(which(y == st)) > 0)})
        for(b in 1:length(state.present)){
            if(length(state.present[[b]]) > 1){
                state.mat[,b] <- as.numeric(state.present[[b]])
            }
        }
        state.bin.mats[[st]] <- state.mat
    }

    #create a similar matrix for the DNA methylation with the average
    #methylation across the bin.
    rrbs.bin.mat <- sapply(binned.rrbs, function(x) if(length(x) > 1){rowMeans(x, na.rm = TRUE)}else{rep(NA, nrow(rrbs.mat))})
    colnames(rrbs.bin.mat) <- bin.mids
    rownames(rrbs.bin.mat) <- rownames(rrbs.mat)
    result <- list("chromatin_mats" = state.bin.mats, "rrbs_mat" = rrbs.bin.mat)
    return(result)
}

aligned.epi.chrom.file <- here("Results", "RRBS", "Aligned.Chromatin.and.RRBS.RDS")
if(!file.exists(aligned.epi.chrom.file)){
    aligned_epi <- lapply_pb(1:length(matched.chrom), 
        function(x) if(length(matched.chrom[[x]]) > 1){align_chrom_rrbs(t(matched.chrom[[x]]), matched.norm.rrbs[[x]], std.coord)}else{NA})
    names(aligned_epi) <- names(matched.chrom)
    saveRDS(aligned_epi, aligned.epi.chrom.file)
}else{
    aligned_epi <- readRDS(aligned.epi.chrom.file)
}

```

For each chromatin state, we looked at the percent methylation of CpG 
sites in the region of the state. We treated missing CpG sites in strains 
as unmethylated. 

The following plots show the median percent methylation of CpG sites in 
each chromatin state as they appear along the gene body. There do appear 
to be differential levels of methylation in the different states. All states 
have roughly the same pattern of being hypomethylated at the TSS, and 
hypermethylated in the gene body. However the degree to which that is 
true varies. 

```{r methyl_by_state, fig.width = 10, fig.height = 10}
get_methyl_by_state <- function(bin_state_mat, rrbs_mat){
    state_rrbs <- matrix(NA, nrow = nrow(rrbs_mat), ncol = ncol(rrbs_mat))
    rownames(state_rrbs) <- rownames(rrbs_mat)
    colnames(state_rrbs) <- colnames(rrbs_mat)
    state.locale <- which(bin_state_mat > 0)
    state_rrbs[state.locale] <- rrbs_mat[state.locale]
    return(state_rrbs)
}

full_rrbs <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){aligned_epi[[x]][[2]]}else{NA})
has.methyl <- which(sapply(full_rrbs, function(x) !all(is.na(x))))

#pdf("~/Desktop/RRBS_in_States.pdf")
all_state_methyl_levels <- vector(mode = "list", length = num.states)
names(all_state_methyl_levels) <- 1:num.states

if(is.interactive){quartz(width = 10, height = 7)}
par(mfrow = c(2,2))
for(state in 1:num.states){
    state_methyl <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){get_methyl_by_state(aligned_epi[[x]][[1]][[state]], aligned_epi[[x]][[2]])}else{NA})
    has.methyl <- which(sapply(state_methyl, function(x) !all(is.na(x))))

    all_strain_list <- lapply(1:9, function(strain) t(sapply(state_methyl[has.methyl], function(x) x[strain,])))

    coords <- as.numeric(colnames(all_strain_list[[1]]))
    all_strain_mat <- Reduce("rbind", all_strain_list)
    row.has.val <- which(apply(all_strain_mat, 1, function(x) !all(is.na(x))))

    percent_list <- lapply(1:ncol(all_strain_mat), function(x) all_strain_mat[,x])
    has.vals <- which(sapply(percent_list, function(x) length(which(!is.na(x)))) > 3)
    all_state_methyl_levels[[state]] <- plot_mean_conf(x = coords[has.vals], 
        y.list = percent_list[has.vals], spar = 0.4,
        xlab = "Relative Position", ylab = "Methylation Percent Mean", 
        main = paste("State", state), ylim = c(0, 100))
}
saveRDS(all_state_methyl_levels, here("Results", "RRBS", 
    "Methylation_Level_by_Chromatin_State.RDS"))
#dev.off()
```


```{r max_min_methyl, eval = FALSE}

u_pos <- sort(unique(unlist(sapply(all_state_methyl_levels, function(x) as.numeric(colnames(x))))))
state.methyl.mat <- matrix(NA, ncol = length(u_pos), nrow = length(all_state_methyl_levels))
rownames(state.methyl.mat) <- 1:num.states
colnames(state.methyl.mat) <- u_pos

for(st in 1:length(all_state_methyl_levels)){
    state.methyl.mat[st,colnames(all_state_methyl_levels[[st]])] <- all_state_methyl_levels[[st]]["mean",]
}


pheatmap(state.methyl.mat, cluster_rows = FALSE, cluster_cols = FALSE)
boxplot(state.methyl.mat)

intragenic.idx <- intersect(which(u_pos >= 0), which(u_pos <= 1))
barplot_with_num(sort(round(rowMeans(state.methyl.mat[,intragenic.idx]))))

methyl.min <- apply(state.methyl.mat, 2, function(x) min(x, na.rm = TRUE))
methyl.max <- apply(state.methyl.mat, 2, function(x) max(x, na.rm = TRUE))
plot.poly.xy(coords, methyl.min, coords, methyl.max, new.plot = TRUE, col = "gray")
axis(1);axis(2)
abline(v = c(0,1))
```

## Chromatin state, methylation, and expression {.tabset .tabset-fade .tabset-pills}

We next investigated whether the chromatin state housing the 
methylated DNA had any effect on the methylation's effect on 
expression (within each strain).

For example, methylating DNA at the TSS in a repressor state
may have a positive effect on expression, whereas methylating
DNA in an enhancer state may reduce gene expression.


```{r chrom_methyl_in_strain, results = "asis"}
methyl.expr.state.file <- here("Results", "RRBS", "Methylation_and_Expression_by_State.RDS")

if(!file.exists(methyl.expr.state.file)){
    strains <- colnames(matched.chrom[[1]])
    expr.order <- match.order(rownames(matched.norm.rrbs[[1]]), names(strain.expr[[1]]), col.table)
    #cbind(rownames(matched.norm.rrbs[[1]]), names(strain.expr[[1]])[expr.order])

    #set up vector to hold effects of methylation on expression by state
    methyl_expr_effect <- vector(mode = "list", length = num.states)
    names(methyl_expr_effect) <- 1:num.states

    #for each state
    for(state in 1:num.states){

        #get the methylation percent for all CpG sites in this state
        state_methyl <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){get_methyl_by_state(aligned_epi[[x]][[1]][[state]], aligned_epi[[x]][[2]])}else{NA})

        #find which genes actually have DNA methylation in this state
        has.methyl <- which(sapply(state_methyl, function(x) !all(is.na(x))))
        
        #set up objects to hold the methylation and expression for each strain
        strain_cor_list <- vector(mode = "list", length = length(strains))
        names(strain_cor_list) <- strains

        all.strain.methyl <- vector(mode = "list", length = length(strains))
        all.strain.expr <- vector(mode = "list", length = length(strains))

        #test the correlation between DNA methylation in this state within each strain
        #pdf("~/Desktop/test.pdf")
        for(strain in 1:length(strains)){
            #pull out methylation values for just this strain
            methyl_by_strain <- t(sapply(state_methyl[has.methyl], function(x) x[strain,]))
            rownames(methyl_by_strain) <- names(aligned_epi)[has.methyl]
            #pheatmap(methyl_by_strain, cluster_rows = FALSE, cluster_cols = FALSE, show_rownames = FALSE)

            #replace NA's with 0's because non-CpG sites are not methylated
            methyl_by_strain[which(is.na(methyl_by_strain))] <- 0
            #pheatmap(methyl_by_strain, cluster_rows = FALSE, cluster_cols = FALSE, show_rownames = FALSE)

            #store the methylation values for this strain in this state
            all.strain.methyl[[strain]] <- methyl_by_strain
            
            #find the corresponding expression for the genes with methylation in this state
            expr.locale <- match(rownames(methyl_by_strain), names(strain.expr))
            comp.expr <- sapply(strain.expr[expr.locale], function(x) x[expr.order[strain]]) 

            #normalize the expression within-strain and store
            all.strain.expr[[strain]] <- rankZ(comp.expr)

            methyl.cor <- apply(methyl_by_strain, 2, function(x) if(var(x) > 0){cor(x, all.strain.expr[[strain]])}else{NA})
            plot(methyl.cor, type = "l", ylim = c(-0.2, 0.2));abline(h = 0)

        }
        #dev.off()

        #for each strain, calculate whether DNA methylation within each
        #state has an effect on gene expression at all positions along
        #the gene body
        strain_methyl_effect <- lapply(1:length(all.strain.methyl), 
            function(y) sapply(1:ncol(all.strain.methyl[[y]]), 
            function(x) if(length(unique(all.strain.methyl[[y]][,x])) > 1){plot.with.model(all.strain.methyl[[y]][,x], all.strain.expr[[y]], 
            confidence = 0.95, plot.results = FALSE)}else{rep(NA, 5)}))
        #plot(strain_methyl_effect[[1]]["slope.fit",], type = "l");abline(h = 0)

        strain.effect.mat <- Reduce("+", strain_methyl_effect)/length(strain_methyl_effect)
        colnames(strain.effect.mat) <- colnames(all.strain.methyl[[1]])
        rownames(strain.effect.mat) <- c("r", "p.value", "slope.fit", "slope.lwr", "slope.upr")

        methyl_expr_effect[[state]] <- strain.effect.mat
    }
    saveRDS(methyl_expr_effect, here("Results", "RRBS", "Methylation_and_Expression_by_State.RDS"))
}else{
    methyl_expr_effect <- readRDS(methyl.expr.state.file)
}

coords <- as.numeric(colnames(methyl_expr_effect[[1]]))
state.col <- colors.from.values(1:num.states, use.pheatmap.colors = TRUE, 
    global.color.scale = TRUE, global.min = 1, global.max = num.states)

#pdf("~/Desktop/Effect_of_DNA_Methylation_By_State.pdf")
if(is.interactive){quartz(width = 9, height = 9)}
par(mfrow = c(3,3), mar = c(2,4,2,2))
for(state in 1:num.states){
    plot.new()
    plot.window(xlim = c(-0.1, 1.1), ylim = c(-0.02, 0.02))
    points(coords, methyl_expr_effect[[state]]["slope.fit",], col = state.col[state], type = "l")    
    plot.poly.xy(coords, methyl_expr_effect[[state]]["slope.lwr",],
    coords, methyl_expr_effect[[state]]["slope.upr",], col = state.col[state])
    abline(h = 0)
    axis(2)
    mtext(side = 2, "Effect", line = 2.5)
    mtext(side = 3, paste("State", state), line = 0)
    abline(v = c(0,1), lty = 2, col = "darkgray")
}
#dev.off()

```

## DNA methylation effects across strains

We also looked for any effect of DNA methylation across strains.
This is the same code above, but draws on the scaled expression 
instead of the TPM data.

```{r across_strain_effect}
methyl_cor_file <- here("Results", "RRBS", "Methylation_percent_by_state.RDS")

if(!file.exists(methyl_cor_file)){

    strains <- colnames(matched.chrom[[1]])
    expr.order <- match.order(rownames(matched.norm.rrbs[[1]]), names(strain.expr[[1]]), col.table)
    min.n.gene = 50
    methyl_expr_cor <- methyl_bin_cor <- vector(mode = "list", length = num.states)
    names(methyl_expr_cor) <- names(methyl_bin_cor) <- 1:num.states

    for(state in 1:num.states){
        #pull out the CpG sites that are located in this chromatin state
        state_methyl <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){get_methyl_by_state(aligned_epi[[x]][[1]][[state]], aligned_epi[[x]][[2]])}else{NA})
        #identify which ones have CpG sites in this chromatin state
        has.methyl <- which(sapply(state_methyl, function(x) !all(is.na(x))))
        strain_cor_list <- vector(mode = "list", length = length(strains))
        names(strain_cor_list) <- strains

        #make objects to hold all methylation and expression data across the strains
        #expression is nromalized across strains
        all.strain.methyl <- vector(mode = "list", length = length(strains))
        all.strain.expr <- vector(mode = "list", length = length(strains))

        for(strain in 1:length(strains)){
            #pull out centered methylation values for this strain
            methyl_by_strain <- t(sapply(state_methyl[has.methyl], function(x) x[strain,]))
            rownames(methyl_by_strain) <- names(aligned_epi)[has.methyl]
            #store these value
            all.strain.methyl[[strain]] <- methyl_by_strain

            #find all corresponding expression
            expr.locale <- match(rownames(methyl_by_strain), names(scaled.expr))
            #pull out the normalized expression for this strain
            comp.expr <- sapply(scaled.expr[expr.locale], function(x) x[expr.order[strain]]) 
            #store this expression vector
            all.strain.expr[[strain]] <- comp.expr
        }

        #put data from all strains together into one big matrix
        #expression is scaled across strains
        #methylation is left as 0, 50, 100
        overall_methyl <- Reduce("rbind", all.strain.methyl)
        overall_expr <- Reduce("c", all.strain.expr)

        #replace NA's with 0's in overall_methyl because non-CpG sites are not methylated
        overall_methyl[which(is.na(overall_methyl))] <- 0

        #initialize a matrix to hold statistics
        conf_mat <- bin.conf.mat <- matrix(NA, nrow = 5, ncol = ncol(methyl_by_strain))
        colnames(conf_mat) <- colnames(bin.conf.mat)  <- colnames(methyl_by_strain)
        rownames(conf_mat) <- rownames(bin.conf.mat) <- c("r", "p.value", "slope.fit", "slope.lwr", "slope.upr")

        #only use columns in which there are a minimum number of genes present
        min.samples <- which(apply(overall_methyl, 2, function(x) length(which(x > 0))) >= min.n.gene)
        strain.cor.mat <- apply(overall_methyl[,min.samples], 2, 
            function(x) plot.with.model(x, rankZ(overall_expr), 
            confidence = 0.95, plot.results = FALSE, report = "cor.test")) 
        #plot(strain.cor.mat["slope.fit",], type = "l")

        conf_mat[,min.samples] <- strain.cor.mat
        methyl_expr_cor[[state]] <- conf_mat
    }

    saveRDS(methyl_expr_cor, methyl_cor_file)
}else{
    methyl_expr_cor <- readRDS(methyl_cor_file)
}
```

### DNA methylation percent

The following plots show the association between DNA methylation
percent and gene expression using only DNA methylation in each
state. Missing CpG sites are treated as unmethylated.

```{r plot_cor}

coords <- as.numeric(colnames(methyl_expr_cor[[1]]))
state.col <- colors.from.values(1:num.states, use.pheatmap.colors = TRUE, 
	global.color.scale = TRUE, global.min = 1, global.max = num.states)

#pdf("~/Desktop/percent_methylation.pdf")
if(is.interactive){quartz(width = 9, height = 9)}
par(mfrow = c(3,3), mar = c(2,4,2,2))
for(state in 1:num.states){
    #plot.new()
    #plot.window(xlim = c(-0.1, 1.1), ylim = c(-0.5, 0.6))
    plot.poly.xy(coords, methyl_expr_cor[[state]]["slope.lwr",],
    coords, methyl_expr_cor[[state]]["slope.upr",], col = state.col[state], new.plot = TRUE)
    points(coords, methyl_expr_cor[[state]]["slope.fit",], col = state.col[state], type = "l")    
    
    #ymin <- max(abs(methyl_expr_cor[[state]][1,]), na.rm = TRUE)*-1
    #ymax <- max(abs(methyl_expr_cor[[state]][1,]), na.rm = TRUE)
    #plot(coords, methyl_expr_cor[[state]][1,], type = "l", ylim = c(ymin, ymax), lwd = 2)

    abline(h = 0)
    axis(2)
    mtext(side = 2, "Effect", line = 2.5)
    mtext(side = 3, paste("State", state), line = -1.5)
    abline(v = c(0,1), lty = 2, col = "darkgray")
}
#dev.off()
```

## Examples

The following plots show the correlations between DNA methylation
and gene expression where the chromatin state makes a large difference

## Strong Correlation Examples {.tabset .tabset-fade .tabset-pills}

```{r plot_example, results = "asis", fig.width = 8, fig.height = 8}

plot.example.cor <- function(state, position, confidence = 0.95){
    pos <- which(coords == position)
    state_chrom <- lapply(aligned_epi, function(x) if(length(x) > 1){x[[1]][[state]]}else{NA})
    full_methyl <- lapply(aligned_epi, function(x) if(length(x) > 1){x[[2]]}else{NA})
    state_methyl <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){get_methyl_by_state(aligned_epi[[x]][[1]][[state]], aligned_epi[[x]][[2]])}else{NA})
    has.methyl <- which(sapply(state_methyl, function(x) !all(is.na(x))))

    all.strain.all.methyl <- all.strain.all.expr <- all.strain.all.chrom <- NULL

    #combine data for all strains
    for(strain in 1:length(strains)){
        chrom_by_strain <- t(sapply(state_chrom[has.methyl], function(x) x[strain,]))
        methyl_by_strain <- t(sapply(full_methyl[has.methyl], function(x) x[strain,]))
        rownames(methyl_by_strain) <- names(aligned_epi)[has.methyl]
        expr.locale <- match(rownames(methyl_by_strain), names(strain.expr))
        comp.expr <- sapply(strain.expr[expr.locale], function(x) x[expr.order[strain]])

        all.strain.all.methyl <- rbind(all.strain.all.methyl, methyl_by_strain)
        all.strain.all.chrom <- rbind(all.strain.all.chrom, chrom_by_strain)
        scaled.vector <- rankZ(comp.expr)
        all.strain.all.expr <- c(all.strain.all.expr, scaled.vector)
    }

    #replace NA's with 0's becahse non-CpGs will not be methylated
    all.strain.all.methyl[which(is.na(all.strain.all.methyl))] <- 0

    if(is.interactive){quartz(width = 8, height = 8)}
    par(mfrow = c(2,2), mar = c(4, 4, 4, 2))
    #gene.strain <- sapply(strsplit(names(all.strain.all.expr), "[.]"), function(x) x[2])
    #strain.col <- col.table[match(gene.strain, col.table[,4]),3]
    has.state <- which(all.strain.all.chrom[,pos] == 1)
    no.state <- which(all.strain.all.chrom[,pos] == 0)
    
    plot.with.model(rankZ(all.strain.all.methyl[has.state,pos]), 
    rankZ(all.strain.all.expr[has.state]), 
    xlab = "Percent DNA Methylation", ylab = "Gene Expression", report = "cor.test",
    main = paste0("State: ", state, "; Relative position: ", coord.mids[pos], "; n = ", 
    length(has.state)), confidence = confidence, col = "darkgray")


    test <- apply(all.strain.all.chrom[has.state,], 2, function(x) cor(x, rankZ(all.strain.all.expr[has.state]), use = "pairwise.complete.obs"))
    plot(as.numeric(names(test)), test, type = "l")
    abline(h = 0)

    plot.with.model(rankZ(all.strain.all.methyl[no.state,pos]), 
    rankZ(all.strain.all.expr[no.state]), 
    xlab = "Percent DNA Methylation", ylab = "Gene Expression", report = "cor.test",
    main = paste0("All other states: Relative position: ", coord.mids[pos], "; n = ", 
    length(no.state)), confidence = confidence, col = "darkgray")

    binned.methyl <- bin.vector(all.strain.all.methyl[,pos], c(0, 50, 100))
    plot_lines(rankZ(all.strain.all.expr), all.strain.all.chrom[,pos], binned.methyl, 
    error_bars = "se", marker1_label = paste("Chromatin State", state), 
    marker2_label = "DNA Methylation", pheno_name = "")

    plot_lines(phenoV = rankZ(all.strain.all.expr), marker1_vals = binned.methyl, 
    marker2_vals = all.strain.all.chrom[,pos], 
    error_bars = "se", marker2_label = paste("Chromatin State", state), 
    marker1_label = "DNA Methylation", pheno_name = "")

    #plot_lines(all.strain.all.expr, binned.methyl, all.strain.all.chrom[,pos], error_bars = "se",marker1_label = paste("Chromatin State", state), marker2_label = "DNA Methylation",pheno_name = "")

    #boxplot(all.strain.all.expr~binned.methyl*all.strain.all.chrom[,pos])
}


state = 2; pos = coords[11]
plot.example.cor(state = state, position = pos, confidence = 0.99)

state = 6; pos = coords[13]
plot.example.cor(state = state, position = pos, confidence = 0.99)

```


## debugging 

Something weird is going on. I'm getting different results in the 
example plots and the overall plots.


```{r testing}

pdf("~/Desktop/alternate_methyl_effects.pdf")

state.stats <- vector(mode = "list", length = num.states)
names(state.stats) <- 1:num.states

for(state in 1:num.states){

    state_chrom <- lapply(aligned_epi, function(x) if(length(x) > 1){x[[1]][[state]]}else{NA})
    full_methyl <- lapply(aligned_epi, function(x) if(length(x) > 1){x[[2]]}else{NA})
    state_methyl <- sapply(1:length(aligned_epi), function(x) if(length(aligned_epi[[x]][[1]]) > 1){get_methyl_by_state(aligned_epi[[x]][[1]][[state]], aligned_epi[[x]][[2]])}else{NA})
    has.methyl <- which(sapply(state_methyl, function(x) !all(is.na(x))))

    all.strain.all.methyl <- all.strain.all.expr <- all.strain.all.chrom <- NULL

    #combine data for all strains
    for(strain in 1:length(strains)){
        chrom_by_strain <- t(sapply(state_chrom[has.methyl], function(x) x[strain,]))
        methyl_by_strain <- t(sapply(full_methyl[has.methyl], function(x) x[strain,]))
        rownames(methyl_by_strain) <- names(aligned_epi)[has.methyl]
        expr.locale <- match(rownames(methyl_by_strain), names(strain.expr))
        comp.expr <- sapply(strain.expr[expr.locale], function(x) x[expr.order[strain]])

        all.strain.all.methyl <- rbind(all.strain.all.methyl, methyl_by_strain)
        all.strain.all.chrom <- rbind(all.strain.all.chrom, chrom_by_strain)
        scaled.vector <- rankZ(comp.expr)
        all.strain.all.expr <- c(all.strain.all.expr, scaled.vector)
    }

    #replace NA's with 0's becahse non-CpGs will not be methylated
    all.strain.all.methyl[which(is.na(all.strain.all.methyl))] <- 0

    methyl.stats <- matrix(NA, nrow = ncol(all.strain.all.methyl), ncol = 5)
    rownames(methyl.stats) <- colnames(all.strain.all.methyl)
    for(pos in 1:ncol(all.strain.all.methyl)){
        instate <- which(all.strain.all.chrom[,pos] == 1)
        if(length(instate) < 5){next()}
        chrom.methyl <- all.strain.all.methyl[instate,pos]
        if(var(chrom.methyl) > 0){
            pos.stats <- plot.with.model(chrom.methyl, 
                rankZ(all.strain.all.expr[instate]), 
                confidence = 0.95, plot.results = FALSE, report = "cor.test")
            methyl.stats[pos,] <- pos.stats
        }
    }
    colnames(methyl.stats) <- names(pos.stats)
    state.stats[[state]] <- t(methyl.stats)

    plot.new()
    plot.window(xlim = c(-0.2, 1.2), ylim = c(-0.03, 0.03))
    plot.poly.xy(as.numeric(colnames(all.strain.all.chrom)), methyl.stats[,"slope.upr"], 
        as.numeric(colnames(all.strain.all.chrom)), methyl.stats[,"slope.lwr"],
        col = "lightgray")
    axis(1);axis(2)    
    abline(h = 0, v = c(0,1))
    mtext(paste("State", state), side = 3, line = 2.5)

}
dev.off()

saveRDS(state.stats, here("Results", "RRBS", "Methylation_and_Expression_by_State.RDS"))
```