---
title: "DNA Methylation in the DO"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

The purpose of this workflow is to impute DNA methylation states
in DO mice and map local gene expression.

```{r load_libraries}
library(here)
is.interactive = FALSE
#is.interactive = TRUE

delete_previous = FALSE
#delete_previous = TRUE
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```


```{r load_other_lib}
needed.packages <- c("qtl2", "abind", "pheatmap", "parallel", "doParallel")
load_libraries(needed.packages)
```


```{r read_data}
gene.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
strain.table <- as.matrix(read.delim(here("Data", "support_files", "strain.color.table.txt"),
comment.char = "!", header = FALSE))

vars <- load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData"))
expr <- dataset.mrna$data$rz
covar <- dataset.mrna$covar.matrix

col.table <- as.matrix(read.table(here("Data", "support_files", "strain.color.table.txt"), 
sep = "", comment.char = "%", stringsAsFactors = FALSE))
strains <- col.table[,6]
```

Read in the gene methylation data from 3.1.1_RRBS_Analysis.Rmd

```{r gene_methyl}
methyl.mat.file <- here("Results", "RRBS", "Aligned.Methyl.Mats.RDS")
methyl.mats <- readRDS(methyl.mat.file)

#transcripts that are shared between RRBS and DO expression data
methyl.id <- gene.info[match(names(methyl.mats), gene.info[,"external_gene_name"]),"ensembl_gene_id"]
methyl.id <- methyl.id[which(!is.na(methyl.id))]
ordered.transcript.info <- gene.info[match(methyl.id, gene.info[,"ensembl_gene_id"]),]
```

```{r fun}
get_nearest_marker <- function(chr, start.pos){
    if(is.na(chr)){return(NA)}
    if(chr == "MT"){return(NA)}
    chr.locale <- which(names(map) == chr)
    nearest.marker <- get.nearest.pt(map[[chr.locale]], start.pos/1e6)
    marker.geno <- genoprobs[[chr.locale]][,,nearest.marker]
    return(marker.geno)
}

rrbs_geno <- function(gene.rrbs.array, haplotype.matrix, gene.chr, perm.order = NULL){
    if(length(gene.rrbs.array) == 1){return(NA)}
    if(length(haplotype.matrix) <= 1){return(NA)}
    #filter out NAs
    has.all <- which(apply(gene.rrbs.array, 2, function(x) all(!is.na(x))))
    if(length(has.all) <= 1){return(NA)}
    #plot.methyl.mat(gene.rrbs.array[,has.all])
    #boxplot(gene.rrbs.array[,has.all])
    #pheatmap(cor(gene.rrbs.array[,has.all]), cluster_rows = FALSE, cluster_cols = FALSE)

    #bin the methylation into 0, 50, and 100
    #this prevents transferring haplotypes directly
    #onto very small variations in methylation
    binned.rrbs <- t(apply(gene.rrbs.array[,has.all,drop=FALSE], 1, function(x) bin.vector(x, c(0, 50, 100))))
        
    #do the imputation, with a shuffled haplotype order if requested
    if(is.null(perm.order)){
      all.rrbs <- haplotype.matrix %*% binned.rrbs
    }else{
      perm.rrbs <- binned.rrbs[perm.order,]
      rownames(perm.rrbs) <- rownames(binned.rrbs)
      all.rrbs <- haplotype.matrix %*% perm.rrbs
    }
    #pheatmap(all.rrbs, cluster_cols = FALSE)
    #boxplot(all.rrbs) 
    #pheatmap(haplotype.matrix)
    #hap.cor <- matrix(NA, nrow = ncol(haplotype.matrix), ncol = ncol(all.rrbs))
    #rownames(hap.cor) <- colnames(haplotype.matrix)
    #colnames(hap.cor) <- colnames(all.rrbs)
    #for(i in 1:ncol(haplotype.matrix)){    
    #  hap.cor[i,] <- apply(all.rrbs, 2, function(x) cor(x, haplotype.matrix[,i]))
    #}
    #pheatmap(hap.cor, cluster_rows = FALSE, cluster_cols = FALSE)
    #which(hap.cor[1,] == 1)
    #gene.rrbs.array[,"67245515"]
    
    all.rrbs.mat <- abind(all.rrbs, round(100-all.rrbs), along = 3)
    dimnames(all.rrbs.mat)[[3]] <- c("A", "B")
    all.rrbs.array <- aperm(all.rrbs.mat, c(1,3,2))
    one.geno <- list(all.rrbs.array)

    #add attributes
    names(one.geno) <- gene.chr
    if(gene.chr == "X"){
      attr(one.geno, "is_x_chr") <- TRUE
    }else{
      attr(one.geno, "is_x_chr") <- FALSE
    }
    attr(one.geno, "crosstype") <- "do"
    attr(one.geno, "alleles") <- dimnames(all.rrbs.array)[[3]]
    attr(one.geno, "alleleprobs") <- TRUE
    attr(one.geno, "class") <- c("calc_genoprob", "list")
    return(one.geno)
}

```


## Get Genotypes for Nearest Markers

For every gene in our list, pull out the genotype matrix for the marker
nearest the TSS.

```{r set_gene}
gene.marker <- vector(mode = "list", length = length(methyl.id))
names(gene.marker) <- methyl.id
for(i in 1:length(methyl.id)){
  if(is.interactive){report.progress(i, length(methyl.id))}
  gene.chr <- ordered.transcript.info[i,"chromosome_name"]
  gene.pos <- ordered.transcript.info[i, "start_position"]
  if(gene.chr != "Y" && gene.chr != "MT"){
    gene.marker[[i]] <- get_nearest_marker(gene.chr,gene.pos) 
  }
}
```

## Impute DO methylation percent and map

Derive DNA methylation percentages for each mouse given 
the genoprobs and the methylation data compiled from 3.1.1_RRBS_Analysis.Rmd. 
This array will have DNA methylation percentage estimates for each 
individual by position.

To generate this array we multiply each DNA methylation matrix along 
the genome by the haplotype matrix for the gene.

```{r scan_rrbs_geno}
#convert to RRBS arrays and scan.
#The arrays are too big to store, so we
#scan them right away and only keep the 
#LOD scores.

strain.order <- order.strains(dimnames(genoprobs[[1]])[[2]], rownames(methyl.mats[[1]]), col.table)
common.ind <- intersect(rownames(expr), rownames(genoprobs[[1]]))
common.expr.locale <- match(common.ind, rownames(expr))
common.genoprob.locale <- match(common.ind, rownames(genoprobs[[1]]))
rrbs.r2.file <- here("Results", "RRBS", "RRBS.R2.RDS")
rrbs.coef.file <- here("Results", "RRBS", "RRBS.coef.RDS")
if(!file.exists(rrbs.r2.file) || delete_previous){
    sink(here("Results", "RRBS", "progress.txt"))
    all.rrbs.r2 <- all.rrbs.coef <- vector(mode = "list", length = length(methyl.id))
    names(all.rrbs.r2) <- names(all.rrbs.coef) <- methyl.id
    for(i in 1:length(all.rrbs.r2)){
        cat(i, ":")
        methyl.mat.locale <- which(names(methyl.mats) == ordered.transcript.info[i,"external_gene_name"])
        if(length(methyl.mats[[methyl.mat.locale]]) <= 1 || nrow(methyl.mats[[methyl.mat.locale]]) >= 2000){cat("No DNA methylation matrix\n");next()}

        transcript.chr <- ordered.transcript.info[i,"chromosome_name"]
        if(is.na(transcript.chr) || transcript.chr == "X" || transcript.chr == "MT"){cat(" Not on an autosome\n");next()}

        rrbs.mat <- methyl.mats[[methyl.mat.locale]][strain.order,,drop=FALSE]
        hap.mat <- gene.marker[[i]]
        rrbs.array <- rrbs_geno(gene.rrbs.array = rrbs.mat, 
          haplotype.matrix = hap.mat, 
          gene.chr = ordered.transcript.info[i,"chromosome_name"])

        if(is.na(rrbs.array)){cat(" Cannot create RRBS array\n");next()}

        gene.expr <- expr[,which(colnames(expr) == methyl.id[i]),drop=FALSE]
        
        if(length(gene.expr) == 0){cat(" No gene expression\n");next()}
        
        adj.expr <- adjust(gene.expr, covar)
        
        regression.models <- apply(rrbs.array[[1]][common.genoprob.locale,1,], 2, function(x) if(var(x) > 0){lm(adj.expr[common.expr.locale,]~x)}else{NA})
        rrbs.r2 <- sapply(regression.models, function(x) if(length(x) > 1){summary(x)$adj.r.squared}else{NA})
        all.rrbs.r2[[i]] <- rrbs.r2  

        rrbs.coef <- sapply(regression.models, function(x) if(length(x) > 1){coef(x)[2]}else{NA})
        all.rrbs.coef[[i]] <- rrbs.coef
        
        cat("\n")
        #quartz();barplot(all.rrbs.r2[[i]])
    }
    sink()
    saveRDS(all.rrbs.r2, rrbs.r2.file)
    saveRDS(all.rrbs.coef, rrbs.coef.file)
}else{
  all.rrbs.r2 <- readRDS(rrbs.r2.file)
  all.rrbs.coef <- readRDS(rrbs.coef.file)
}
```

## Position Effects

As we did with the inbred expression, look for 
position-based effects in the outbred expression.

```{r center_coef}
gene.names <- gene.info[match(names(all.rrbs.coef), gene.info[,"ensembl_gene_id"]),"external_gene_name"]
centered.coord <- lapply(1:length(gene.names),
  function(x) if(length(all.rrbs.coef[[x]]) > 1){center.on.feature(gene.names[x], 
  gene.info, as.numeric(gsub(".x", "", names(all.rrbs.coef[[x]]))), 
  feature = "full")}else{NA})

ymin <- min(unlist(all.rrbs.coef), na.rm = TRUE)
ymax <- max(unlist(all.rrbs.coef), na.rm = TRUE)

nbins = 15
coord.bins <- segment.region(-0.5, 1.5, nbins)
plot.coord <- coord.bins[1:(length(coord.bins)-1)]
all.coord <- unlist(lapply(centered.coord, function(x) as.numeric(names(x))))
all.coef <- unlist(all.rrbs.coef)

binned.coef <- lapply(1:(length(coord.bins)-1), 
  function(x) all.coef[intersect(which(all.coord >= coord.bins[x]), 
  which(all.coord < coord.bins[(x+1)]))])

coef.mean <- sapply(binned.coef, function(x) mean(x, na.rm = TRUE))
coef.se <- sapply(binned.coef, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))

plot.new()
plot.window(xlim = c(-0.5, 1.5), ylim = c(min(coef.mean-coef.se), max(coef.mean+coef.se)))
plot.poly.xy(poly.top.x = plot.coord, poly.top.y = coef.mean+coef.se, 
  poly.bottom.x = plot.coord, poly.bottom.y = coef.mean-coef.se,
  border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
abline(h = 0)
axis(1)
axis(2)
abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
mtext("Mean DNA Methylation Effect on Transcription in DO", side = 3, line = 1.5)
```

```{r centered_r2}
ymin <- min(unlist(all.rrbs.r2), na.rm = TRUE)
ymax <- max(unlist(all.rrbs.r2), na.rm = TRUE)

all.r2 <- unlist(all.rrbs.r2)

binned.r2 <- lapply(1:(length(coord.bins)-1), 
  function(x) all.r2[intersect(which(all.coord >= coord.bins[x]), 
  which(all.coord < coord.bins[(x+1)]))])

r2.mean <- sapply(binned.r2, function(x) mean(x, na.rm = TRUE))
r2.se <- sapply(binned.r2, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))

plot.new()
plot.window(xlim = c(-0.5, 1.5), ylim = c(min(r2.mean-r2.se), max(r2.mean+r2.se)))
plot.poly.xy(poly.top.x = plot.coord, poly.top.y = r2.mean+r2.se, 
  poly.bottom.x = plot.coord, poly.bottom.y = r2.mean-r2.se,
  border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
abline(h = 0)
axis(1)
axis(2)
abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
mtext("Mean DNA Methylation Variance Explained in DO", side = 3, line = 1.5)
```