---
title: "SNPs in DO"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


## Introduction

The purpose of this workflow is to impute SNPs from the DO founders
into the current DO population to perform association mapping using
local SNPs and expression values. We will then overlay the SNP 
associations with the chromatin associations to see if we can annotate
SNPs with chromatin state functions. This function performs the imputation
for all genes we have data for.

```{r set_param}
num.states = 14
gene.buffer = 1000 #base pair buffer around gene when looking for SNPs
snp.alleles <- c("A", "C", "G", "T") #alleles of SNPs
save.every = 100
is.interactive = FALSE
#is.interactive = TRUE

females_only = FALSE
chow_only = TRUE
```

```{r load_libraries}
library(here)
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_other_lib}
needed.packages <- c("curl", "stringr", "qtl2", "abind", "parallel", "doParallel")
load_libraries(needed.packages)
```

```{r read_data}
gene.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
strain.table <- as.matrix(read.delim(here("Data", "support_files", "strain.color.table.txt"),
comment.char = "!", header = FALSE))

vars <- load(here("Data", "DOQTL", "Svenson_DO850_for_eQTL_viewer_v9.RData"))
expr <- dataset.mrna$data$rz
covar <- dataset.mrna$covar.matrix

#if not mapping, you can load smaller data sets
#expr.data <- readRDS(here("Data", "DOQTL", "DO_Expression.RDS"))
#expr <- expr.data$expr
#covar <- expr.data$covar

chrom.mats <- readRDS(here("Results", "ChromHMM", paste0(num.states, "_states_C"), 
paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

#transcripts that are shared between chromatin and DO expression data
common.transcripts <- intersect(colnames(expr), names(chrom.mats)) 
transcript.info.table <- gene.info[match(common.transcripts, gene.info[,"ensembl_gene_id"]),]
```


```{r fun}
table_to_snp_mat <- function(snp.table, snp.alleles, strain.key, strain.locale){
    #The SNP matrix holds genotypes for each strain at 
    #each position in the gene. Each position is in a row
    #A 1 indicates that the strain at that position has 
    #the particular allele
    if(length(snp.table) == 0){return(NA)}
    snp.mat <- array(0, dim = c(nrow(snp.table), 8, 4), 
        dimnames = list(snp.table[,"Position"], LETTERS[1:8], snp.alleles))
    strain.names <- gsub("/", "_", str_trim(strain.key[,2]))
    for(i in 1:nrow(snp.table)){
        ref.allele <- which(snp.alleles == snp.table[i,"Ref"])
        snp.mat[i,,ref.allele] <- 1 #start with everyone having the reference allele
        alt.allele <- trim_alt_allele(snp.table[i,strain.locale])
        alt.strain <- strain.table[match(names(alt.allele), strain.names),5]
        #take out DBA, because this isn't in the DO
        alt.strain <- alt.strain[which(!is.na(alt.strain))]
        #set the strains with SNPs to 0 for the referece allele 
        snp.mat[i,alt.strain,ref.allele] <- 0
        #and set the alternate allele to 1 for these strains
        snp.mat[i,alt.strain, alt.allele] <- 1
    }
    return(snp.mat)
}

get_nearest_marker <- function(chr, start.pos){
    if(is.na(chr)){return(NA)}
    if(chr == "MT"){return(NA)}
    chr.locale <- which(names(map) == chr)
    nearest.marker <- get.nearest.pt(map[[chr.locale]], start.pos/1e6)
    marker.geno <- genoprobs[[chr.locale]][,,nearest.marker]
    return(marker.geno)
}

trim_alt_allele <- function(alt_allele, allowable_alleles = snp.alleles){
    allele.names <- names(alt_allele)
    caps_allele <- str_to_upper(alt_allele)
    names(caps_allele) <- allele.names
    allowable_locale <- which(caps_allele %in% allowable_alleles)
    trimmed_allele <- caps_allele[allowable_locale]
    return(trimmed_allele)
}


snp_geno <- function(gene.snp.array, haplotype.matrix, gene.chr, perm.order = NULL){
    
    if(length(gene.snp.array) == 1){return(NA)}
    if(length(haplotype.matrix) == 1){return(NA)}

    if(!is.null(perm.order)){
        #reassign haplotypes if requested.
        new.snps <- gene.snp.array[,perm.order,,drop=FALSE]
        colnames(new.snps) <- colnames(gene.snp.array)
        gene.snp.array <- new.snps
    }

    all.snp <- lapply(1:nrow(gene.snp.array), 
        function(x) haplotype.matrix %*% gene.snp.array[x,,])
    all.snp.mat <- abind(all.snp, along = 3)
    dimnames(all.snp.mat)[[3]] <- paste(gene.chr, rownames(gene.snp.array), sep = "_")
    one.geno <- list(all.snp.mat)

    #add attributes
    names(one.geno) <- gene.chr
    if(gene.chr == "X"){
      attr(one.geno, "is_x_chr") <- TRUE
    }else{
      attr(one.geno, "is_x_chr") <- FALSE
    }
    attr(one.geno, "crosstype") <- "do"
    attr(one.geno, "alleles") <- dimnames(gene.snp.array)[[3]]
    attr(one.geno, "alleleprobs") <- TRUE
    attr(one.geno, "class") <- c("calc_genoprob", "list")
    return(one.geno)
}

one.gene.snps <- function(i, gene.buffer){
    if(is.na(transcript.info.table[i,"chromosome_name"])){return(NA)}
    if(transcript.info.table[i,"chromosome_name"] == "MT"){return(NA)}
    gene.snps <- get.sanger.snps(transcript.info.table[i,"chromosome_name"], 
        start.pos = transcript.info.table[i,"start_position"] - gene.buffer, 
        end.pos = transcript.info.table[i,"end_position"] + gene.buffer)
    return(gene.snps)
}
```

## Get Genotypes for Nearest Markers

For every gene in our list, pull out the genotype matrix for the marker
nearest the TSS.

```{r set_gene}
gene.marker <- vector(mode = "list", length = length(common.transcripts))
names(gene.marker) <- common.transcripts
for(i in 1:nrow(transcript.info.table)){
    gene.marker[[i]] <- get_nearest_marker(transcript.info.table[i,"chromosome_name"], 
    transcript.info.table[i, "start_position"])
}
```

## Download Sanger SNPs

Get the Sanger SNPs in the region of each gene. Build a SNP matrix
for the 8 founder strains.

```{r snp_tables}
snp.table.file <- here("Results", "SNPs", "Sanger.SNP.Tables.RDS")
snp.tables <- readRDS(snp.table.file)
is.done <- length(snp.tables[[length(snp.tables)]]) > 0
if(!is.done){
    start.idx <- min(which(sapply(snp.tables, length) == 0))
    for(i in start.idx:length(snp.tables)){
        print(i)
        snp.tables[[i]] <- one.gene.snps(i)
        if(i %% 100 == 0){
            saveRDS(snp.tables, snp.table.file)        
        }
    }
    #this loop stops a lot. The following lines
    #are useful for manual restart
    #start.idx <- i
    #start.idx <- i + 1
    names(snp.tables) <- common.transcripts
    saveRDS(snp.tables, snp.table.file)
}
```

```{r snp_arrays}
#convert to binary SNP arrays
snp.by.position.file <- here("Results", "SNPs", "SNP.by.Position.RDS")
if(!file.exists(snp.by.position.file)){
    snp.mats <- vector(mode = "list", length = length(snp.tables))
    #find the columns where strain information is stored
    strain.locale <- unlist(sapply(strain.table[,9], 
        function(x) which(colnames(snp.tables[[1]]) == x)))
    snp.mats <- lapply_pb(snp.tables, 
        function(x) if(length(x) > 1){table_to_snp_mat(snp.table = x, snp.alleles, 
        strain.table, strain.locale)}else{NA})
    saveRDS(snp.mats, snp.by.position.file)
}else{
    snp.mats <- readRDS(snp.by.position.file)
}
```

Derive a set of SNPs for each mouse given the genoprobs and the 
SNPs from Sanger. This array will have SNP estimates for each 
individual by position by SNP allele.

To generate this array we multiply each SNP genotype matrix along the 
genome by the haplotype matrix for the gene.

```{r get_snp_geno}
#conver to SNP arrays and scan.
#The arrays are too big to store, so we
#scan them right away and only keep the 
#LOD scores.

snp.coef.file <- here("Results", "SNPs", "SNP.coef.scores.RDS")
snp.r2.file <- here("Results", "SNPs", "SNP.R2.RDS")

if(females_only){
  snp.coef.file <- gsub(".RDS", "_females.RDS", snp.coef.file)
  snp.r2.file <- gsub(".RDS", "_females.RDS", snp.r2.file)
}
if(chow_only){
  snp.coef.file <- gsub(".RDS", "_chow.RDS", snp.coef.file)
  snp.r2.file <- gsub(".RDS", "_chow.RDS", snp.r2.file)
}

if(!file.exists(snp.r2.file)){
    all.snp.coef <- all.snp.r2 <- vector(mode = "list", length = length(snp.mats))
    names(all.snp.coef) <- names(all.snp.r2) <- common.transcripts
    sink(here("Results", "SNPs", "progress.txt"))
    for(i in 1:length(all.snp.coef)){
        cat(i, ":")
        if(length(snp.mats[[i]]) <= 1){cat(" No SNP mat\n");next()}
        if(nrow(snp.mats[[i]]) > 2000){cat(" SNP mat too big\n");next()}
        snp.array <- snp_geno(gene.snp.array = snp.mats[[i]], 
          haplotype.matrix = gene.marker[[i]], 
          gene.chr = transcript.info.table[i,"chromosome_name"])
        
        if(is.na(snp.array)){cat(" Cannot generate genotype array\n");next()}

        gene.expr <- expr[,which(colnames(expr) == common.transcripts[i]),drop=FALSE]

        if(length(gene.expr) == 0){cat(" No expression\n");next()}

        sub.covar <- covar #duplicate covar in case we want to subset by both sex and diet
        if(females_only){
          sex.idx <- which(covar[,"sexM"] == 0)
        }else{
          sex.idx <- 1:nrow(covar)
        }
        if(chow_only){
          diet.idx <- which(covar[,"diethf"] == 0)
        }else{
          diet.idx <- 1:nrow(covar)
        }
        sub.idx <- intersect(sex.idx, diet.idx)

        adj.expr <- adjust(gene.expr[sub.idx,,drop=FALSE], covar[sub.idx,])
        
        common.ind <- intersect(rownames(adj.expr), rownames(snp.array[[1]]))
        common.expr.locale <- match(common.ind, rownames(adj.expr))
        common.genoprob.locale <- match(common.ind, rownames(snp.array[[1]]))

        regression.models  <- apply(snp.array[[1]], 3, function(x) lm(adj.expr[common.expr.locale,]~x[common.genoprob.locale,]))
        snp.r2.test <- sapply(regression.models, function(x) if(length(x) > 1){summary(x)$adj.r.squared}else{NA})
        all.snp.r2[[i]] <- snp.r2.test
        snp.coef <- sapply(regression.models, function(x) if(length(x) > 1){coef(x)[2:length(coef(x))]}else{rep(NA, 4)})
        all.snp.coef[[i]] <- snp.coef
        cat("\n")
    }
    sink()
    saveRDS(all.snp.coef, snp.coef.file)
    saveRDS(all.snp.r2, snp.r2.file)
}else{
    all.snp.coef <- readRDS(snp.coef.file)
    all.snp.r2 <- readRDS(snp.r2.file)
}
```

## Positions of effects

```{r position_effects}

gene.names <- gene.info[match(names(all.snp.coef), gene.info[,"ensembl_gene_id"]),"external_gene_name"]
centered.coord <- lapply(1:length(gene.names),
  function(x) if(length(all.snp.coef[[x]]) > 1){center.on.feature(gene.names[x], 
  gene.info, as.numeric(sapply(strsplit(colnames(all.snp.coef[[x]]), "_"), function(y) y[2])), 
  feature = "full")}else{NA})

ymin <- min(unlist(all.snp.coef), na.rm = TRUE)
ymax <- max(unlist(all.snp.coef), na.rm = TRUE)

nbins = 15
coord.bins <- segment.region(-0.5, 1.5, nbins)
plot.coord <- coord.bins[1:(length(coord.bins)-1)]
all.coord <- unlist(lapply(centered.coord, function(x) if(length(x) > 1){as.numeric(names(x))}else{NA}))
all.coef <- unlist(lapply(1:length(all.snp.coef), function(x) if(length(centered.coord[[x]]) > 1){colSums(all.snp.coef[[x]], na.rm = TRUE)}else{NA}))

binned.coef <- lapply(1:(length(coord.bins)-1), 
  function(x) all.coef[intersect(which(all.coord >= coord.bins[x]), 
  which(all.coord < coord.bins[(x+1)]))])
#boxplot(binned.coef)

coef.mean <- sapply(binned.coef, function(x) mean(x, na.rm = TRUE))
coef.se <- sapply(binned.coef, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))

plot.new()
plot.window(xlim = c(-0.5, 1.5), ylim = c(min(coef.mean-coef.se, na.rm = TRUE), 
  max(coef.mean+coef.se, na.rm = TRUE)))
plot.poly.xy(poly.top.x = plot.coord, poly.top.y = coef.mean+coef.se, 
  poly.bottom.x = plot.coord, poly.bottom.y = coef.mean-coef.se,
  border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
abline(h = 0)
axis(1)
axis(2)
abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
mtext("Mean DNA Methylation Effect on Transcription in DO", side = 3, line = 1.5)
```

```{r var_exp}
ymin <- min(unlist(all.snp.r2), na.rm = TRUE)
ymax <- max(unlist(all.snp.r2), na.rm = TRUE)

nbins = 15
coord.bins <- segment.region(-0.5, 1.5, nbins)
plot.coord <- coord.bins[1:(length(coord.bins)-1)]
all.coord <- unlist(lapply(centered.coord, function(x) if(length(x) > 1){as.numeric(names(x))}else{NA}))
all.r2 <- unlist(lapply(1:length(centered.coord), function(x) if(length(centered.coord[[x]]) > 1){all.snp.r2[[x]]}else{NA}))

binned.r2 <- lapply(1:(length(coord.bins)-1), 
  function(x) all.r2[intersect(which(all.coord >= coord.bins[x]), 
  which(all.coord < coord.bins[(x+1)]))])

#boxplot(binned.r2);abline(h = 0)

r2.mean <- sapply(binned.r2, function(x) mean(x, na.rm = TRUE))
r2.se <- sapply(binned.r2, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))

plot.new()
plot.window(xlim = c(-0.5, 1.5), ylim = c(min(r2.mean-r2.se, na.rm = TRUE), 
  max(r2.mean+r2.se, na.rm = TRUE)))
plot.poly.xy(poly.top.x = plot.coord, poly.top.y = r2.mean+r2.se, 
  poly.bottom.x = plot.coord, poly.bottom.y = r2.mean-r2.se,
  border = NULL, col = "gray", lwd = 1, new.plot = FALSE)
abline(h = 0)
axis(1)
axis(2)
abline(v = c(0, 1), col = "gray", lty = 2, lwd = 2)
mtext("Mean SNP Variance Explained on Transcription in DO", side = 3, line = 1.5)
```

## Permutations for SNPs

We permuted the SNPs as we did for chromatin state.
We permute the haplotype that is assigned to each SNP 
vector, re-do the imputation, and recalculate the LOD
scores and R2 values.

```{r perm_fun}
#This function and the perm data will be exported to the
#clusters
scan_one_snp <- function(perm.data, p){

    perm.order <- order.mat[p,]
  
    snp.array <- snp_geno(gene.snp.array = perm.data$snp.mat, 
        haplotype.matrix = perm.data$haps, 
        gene.chr = perm.data$chr, perm.order = perm.order)

    if(is.na(snp.array)){return(NA)}

    one.lod <- scan1(snp.array, perm.data$expr)

    common.ind <- intersect(rownames(perm.data$gene.expr), rownames(snp.array[[1]]))
    common.expr.locale <- match(common.ind, rownames(perm.data$gene.expr))
    common.genoprob.locale <- match(common.ind, rownames(snp.array[[1]]))
    regression.models  <- apply(snp.array[[1]], 3, function(x) lm(perm.data$expr[common.expr.locale,]~x[common.genoprob.locale,]))
    one.r2 <- sapply(regression.models, function(x) if(length(x) > 1){summary(x)$adj.r.squared}else{NA})
              
    results <- list("one.lod" = one.lod, 
        "one.r2" = one.r2)

  return(results)
}
```

```{r select_nperm}
nperm = 1000
```


```{r set_perm}
#select a set of permutations for which none of 
#the haplotypes remain in their original position.
#these are called derrangements
perm.mat <- derrangements(8)
#sample a random set of derraangements to use
#for all transcripts
perm.rows <- sample(1:nrow(perm.mat), nperm)
order.mat <- perm.mat[perm.rows,]
```



```{r perm_files}

perm.lod.snp.file <- here("Results", "SNPs", "Perm.SNP.LOD.RDS")
perm.r2.snp.file <- here("Results", "SNPs", "Perm.SNP.R2.RDS")


if(females_only){
  perm.lod.snp.file <- gsub(".RDS", "_females.RDS", perm.lod.snp.file)
  perm.r2.snp.file <- gsub(".RDS", "_females.RDS", perm.r2.snp.file)
}
if(chow_only){
  perm.lod.snp.file <- gsub(".RDS", "_chow.RDS", perm.lod.snp.file)
  perm.r2.snp.file <- gsub(".RDS", "_chow.RDS", perm.r2.snp.file)
}

if(file.exists(perm.lod.snp.file)){
  perm.snp.lod <- readRDS(perm.lod.snp.file)
  perm.snp.r2 <- readRDS(perm.r2.snp.file)

  #figure out where we left off.
  not.na <- which(!is.na(perm.snp.lod[,1]))  
  left.off <- max(not.na)
  start.at <- left.off + 1
}else{
  start.at <- 1
  #if we are at the beginning, intialize all the objects for holding
  #results. 
  perm.snp.lod <- perm.snp.r2 <- matrix(NA, nrow = length(all.snp.r2), ncol = nperm)
  rownames(perm.snp.lod) <- rownames(perm.snp.r2) <- names(all.snp.r2)
}

```


```{r set_cl, warning = FALSE}
save.every = 100
if(start.at < length(all.snp.r2)){
  #set up the clusters if we need to run perutations
  n.cores <- detectCores() - 1
  to.export <- c("scan1", "scan1coef", "scan_one_snp", "order.mat",
    "snp_geno", "abind", "common.expr.locale", "common.genoprob.locale")

  #create and register multiple cores for parallele permutations
  cl <- makeCluster(n.cores)
  registerDoParallel(cl)
  #export everything we need to the cluster
  #this takes time upfront, but the parallelization
  #saves a huge amount of time overall
  clusterExport(cl, to.export) 
}

```


```{r perm_snps}

geno.lod.file <- here("Results", "ChromHMM", "14_states_C", "eQTL.lod.RDS")
geno.lod <- readRDS(geno.lod.file)

lod.thresh = 6
to.perm <- which(geno.lod[,"Additive"] > lod.thresh)
do.perm <- rep(FALSE, length(common.transcripts))
do.perm[to.perm] <- TRUE

if(start.at < length(all.snp.r2)){
  sink(here("Results", "SNPs", "progress.txt"))
  for(tr in start.at:length(perm.snp.lod)){
    cat(paste0(tr, ":"))

    if(!do.perm[tr]){cat(" LOD score < 6\n");next()}

    if(length(snp.mats[[tr]]) == 1){cat(" No SNP matrix\n");next()}
    if(nrow(snp.mats[[tr]]) > 2000){cat(" SNP matrix too large\n");next()}

    gene.expr <- expr[,which(colnames(expr) == common.transcripts[tr]),drop=FALSE]
    adj.expr <- adjust(gene.expr, covar)

    if(nrow(snp.mats[[tr]]) < 2){cat(" Not enough SNPs to test\n");next()}

    sub.covar <- covar #duplicate covar in case we want to subset by both sex and diet

    if(females_only){
      sex.idx <- which(covar[,"sexM"] == 0)
    }else{
      sex.idx <- 1:nrow(covar)
    }
    if(chow_only){
      diet.idx <- which(covar[,"diethf"] == 0)
    }else{
      diet.idx <- 1:nrow(covar)
    }
    sub.idx <- intersect(sex.idx, diet.idx)

    perm.data <- list("snp.mat" = snp.mats[[tr]],
      "haps" = gene.marker[[tr]], 
      "chr" = transcript.info.table[tr,"chromosome_name"],
      "expr" = adj.expr[sub.idx,,drop=FALSE])

    #send the new gene ID to the clusters
    clusterExport(cl, "perm.data")

    #do the permutations in parallel
    perm.snp.results <- parLapply(cl, 1:nperm, 
        function(p) scan_one_snp(perm.data, 
        p = p))

    #check for NAs
    if(!all(sapply(perm.snp.results, length) > 1)){cat(" No data returned from permutations\n"); next()}

    perm.tr.lod <- lapply(perm.snp.results, function(x) x$one.lod)
    perm.tr.r2 <- lapply(perm.snp.results, function(x) x$one.r2)

    #keep the maximum r2 and LOD values from each permutation.
    perm.snp.lod[tr,] <- t(sapply(perm.tr.lod, function(x) max(x[,1])))
    perm.snp.r2[tr,] <- t(sapply(perm.tr.r2, function(x) max(x, na.rm = TRUE)))

    if(tr %% save.every == 0){
        saveRDS(perm.snp.lod, perm.lod.snp.file)
        saveRDS(perm.snp.r2, perm.r2.snp.file)
    }

    cat(" Finished permutations\n")

  } #end looping over transcripts
  #save again at the very end
  saveRDS(perm.snp.lod, perm.lod.snp.file)
  saveRDS(perm.snp.r2, perm.r2.snp.file)

  sink()

  #close the clusters at the end and save the final objects
  stopCluster(cl)

} #end case for needing to do more permutations.
```

## Assess permutations

To assess the permutations, we look at the distribution
of empirical p values. If this distribution is uniform,
then it is the imputation itself that is explaining the
LOD scores we see. If the distribution is skewed toward
0, then the DNA methylation is adding something beyond
the imputation itself.

The plots below show the distributions of p values for 
the R2 statistic. The distribution is weirdly bumpy. 
I'm not sure why there is build-up in discrete bins.

```{r assess_lod_perm, fig.width = 8, fig.height = 8}

geno.lod <- readRDS(here("Results", "ChromHMM", "14_states_C", "eQTL.lod.RDS"))
low.lod <- which(geno.lod[,"Additive"] <= 6)
low.lod.idx <- match(rownames(geno.lod)[low.lod], names(all.snp.r2))

#many of the observed values match the maximum value of the 
#permutations. If we use a cumulative distribution function
#these cases give us a p value of 0 even if the null distribution
#matched the observed value hundreds of times. 
#emp.p() takes that matching into account it counts the number
#of times the permuted values were equal to or more extreme than
#the observed value and divides by the number of permutations.
max.snp.r2 <- sapply(all.snp.r2, function(x) if(!all(is.na(x))){max(x, na.rm = TRUE)}else{NA})
r2.p <- sapply(1:length(all.snp.r2), 
  function(x) if(!all(is.na(perm.snp.r2[x,]))){emp.p(max.snp.r2[x], perm.snp.r2[x,])}else{NA})
names(r2.p) <- names(all.snp.r2)

results.table <- cbind(max.snp.r2, r2.p)

perm.results.file <- here("Results", "SNPs", "Perm_Results.RDS")
if(females_only){
  perm.results.file <- gsub(".RDS", "_females.RDS", perm.results.file)
}
if(chow_only){
  perm.results.file <- gsub(".RDS", "_chow.RDS", perm.results.file)
}

saveRDS(results.table, perm.results.file)

r2.p[low.lod.idx] <- NA
hist(r2.p, main = "R2 p",
  xlab = "p value")
```

The following plots show each test statistic vs. its
p values. In general higher test statistics have lower
p values, but this is not strictly true. 


```{r stat_v_p, fig.width = 8, fig.height = 4}
plot(max.snp.r2, -log10(r2.p+1e-3), xlab = "Variance Explained", 
  ylab = "-log10(p)", main = "R2 vs. R2-log10(p)")
```

```{r example, eval = FALSE}
examples <- which(r2.p == 0)
i <- examples[1]
gene.name = gene.info[which(gene.info[,"ensembl_gene_id"] == names(lod.p)[i]),"external_gene_name"][1]
xlim <- c(min(c(perm.snp.r2[i,], max.snp.r2[i])), max(c(perm.snp.r2[i,], max.snp.r2[i])))
hist_with_points(c(perm.snp.r2[i,], max.snp.r2[i]), 
  col = c(rep("gray", ncol(perm.snp.r2)), "red"), xlab = "LOD", main = gene.name)
abline(v = max.snp.r2[i], col = "red")
```