---
title: "SNPs in DO"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


## Introduction

The purpose of this workflow is to compare the eQTL scans with imputed SNPs
to eQTL scans with chromatin states.

```{r set_param}
num.states = 9
```

```{r load_libraries}
library(here)
```


```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_other_lib}
needed.packages <- c("curl", "stringr", "qtl2", "abind")
load_libraries(needed.packages)
```

```{r read_data}
gene.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
strain.table <- as.matrix(read.delim(here("Data", "support_files", "strain.color.table.txt"),
comment.char = "!", header = FALSE))

expr.data <- readRDS(here("Data", "DOQTL", "DO_Expression.RDS")) 
expr <- expr.data$expr
covar <- expr.data$covar

chrom.mats <- readRDS(here("Results", "ChromHMM", paste0(num.states, "_states_C"), 
paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

chrom.lod.file <- here("Results", "chQTL", "chQTL.lod.RDS")
chrom.lod <- readRDS(chrom.lod.file)

inbred.expr <- readRDS(here("Data", "RNASeq", "Strain_Scaled_C_Expression.RData"))

snp.lod.file <- here("Results", "SNPs", "SNP.LOD.RDS")
```

```{r fun}
get_nearest_marker <- function(chr, start.pos){
    chr.locale <- which(names(map) == chr)
    nearest.marker <- get.nearest.pt(map[[chr.locale]], start.pos)
    marker.geno <- genoprobs[[chr.locale]][,,nearest.marker]
    return(marker.geno)
}

trim_alt_allele <- function(alt_allele){
    alt_allele <- alt_allele[which(alt_allele != "-")]
    alt_allele <- alt_allele[which(alt_allele != "~")]
    alt_allele <- sapply(alt_allele, function(x) strsplit(x, "")[[1]][1])
    allele_names <- names(alt_allele)
    alt_allele <- str_to_upper(alt_allele)
    names(alt_allele) <- allele_names
    return(alt_allele)
}

snp_geno <- function(gene.snp.array, haplotype.matrix, gene.chr){
    all.snp <- lapply(1:nrow(gene.snp.array), 
        function(x) haplotype.matrix %*% gene.snp.array[x,,])
    all.snp.mat <- abind(all.snp, along = 3)
    dimnames(all.snp.mat)[[3]] <- paste(gene.chr, rownames(gene.snp.array), sep = "_")
    one.geno <- list(all.snp.mat)

    #add attributes
    names(one.geno) <- gene.chr
    if(gene.chr == "X"){
      attr(one.geno, "is_x_chr") <- TRUE
    }else{
      attr(one.geno, "is_x_chr") <- FALSE
    }
    attr(one.geno, "crosstype") <- "do"
    attr(one.geno, "alleles") <- dimnames(gene.snp.array)[[3]]
    attr(one.geno, "alleleprobs") <- TRUE
    attr(one.geno, "class") <- c("calc_genoprob", "list")
    return(one.geno)
}
```

We are going to start with a single gene as a test case.

```{r set_gene}
gene.name = "Irf5"
gene.locale <- which(gene.info[,"external_gene_name"] == gene.name)

gene.id <- unique(gene.info[gene.locale,"ensembl_gene_id"])
gene.chr <- unique(gene.info[gene.locale,"chromosome_name"])
gene.start <- unique(gene.info[gene.locale,"start_position"])
gene.end <- unique(gene.info[gene.locale,"end_position"])
gene.buffer = 1000
```

Get the genotypes at the marker nearest the TSS of the gene.

```{r get_geno}
gene.marker <- get_nearest_marker(gene.chr, gene.start)
```

Get the Sanger SNPs in the region of the gene. Build a SNP matrix
for the 8 founder strains.

```{r get_snps}
snp.alleles <- c("A", "C", "G", "T")
gene.snps <- get.sanger.snps(gene.chr, gene.start - gene.buffer, gene.end + gene.buffer)
strain.locale <- unlist(sapply(strain.table[,9], function(x) which(colnames(gene.snps) == x)))

#This SNP matrix holds genotypes for each strain at 
#each position in the gene. Each position is in a row
#A 1 indicates that the strain at that position has 
#the particular allele
snp.mat <- array(0, dim = c(nrow(gene.snps), 8, 4), 
    dimnames = list(gene.snps[,"Position"], LETTERS[1:8], snp.alleles))

strain.names <- gsub("/", "_", str_trim(strain.table[,2]))
for(i in 1:nrow(gene.snps)){
    ref.allele <- which(snp.alleles == gene.snps[i,"Ref"])
    snp.mat[i,,ref.allele] <- 1 #start with everyone having the reference allele
    alt.allele <- trim_alt_allele(gene.snps[i,strain.locale])
    alt.strain <- strain.table[match(names(alt.allele), strain.names),5]
    #set the strains with SNPs to 0 for the referece allele 
    snp.mat[i,alt.strain,ref.allele] <- 0
    #and set the alternate allele to 1 for these strains
    snp.mat[i,alt.strain, alt.allele] <- 1
}
```

Derive a set of SNPs for each mouse given the genoprobs and the 
SNPs from Sanger. This array will have SNP estimates for each 
individual by position by SNP allele.

To generate this array we multiply each SNP genotype matrix along the 
genome by the haplotype matrix for the gene.


```{r get_snp_geno}
all.snp.mat <- snp_geno(snp.mat, gene.marker, gene.chr)
```

```{r map_expression}
gene.expr <- expr[,which(colnames(expr) == gene.id),drop=FALSE]
expr.scan <- scan1(genoprobs = all.snp.mat, pheno = gene.expr, addcovar = covar)
```

Compare to the chromatin based LOD scores generated in 2.1_chromaprobs.Rmd

```{r chrom_lod}
chrom.locale <- which(names(chrom.lod) == gene.id)
gene.chrom.lod <- chrom.lod[[chrom.locale]][,1,drop=FALSE]
```


```{r get_chromatin_matrix}
gene.chrom.idx <- which(names(chrom.mats) == gene.id)
gene.chrom.mat <- chrom.mats[[gene.chrom.idx]]
```

```{r get_inbred_expression_order}
inbred.expr.idx <- which(names(inbred.expr) == gene.id)
gene.inbred.expr <- inbred.expr[[inbred.expr.idx]]
inbred.expr.order <- rownames(gene.inbred.expr)[order(gene.inbred.expr[,1])]
```

```{r overlay_results}
all.pos <- c(as.numeric(rownames(snp.mat)),
                        as.numeric(rownames(gene.chrom.lod)),
                        as.numeric(rownames(gene.chrom.mat)))
global.max.pos <- max(all.pos)
global.min.pos <- min(all.pos)

par(mfrow = c(3,1))
par(mar = c(4,4,4,4))
plot(x = as.numeric(rownames(snp.mat)), expr.scan[,1], 
type = "h", ylab = "SNP association LOD Score", xlab = "Genomic Position", axes = FALSE,
xlim = c(global.min.pos, global.max.pos))
axis(1);axis(2)

par(mar = c(0,4,0,4))
strain.order <- rev(order.strains(inbred.expr.order, colnames(gene.chrom.mat), strain.table))
plot.chrom.mat(t(gene.chrom.mat[,strain.order]), num.states = num.states, 
    xlim = c(global.min.pos, global.max.pos))

par(mar = c(4,4,0,4))
plot(x = as.numeric(rownames(gene.chrom.lod)), gene.chrom.lod[,1], type = "h",
    xlab = "Genomic Position", ylab = "Chromatin LOD Score", axes = FALSE,
    xlim = c(global.min.pos, global.max.pos))
axis(1);axis(2)

```