---
title: "SNPs in DO"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


## Introduction

The purpose of this workflow is to compare the eQTL scans with imputed SNPs
to eQTL scans with chromatin states.

```{r set_param}
num.states = 9
```

```{r load_libraries}
library(here)
```


```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_other_lib}
needed.packages <- c("curl", "stringr", "qtl2", "abind")
load_libraries(needed.packages)
```

```{r read_data}
gene.info <- readRDS(here("Data", "RNASeq", "RNASeq_gene_info.RData"))
strain.table <- as.matrix(read.delim(here("Data", "support_files", "strain.color.table.txt"),
comment.char = "!", header = FALSE))

expr.data <- readRDS(here("Data", "DOQTL", "DO_Expression.RDS")) 
expr <- expr.data$expr
covar <- expr.data$covar

chrom.mats <- readRDS(here("Results", "ChromHMM", paste0(num.states, "_states_C"), 
paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

chrom.lod.file <- here("Results", "chQTL", "chQTL.lod.RDS")
chrom.lod <- readRDS(chrom.lod.file)

inbred.expr <- readRDS(here("Data", "RNASeq", "Strain_Scaled_C_Expression.RData"))

snp.lod.file <- here("Results", "SNPs", "SNP.LOD.RDS")
snp.lod <- readRDS(snp.lod.file)

```

```{r plot_a_few}
for(i in 1:length(snp.lod)){
    gene.id <- common.transcripts[i]
    chrom.mat.idx <- which(names(chrom.mats) == gene.id)
    gene.chrom <- chrom.mats[[chrom.mat.idx]]

    chrom.lod.idx <- which(names(chrom.lod) == gene.id)
    gene.chrom.scan <- chrom.lod[[chrom.lod.idx]]

    snp.lod.idx <- which(names(snp.lod) == gene.id)
    
}

```

```{r get_chromatin_matrix}
gene.chrom.idx <- which(names(chrom.mats) == gene.id)
gene.chrom.mat <- chrom.mats[[gene.chrom.idx]]
```

```{r get_inbred_expression_order}
inbred.expr.idx <- which(names(inbred.expr) == gene.id)
gene.inbred.expr <- inbred.expr[[inbred.expr.idx]]
inbred.expr.order <- rownames(gene.inbred.expr)[order(gene.inbred.expr[,1])]
```

```{r overlay_results}
all.pos <- c(as.numeric(rownames(snp.mat)),
                        as.numeric(rownames(gene.chrom.lod)),
                        as.numeric(rownames(gene.chrom.mat)))
global.max.pos <- max(all.pos)
global.min.pos <- min(all.pos)

par(mfrow = c(3,1))
par(mar = c(4,4,4,4))
plot(x = as.numeric(rownames(snp.mat)), expr.scan[,1], 
type = "h", ylab = "SNP association LOD Score", xlab = "Genomic Position", axes = FALSE,
xlim = c(global.min.pos, global.max.pos))
axis(1);axis(2)

par(mar = c(0,4,0,4))
strain.order <- rev(order.strains(inbred.expr.order, colnames(gene.chrom.mat), strain.table))
plot.chrom.mat(t(gene.chrom.mat[,strain.order]), num.states = num.states, 
    xlim = c(global.min.pos, global.max.pos))

par(mar = c(4,4,0,4))
plot(x = as.numeric(rownames(gene.chrom.lod)), gene.chrom.lod[,1], type = "h",
    xlab = "Genomic Position", ylab = "Chromatin LOD Score", axes = FALSE,
    xlim = c(global.min.pos, global.max.pos))
axis(1);axis(2)

```