---
title: "5.0 Compare Permutations"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---


## Introduction
The purpose of this workflow is to compare the permuted
effects of the different genomic features: chromatin state,
SNPs, and DNA methylation.

```{r libraries}
library(here)
num.states = 14
```

```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE)
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_other_lib}
needed.packages <- c("gprofiler2", "pheatmap")
load_libraries(needed.packages)
```


```{r read_data}
gene.info <- read.csv(here("Data", "DOQTL", "mRNA_Info.csv"))

chrom.results <- readRDS(here("Results", "ChromHMM", paste0(num.states, "_states_C"), "Perm_results.RDS")) 
rrbs.results <- readRDS(here("Results", "RRBS", "Perm_results.RDS"))
snp.results <- readRDS(here("Results", "SNPs", "Perm_results.RDS"))

all.results <- list("Chromatin" = chrom.results, "DNA Methylation" = rrbs.results,
    "SNPs" = snp.results)
common.transcripts <- Reduce("intersect", lapply(all.results, rownames))
matched.idx <- lapply(all.results, 
    function(x) match(common.transcripts, rownames(x)))
feature.pairs <- pair.matrix(1:length(all.results))

matched.results <- lapply(1:length(all.results), 
    function(x) all.results[[x]][matched.idx[[x]],])
names(matched.results) <- names(all.results)
```

```{r alpha}
alpha = 1/1000
```

The following Venn diagram shows the number of 
transcripts that had significant (p = `r alpha`) 
effects from each feature according to the permutations.

DNA methylation actually has the largest number of 
significant transcripts. And there is very little
overlap among the groups.

```{r sig_idx}
sig.idx <- lapply(matched.results, function(x) which(x[,2] <= alpha))
plotVenn(sig.idx)

all.intersections <- lapply(1:nrow(feature.pairs), 
    function(x) common.transcripts[intersect(sig.idx[[feature.pairs[x,1]]], sig.idx[[feature.pairs[x,2]]])])
names(all.intersections) <- apply(feature.pairs, 1, 
    function(x) paste0(names(matched.results)[x[1]], "_", names(matched.results)[x[2]]))

intersected.effects <- lapply(matched.results, function(x) x[match(unlist(all.intersections), rownames(x)),1])
boxplot(intersected.effects)
```

```{r eval = FALSE}
#run Figures.Rmd to load data and code and then check the following

plot_together("ENSMUSG00000007029")

```

```{r enrich}
sig.enrich <- lapply(sig.idx, function(x) gost(names(x), organism = "mmusculus"))
plot.enrichment.group(sig.enrich, transformation = function(x) log(x+1e-6))
```

What are the effect sizes of these significant effects?

```{r lod_comp}
sig.effect <- lapply(1:length(sig.idx), function(x) matched.results[[x]][sig.idx[[x]],1])
names(sig.effect) <- names(matched.results)
boxplot(sig.effect, main = "LOD scores of significant effects")
```

```{r compare}
par(mfrow = c(2,2))
for(i in 1:nrow(feature.pairs)){
    pair1 <- feature.pairs[i,1]
    pair2 <- feature.pairs[i,2]
    plot(all.results[[pair1]][matched.idx[[pair1]],1], 
        all.results[[pair2]][matched.idx[[pair2]],1], 
        xlab = names(all.results)[pair1], ylab = names(all.results)[pair2])
    abline(0,1,col = "red")
}
```

## Compare Degrees of Freedom

Do chromatin state matrices explain more variance
because they have more degrees of freedom?

```{r df}
chrom.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))
chrom.mats <- readRDS(file.path(chrom.dir, paste0("Chromatin_States_", num.states, "_full_gene_1000.RData")))

methyl.mat.file <- here("Results", "RRBS", "Aligned.Methyl.Mats.RDS")
methyl.mats <- readRDS(methyl.mat.file)

snp.by.position.file <- here("Results", "SNPs", "SNP.by.Position.RDS")
snp.mats <- readRDS(snp.by.position.file)

gene.names <- gene.info[match(common.transcripts, gene.info[,1]),"symbol"]

for(tr in 1:length(common.transcripts)){
    chrom.idx <- which(names(chrom.mats) == common.transcripts[tr])
    methyl.idx <- which(names(methyl.mats) == gene.names[tr])
    snp.idx <- which(names(snp.mats) == common.transcripts[tr])

    if(length(chrom.idx) > 0){
        chrom.df <- apply(chrom.mats[[chrom.idx]], 1, function(x) length(unique(x))-1)
    }
    if(length(methyl.idx)){
        rounded.mat <- round(methyl.mats[[methyl.idx]]/50)*50
        methyl.df <- apply(rounded.mat, 2, function(x) length(unique(x))-1)
    }
    if(length(snp.idx) > 0){
        snp.df <- apply(snp.mats[[snp.idx]], 1, function(x) length(which(colSums(x) > 0))-1)
    }

}

```