

## Differential Expression
Look for differential expression between treatment and control.

```{r diff_exp}

u_strains <- unique(strains)
c.locale <- which(treat.label == "C")
t.locale <- which(treat.label == "T")

c.strains <- lapply(u_strains, function(x) intersect(which(strains == x), c.locale))
names(c.strains) <- u_strains

t.strains <- lapply(u_strains, function(x) intersect(which(strains == x), t.locale))
names(t.strains) <- u_strains

both.treat <- which(sapply(t.strains, length) > 0)
c.comp <- c.strains[both.treat]
t.comp <- t.strains[both.treat]

c.expr <- lapply(c.comp, function(x) assay(vsd)[,x])
t.expr <- lapply(t.comp, function(x) assay(vsd)[,x])
```

The following figure shows example gene expression with significant differences
in expression upon treatment with dexamethasone.

```{r example_diff, fig.width = 7, fig.height = 5}
gene.idx <- 42
c.gene.expr <- lapply(c.expr, function(x) x[gene.idx,])
t.gene.expr <- lapply(t.expr, function(x) x[gene.idx,])
plot.grouped.boxes(list(c.gene.expr, t.gene.expr), group.labels = c("Control", "Treatment"))
```

```{r, test_all, warning = FALSE, message = FALSE, error = FALSE}
p.file <- here("results", "Diff.Expr.p.val.RDS")
mean.file <- here("results", "Diff.Expr.Mean.RDS")

if(!file.exists(p.file)){
    all.gene.p <- matrix(NA, nrow = nrow(assay(vsd)), ncol = length(c.comp))
    colnames(all.gene.p) <- names(c.comp)
    rownames(all.gene.p) <- rownames(assay(vsd))

    all.gene.mean <- vector(mode = "list", length = nrow(assay(vsd)))
    names(all.gene.mean) <- rownames(assay(vsd))

    grps <- as.factor(c(rep("C", 3), rep("T", 3)))
    for(i in 1:nrow(assay(vsd))){
        if(is.interactive){report.progress(i, nrow(assay(vsd)))}
        c.gene.expr <- lapply(c.expr, function(x) x[i,])
        t.gene.expr <- lapply(t.expr, function(x) x[i,]) 

        c.gene.mean <- sapply(c.gene.expr, mean)
        t.gene.mean <- sapply(t.gene.expr, mean)

        diff.table <- rbind(c.gene.mean, t.gene.mean)
        all.gene.mean[[i]] <- diff.table
        #plot.grouped.boxes(list(c.gene.expr, t.gene.expr), group.labels = c("Control", "Treatment"))
        grp.expr <- lapply(1:length(c.gene.expr), function(x) c(c.gene.expr[[x]], t.gene.expr[[x]]))
        grp.tests <- sapply(grp.expr, function(x) anova(aov(x~grps))$"Pr(>F)"[1])
        all.gene.p[i,] <- grp.tests
    }
    saveRDS(all.gene.p, p.file)
    saveRDS(all.gene.mean, mean.file)
}else{
all.gene.p <- readRDS(p.file)
all.gene.mean <- readRDS(mean.file)
}
```

```{r volcano, fig.width = 12, fig.height = 6}
log.p <- -log10(all.gene.p)
diff.mag <- t(sapply(all.gene.mean, function(x) x[1,] - x[2,]))

par(mfrow = c(2,3))
for(i in 1:ncol(log.p)){
    plot(diff.mag[,i], log.p[,i], main = colnames(diff.mag)[i],
    xlab = "Expression Difference", ylab = "-log10(p)")
}
```

The following figures show the distributions of p values for differential expression
compared to the uniform distribution. These distributions look great.

```{r qqplot, fig.width = 12, fig.height = 6}
par(mfrow = c(2,3))
for(i in 1:ncol(all.gene.p)){
    qqunif.plot(as.numeric(all.gene.p[,1]))
}
```

```{r sig}
gene.fdr <- apply(all.gene.p, 2, function(x) p.adjust(as.numeric(x), "fdr"))
#sig.genes <- rownames(assay(vsd))[unique(which(gene.fdr < 0.1, arr.ind = TRUE)[,1])]
sig.genes <- rownames(assay(vsd))[unique(which(all.gene.p < 0.01, arr.ind = TRUE)[,1])]

enrich <- gprofiler(sig.genes, organism = "mmusculus", src_filter = "GO")
plot.enrichment.vis(enrich, num.terms = 20, order.by = "p.value")

treat.diff <- t(sapply(all.gene.mean, function(x) x[2,] - x[1,]))
strain.upreg <- apply(treat.diff, 2, function(x) rownames(assay(vsd))[which(x > 0)])
strain.downreg <- apply(treat.diff, 2, function(x) rownames(assay(vsd))[which(x < 0)])

upreg.sig <- lapply(strain.upreg, function(x) intersect(x, sig.genes))
downreg.sig <- lapply(strain.downreg, function(x) intersect(x, sig.genes))

saveRDS(upreg.sig, here("results", "Sig.Upregulated.RDS"))
saveRDS(downreg.sig, here("results", "Sig.Downregulated.RDS"))
```

The following figure shows the shared significantly upregulated genes among the strains.

```{r upreg_venn}
vt <- plotVenn(upreg.sig)
```

The following figure shows the shared significantly downregulated genes among the strains.

```{r downreg_venn}
vt <- plotVenn(downreg.sig)
```

The following figure shows the enrichment for significantly upregulated genes by strain.

```{r upgreg_enrich}
enrich.upreg <- lapply(upreg.sig, function(x) gprofiler(x, organism = "mmusculus", src_filter = "GO"))
plot.enrichment.group(enrich.upreg, n.terms = 30, sort.by = "p.value")

all.enrich.upreg <- gprofiler(Reduce("intersect", upreg.sig), organism = "mmusculus", src_filter = "GO")
plot.enrichment.vis(all.enrich.upreg, order.by = "p.value", num.terms = 30)
```

The following figure shows the enrichment for significantly upregulated genes by strain.

```{r downgreg_enrich}
enrich.downreg <- lapply(downreg.sig, function(x) gprofiler(x, organism = "mmusculus", src_filter = "GO"))
enrich.mat <- plot.enrichment.group(enrich.downreg, n.terms = 30, sort.by = "p.value")
pheatmap(cor(enrich.mat))

all.enrich <- gprofiler(Reduce("intersect", downreg.sig), organism = "mmusculus", src_filter = "GO")
plot.enrichment.vis(all.enrich, order.by = "p.value", num.terms = 30)
```



```{r chrom_bin_LOD}
get_cor_at_lod_bin <- function(lod.min, lod.max, sorted.lod, sorted.cor){
    gene.idx <- intersect(which(sorted.lod >= lod.max), which(sorted.lod >= lod.min))
    gene.cor <- sorted.cor[gene.idx]
    return(gene.cor)
}

#bin LOD scores, look at correlations of genes with chromatin state
#in each bin.
lod.seq <- seq.int(min(all.cis.lod), max(all.cis.lod), length.out = 100)
par(mfrow = c(2,4))
for(s in 1:num.states){
    binned.chrom.cor <- lapply(1:(length(lod.seq)-1), 
    function(x) get_cor_at_lod_bin(lod.seq[x], lod.seq[(x+1)], all.cis.lod, 
    all.prop.r[common.chrom.locale,s]))
    chrom.cor.med <- sapply(binned.chrom.cor, function(x) median(abs(x), na.rm = TRUE))
    #boxplot(binned.chrom.cor, main = paste("State", s))
    plot(lod.seq[-1], chrom.cor.med, main = paste("State", s), xlab = "eQTL LOD score",
    ylab = "Abs. Median State Correlation", type = "l", ylim = c(0, 0.85))
}
```



## Comparing Genetics and Chromatin

What is the correlation between cis LOD scores and the correlation between
chromatin state and expression?

The following plots show the absolute value of the correlation between each 
transcript in the inbred mice and its expression vs. the LOD score in the DO for
the same transcript. We expect that if chromatin and genetics are providing redundant
information that this two measures would be very highly correlated. They are actually
positively correlated for all states, particularly for states 1 and 7. 

However, there are transcripts whose chromatin is very highly correlated with 
inbred expression but have low LOD scores. There are also many genes with very
high LOD scores but very low correlation between chromatin state and inbred expression.

```{r chrom_lod_cor}

#get the cis lod for each transcript
get_cis_lod <- function(eqtl.results){
    trans.start <- eqtl.results$gene.info$Start.Mbp * 1e6
    marker.coord <- as.numeric(sapply(strsplit(rownames(eqtl.results$DO.coefficients), "_"), function(x) x[2]))
    nearest.marker <- get.nearest.pt(marker.coord, trans.start)
    nearest.lod <- eqtl.results$LOD[nearest.marker]
    return(nearest.lod)
}

common.genes <- intersect(names(do.eqtl), rownames(all.prop.r))
common.eqtl.locale <- match(common.genes, names(do.eqtl))
common.chrom.locale <- match(common.genes, rownames(all.prop.r))

cis.lod.file <- here("Data", "DOQTL", "cis.eqtl.lod.RDS")
if(!file.exists(cis.lod.file)){
    all.cis.lod <- unlist(lapply(do.eqtl[common.eqtl.locale], get_cis_lod))
    saveRDS(all.cis.lod, cis.lod.file)
}else{
    all.cis.lod <- readRDS(cis.lod.file)
}

lod.thresh <- 0
lod.locale <- which(all.cis.lod >= lod.thresh)
par(mfrow = c(2,4))
for(i in 1:ncol(all.prop.r)){
    plot.with.model(all.cis.lod[lod.locale], abs(all.prop.r[common.chrom.locale[lod.locale],i]), 
    main = paste("State", i), xlab = "|r| Chromatin vs. Expression",
    ylab = "eQTL LOD score", report = "cor.test")
}
```

The following plot shows the same as the above plots, except for the correlation 
between the overall matrix and inbred expression. For this test, we dimension-reduced
the chromatin matrix to a single dimension using multi-dimensional scaling and correlated
that dimension with inbred expression. This correlation is higher than any of the correlation
with any of the individual states. 

```{r mds_model}
plot.with.model(all.cis.lod, abs(unlist(all.mds.r)[common.chrom.locale]), report = "cor.test",
xlab = "eQTL LOD Score", ylab = "|r| MDS chromatin v. Expression")
```

## Low Lod High Chromm {.tabset .tabset-fade .tabset-pills}

Let's spot-check a few of these genes.

```{r spot_check, results = "asis"}

find_genes <- function(sorted.cor, sorted.lod, lod.lim, cor.lim){
    lod.locale <- intersect(which(sorted.lod >= lod.lim[1]), which(sorted.lod <= lod.lim[2]))
    cor.locale <- intersect(which(sorted.cor >= cor.lim[1]), which(sorted.cor <= cor.lim[2]))
    inter.locale <- intersect(lod.locale, cor.locale)
}

min.lod <- 0; max.lod <- 5; min.cor <- 0.9; max.cor <- 1
sorted.cor <- unlist(all.mds.r)[common.chrom.locale]
trimmed.names <- sapply(strsplit(names(sorted.cor), ".", fixed = TRUE), function(x) x[1])
names(sorted.cor) <- trimmed.names
sorted.lod <- unlist(all.cis.lod)
low.lod.high.cor <- trimmed.names[find_genes(sorted.cor, sorted.lod, c(min.lod, max.lod), c(min.cor, max.cor))]
low.lod.high.cor.names <- rnaseq.gene.info[match(low.lod.high.cor, rnaseq.gene.info[,"ensembl_gene_id"]), "external_gene_name"]

for(g in 1:length(low.lod.high.cor)){
    cat("###", low.lod.high.cor.names[g], "\n")
    plot.one.gene.chromatin(do.eqtl, rnaseq.gene.info, rna.seq, all.bed, col.table, 
    gene.name = low.lod.high.cor.names[g], T.or.C, start.feature, end.feature, upstream.buffer, 
    downstream.buffer, separate.windows = is.interactive, dim.rd = "mds", state = NULL, 
    total.states = num.states, show.state.numbers = FALSE)
    cat("\n")
}

```
