
## Correlation with MDS
Also calculate the correlation between the multi-dimensionally scaled
state matrix and gene expression. This won't tell us which states
have good correlations with expression, but will tell us overall
how well the states capture expression.

The barplot below shows the median aboslute value of the correlation 
between the MDS matrix and expression. All numbers of states are very
similar, but the five-state model has a slight edge. 

```{r mds}
all.mds <- vector(mode = "list", length = length(state.num))
names(all.mds) <- basename(state.num)[state_order]
for(i in 1:length(state.num)){
	mds.file <- list.files(path = state.num[state_order[i]], pattern = "MDS_Expression_Cor", full.names = TRUE)
	if(length(mds.file) > 0){all.mds[[i]] <- readRDS(mds.file)}
}

all.mds.r <- lapply(all.mds, function(x) signif(unlist(x$all.r), 2))
all.mds.p <- lapply(all.mds, function(x) x$all.p)

mds.mat <- Reduce("cbind", all.mds.r)
mds.mat <- apply(mds.mat, 2, abs)
colnames(mds.mat) <- names(all.mds)
barplot(apply(mds.mat, 2, function(x) median(x, na.rm = TRUE)), las = 2)
```

We also looked gene by gene for which model had the best correlation.
The following barplot shows the number of times each model had the 
best correlation for the genes. The lowest number of states is clearly
the best most of the time.

```{r best_state}
best.state <- unlist(apply(mds.mat, 1, function(x) if(all(is.na(x))){NA}else{which.max(x)}))
best.table <- table(best.state)
names(best.table) <- names(all.mds)
par(mar = c(10, 4, 4, 4))
barplot(best.table, las = 2)
```

## DNA methylation density by methylation percent

We also looked individually at the density of sites that were 0, 50, or 100% methylated.
All are sparse in the gene body and dense at the TSS. There are no differences in patterns
across the different types, just in degree.

This also can't distinguish from sites that have variable methylation percentages
across strains, so it might not be a good measure. 

```{r sub_methyl_dens}
rounded.methyl <- lapply(methyl.mats, function(x) round(x/50)*50)


bins <- c(0, 50, 100)
for(i in 1:length(bins)){
  sub_methyl_idx <- lapply(rounded.methyl, function(x) unique(which(x == bins[i], arr.ind = TRUE)[,2]))
  sub_methyl_consec <- lapply(1:length(sub_methyl_idx), function(x) if(length(sub_methyl_idx[[x]]) > 1){consec.pairs(as.numeric(colnames(rounded.methyl[[x]])[sub_methyl_idx[[x]]]))}else{NA})
  sub_methyl_dist <- lapply(sub_methyl_consec, function(x) if(length(x) >  1){x[,2] - x[,1]}else{NA})
  
  for(j in 1:length(methyl_dist)){
    if(length(sub_methyl_idx[[j]]) > 1 && !is.null(colnames(norm_rrbs[[j]]))){
      consec.coord <- consec.pairs(as.numeric(colnames(norm_rrbs[[j]])[sub_methyl_idx[[j]]]))
      mean.coords <- rowMeans(consec.coord)
      names(sub_methyl_dist[[j]]) <- mean.coords
    }
  }

 avg_sub_methyl_dist <- plot.centered.vals(val.list = sub_methyl_dist, 
    seq.by = seq.by, min.representation = 10, ylim = c(0, 10000),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position", return.means = FALSE)
}

if(is.interactive){quartz()}
methyl_dens_file <- here("Results", "RRBS", "Methylation_Density.RDS")
if(!file.exists(methyl_dens_file) || delete.previous){
  avg_methyl_dist <- plot.centered.vals(val.list = methyl_dist, 
    seq.by = seq.by, min.representation = 10, ylim = c(0, 3500),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position", return.means = FALSE)
  abline(v = c(0,1))
  saveRDS(avg_methyl_dist, methyl_dens_file)
}else{
  avg_methyl_dist <- readRDS(methyl_dens_file)
  plot.centered.mat(avg_methyl_dist, ylim = c(0, 3500),
    plot.label = paste("Average Distance to Next Methylation Position"),
    ylab = "Distance to Next Methylation Position")
      abline(v = c(0,1))
}
```




