---
title: "Figures"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

## Introduction

The purpose of this workflow to house all the code for generating 
figures for the manuscript. It reads in data generated by other
markdown files and builds figures.

```{r set_param}
library(here)
param <- read.delim(here("Data", "support_files", "param.txt"), header = FALSE, 
    stringsAsFactors = FALSE, row.names = 1)
num.states = as.numeric(param["num_states",])
ordered.marks <- strsplit(param["ordered_marks",], ",")[[1]]

is.interactive = FALSE
#is.interactive = TRUE

data.dir <- here("Data", "ChromHMM", paste0(num.states, "_states_C"))
results.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))

#use.states <- c(1,2,5,7,13,14,15,16) #choose to display only a subset of states
use.states <- 1:num.states
#default.mar <- par()$mar
```


```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE, pattern = ".R")
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
needed.packages <- c("pheatmap", "RColorBrewer", "gridExtra", "grid")
load_libraries(needed.packages)
```


```{r read_info}
key.file <- here("Data", "support_files", "strain.color.table.txt") #generated by hand
col.table <- as.matrix(read.table(key.file, sep = "\t", comment.char = "%", 
    stringsAsFactors = FALSE))
state.col <- read.table(here("Data", "support_files", "state_colors.txt"), 
    comment.char = "%")[,1] #generated by hand
```

## Figure 1 - Gene expression PCs

The first two principle components of the hepatocyte transcriptome
across nine inbred strains of mouse. Each point is an individual 
mouse. Strain is represented by color. 

```{r pc_plot}
rna.seq.file <- here("Data", "RNASeq", "StrainsEffCts9_norm.RDS") #generated by 1.1_Expression.Rmd
rna.seq <- readRDS(rna.seq.file)
#boxplot(rna.seq, las = 2)

layout.mat <- matrix(c(1,2), nrow = 1)
col.order <- match.order(substr(colnames(rna.seq), 1, 2), col.table[,1], col.table)
pdf(here("Documents", "3.Manuscript", "Figures", "Fig1_Decomposition.pdf"), width = 5, height = 5)
expr.decomp <- plot.decomp(t(rna.seq), cols = col.table[col.order,3], cex = 1.5)
legend("bottomright", legend = col.table[,2], pch = 16, col = col.table[,3])
dev.off()
```

## Figure 2 - Chromatin state overview

Overview of chromatin state composition, genomic distribution,
and effect on expression. The left most panel shows the emission
probabilites for each histone modification in each chromatin state. 
Blue indicates the absence of the histone modification, and red 
indicates the presence of the modification. The panel labeled
genomic enrichment shows the distribution of each state around 
functional elements in the genome. Red indicate that the
state is more likely to be found near the annotated functional 
element than expected by chance. Blue indicates that the state
is less likely to be found near the annotated functional element
than expected by chance. Abbreviations are as follows: TFBS = 
transcription factor binding sites, cCRE = candidate cis-regulatory
elements [32728249], TSS = transcription start site, TES = transcription 
end site. The panel labeled Expression Effect shows the effect of the 
presence of each state on gene expression across mouse strains. Positive
values indicate that strains with more of that state had higher gene 
expression than strains with less of the state. The final column of the
figure shows plausible annotations for each state based on combining the 
data in the previous three panels. 

```{r overview_read_data}
#emissions
emissions.mat <- as.matrix(read.delim(file.path(data.dir, paste0("emissions_", num.states, ".txt")), 
    row.names = 1))
emission.order <- match(ordered.marks, colnames(emissions.mat))

#enrichments
enrichment.mat <- as.matrix(read.delim(here("Results", "Enrichments", 
    paste0(num.states, "_states_C_enrichments.txt")), row.names = 1))
enrichment.col <- c("Intergenic" = "Genome..", "CpG Island" = "CpGIsland.mm10.bed.gz", 
    "cCRE" = "Encode_cCRE.bed.gz", "TES" = "RefSeqTES.mm10.bed.gz", "TSS" = "RefSeqTSS.mm10.bed.gz", 
    "TFBS" = "TFbs.bed.gz", "Enhancers" = "enhancers.bed.gz", "Promoters" = "promoters.bed.gz")
col.idx <- match(enrichment.col, colnames(enrichment.mat))
sub.enrichment <- enrichment.mat[use.states,col.idx]
col.order <- hclust(dist(t(sub.enrichment)))$order
#scaled.enrichment <- log10(sub.enrichment[,col.order])
scaled.enrichment <- t(apply(sub.enrichment[,col.order], 1, scale))

#across-strain effects
all.state.effects <- readRDS(here("Results", "ChromHMM", "across.strain.models.RDS")) #generated by 1.3_Compare_ChromHMM_States.Rmd
state.effects <- all.state.effects[[which(names(all.state.effects) == num.states)]]

annotations <- as.matrix(read.delim(here("Data", "support_files", "annotation.table.txt"),
stringsAsFactors = FALSE)) 
```

```{r annotate_states}
#this function annotates the states based on the emission matrix.
annotate_state <- function(state, annotations, thresh = 0.3){
    has.mark <- which(state >= thresh)
    no.mark <- which(state < thresh)

    has.mark.idx <- Reduce("intersect", lapply(names(has.mark), function(x) which(annotations[,x] == "1")))
    no.mark.idx <- Reduce("intersect", lapply(names(no.mark), function(x) which(annotations[,x] == "0")))
    if(length(has.mark) > 0 && length(no.mark) > 0){
        state.idx <- intersect(has.mark.idx, no.mark.idx)
    }
    if(length(has.mark) == 0){
        state.idx <- no.mark.idx
    }
    if(length(no.mark) == 0){
        state.idx <- has.mark.idx
    }
    #pheatmap(all.states[state.idx,])
    return(annotations[state.idx,"annotations"])
}

state.annotations <- rev(apply(emissions.mat, 1, function(x) annotate_state(x, annotations, 0.3)))
```

```{r overview_fig}

layout.mat <- matrix(c(0,1,2,3,4,0,5,6,0,0,7,8,9,10,11), nrow = 3, byrow = TRUE)

pdf(here("Documents", "3.Manuscript", "Figures", "Fig2_Overview.pdf"), width = 10, height = 6)
#quartz(width = 10, height = 6)
layout(layout.mat, widths = c(0.2, 0.3, 0.5, 0.7, 0.3), heights = c(0.1, 0.02, 1))
#layout.show(11)
#1-4. state titles
par(mar = c(0,0,0,0))
plot.text("Modification", cex = 1.5)
plot.text("Genomic Enrichment", cex = 1.5)
plot.text("Expression Effect", cex = 1.5)
plot.text("Annotation", cex = 1.5)

#5-6, scales
par(mar = c(0,0,0,0))
imageWithTextColorbar(emissions.mat, global.color.scale = TRUE, global.min = 0,
global.max = 1, use.pheatmap.colors = TRUE, orientation = "h", cex = 1, n.ax.ticks = 3,
axis.line = 0)

par(mar = c(0,2,0,2))
enrichment.bounds  <- get_plot_bounds(min(scaled.enrichment), max(scaled.enrichment), 
    scale.factor = 10, return.even = TRUE, center = TRUE)
imageWithTextColorbar(scaled.enrichment, use.pheatmap.colors = TRUE, 
global.color.scale = TRUE, global.min = enrichment.bounds[1], 
global.max = enrichment.bounds[2], orientation = "h", cex = 1, 
n.ax.ticks = 3, axis.line = 0)

#7 state labels
par(mar = c(8,0,2,0))
plot.new()
plot.window(xlim = c(0,1), ylim = c(0,1))
text.y <- segment.region(0, 1, length(use.states), "ends")
text(x = rep(0.95, length(use.states)), y = text.y, adj = 1, cex = 1.5,
    labels = paste("State", rev(use.states)), col = rev(state.col[use.states]), font = 2)

#8 emissions matrix
par(mar = c(8,0,2,1))
imageWithText(emissions.mat[,emission.order], global.color.scale = TRUE, global.min = 0,
global.max = 1, use.pheatmap.colors = TRUE, show.text = FALSE, 
row.names = rep("", nrow(emissions.mat)), col.text.shift = -0.7, col.text.cex = 1.5)

#9 enrichment matrix
par(mar = c(8,1,2,0))
imageWithText(scaled.enrichment, 
    use.pheatmap.colors = TRUE, show.text = FALSE, col.names = names(enrichment.col)[col.order],
    row.names = rep("", length(use.states)), col.text.shift = -0.7, col.text.cex = 1.5)

#10 effects
par(xpd = FALSE)
par(mar = c(8,1,2,1))
ylim  <- get_plot_bounds(min(state.effects[,"slope.lwr"]), max(state.effects[,"slope.upr"]), 
    scale.factor = 100, return.even = TRUE, center = TRUE)
ymin <- ylim[1]
ymax <- ylim[2]
full.range <- ymax - ymin
effect.grid <- round(segment.region(ymin, ymax, 8, "ends"), 2)
plot.new()
plot.window(ylim = c(1,num.states), xlim = c(ymin, ymax))
#add grid lines
abline(h = 1:num.states, v = effect.grid, col = "gray80")
state.y <- length(use.states):1

for(state in 1:nrow(state.effects)){

    line.col <- state.col[use.states][state]

    points(y = state.y[state], x = state.effects[state,"slope.fit"],
    col = line.col, cex = 1.5, pch = 16)

    segments(y0 = state.y[state], x0 = state.effects[state,"slope.lwr"],
    x1 = state.effects[state, "slope.upr"], lwd = 4, 
    col = line.col) #full range

    segments(y0 = state.y[state]-0.2, y1 = state.y[state] + 0.2, 
    x0 = state.effects[state,"slope.upr"], lwd = 4, 
    col = line.col) #upper bound

    segments(y0 = state.y[state]-0.2, y1 = state.y[state] + 0.2, 
    x0 = state.effects[state,"slope.lwr"], lwd = 4, 
    col = line.col) #lower bound

    }
abline(v = 0)
axis(1, at = effect.grid)
mtext("State Effect", side = 1, line = 3)

#11 Annotations
par(xpd = TRUE)
par(mar = c(8,0,2,0))
plot.new()
plot.window(xlim = c(0,1), ylim = c(0,1))
text(x = rep(-0.03, length(use.states)), y = text.y, adj = 0, cex = 1.4,
    labels = state.annotations[use.states], col = rev(state.col[use.states]), font = 2)
par(xpd = FALSE)
dev.off()

```

## Figure 3 - Epigenetic state abundance

Relative abundance of chromatin states and methylated DNA. A. Each panel shows
the abundance of a single chromatin state relative to gene TSS and TES. The 
$y$-axis in each panel is the proportion of genes containing the state. Each
panel has an independent $y$-axis to better show the shape of each curve.
The $x$-axis is the relative gene position. The TSS and TES are marked as vertical
gray dashed lines. B. The same data shown in panel A, but with all states overlayed
onto a single $x$- and $y$-axis to show the relative abundance of the states. 
C. The density of CpG sites relative to the gene body. The $y$-axis shows the 
distance, in base pairs, to the next CpG site. This number goes down to almost
0 near the TSS showing that CpG sites are very densely packed in this region. 
CpG sites are less dense within the gene body than in the intergenic space. 
The blue polygon shows the 95\% confidence interval around the estimate. D. 
Percent methylation relative to the gene body. The $y$-axis shows the median 
percent methylation at CpG sites, and the $x$-axis shows relative gene position. 
CpG sites near the TSS are unmethylated relative to intragenic and intergenic
CpG sites.

```{r abundance_data}
state.abundance.file <- file.path(results.dir, "Chromatin.State.Position.Far.Genes.RDS") #generated in 1.4_Chromatin_States_and_Expression.Rmd
state.abundance <- readRDS(state.abundance.file)

methyl.dens.file <- here("Results", "RRBS", "Methylation_Density.RDS") #generated in 3.1.1_RRBS_Analysis.Rmd
avg.methyl.dist <- readRDS(methyl.dens.file)

strain.methyl.file <- here("Results", "RRBS", "Strain.RRBS.Percent.RDS") #generated in 3.1.1_RRBS_Analysis.Rmd
strain.methyl.mat <- readRDS(strain.methyl.file)
```

```{r abundance_plot}
chrom.layout <- 1:(length(use.states)+1)
big.panel.len <- floor((length(use.states)+1)/3)
big.panel.idx <- (length(use.states)+2):(length(use.states)+4)
other.layout <- rep(length(use.states)+2, (length(use.states)+1))

start.y <- 1

for(i in 1:3){
    end.y <- start.y + big.panel.len - 1
    panel.layout <- layout.with.inset(nrow = (length(use.states)+1), ncol = 1, 
    inset.x.min = 1, inset.x.max = 1, inset.y.min = start.y, inset.y.max = end.y)
    other.layout[which(panel.layout == 2)] <- rev(big.panel.idx)[i]
    start.y <- end.y + 1
}
layout.mat <- cbind(chrom.layout, other.layout)

xmin = -0.5; xmax = 1.5

pdf(here("Documents", "3.Manuscript", "Figures", "Fig3_Abundance.pdf"), height = 9, width = 6.5)

panel.height <- 1/(length(use.states)+1)
layout(layout.mat, widths = c(0.5, 1), heights = c(panel.height/2, rep(panel.height, length(use.states))))

# add TSS/TES labels for chromatin state plots
par(mar = c(0.5,3,0.5,2))
plot.new()
plot.window(xlim = c(xmin, xmax), ylim = c(0,1))
par(xpd = TRUE)
text(x = 0, y = 0, labels = "TSS", font = 2, col = "gray40", cex = 1.5)
text(x = 1, y = 0, labels = "TES", font = 2, col = "gray40", cex = 1.5)
par(xpd = FALSE)

#chromatin state abundance
for(s in 1:length(state.abundance)){
    state.avg <- state.abundance[[s]]$state.avg
    state.pos <- state.abundance[[s]]$position
    
    #trim
    idx <- intersect(which(state.pos >= xmin), which(state.pos <= xmax))
    state.avg <- state.avg[idx]
    state.pos <- state.pos[idx]

    max.abundance <- max(state.avg, na.rm = TRUE)
    min.abundance <- min(state.avg, na.rm = TRUE)
    abundance.range <- max.abundance - min.abundance

    ymin <- (min.abundance-(abundance.range*0.08))
    ymax <- (max.abundance + (abundance.range*0.08))

    plot(state.pos, state.avg, type = "l", ylim = c(ymin, ymax), 
        col = state.col[s], xlim = c(-0.5, 1.5), lwd = 4, axes = FALSE)
    
    draw.rectangle(-0.57, 1.57, ymin, ymax, border.col = "gray80", lwd = 2)
    #axis(2)
    abline(v = c(0,1), col = "darkgray", lty = 2, lwd = 3)
    mtext(side = 2, paste("State", s), cex = 1, line = 0.5)
}

#chromatin state abundance all together
all.abund <- t(sapply(state.abundance, function(x) x$state.avg))[,idx]
ymax <- ceiling(max(all.abund, na.rm = TRUE)*10)/10

par(mar = c(2,4,2,2))
plot.new()
plot.window(xlim = c(-0.5, 1.5), ylim = c(0,ymax))
segments(x0 = c(0,1), x1 = c(0,1), y0 = c(0,0), y1 = c(ymax*0.95, ymax*0.95), 
    lty = 2, col = "gray80", lwd = 4)
for(s in use.states){ #ordered for ease of visualization
    points(state.abundance[[s]]$position, y = state.abundance[[s]]$state.avg, 
    col = state.col[s], type = "l", lwd = 4)
}
axis(2, cex.axis = 1.5)
mtext("Proportion", side = 2, line = 3, cex = 1.5)
text(x = 0, y = ymax, labels = "TSS", font = 2, col = "gray40", cex = 2)
text(x = 1, y = ymax, labels = "TES", font = 2, col = "gray40", cex = 2)

#DNA methylation density
par(mar = c(2,4,1,2))
dist.pos <- as.numeric(colnames(avg.methyl.dist))
dist.avg <- apply(avg.methyl.dist, 2, function(x) median(x, na.rm = TRUE))
dist.se <- apply(avg.methyl.dist, 2, function(x) sd(x, na.rm = TRUE)/sqrt(length(which(!is.na(x)))))
ytop <- dist.avg + dist.se
plot.new()
plot.window(xlim = c(xmin, xmax), ylim = c(0, (max(ytop)*1.05)))
plot.poly.xy(poly.top.x = dist.pos, poly.top.y = dist.avg+dist.se, 
    poly.bottom.x = dist.pos, poly.bottom.y = dist.avg-dist.se, 
    col = "lightblue", new.plot = FALSE, border = NA)
points(dist.pos, dist.avg, col = "gray40", type = "l", lwd = 4)
segments(x0 = c(0,1), x1 = c(0,1), y0 = c(0,0), y1 = c(max(ytop)*0.95, max(ytop)*0.95), 
    lty = 2, col = "gray80", lwd = 3)
text(x = 0, y = max(ytop), labels = "TSS", font = 2, col = "gray40", cex = 2)
text(x = 1, y = max(ytop), labels = "TES", font = 2, col = "gray40", cex = 2)
axis(2, cex.axis = 1.5)
mtext("Distance (pb)", side = 2, line = 3, cex = 1.5)

#DNA methylation percent
perc.pos <- as.numeric(colnames(strain.methyl.mat[[1]]))
methyl.mean.perc <- rowMeans(sapply(strain.methyl.mat, function(x) apply(x, 2, function(y) median(y, na.rm = TRUE))))
methyl.mean.se <- rowMeans(sapply(strain.methyl.mat, function(x) apply(x, 2, function(y) sd(y, na.rm = TRUE)/sqrt(length(which(!is.na(y)))))))

ytop <- methyl.mean.perc + methyl.mean.se
plot.new()
plot.window(xlim = c(xmin, xmax), ylim = c(0, (max(ytop)*1.05)))
plot.poly.xy(poly.top.x = perc.pos, poly.top.y = methyl.mean.perc+methyl.mean.se,
    poly.bottom.x = perc.pos, poly.bottom.y = methyl.mean.perc-methyl.mean.se, 
    col = "lightblue", new.plot = FALSE, border = NA)
points(perc.pos, methyl.mean.perc, col = "gray40", type = "l", lwd = 4)
segments(x0 = c(0,1), x1 = c(0,1), y0 = c(0,0), y1 = c(max(ytop)*0.95, max(ytop)*0.95), 
    lty = 2, col = "gray80", lwd = 3)
text(x = 0, y = (max(ytop)*1.05), labels = "TSS", font = 2, col = "gray40", cex = 2)
text(x = 1, y = (max(ytop)*1.05), labels = "TES", font = 2, col = "gray40", cex = 2)
axis(2, cex.axis = 1.5)
mtext("Percent Methylated", side = 2, line = 3, cex = 1.5)
dev.off()
```

## Figure 4 - Epigenetic Effects

Effects of chromatin states on gene expression. Each column shows the effect
of each chromatin state on gene expression in a different context. The first
column shows the effect across genes in the inbred mice showing how chromatin
states are used within a single organism to increase the expression of some genes
and decrease the expression of other genes. The second column shows the effect
of chromatin state on gene expression across strains, showing how variation in
chromatin state across strains leads to variation in expression of individual
genes across strains. The third column shows the effect of imputed chromatin 
state on gene expression in a population of diversity outbred mice showing the
effect of variation in chromatin state across genetically diverse individuals
on local gene expression. Each column of panels is plotted on a single scale
for the $y$-axis so the magnitude of the effects in a single column can be 
compared directly to each other. Across a single row, the scale of the $y$-axis 
varies to highlight the similarity of the shape of each curve in each different 
setting. The final column shows the annotation of each state for comparison with
its effects on gene expression. All $y$-axes is the $\beta$ coefficient from the 
linear model shown in equation \ref{eqn:linear}. All $x$-axes show the relative 
position along the gene body running from just upstream of the TSS to just downstream 
of the TES. Vertical gray dashed lines mark the TSS and TES in all panels.

```{r read_effect_data}
#within-strain coefficients
within.strain.window.state.file <- file.path(results.dir, "State.Window.Effects.Within.Strains.RDS") #generated in 1.4_Chromatin_States_and_Expression.Rmd
within.strain.window.results  <- readRDS(within.strain.window.state.file)

#across-strain coefficients
across.strain.window.state.file <- file.path(results.dir, "State.Window.Effects.Across.Strains.RDS") #generated in 1.4_Chromatin_States_and_Expression.Rmd
across.strain.window.results <- readRDS(across.strain.window.state.file)

#imputed coefficients
ch.coef <- readRDS(file.path(results.dir, "Chromatin_State_DO_Coef.RDS")) #generated in 2.2_chromaprobs_analysis.Rmd

methyl.effect.file <- here("Results", "RRBS", "RRBS.Expr.Cor.Across.Strains.RDS") #generated in 3.1.1_RRBS_Analysis.Rmd
methyl.effect <- readRDS(methyl.effect.file)

gene.windows <- readRDS(here("Data", "support_files", "gene_windows.RDS"))
```

```{r panel_layout}
label.row <- 0:4
tss.row <- c(0,5,6,7,0)
num.label.panels <- last.panel <- max(tss.row)
panel.col <- vector(mode = "list", length = 4)
for(i in 1:5){
    next.panel <- last.panel + 1
    one.col <- next.panel:((i*num.states)+num.label.panels)
    panel.col[[i]] <- one.col
    last.panel <- max(one.col)
}

layout.mat <- rbind(label.row, tss.row, Reduce("cbind", panel.col))
```

```{r state_effects_figure}
if(is.interactive){quartz(height = 9, width = 6.5)}

pdf(here("Documents", "3.Manuscript", "Figures", "Fig4_Effects.pdf"), width = 6.5, height = 9)
xmin = -0.2; xmax = 1.2

panel.height <- 1/(length(use.states)+2)
layout(layout.mat, heights = c(rep(panel.height/3, 2), rep(panel.height, length(use.states))),
widths = c(0.2, rep(1, 4)))
#layout.show(max(layout.mat))
#plot column labels
par(mar = c(0,1,0,1))
plot.text("Within-Strain", font = 2, col = "gray40", cex = 1.5)
plot.text("Across-Strain", font = 2, col = "gray40", cex = 1.5)
plot.text("Outbred", font = 2, col = "gray40", cex = 1.5)
plot.text("Annotation", font = 2, col = "gray40", cex = 1.5)

#plot TSS/TES labels
par(mar = c(1,1,0,1))
par(xpd = NA)
for(i in 1:3){
    plot.new()
    plot.window(xlim = c(xmin, xmax), ylim = c(0,1))
    text(x = 0, y = 0, labels = "TSS", font = 2, col = "gray40", cex = 1.2)
    text(x = 1, y = 0, labels = "TES", font = 2, col = "gray40", cex = 1.2)
}
par(xpd = FALSE)

#plot state labels
par(mar = c(0,0,0,0))
for(i in 1:length(use.states)){
    plot.text(paste("State", use.states[i]), srt = 90, col = "gray40", font = 2, cex = 1.3)
}

#plot within-strain effects
par(mar = c(0,0,0,0))
all.lower.y <- sapply(within.strain.window.results, function(x) x[,"slope.lwr"])
all.upper.y <- sapply(within.strain.window.results, function(x) x[,"slope.upr"])
#make one y axis for all plots, center it so 0 line lines up across
#the full row
ylim <- get_plot_bounds(min(all.lower.y), max(all.upper.y), scale.factor = 100, 
    return.even = TRUE, center = TRUE)
ymin <- ylim[1]
ymax = ylim[2]

window.mids <- sapply(gene.windows, max)
for(state in use.states){
    state.effects <- sapply(within.strain.window.results, function(x) x[state,])
    min.effect <- min(state.effects["slope.lwr",which(is.finite(state.effects["slope.lwr",]))])
    max.effect <- max(state.effects["slope.upr",which(is.finite(state.effects["slope.upr",]))])
    plot.new()
    plot.window(xlim = c(xmin, xmax), ylim = c(ymin, ymax))
    points(window.mids, state.effects["slope.fit",], col = state.col[state], type = "l")
    plot.poly.xy(window.mids, state.effects["slope.lwr",],
    window.mids, state.effects["slope.upr",], col = state.col[state])
    segments(xmin, 0, xmax, col = "gray40", lwd = 2)
    abline(v = c(0,1), lty = 2, col = "darkgray", lwd = 2)
    draw.rectangle(xmin, xmax, ymin, ymax, border.col = "gray80", lwd = 2)
}

#plot across-strain effects
all.lower.y <- sapply(across.strain.window.results, function(x) x[,"slope.lwr"])
all.upper.y <- sapply(across.strain.window.results, function(x) x[,"slope.upr"])
#make one y axis for all plots, center it so 0 line lines up across
#the full row
ylim <- get_plot_bounds(min(all.lower.y), max(all.upper.y), scale.factor = 100, 
    return.even = TRUE, center = TRUE)
ymin <- ylim[1]
ymax = ylim[2]

for(state in use.states){
    state.effects <- sapply(across.strain.window.results, function(x) x[state,])
    min.effect <- min(state.effects["slope.lwr",which(is.finite(state.effects["slope.lwr",]))])
    max.effect <- max(state.effects["slope.upr",which(is.finite(state.effects["slope.upr",]))])
    plot.new()
    plot.window(xlim = c(xmin, xmax), ylim = c(ymin, ymax))
    points(window.mids, state.effects["slope.fit",], col = state.col[state], type = "l")
    plot.poly.xy(window.mids, state.effects["slope.lwr",],
    window.mids, state.effects["slope.upr",], col = state.col[state])
    segments(xmin, 0, xmax, col = "gray40", lwd = 2)
    abline(v = c(0,1), lty = 2, col = "darkgray", lwd = 2)
    draw.rectangle(xmin, xmax, ymin, ymax, border.col = "gray80", lwd = 2)
}

# plot DO effects
idx <- lapply(all.state.coef, function(x) intersect(which(as.numeric(colnames(x)) >= xmin), which(as.numeric(colnames(x)) <= xmax)))
all.state.mean <- lapply(1:length(all.state.coef), function(x) colMeans(all.state.coef[[x]][,idx[[x]]], na.rm = TRUE))
all.state.se <- lapply(1:length(all.state.coef), function(x) apply(all.state.coef[[x]][,idx[[x]]], 2, function(y) sd(y, na.rm = TRUE)/sqrt(length(which(!is.na(y))))))
all.lower.y <- unlist(all.state.mean) - unlist(all.state.se)
all.upper.y <- unlist(all.state.mean) + unlist(all.state.se)
#make one y axis for all plots, center it so 0 line lines up across
#the full row
ylim <- get_plot_bounds(min(all.lower.y), max(all.upper.y), scale.factor = 100, 
    return.even = TRUE, center = TRUE)
ymin <- ylim[1]
ymax = ylim[2]

for(state in use.states){
    plot.new()
    plot.window(xlim = c(xmin, xmax), ylim = c(ymin, ymax))
    coef.pos <- as.numeric(names(all.state.mean[[state]]))
    #points(coef.pos, all.state.mean[[state]], col = state.col[state], type = "l")
    plot.poly.xy(coef.pos, all.state.mean[[state]]-all.state.se[[state]],
    coef.pos, all.state.mean[[state]]+all.state.se[[state]], col = state.col[state])
    segments(xmin, 0, xmax, col = "gray40", lwd = 2)
    abline(v = c(0,1), lty = 2, col = "darkgray", lwd = 2)
    draw.rectangle(xmin, xmax, ymin, ymax, border.col = "gray80", lwd = 2)
}  

# plot annotations
par(mar = c(0,0,0,0))
for(state in use.states){
    #plot.text(rev(state.annotations)[state], x = 0, y = 0.5, adj = 0, cex = 1.4, col = rev(state.col[state]), font = 2)
    plot.text(rev(state.annotations)[state], x = 0.5, y = 0.85, adj = 0.5, cex = 1.3, 
    col = "gray40", font = 2)
    draw.rectangle(0, 1, 0, 1, border.col = "gray80", lwd = 2)
}
dev.off()

```




```{r within_strain_methyl_fig, eval = FALSE}
upstream.limit <- -0.5
downstream.limit <- 1.5
min.fit = -0.01
max.fit = 0.01

idx <- intersect(which(rel.pos >= upstream.limit), which(rel.pos <= downstream.limit))

pdf(here("Documents", "3.Manuscript", "Figures", "DNA_methylation_within_strain.pdf"), 
  height = 5, width = 6)
plot.new()
plot.window(xlim = c(upstream.limit, downstream.limit), ylim = c(min.fit, max.fit)) 
plot.poly.xy(rel.pos[idx], all.lwr[idx], rel.pos[idx], all.upr[idx], col = "gray80", 
  border = "gray80")
points(rel.pos[idx], all.fit[idx], type = "l", col = "gray40", lwd = 2)
abline(v = c(0,1), lwd = 4, col = "gray60", lty = 2)
abline(h = 0, col = "gray40", lty = 1, lwd = 4)
#mtext(side = 2, "Effect", cex = 2, line = 0.5)
draw.rectangle(upstream.limit, downstream.limit, min.fit, max.fit, 
  border.col = "gray80", lwd = 2)
dev.off()
```


```{r across_strain_methyl_fig, eval = FALSE}
upstream.limit <- -0.5
downstream.limit <- 1.5
min.fit = -0.01
max.fit = 0.01

idx <- intersect(which(rel.pos >= upstream.limit), which(rel.pos <= downstream.limit))

pdf(here("Documents", "3.Manuscript", "Figures", "DNA_methylation_across_strains.pdf"), 
  height = 5, width = 6)
plot.new()
plot.window(xlim = c(upstream.limit, downstream.limit), ylim = c(min.fit, max.fit)) 
plot.poly.xy(rel.pos[idx], stat.mat[idx,"slope.lwr"], rel.pos[idx], stat.mat[idx,"slope.upr"],
col = "gray80", border = "gray80")
points(rel.pos[idx], stat.mat[idx,"slope.fit"], type = "l", col = "gray40", lwd = 2)
abline(v = c(0,1), lwd = 4, col = "gray60", lty = 2)
abline(h = 0, col = "gray40", lty = 1, lwd = 4)
#mtext(side = 2, "Effect", cex = 2, line = 0.5)
draw.rectangle(upstream.limit, downstream.limit, min.fit, max.fit, 
  border.col = "gray80", lwd = 2)
dev.off()
```
