---
title: "Figures"
author: Anna L Tyler
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    code_folding: hide
    collapsed: no
    toc: yes
    toc_float: yes
---

## Introduction

The purpose of this workflow to house all the code for generating 
figures for the manuscript. It reads in data generated by other
markdown files and builds figures.

```{r set_param}
library(here)
param <- read.delim(here("Data", "support_files", "param.txt"), header = FALSE, 
    stringsAsFactors = FALSE, row.names = 1)
num.states = as.numeric(param["num_states",])
ordered.marks <- strsplit(param["ordered_marks",], ",")[[1]]

is.interactive = FALSE
#is.interactive = TRUE

data.dir <- here("Data", "ChromHMM", paste0(num.states, "_states_C"))
results.dir <- here("Results", "ChromHMM", paste0(num.states, "_states_C"))

#use.states <- c(1,2,5,7,13,14,15,16) #choose to display only a subset of states
use.states <- 1:num.states
#default.mar <- par()$mar
```


```{r source_code}
all.code.dir <- list.files(here("Code"), full.names = TRUE)
for(i in 1:length(all.code.dir)){
	all.fun <- list.files(all.code.dir[i], full.names = TRUE, pattern = ".R")
	for(j in 1:length(all.fun)){source(all.fun[j])}
}
```

```{r load_libraries, message = FALSE, warning = FALSE, error = FALSE}
needed.packages <- c("pheatmap", "RColorBrewer", "gridExtra", "grid")
load_libraries(needed.packages)
```


```{r read_info}
key.file <- here("Data", "support_files", "strain.color.table.txt")
col.table <- as.matrix(read.table(key.file, sep = "\t", comment.char = "%", 
    stringsAsFactors = FALSE))
state.col <- read.table(here("Data", "support_files", "state_colors.txt"), 
    comment.char = "%")
```

## Figure 1 - Gene expression PCs

The first two principle components of the hepatocyte transcriptome
across nine inbred strains of mouse. Each point is an individual 
mouse. Strain is represented by color. 

```{r pc_plot}
rna.seq.file <- here("Data", "RNASeq", "StrainsEffCts9_norm.RDS")
rna.seq <- readRDS(rna.seq.file)
#boxplot(rna.seq, las = 2)

layout.mat <- matrix(c(1,2), nrow = 1)
col.order <- match.order(substr(colnames(rna.seq), 1, 2), col.table[,1], col.table)
pdf(here("Documents", "3.Manuscript", "Figures", "Fig_1_Decomposition.pdf"), width = 5, height = 5)
expr.decomp <- plot.decomp(t(rna.seq), cols = col.table[col.order,3], cex = 1.5)
legend("bottomright", legend = col.table[,2], pch = 16, col = col.table[,3])
dev.off()
```

## Figure 2 - Chromatin state overview

Overview of chromatin state composition, genomic distribution,
and effect on expression. The left most panel shows the emission
probabilites for each histone modification in each chromatin state. 
Blue indicates the absence of the histone modification, and red 
indicates the presence of the modification. The panel labeled
genomic enrichment shows the distribution of each state around 
functional elements in the genome. Red indicate that the
state is more likely to be found near the annotated functional 
element than expected by chance. Blue indicates that the state
is less likely to be found near the annotated functional element
than expected by chance. Abbreviations are as follows: TFBS = 
transcription factor binding sites, cCRE = candidate cis-regulatory
elements [32728249], TSS = transcription start site, TES = transcription 
end site. The panel labeled Expression Effect shows the effect of the 
presence of each state on gene expression across mouse strains. Positive
values indicate that strains with more of that state had higher gene 
expression than strains with less of the state. The final column of the
figure shows plausible annotations for each state based on combining the 
data in the previous three panels. 

```{r overview}
emissions.mat <- as.matrix(read.delim(file.path(data.dir, paste0("emissions_", num.states, ".txt")), 
    row.names = 1))
emission.order <- match(ordered.marks, colnames(emissions.mat))

enrichment.mat <- as.matrix(read.delim(here("Results", "Enrichments", 
    paste0(num.states, "_states_C_enrichments.txt")), row.names = 1))
enrichment.col <- c("Intergenic" = "Genome..", "CpG Island" = "CpGIsland.mm10.bed.gz", 
    "cCRE" = "Encode_cCRE.bed.gz", "TES" = "RefSeqTES.mm10.bed.gz", "TSS" = "RefSeqTSS.mm10.bed.gz", 
    "TFBS" = "TFbs.bed.gz", "Enhancers" = "enhancers.bed.gz", "Promoters" = "promoters.bed.gz")
col.idx <- match(enrichment.col, colnames(enrichment.mat))
sub.enrichment <- enrichment.mat[,col.idx]
col.order <- hclust(dist(t(sub.enrichment)))$order
scaled.enrichment <- log10(sub.enrichment[,col.order])

layout.mat <- matrix(c(0,1,2,3,4,0,5,6,0,0,7,8,9,10,11), nrow = 3, byrow = TRUE)

quartz(width = 10, height = 6)
layout(layout.mat, widths = c(0.2, 0.3, 0.5, 0.7, 0.3), heights = c(0.1, 0.02, 1))
#layout.show(11)
#1-4. state titles
par(mar = c(0,0,0,0))
plot.text("Modification", cex = 1.5)
plot.text("Genomic Enrichment", cex = 1.5)
plot.text("Expression Effect", cex = 1.5)
plot.text("Annotation", cex = 2)

#5-6, scales
par(mar = c(0,0,0,0))
imageWithTextColorbar_test(emissions.mat, global.color.scale = TRUE, global.min = 0,
global.max = 1, use.pheatmap.colors = TRUE, orientation = "h", cex = 1, n.ax.ticks = 3,
axis.line = 0)

par(mar = c(0,2,0,2))
imageWithTextColorbar(scaled.enrichment, use.pheatmap.colors = TRUE, 
orientation = "h", cex = 1, n.ax.ticks = 3, axis.line = 0)

#7 state labels
par(mar = c(0,0,0,0))
plot.new()
plot.window(xlim = c(0,1), ylim = c(0,1))
text.y <- segment.region(0.95, 0.12, length(use.states), "ends")
text(x = rep(0.95, length(use.states)), y = text.y, adj = 1, cex = 1.5,
    labels = paste("State", use.states), col = state.col[,1], font = 2)

par(mar = c(4,0,2,1))
imageWithText(emissions.mat[,emission.order], global.color.scale = TRUE, global.min = 0,
global.max = 1, use.pheatmap.colors = TRUE, show.text = FALSE, 
row.names = rep("", nrow(emissions.mat)), col.text.shift = -0.7)

par(mar = c(4,1,2,1))
imageWithText(apply(sub.enrichment[1:num.states,col.order], 2, scale), 
    use.pheatmap.colors = TRUE, show.text = FALSE, col.names = names(enrichment.col)[col.order],
    row.names = rep("", length(use.states)), col.text.shift = -0.7)

```
