---
title: Correcting for relatedness in standard mouse mapping populations; and something about epistasis
author:
  - name: Catrina Spruce
    email: Catrina.Spruce@jax.org
    affiliation: The Jackson Laboratory
  - name: Anna L. Tyler
    email: Anna.Tyler@jax.org
    affiliation: The Jackson Laboratory
  - name: Many more people
    email: many more emails
    affiliation: JAX-MG and JAX-GM
  - name: Gregory W. Carter
    email: Gregory.Carter@jax.org
    affiliation: The Jackson Laboratory
    corresponding: Gregory.Carter@jax.org
address:
  - code: The Jackson Laboratory
    address: 600 Main St. Bar Harbor, ME, 04609
abstract: |
  

author_summary: |
  Epigenetics in mice
bibliography: epigenetics.bib
output: rticles::plos_article
csl: plos.csl
---

```{r read_results, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
library(here)
```

# Introduction


# Materials and Methods

## Chromatin Data

## RRBS Data

## Mice

## Data Processing

## Identifying chromatin states

## RRBS processing

# Results

## Eight-state model correlated best with gene expression

## State proportion correlated with gene expression
higher proportions of state 7 were correlated with increased expression
higher proportions of state 2 were correlated with decreased expression
figure shows the number of whose expression was significantly correlated
with each state at a nominal $p$ value of 0.01. State 7 was associated 
with the expression of the largest number of genes.

We see activating and suppressing marks at locations all through the gene.
not necessarily just at the TSS
(check to see if we can get coordinates of different transcripts)

## Spatial distribution of states around gene bodies
Each state had its own distribution around gene bodies.
The following distributions were for a subset of genes. Need to do genes overall
and look at subsets that might be interesting either for regulatory reasons
or GO term enrichment, or something else.


State 1: trough at TSS, climbs through gene body 
State 2: low throughout
State 3: peak at TSS, otherwise low
State 4: low throughout
State 5: trough at TSS, peak after TES, moderate in gene body
State 6: troush at TSS, climbs through gene body, peak after TES
State 7: big spike at TSS, declines through gene body
State 8: slight peak after TSS, declines through gene body 

# Discussion


# Acknowledgements
This work was funded by XXX.

# Data and Software Availability
All data used in this study and the code used to analyze it are avalable as part of 
a reproducible workflow located at... (Figshare?).

CAPE is available at CRAN...

# Supplemental Figure Legends

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first principal component (PC) of
the kinship matrix. Traits with high correlation to the kinship matrix may 
be highly polygenic and thus be susceptible to test statistic inflation due to 
many true positives. To reduce this risk, we selected traits with low correlation
with the first kinship matrix PC. This figure shows the distribution of correlations
between traits and the first kinship PC across populations. 
}
\label{fig:trait_cor}
\end{figure}


\begin{figure}[ht]
\centering
\caption{Reducing $n$ reduces inflation. 
This figure is identical to Fig. \ref{LD} except that we have added a column 
for the F2 that has been subsampled to the same $n$ as the Backcross. This 
subsampling reduces power to detect effects, and thus reduces inflation to 
roughly the same level as that seen in the backcross.
}
\label{subsampled}
\end{figure}

# Supplemental Table Descriptions

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first PC of the kinship matrix.
}
\label{table:trait_cor}
\end{figure}


# References {#references .unnumbered}
