% Template for PLoS
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended
% to minimize problems and delays during our production
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% Once your paper is accepted for publication,
% PLEASE REMOVE ALL TRACKED CHANGES in this file
% and leave only the final text of your manuscript.
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file.
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission.
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.
%
% Do not include text that is not math in the math environment.
%
% Please add line breaks to long display equations when possible in order to fit size of the column.
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
% \usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace}
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
% \bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}


% Pandoc citation processing




\usepackage{forarray}
\usepackage{xstring}
\newcommand{\getIndex}[2]{
  \ForEach{,}{\IfEq{#1}{\thislevelitem}{\number\thislevelcount\ExitForEach}{}}{#2}
}

\setcounter{secnumdepth}{0}

\newcommand{\getAff}[1]{
  \getIndex{#1}{The Jackson Laboratory}
}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Correcting for relatedness in standard mouse mapping
populations; and something about
epistasis} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Catrina Spruce\textsuperscript{\getAff{The Jackson Laboratory}},
Anna L. Tyler\textsuperscript{\getAff{The Jackson Laboratory}},
Many more people\textsuperscript{\getAff{JAX-MG and JAX-GM}},
Gregory W. Carter\textsuperscript{\getAff{The Jackson
Laboratory}}\textsuperscript{*}\\
\bigskip
\textbf{\getAff{The Jackson Laboratory}}600 Main St.~Bar Harbor, ME,
04609\\
\bigskip
* Corresponding author: Gregory.Carter@jax.org\\
\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
The abstract goes here

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step.
% Author Summary not valid for PLOS ONE submissions.
\section*{Author summary}
The author summary goes here

\linenumbers

% Use "Eq" instead of "Equation" for equation citations.
\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

There is evidence that, especially early in life, chromatin
modifications are genetically determined {[}cite{]}.

\hypertarget{materials-and-methods}{%
\section{Materials and Methods}\label{materials-and-methods}}

\hypertarget{mice}{%
\subsection{Mice}\label{mice}}

\hypertarget{inbred-founder-mice}{%
\subsubsection{Inbred Founder Mice}\label{inbred-founder-mice}}

\hypertarget{diversity-outbred-mice}{%
\subsubsection{Diversity Outbred mice}\label{diversity-outbred-mice}}

The genomic features we collected from inbred founders: chromatin state,
percent DNA methylation, and SNPs were imputed into a population of DO
mice based on local haplotypes. These mice were described previously in
(Svenson 2012). The study population included males and females from DO
generations four through eleven. Mice were randomly assigned to either a
chow diet (6\% fat by weight, LabDiet 5K52, LabDiet, Scott Distributing,
Hudson, NH), or a high-fat, high-sucrose (HF/HS) diet (45\% fat, 40\%
carbohydrates, and 15\% protein) (Envigo Teklad TD.08811, Envigo,
Madison, WI). Mice were maintained on this diet for 26 weeks (CITE).

\hypertarget{genotyping}{%
\subsection{Genotyping}\label{genotyping}}

\hypertarget{diversity-outbred-mice-1}{%
\subsubsection{Diversity outbred mice}\label{diversity-outbred-mice-1}}

All DO mice were genotyped as described in Svenson et al.~(2012) using
the Mouse Universal Genotyping Array (MUGA) (7854 markers), and the
MegaMUGA (77,642 markers) (GeneSeek, Lincoln, NE). All animal procedures
were approved by the Animal Care and Use Committee at The Jackson
Laboratory (Animal Use Summary \# 06006).

\hypertarget{measurement-of-gene-expression}{%
\subsection{Measurement of gene
expression}\label{measurement-of-gene-expression}}

\hypertarget{inbred-founders}{%
\subsubsection{Inbred Founders}\label{inbred-founders}}

\hypertarget{diversity-outbred-mice-2}{%
\subsubsection{Diversity outbred mice}\label{diversity-outbred-mice-2}}

At sacrifice, whole livers were collected and gene expression was
measured using RNA-Seq as described in (Chick, Munger et al.~2016, and
Tyler et al.~2017)

\hypertarget{measurement-of-chromatin-modifications}{%
\subsection{Measurement of Chromatin
Modifications}\label{measurement-of-chromatin-modifications}}

\hypertarget{measurement-of-dna-methylation}{%
\subsection{Measurement of DNA
Methylation}\label{measurement-of-dna-methylation}}

Percent DNA methylation was measured using reduced representation
bisulfite sequencing.

\hypertarget{data-processing}{%
\subsection{Data Processing}\label{data-processing}}

\hypertarget{chromatin-modifications}{%
\subsection{Chromatin modifications}\label{chromatin-modifications}}

Annat's stuff to get fastq files to bam files bam to bed binarize bed
files

\hypertarget{dna-methylation}{%
\subsection{DNA methylation}\label{dna-methylation}}

Vivek's stuff to get bed files.

\hypertarget{analysis}{%
\subsection{Analysis}\label{analysis}}

\hypertarget{identification-and-characterization-of-chromatin-states}{%
\subsection{Identification and characterization of chromatin
states}\label{identification-and-characterization-of-chromatin-states}}

We used ChromHMM {[}29120462{]} to identify chromatin states
corresponding to the presence and absence of the four chromatin
modifications. We calculated states for all numbers of states between
four and 16, which is the maximum number of states possible with four
binary chromatin modifications.

\hypertarget{selecting-the-number-of-chromatin-states}{%
\subsubsection{Selecting the number of chromatin
states}\label{selecting-the-number-of-chromatin-states}}

To identify the ChromHMM model that corresponded best with gene
expression, we compared the correlation of each state with gene
expression across all ChromHMM models (Supp Fig. XXX). Across all
models, the correlations between gene expression and chromatin state
could be binned roughly into five bins: low, moderately low, no
correlation, moderately high correlation, and high correlation. The
nine-state model had states in each of these categories with the lowest
redundancy. Furthermore, state seven in the nine-state model had the
maximum correlation with gene expression. Therefore, we chose the
nine-state model for downstream analysis.

\hypertarget{chromatin-state-composition}{%
\subsubsection{Chromatin state
composition}\label{chromatin-state-composition}}

Emissions probabilities derived from ChromHMM indicate the contribution
of each histone modification to each chromatin state (Figure XXX).

\hypertarget{positional-enrichments-of-chromatin-states}{%
\subsubsection{Positional enrichments of chromatin
states}\label{positional-enrichments-of-chromatin-states}}

We used the ChromHMM function OverlapEnrichment to identify correlations
between chromatin state position and annotated functional elements of
the genome. We used functional annotations for the following features:

\begin{itemize}
\tightlist
\item
  \textbf{Transcription start sites (TSS)} - Annotations of TSS in the
  mouse genome were provided by RefSeq {[}26553804{]} and included with
  the release of ChromHMM, which we downloaded on December 9, 2019
  {[}29120462{]}.
\item
  \textbf{Transcription end sites (TES)} - Annotations of TES in the
  mouse genome were provided by RefSeq and included with the release of
  ChromHMM.
\item
  \textbf{Transcription factor binding sites (TFBS)} - We downloaded
  TFBS coordinates from OregAnno {[}26578589{]} using the UCSC genome
  browser {[}12045153{]} on May 4, 2021.
\item
  \textbf{Promoters} - We downloaded promoter coordinates provided by
  the eukaryotic promoter database {[}27899657,25378343{]}, through the
  UCSC genome browser on April 26, 2021.
\item
  \textbf{Enhancers} - We downloaded annotated enhancers provided by
  ChromHMM through the UCSC genome browser on April 26, 2021.
\item
  \textbf{Candidates of cis regulatory elements in the mouse genome
  (cCREs)} - We downloaded cCRE annotations provided by ENCODE
  {[}22955616{]} through the UCSC genome browser on April 26, 2021.
\item
  \textbf{CpG Islands} - Annotations of CpG islands in the mouse genome
  were included with the release of ChromHMM.
\end{itemize}

\hypertarget{downloading-snp-data}{%
\subsection{Downloading SNP data}\label{downloading-snp-data}}

We downloaded SNP data for the eight inbred DO/CC founders from the
Sanger SNP database {[}1921910, 21921916{]} on July 6, 2021. We
downloaded SNPs ranging from 1kb upstream of the TSS to 1kb downstream
of the TES for each gene in our expression data set.

\hypertarget{aligning-positions-relative-to-gene-bodies}{%
\subsection{Aligning positions relative to gene
bodies}\label{aligning-positions-relative-to-gene-bodies}}

For multiple analyses in this paper, we quantified genomic feature
abundance or correlation to gene expression, based on the feature's
relative position to the gene body. To do this, we normalized all gene
coordinates to run from 0 at the transcription start site to 1 at the
transcription end site. Upstream regulatory regions were assigned
negative coordinates and downstream regulatory regions were assigned
coordinates greater than 1. Base pair positions of genomic features were
first centered on the TSS by subtracting the base pair position of the
TSS. Centered positions were then divided by the length of the gene in
base pairs, defined as the distancr from the gene TSS to the gene TES.
These relative positions were grouped into 41 positions defined by the
sequence from -2 to 2 incremented by 0.1. If multiple positions were
grouped together, the mean value across positions was used.

To avoid potential contamination from regulatory regions of nearby
genes, we only included genes that were at least 2kb from their nearest
neighbor, for a final set of 14048 genes.

\hypertarget{assessing-physical-distributions-of-epigenetic-modifications}{%
\subsection{Assessing physical distributions of epigenetic
modifications}\label{assessing-physical-distributions-of-epigenetic-modifications}}

We used the above normalized base pair positions to calculate the
relative abundance of each chromatin state across all gene bodies. For
each relative position, we calculated the proportion of genes that
contained each chromatin state.

We also calculated the median percent DNA methylation across all gene
bodies at each normalized position. To calculate density of CpG sites,
we first used absolute base pair positions to calculate the distance
from each CpG site to the next. We then normalized the positions
relative to gene TSS and TES as described above. The final result was a
distance to the next CpG site in base pairs for each normalized position
across the gene bodies.

\hypertarget{correlating-genomic-features-with-gene-expression-in-inbred-mice}{%
\subsection{Correlating genomic features with gene expression in inbred
mice}\label{correlating-genomic-features-with-gene-expression-in-inbred-mice}}

We correlated both chromatin state and percent DNA methylation with gene
expression in nine strains of inbred mice which included the DO/CC
founders and DBA/2J. We looked for relationships between epigenetic
features and gene expression both within-strain, and across strains.

Within-strain correlations between epigenetic features and gene
expression suggest some role for the epigenetic feature in regulation of
expression. Across-strain correlations of epigenetic features and gene
expression further suggest a role for local genotype in determining the
regulatory epigenetic features.

To assess the within-strain relationship between chromatin state and
gene expression, we scaled transcript levels of all genes in each
strain. We used the normalized base pair positions calculated above to
average expression of all genes assigned to each chromatin state at each
position along the gene body.

Similarly, we calculated the correlation between gene expression and
percent methylation at each normalized position across the gene body.

To identify across-strain correlations between epigenetic modifications
and transcript abundance, we scaled transcript abundance across strains
separately for each transcript. We then correlated this scaled abundance
with either chromatin state or percent DNA methylation at each
normalized position across the gene body.

\hypertarget{imputing-genomic-features-in-diversity-outbred-mice}{%
\subsection{Imputing genomic features in Diversity Outbred
mice}\label{imputing-genomic-features-in-diversity-outbred-mice}}

To further investigate the effect of genetic and epigenetic features on
local gene expression, we imputed chromatin state, SNPs, and DNA
methylation into a population of diversity outbred (DO) mice (CITE).

Each imputation followed the same basic procedure: For each transcript,
we identified the haplotype probabilities in the DO mice at the genetic
marker nearest the gene transcription start site. This matrix held DO
individuals in rows and DO founder haplotypes in columns.

For each transcript, we also generated a three-dimensional array
representing the genomic features derived from the DO founders. This
array held DO founders in rows, feature state in columns, and genomic
position in the third dimension. The feature state for chromatin
consisted of states one through nine, for SNPs feature state consisted
of the genotypes A,C,G, and T, and for DNA methylation, feature state
consisted of percent DNA methylation rounded to the nearest 0\%, 50\%,
or 100\%. Rounding reduced the direct transfer of haplotype effects to
functionally irrelevant variation in DNA methylation across strains at
individual positions.

We then matrix multiplied the haplotype probabilities by each genomic
feature array to obtain the imputed genomic feature for each DO mouse.
This final array held DO individuals in rows, genomic feature in the
second dimension, and genomic position in the third dimension. This
array is analagous to the genoprobs object in R/qtl2 (CITE). The genomic
position dimension included all positions between the transcription
start site and the transcription end site (\(\pm 1kb\)). SNP data for
the DO founders in mm10 coordinates were downloaded from the Sanger SNP
database (CITE), on July 6, 2021.

To calculate the effect of each genomic feature on gene expression, we
fit the following linear model:

\begin{equation*}
y=\beta_0
\end{equation*}

where blah blah blah

From this linear model, we calculated the variance explained (\(R^2\))
by each genomic feature. We thus related gene expression in the DO to
each position of imputed chromatin state, SNPs, or DNA methylation in
and around the gene body.

\hypertarget{results}{%
\section{Results}\label{results}}

\hypertarget{description-of-chromatin-states}{%
\subsection{Description of Chromatin
States}\label{description-of-chromatin-states}}

We identified nine chromatin states corresponding to nine distinct
combinations of histone modifications (Figure XXXA). These states were
differentially distributed near functionally annotated genomic elements
(Figure XXXB). For example, State 1, which corresponded to the absence
of all four histone modifications, found mainly in intergenic regions.
States 5 and 9 were enriched near enhancers and TES respectively.
Finally, states 3 and 7 were highly enriched near the TSS and other
functional elements that also occur near the TSS, such as cis-regulatory
regions (cCREs), transcription factor binding sites (TFBS), and
promoters.

A subset of the chromatin states was also correlated with gene
expression across strains (Figure XXXC) in a manner that concorded with
both their histone modification profiles and their enrichments near
functional genomic elements. For example, the histone modification
H3K27me3 has been previously shown to be associated with reduced
transcription {[}CITE{]}. We found this modification in both states 2
and 3 (Figure XXXA), which were both correlated with reduced
transcription (Figure XXXC). State 3, furthermore, was enriched near the
TSS, promoters, TFBS, and other regulatory elements supporting a
possible role in transcriptional regulation.

State 5 was characterized by the presence of two histone modifications
previously associated with increased transcription: H3K27ac {[}CITE{]},
and H3K4me1 {[}CITE{]}. The enrichment of this state in enhancers
coincides with previous work showing the presence of these two
modifications in active enhancers {[}21106759, 21160473, 29273804{]},
and supports the role of this state in upregulation of gene
transcription.

States 6 and 7 were also associated with increased transcription and the
presence of transcriptionally activating histone modifications. State 6
was modestly enriched in enhancers, while state 7 was enriched not only
in enhancers, but also stongly near the TSS and other associated
functional elements (Figure XXXB).

\hypertarget{spatial-distribution-of-chromatin-states-and-dna-methylation}{%
\subsection{Spatial distribution of chromatin states and DNA
methylation}\label{spatial-distribution-of-chromatin-states-and-dna-methylation}}

In addition to looking for enrichment of chromatin states near annotated
functional elements, we also characterized the relative spatial
distribution around gene bodies of both chromatin states and DNA
methylation. To do this, we scaled base pair positions of our measured
genomic features to run between 0 at the TSS and 1 at the TES of their
containing gene (Methods) (Figure XXX).

Each chromatin state had a characteristic distribution pattern relative
to gene bodies (Figure XXXA and B). For example, state 1 was strongly
depleted near the TSS, indicating that this region is commonly subject
to chromatin modification. However, its abundance increased steadily to
a peak at the TES. In contrast, state 7 was present in over 60\% of TSS,
but decreased to almost 0\% near the TES.

The remaining states were relatively low in abundance compared to states
1 and 7, but also showed specific distributions relative to the gene
body. State 8, was depleted at the TSS, but enriched immediately
downstream of the TSS. State 9 had slight enrichments immediately
upstream of the TSS and immediately downstream of the TES. These
patterns agree with the enrichments around functional genomic elements
shown in Figure XXXB, and add interesting resolution around both the TSS
and TES.

DNA methylation also showed strong positional enrichments (Figure XXXC
and D). Across all genes, the TSS had densely packed CpG sites cytosines
relative to the region between the TSS and TES (Figure XXXC). As
expected, the median CpG site near TSS were consistently hypomethylated
relative to the median CpG site in intergenic regions. CpG sites within
the gene body were slightly hypermethylated compared to intergenic CpGs
(Figure XXXD).

\hypertarget{within-strain-correlation-of-epigenetic-features-and-gene-expression}{%
\subsection{Within-strain correlation of epigenetic features and gene
expression}\label{within-strain-correlation-of-epigenetic-features-and-gene-expression}}

To investigate the relationship between epigenetic modifications and
gene expression, we calculated the relationship both chromatin state and
DNA methylation and transcript abundance (Methods) (Figure XXX). Across
all genes within individual inbred strains of mice, the percent
methylation at CpG sites was negatively correlated with transcript
abundance primarily at the TSS.

The correlation of each chromatin state matched that seen in Figure
XXXC, and the spatial resolution of the relationships added an
interesting dimension to this observation. For example, state 3, which
was overall correlated with lower transcript abundance in Figure XXXC,
was shown to have this negative effect primarily when it was positioned
at the TSS. In contrast, state 2, was also negatively correlated with
transcription overall, but this effect was constant throughout the gene
body. State 2 is characterized by H3K27me3, which has previously been
associated with reduced transcription {[}CITE{]}, but was not enriched
near any annotated functional genomic elements.

\hypertarget{across-strain-correlation-of-epigenetic-features-and-gene-expression}{%
\subsection{Across-strain correlation of epigenetic features and gene
expression}\label{across-strain-correlation-of-epigenetic-features-and-gene-expression}}

To investigate whether the previously identified relationships between
epigenetic features and gene expression were related to local genotype,
we looked for correlations between gene expression and epigenetic
features for each transcript across all inbred strains. That is, for any
given transcript, did variation in chromatin state or DNA methylation
across strains correlate with gene expression?

Interestingly, the relationship between gene expression and chromatin
state across strains was nearly identical to the relationship
within-strain, although positional effects were perhaps even more
pronounced. States 3 and 9 had negative correlations with gene
expression across strains that were localized to the TSS. States 5 and 7
had strong positive correlations with gene expression throughout the
gene body, and states 1 and 2 had strong negative correlations with gene
expression throughout the gene body.

In stark contrast, DNA methylation was completely uncorrelated with
variation in gene expression across strains (Figure XXXB). This lack of
correlation is likely due to the low variability of percent methylation
across strains at any given position. Figure XXXC shows the standard
deviation in percent DNA methylation at normalized positions across the
gene body. It is strikingly low everywhere, with the standard deviation
being around 6\%, which is likely below any biologically functional
threshold. The variation dips even lower, to around 4\% at the TSS,
indicating that for the most part that DNA methylation does not vary
across strains and is not contributing to strain difference in gene
expression.

\hypertarget{imputed-chromatin-state-correlated-with-local-gene-expression-in-diversity-outbred-mice}{%
\subsection{Imputed chromatin state correlated with local gene
expression in Diversity Outbred
mice}\label{imputed-chromatin-state-correlated-with-local-gene-expression-in-diversity-outbred-mice}}

To further investigate the relationship between genotype, epigenetic
features, and gene expression, we imputed chromatin state, DNA
methylation, and SNPs into a population of DO mice described previously
{[}Svenson, Tyler{]}.

We investigated the extent to which chromatin state imputed into DO mice
explained variation in expression across individuals. Although local
genetic variation explains a large amount of variation in gene
expression {[}cite{]}, chromatin state may offer further insight into
regulation of gene expression at the local level. {[}more compelling
stuff here{]}

We imputed genome-wide chromatin states in a population of DO mice based
on their genotype (Methods) and compared the percent variance explained
by local genotype to the maximum percent variance explained by local
chromatin state for each transcript (Figure XXX.) The two measurements
were very tightly correlated (Pearson R = 0.95) indicating that
chromatin state determined by genetics is an excellent approximation of
the genetic effect on gene expression. The imputation further allowed us
to observe the effects of chromatin state across {[}500{]} genetically
diverse mice by measuring chromatin modifications in a handful of inbred
mice. Further, because chromatin modifications are measured at extremely
high density, we can map high-density chromatin effects in the DO mice,
which may help prioritize functional SNPs within gene bodies and in
regulatory regions.

For example, Figure XXX shows chromatin states across the gene Irf5 in
the inbred founders along with the LOD score and chromatin state effects
at each position along the gene body as calculated in the DO population.
The LOD scores and allele effects highlight variation at the TSS, and at
several internal positions in the gene as potentially regulating gene
expression.

\hypertarget{dna-methylation-varied-across-the-gene-body}{%
\subsection{DNA methylation varied across the gene
body}\label{dna-methylation-varied-across-the-gene-body}}

In addition to chromatin state, we examined the distribution of DNA
methylation across the gene body, as well as the relationship between
DNA methylation and gene expression in both inbred mice and DO mice.

As expected, methylated cytosines were densely packed near the gene TSS
(Figure XXX). They were relatively sparse within the gene body, and had
intermediate spacing outside of gene bodies.

Outside of gene bodies, percent methylation was measured at an average
of 50\%, whereas there was very low DNA methylation at the gene TSS
(Figure XXX). Percent DNA methylation within gene bodies was higher than
the surrounding intergenic spaces, reaching a maximum of around 80\%
near the gene TES.

Within each strain, percent methylation at the gene TSS was slightly
negatively correlated with gene expression (Pearson r for all strains
was about -0.2). However, there was very little variation in DNA
methylation across strains, particularly at the TSS, and consequently,
there was no relationship between percent methylation and gene
expression across strains.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

The enrichment of these states in regulatory regions indicates the
possibility that these states are used for regulating expression levels
whereas states 7 and 3 at the transcription start site may be primarily
related to switching gene transcription on and off.

Haplotype and chromatin state represent broader regions of genome than
SNPs and DNA methylation, which are measured at the base pair level. The
measurements that represent larger regions of the genome are more
predictive of local gene expression than the point-wise measurements.

While local haplotype is the best predictor of gene expression, it has
poor resolution. SNPs and DNA methylation have very high resolution, but
are relatively poor predictors of gene expression. Chromatin state sits
in the middle ground. It is almost as good a predictor of gene
expression as haplotype, but has resolution down to 200 base pairs, thus
offering the potential for dissecting mechanisms of local gene
expression at a higher resolution than is possible with haplotype alone.

There is clearly a lot going on at the TSS, but there these results show
correlations between gene expression

Perhaps by overlaying all modalities, particularly with measurements of
open chromatin, we can come up with examples of this kind of inference?
Are there any anecdotes that illustrate this?

Local chromatin state was highly correlated with local gene expression
in the DO/CC founders. This was true across genes within each strain, as
well as for individual genes across strains, suggesting that variation
in chromatin modifications may be a major mechanism of local gene
expression regulation.

(Alternatively, chromatin state aligns well with the true local
mechanism of gene regulation, but is not itself a mechanism.)

\hypertarget{positional-information-is-interesting}{%
\subsection{Positional information is
interesting}\label{positional-information-is-interesting}}

We observed interesting spatial patterns of chromatin state distribution
and correlation with gene expression. States 3 and 7 were particularly
abundant around transcription start sites (TSS), while all other states
were depleted at the TSS. State 8 peaked in abundance immediately
downstream of the TSS, and state 9 peaked immediately upstream of the
TSS.

State 5 had relatively low abundance. However, it was concentrated
within gene bodies where it had a relatively strong positive correlation
with gene expression. This indicates that (?)

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

This work was funded by XXX.

\hypertarget{data-and-software-availability}{%
\section{Data and Software
Availability}\label{data-and-software-availability}}

All data used in this study and the code used to analyze it are avalable
as part of a reproducible workflow located at\ldots{} (Figshare?,
Synapse?).

\hypertarget{supplemental-figure-legends}{%
\section{Supplemental Figure
Legends}\label{supplemental-figure-legends}}

\begin{figure}[ht]
\centering
\caption{
}
\label{fig:trait_cor}
\end{figure}

\hypertarget{supplemental-table-descriptions}{%
\section{Supplemental Table
Descriptions}\label{supplemental-table-descriptions}}

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first PC of the kinship matrix.
}
\label{table:trait_cor}
\end{figure}

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\nolinenumbers


\end{document}
