---
title: Correcting for relatedness in standard mouse mapping populations; and something about epistasis
author:
  - name: Catrina Spruce
    email: Catrina.Spruce@jax.org
    affiliation: The Jackson Laboratory
  - name: Anna L. Tyler
    email: Anna.Tyler@jax.org
    affiliation: The Jackson Laboratory
  - name: Many more people
    email: many more emails
    affiliation: JAX-MG and JAX-GM
  - name: Gregory W. Carter
    email: Gregory.Carter@jax.org
    affiliation: The Jackson Laboratory
    corresponding: Gregory.Carter@jax.org
address:
  - code: The Jackson Laboratory
    address: 600 Main St. Bar Harbor, ME, 04609
abstract: |
  

author_summary: |
  Epigenetics in mice
bibliography: epigenetics.bib
output: rticles::plos_article
csl: plos.csl
---

```{r read_results, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
library(here)
```

# Introduction


# Materials and Methods

## Chromatin Data

## RRBS Data

## Mice

## Data Processing

## Identifying chromatin states

## Imputing chromatin states in DO
To better investigate the effect of chromatin state on local gene expression,
we imputed chromatin state into a diversity outbred (DO) mouse population. 


## RRBS processing

# Results

## Eight-state model correlated best with gene expression

## State proportion correlated with gene expression
Higher proportions of state 7 were correlated with increased expression
both across genes (within-strain), and across strains. For example,
Irf5 was expressed at low levels in PWK and CAST. Both strains had state 3
at the TSS, and low levels of state 7 elsewhere in the gene. The other 
strains expressed Irf5 much more highly. All of these strains had state 7 
at the TSS and at higher levels throughout the rest of the gene. Proportion
of state 7 throughout the gene was positively correlated with expression.

Higher proportions of state 2 were correlated with decreased expression
figure shows the number of whose expression was significantly correlated
with each state at a nominal $p$ value of 0.01. 

State 7 was associated 
with the expression of the largest number of genes.

We see activating and suppressing marks at locations all through the gene.
not necessarily just at the TSS
(check to see if we can get coordinates of different transcripts)

## Spatial distribution of states around gene bodies
Each state was distributed unevenly relative to gene bodies.


State 1: trough at TSS, climbs through gene body 
State 2: pretty low throughout, but slight peak at TSS
State 3: pretty low, but a distinct peak at TSS
State 4: low throughout, but higher in gene body than at TSS
State 5: trough at TSS, maybe a peak upstream of TSS
State 6: very low in gene body
State 7: big spike at TSS, declines through gene body
State 8: slight trough right at TSS, but higher just upstream and downstream of TSS
  another peak after TES.

## Correlation of chromatin state and gene expression was differentially distributed across the gene body
We examined whether there was a spatial component to the correlation between
chromatin state and gene expression. To do this we normalized all gene coordinates
to run from 0 at the transcription start site (TSS) and 1 at the transcription 
end site (TES). We then correlated gene expression to chromatin state in a 
sliding window across the gene body. 

Figure XXX shows the average Pearson correlation between expression and chromatin 
state across all windows for each chromatin state. 

States 1 and 2 generally suppress expression regardless of position, but do seem
to have stronger effects at positions near the TSS. 

State 7 generally enhances expression regardless of position, but does seem to have
a stronger effect at positions near the TSS.

The presence of states 3 and 8 suppress expression primarily when they
are present at the TSS. Big swings upstream and downstream of the gene 
for state 3 are primarily due to small numbers of genes in those groups.

States 5 and 6 are moderately positively correlated with expresion when they
are present within the gene body.

State 4 has very little effect anywhere and is generally not associated with
differential expression.

Overall there is very little effect of any state on expression near the TES 
specifically.

## Chromatin states were aligned with eQTLs
We investigated whether chromatin states provided regulatory information on 
top of that provided by the underlying genetics. 

We imputed chromatin state onto a population of DO mice based on their genotype.
There is good evidence that, especially early in life, chromatin modifications are
genetically determined. Twins chromatin modifications are identical when they are
born and slowly drift apart from each other [cite]. 

With high-resolution chromatin state information, we can potentially look at 
fine-grained regulatory structures within cis eQTL.



# Discussion


# Acknowledgements
This work was funded by XXX.

# Data and Software Availability
All data used in this study and the code used to analyze it are avalable as part of 
a reproducible workflow located at... (Figshare?).

CAPE is available at CRAN...

# Supplemental Figure Legends

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first principal component (PC) of
the kinship matrix. Traits with high correlation to the kinship matrix may 
be highly polygenic and thus be susceptible to test statistic inflation due to 
many true positives. To reduce this risk, we selected traits with low correlation
with the first kinship matrix PC. This figure shows the distribution of correlations
between traits and the first kinship PC across populations. 
}
\label{fig:trait_cor}
\end{figure}


\begin{figure}[ht]
\centering
\caption{Reducing $n$ reduces inflation. 
This figure is identical to Fig. \ref{LD} except that we have added a column 
for the F2 that has been subsampled to the same $n$ as the Backcross. This 
subsampling reduces power to detect effects, and thus reduces inflation to 
roughly the same level as that seen in the backcross.
}
\label{subsampled}
\end{figure}

# Supplemental Table Descriptions

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first PC of the kinship matrix.
}
\label{table:trait_cor}
\end{figure}


# References {#references .unnumbered}
