% Template for PLoS
%DIF LATEXDIFF DIFFERENCE FILE
%DIF DEL Epigenetics_Manuscript_orig.tex   Thu Jun 23 10:09:41 2022
%DIF ADD Epigenetics_Manuscript.tex        Thu Jun 23 12:11:17 2022
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended
% to minimize problems and delays during our production
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% Once your paper is accepted for publication,
% PLEASE REMOVE ALL TRACKED CHANGES in this file
% and leave only the final text of your manuscript.
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file.
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission.
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.
%
% Do not include text that is not math in the math environment.
%
% Please add line breaks to long display equations when possible in order to fit size of the column.
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
% \usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace}
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
% \bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}


% Pandoc citation processing
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
% for Pandoc 2.8 to 2.10.1
\newenvironment{cslreferences}%
  {}%
  {\par}
% For Pandoc 2.11+
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1 \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces\fi
  % set entry spacing
  \ifnum #2 > 0
  \setlength{\parskip}{#2\baselineskip}
  \fi
 }%
 {}
\usepackage{calc} % for calculating minipage widths
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}




\usepackage{forarray}
\usepackage{xstring}
\newcommand{\getIndex}[2]{
  \ForEach{,}{\IfEq{#1}{\thislevelitem}{\number\thislevelcount\ExitForEach}{}}{#2}
}

\setcounter{secnumdepth}{0}

\newcommand{\getAff}[1]{
%DIF 212c212-213
%DIF <   \getIndex{#1}{The Jackson Laboratory}
%DIF -------
  \getIndex{#1}{The Jackson Laboratory for Mammalian Genetics,The %DIF > 
Jackson Laboratory for Genomic Medicine} %DIF > 
%DIF -------
}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF UNDERLINE PREAMBLE %DIF PREAMBLE
\RequirePackage[normalem]{ulem} %DIF PREAMBLE
\RequirePackage{color}\definecolor{RED}{rgb}{1,0,0}\definecolor{BLUE}{rgb}{0,0,1} %DIF PREAMBLE
\providecommand{\DIFaddtex}[1]{{\protect\color{blue}\uwave{#1}}} %DIF PREAMBLE
\providecommand{\DIFdeltex}[1]{{\protect\color{red}\sout{#1}}}                      %DIF PREAMBLE
%DIF SAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddbegin}{} %DIF PREAMBLE
\providecommand{\DIFaddend}{} %DIF PREAMBLE
\providecommand{\DIFdelbegin}{} %DIF PREAMBLE
\providecommand{\DIFdelend}{} %DIF PREAMBLE
\providecommand{\DIFmodbegin}{} %DIF PREAMBLE
\providecommand{\DIFmodend}{} %DIF PREAMBLE
%DIF FLOATSAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddFL}[1]{\DIFadd{#1}} %DIF PREAMBLE
\providecommand{\DIFdelFL}[1]{\DIFdel{#1}} %DIF PREAMBLE
\providecommand{\DIFaddbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFaddendFL}{} %DIF PREAMBLE
\providecommand{\DIFdelbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFdelendFL}{} %DIF PREAMBLE
%DIF HYPERREF PREAMBLE %DIF PREAMBLE
\providecommand{\DIFadd}[1]{\texorpdfstring{\DIFaddtex{#1}}{#1}} %DIF PREAMBLE
\providecommand{\DIFdel}[1]{\texorpdfstring{\DIFdeltex{#1}}{}} %DIF PREAMBLE
\newcommand{\DIFscaledelfig}{0.5}
%DIF HIGHLIGHTGRAPHICS PREAMBLE %DIF PREAMBLE
\RequirePackage{settobox} %DIF PREAMBLE
\RequirePackage{letltxmacro} %DIF PREAMBLE
\newsavebox{\DIFdelgraphicsbox} %DIF PREAMBLE
\newlength{\DIFdelgraphicswidth} %DIF PREAMBLE
\newlength{\DIFdelgraphicsheight} %DIF PREAMBLE
% store original definition of \includegraphics %DIF PREAMBLE
\LetLtxMacro{\DIFOincludegraphics}{\includegraphics} %DIF PREAMBLE
\newcommand{\DIFaddincludegraphics}[2][]{{\color{blue}\fbox{\DIFOincludegraphics[#1]{#2}}}} %DIF PREAMBLE
\newcommand{\DIFdelincludegraphics}[2][]{% %DIF PREAMBLE
\sbox{\DIFdelgraphicsbox}{\DIFOincludegraphics[#1]{#2}}% %DIF PREAMBLE
\settoboxwidth{\DIFdelgraphicswidth}{\DIFdelgraphicsbox} %DIF PREAMBLE
\settoboxtotalheight{\DIFdelgraphicsheight}{\DIFdelgraphicsbox} %DIF PREAMBLE
\scalebox{\DIFscaledelfig}{% %DIF PREAMBLE
\parbox[b]{\DIFdelgraphicswidth}{\usebox{\DIFdelgraphicsbox}\\[-\baselineskip] \rule{\DIFdelgraphicswidth}{0em}}\llap{\resizebox{\DIFdelgraphicswidth}{\DIFdelgraphicsheight}{% %DIF PREAMBLE
\setlength{\unitlength}{\DIFdelgraphicswidth}% %DIF PREAMBLE
\begin{picture}(1,1)% %DIF PREAMBLE
\thicklines\linethickness{2pt} %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\framebox(1,1){}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\line( 1,1){1}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,1){\line(1,-1){1}}}% %DIF PREAMBLE
\end{picture}% %DIF PREAMBLE
}\hspace*{3pt}}} %DIF PREAMBLE
} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbegin}{\DIFaddbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddend}{\DIFaddend} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbegin}{\DIFdelbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelend}{\DIFdelend} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbegin}{\DIFOaddbegin \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbegin}{\DIFOdelbegin \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbeginFL}{\DIFaddbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddendFL}{\DIFaddendFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbeginFL}{\DIFdelbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelendFL}{\DIFdelendFL} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbeginFL}{\DIFOaddbeginFL \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbeginFL}{\DIFOdelbeginFL \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
%DIF LISTINGS PREAMBLE %DIF PREAMBLE
\RequirePackage{listings} %DIF PREAMBLE
\RequirePackage{color} %DIF PREAMBLE
\lstdefinelanguage{DIFcode}{ %DIF PREAMBLE
%DIF DIFCODE_UNDERLINE %DIF PREAMBLE
  moredelim=[il][\color{red}\sout]{\%DIF\ <\ }, %DIF PREAMBLE
  moredelim=[il][\color{blue}\uwave]{\%DIF\ >\ } %DIF PREAMBLE
} %DIF PREAMBLE
\lstdefinestyle{DIFverbatimstyle}{ %DIF PREAMBLE
	language=DIFcode, %DIF PREAMBLE
	basicstyle=\ttfamily, %DIF PREAMBLE
	columns=fullflexible, %DIF PREAMBLE
	keepspaces=true %DIF PREAMBLE
} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim}{\lstset{style=DIFverbatimstyle}}{} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim*}{\lstset{style=DIFverbatimstyle,showspaces=true}}{} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF

\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\DIFdelbegin %DIFDELCMD < \newline{Variation in epigenetic state correlates with gene
%DIFDELCMD < expression across nine inbred strains of
%DIFDELCMD < mice} %%%
\DIFdelend \DIFaddbegin \newline{Variation in histone configurations correlates with gene
expression across nine inbred strains of
mice} \DIFaddend % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
\DIFdelbegin \DIFdel{Catrina Spruce}\DIFdelend \DIFaddbegin \DIFadd{Anna L. Tyler\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
Catrina Spruce}\DIFaddend \textsuperscript{\DIFdelbegin %DIFDELCMD < \getAff{The Jackson Laboratory}%%%
\DIFdelend \DIFaddbegin \getAff{The Jackson Laboratory for
Mammalian Genetics}\DIFaddend },
\DIFdelbegin \DIFdel{Anna L. Tyler}\DIFdelend \DIFaddbegin \DIFadd{Annat Haber\textsuperscript{\getAff{The Jackson Laboratory for Genomic
Medicine}},
Robyn L. Ball}\DIFaddend \textsuperscript{\DIFdelbegin %DIFDELCMD < \getAff{The Jackson Laboratory}%%%
\DIFdelend \DIFaddbegin \getAff{The Jackson Laboratory for
Mammalian Genetics}\DIFaddend },
\DIFdelbegin \DIFdel{Many more people}\DIFdelend \DIFaddbegin \DIFadd{Wendy A. Pitman}\DIFaddend \textsuperscript{\DIFdelbegin %DIFDELCMD < \getAff{JAX-MG and JAX-GM}%%%
\DIFdelend \DIFaddbegin \getAff{unknown}\DIFaddend },
\DIFaddbegin \DIFadd{Narayanan Raghupathy\textsuperscript{\getAff{unknown}},
Michael Walker\textsuperscript{\getAff{unknown}},
Vivek M. Philip\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
Gary A. Churchill\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
Jennifer J. Trowbridge\textsuperscript{\getAff{The Jackson Laboratory
for Mammalian Genetics}},
Michael L. Stitzel\textsuperscript{\getAff{The Jackson Laboratory for
Genomic Medicine}},
Kenneth Paigen\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
Christopher L. Baker\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
Petko M. Petkov\textsuperscript{\getAff{The Jackson Laboratory for
Mammalian Genetics}},
}\DIFaddend Gregory W. Carter\textsuperscript{\DIFdelbegin %DIFDELCMD < \getAff{The Jackson
%DIFDELCMD < Laboratory}%%%
\DIFdelend \DIFaddbegin \getAff{The Jackson Laboratory for
Mammalian Genetics}\DIFaddend }\textsuperscript{*}\\
\bigskip
\textbf{\DIFdelbegin %DIFDELCMD < \getAff{The Jackson Laboratory}%%%
\DIFdelend \DIFaddbegin \getAff{The Jackson Laboratory for Mammalian Genetics}\DIFaddend }600 Main
St.~Bar Harbor, ME, 04609\\
\DIFaddbegin \textbf{\getAff{The Jackson Laboratory for Genomic Medicine}}\DIFadd{299
Farmington Ave, Farmington, CT 06032}\\
\DIFaddend \bigskip
* Corresponding author: Gregory.Carter@jax.org\\
\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
Abstract goes here.

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step.
% Author Summary not valid for PLOS ONE submissions.
\section*{Author summary}
The author summary goes here if we submit to a journal that has one.

\linenumbers

% Use "Eq" instead of "Equation" for equation citations.
\hypertarget{abstract}{%
\section{Abstract}\label{abstract}}

It is well established that epigenetic features, such as histone
modifications and DNA methylation, are associated gene expression across
cell types. However, it is not well known how variation in genotype
affects epigenetic state, or to what extent such variation contributes
to variation in gene expression across genetically distinct individuals.
Here we investigated the relationship between heritable epigenetic
variation and gene expression in hepatocytes across nine inbred mouse
strains. Eight of the inbred strains were founders of the diversity
outbred (DO) mice, and the ninth was DBA/2J, which, along with C57BL/6J,
is one of the founders of the BxD recombinant inbred panel of mice. We
surveyed four histone modifications, H3K4me1, H3K4me3, H3K27me3 and
H3K27ac, as well as DNA methylation. We used ChromHMM to identify 14
chromatin states representing distinct combinations of the four measured
histone modifications. We found that variation in chromatin state
mirrored genetic variation across the inbred strains. Furthermore,
epigenetic variation was correlated with gene expression across strains.
The correspondence between epigenetic state and gene expression was
replicated in an independent population of DO mice in which we imputed
local epigenetic state. In contrast, we found that DNA methylation did
not vary across inbred strains and was not correlated with variation in
expression in DO mice. This work suggests that chromatin state is highly
influenced by local genotype and may be a primary mode through which
expression quantitative trait loci (eQTLs) are mediated. We further
demonstrate that strain variation in chromatin state, paired with gene
expression is useful for annotation of functional regions of the mouse
genome. Finally, we provide, to our knowledge, the first data resource
that documents variation in chromatin state across genetically distinct
individuals.

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

Epigenetic modifications to DNA and its associated histone proteins
influence the accessibility of DNA to transcription machinery, and are
associated with up- and down-regulation of gene expression {[}1--3{]}.
Across cell types, unique combinatorial patterns of histone
modifications mark chromatin states that establish cell type-specific
patterns of gene expression {[}4{]}. Similarly, the methylation of CpG
sites around gene promoters and enhancers influences transcription in a
cell type-specific manner {[}6,7{]}.

Patterns of histone modifications and DNA methylation are established
during development. The result is a canonical epigenetic landscape for
coordination of major patterns of gene expression for each cell type
(for review, see {[}8,9{]}). As an organism ages and responds to its
environment, patterns of both histone modifications {[}10--13{]} and of
DNA methylation {[}14,15{]} change. Variation in epigenetic
modifications have been linked to premature aging {[}16,17{]}, autism
{[}18,19{]}, cancer {[}20,21{]}, and neurological diseases among others
{[}22{]}.

Epigenetic modifications coordinate allow a single genome to give rise
to many different types of cells with diverse morphology and physiology.
\DIFdelbegin \DIFdel{This amazing feature of epigenetic modifications has been intensely
studied, and the variation }\DIFdelend \DIFaddbegin \DIFadd{Variation }\DIFaddend in epigenetic landscapes across cell types has been
extensively documented {[}5,23{]}. Less well understood, however, is the
role that genetic variation plays in determining epigenetic landscapes.

Across genetically diverse populations of humans or mice, individual
cell types, such as hepatocytes, or cardiomyocytes, have globally
similar gene expression profiles that define their role within the
greater organism. However, it is also true that across individuals, gene
expression varies within the global constraints of cell type. This
variation can increase or decrease an organism's risk of developing
disease. Variation in gene expression has been extensively mapped to
variation in genetic loci, or expression quantitative trait loci (eQTL).
Large, coordinated efforts, such as the Genotype-Tissue Expression
(GTEx) Project {[}24{]} have identified and catalogued many such loci in
humans, and countless independent studies have identified eQTL in mice
and other model organisms.

Although the link between genetic variation and gene expression has been
well studied, there is relatively little known about inter-individual
variation in epigenetic modifications, and how these variations are
related to variations in genotype and gene expression. The generation of
a more complete picture of inter-individual variation in epigenetic
modifications will increase our understanding of the mechanisms of gene
regulation, provide insights into the mechanisms establishing cell
type-specific epigenetic landscapes, and to improve the functional
annotation of the genome as it relates to the regulation of gene
expression. The vast majority of SNPs associated with human disease
traits are located in non-coding regions, suggesting that they influence
gene regulation, rather than protein function {[}26,27{]}. However,
annotation of these regions is difficult without additional genomic
features, such as histone modifictions and DNA methylation. Overlaying a
map of variation in epigenetic features has the potential to provide a
picture of how genetic variation changes functional elements, like
enhancers and insulators, in the genome {[}4,28{]}.

Advances in chromatin immunoprecipitation (ChIP) and sequencing
technologies now enable genome-wide surveys of histone modifications
with relatively few cells {[}29{]}, thus opening the door to the
possibility of cataloging epigenetic variation across more cell types
and individuals. Here, we performed a survey of epigenetic variation in
hepatocytes across nine inbred mouse strains. We included the eight
founders of the Diversity Outbred/Collaborative Cross (DO/CC) {[}30{]}
mice, as well as DBA/2J, which, along with C57BL/6J, is one of the
founders of the widely used BxD recombinant inbred panel of mice
{[}31{]}. We assayed four histone modifications (H3K4me1, H3K4me3,
H3K27me3, and H3K27ac), as well as DNA methylation. We used ChromHMM
{[}32{]} to identify 14 chromatin states, classified by unique
combinations of the four histone marks, and investigated the association
between variation in these states and variation in gene expression
across the nine strains. We separately investigated the relationship
between DNA methylation and gene expression across strains.

We further investigated the relationship between epigenetic state and
gene expression by imputing the 14 chromatin states and DNA methylation
into a population of DO mice. We then mapped gene expression to the
imputed epigenetic states to assess the extent to which eQTLs \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were
}\DIFaddend driven by variation in epigenetic modification. We \DIFdelbegin \DIFdel{thus linked }\DIFdelend \DIFaddbegin \DIFadd{found that
histone-defined chromatin states were highly predictive of gene
expression while DNA methylation was not, thus linking }\DIFaddend genetically
controlled variation in epigentic modifications to variation in gene
expression in mice\DIFdelbegin \DIFdel{, and we }\DIFdelend \DIFaddbegin \DIFadd{. We also }\DIFaddend provide the first resource documenting
epigenetic variation across a wide panel of genetically diverse mice.

\hypertarget{materials-and-methods}{%
\section{Materials and Methods}\label{materials-and-methods}}

\hypertarget{ethics-statement}{%
\subsection{Ethics Statement}\label{ethics-statement}}

Ethics Statement All animal procedures followed Association for
Assessment and Accreditation of Laboratory Animal Care guidelines and
were approved by Institutional Animal Care and Use Committee (The
Jackson Laboratory, Protocol XXX).

\hypertarget{inbred-mice}{%
\subsection{Inbred Mice}\label{inbred-mice}}

Three female mice from each of nine inbred strains were used. Eight of
these strains (129S1/SvImJ, A/J, C57BL/6J, CAST/EiJ, NOD/ShiLtJ,
NZO/HlLtJ, PWK/PhJ, and WSB/EiJ) are the eight strains that served as
founders of the Collaborative Cross/Diversity Outbred mice {[}33{]}. The
ninth strain, DBA/2J, will facilitate the interpretation of existing and
forthcoming genetic mapping data obtained from the BxD recombinant
inbred strain panel. Samples were harvested from the mice at 12 weeks of
age.

\hypertarget{liver-perfusion}{%
\subsubsection{Liver perfusion}\label{liver-perfusion}}

To purify hepatocytes from the liver cell population, the mouse livers
were perfused with 87 CDU/mL Liberase collagenase with 0.02\% CaCl2 in
Leffert's buffer to digest the liver into a single-cell suspension, and
then isolated using centrifugation.

We aliquoted \(5 \times 10^{6}\) cells for each RNA-Seq and bisulfite
sequencing, and the rest were cross-linked for ChIP assays. Both
aliquots were spun down at 200 rpm for 5 min, and resuspended in
\(1200\mu L\) RTL+BME (for RNA-Seq) or frozen as a cell pellet in liquid
nitrogen (for bisulfite sequencing). In the sample for ChIP-Seq, protein
complexes were cross-linked to DNA using 37\% formaldehyde in methanol.
All cell samples were stored at -80°C until used (See Supplemental
Methods for more detail).

\hypertarget{hepatocyte-histone-binding-and-gene-expression-assays}{%
\subsubsection{Hepatocyte histone binding and gene expression
assays}\label{hepatocyte-histone-binding-and-gene-expression-assays}}

Hepatocyte samples from 30 treatment and control mice were used in the
following assays:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  RNA-seq to quantify mRNA and long non-coding RNA expression, with
  approximately 30 million reads per sample.
\item
  Reduced-representation bisulfate sequencing to identify methylation
  states of approximately two million CpG sites in the genome. The
  average read depth was 20-30x.
\item
  Chromatin immunoprecipitation and sequencing to assess binding of the
  following histone marks:

  \begin{enumerate}
  \def\labelenumii{\alph{enumii}.}
  \tightlist
  \item
    H3K4me3 to map active promoters
  \item
    H3K4me1 to identify active and poised enhancers
  \item
    H3K27me3 to identify closed chromatin
  \item
    H3K27ac, to identify actively used enhancers
  \item
    A negative control (input chromatin)
  \end{enumerate}
\end{enumerate}

Samples are sequenced with \(\sim40\) million reads per sample.

The samples for RNA-Seq in RTL+BME buffer were sent to The Jackson
Laboratory Gene Expression Service for RNA extraction and library
synthesis.

\hypertarget{histone-chromatin-immunoprecipitation-assays}{%
\subsubsection{Histone chromatin immunoprecipitation
assays}\label{histone-chromatin-immunoprecipitation-assays}}

After extraction, hepatocyte cells were lysed to release the nuclei,
spun down, and resuspended in 130ul MNase buffer with 1mM PMSF and 1x
protease inhibitor cocktail (Roche) to prevent histone protein
degradation. The samples were then digested with 15U of micrococcal
nuclease (MNase), which digests the exposed DNA, but leaves the
nucleosome-bound DNA intact. We confirmed digestion of nucleosomes into
150bp fragments with with agarose gel. The digestion reaction was
stopped with EDTA and samples were used immediately in the ChIP assay.
The ChIP assay was performed with Dynabead Protein G beads and histone
antibodies. After binding to antibodies, samples were washed to remove
unboud chromatin and then eluted with high-salt buffer and proteinase K
to digest protein away from DNA-protein complexes. The DNA was purified
using the Qiagen PCR purification kit. Quantification was performed
using the Qubit quantification system (See Supplemental Methods).

\hypertarget{diversity-outbred-mice}{%
\subsection{Diversity Outbred mice}\label{diversity-outbred-mice}}

We used previously published data from a population of 478 diversity
outbred (DO) mice {[}30{]} to compare to the data collected from the
inbred mice. The DO population included males and females from DO
generations four through 11. Mice were randomly assigned to either a
chow diet (6\% fat by weight, LabDiet 5K52, LabDiet, Scott Distributing,
Hudson, NH), or a high-fat, high-sucrose (HF/HS) diet (45\% fat, 40\%
carbohydrates, and 15\% protein) (Envigo Teklad TD.08811, Envigo,
Madison, WI). Mice were maintained on this diet for 26 weeks.

\hypertarget{genotyping}{%
\subsubsection{Genotyping}\label{genotyping}}

All DO mice were genotyped as described in {[}30{]} using the Mouse
Universal Genotyping Array (MUGA) (7854 markers), and the MegaMUGA
(77,642 markers) (GeneSeek, Lincoln, NE). All animal procedures were
approved by the Animal Care and Use Committee at The Jackson Laboratory
(Animal Use Summary \# 06006).

Founder haplotypes were inferred from SNPs using a Hidden Markov Model
as described in {[}34{]}. The MUGA and MegaMUGA arrays were merged to
create a final set of evenly spaced 64,000 interpolated markers.

\hypertarget{tissue-collection-and-gene-expression}{%
\subsubsection{Tissue collection and gene
expression}\label{tissue-collection-and-gene-expression}}

At sacrifice, whole livers were collected and gene expression was
measured using RNA-Seq as described in {[}35,36{]}. Transcript sequences
were aligned to strain-specific genomes, and we used an expectation
maximization algorithm (EMASE) to estimate read counts
(\url{https://github.com/churchill-lab/emase}).

\hypertarget{data-processing}{%
\subsection{Data Processing}\label{data-processing}}

\hypertarget{sequence-processing}{%
\subsubsection{Sequence processing}\label{sequence-processing}}

The raw sequencing data from both RNA-Seq and ChIP-Seq was put through
the quality control program FastQC (version), and duplicate sequences
were removed before downstream analysis. Reads from each sample were
mapped to strain-specific pseudogenomes, which integrate known SNPs and
indels from each strain. The B6 samples were aligned directly to the
reference mouse genome. The pseudogenomes were created using
\href{https://github.com/churchill-lab/emase}{EMASE}. We used Bowtie
{[}37{]} to align and map reads from the RNA-Seq and ChIP-Seq
experiments.

\hypertarget{transcript-quantification}{%
\subsubsection{Transcript
quantification}\label{transcript-quantification}}

We quantified gene expression using edgeR {[}38{]}, and transcripts with
less than 1 CPM in two or more replicates were filtered out. Transcripts
were further filtered to include only protein-coding transcripts.

We used the R package sva {[}39{]} to perform a variance stabilizing
transformation (vst) on the RNA-Seq read counts from both inbred and
outbred mice. In the inbred mice we used a blind transformation, while
in the outbred mice, we included DO wave and sex in the model. For eQTL
mapping, we performed rank Z normalization on the RNA-Seq read counts
across transcripts from the outbred mice.

\hypertarget{chip-seq-quantification}{%
\subsubsection{ChIP-Seq quantification}\label{chip-seq-quantification}}

We used MACS 1.4.2 {[}40{]} to identify peaks in the ChIP-Seq sequencing
data, with a significance threshold of \(p \leq 10^{-5}\). In order to
compare peaks across strains, we converted the MACS output peak
coordinates to common B6 coordinates using g2g tools
(https://churchill-lab.github.io/g2gtools/).

\hypertarget{quantifying-dna-methylation}{%
\subsection{Quantifying DNA
methylation}\label{quantifying-dna-methylation}}

RRBS data were processed using a bismark-based pipeline modified from
{[}41{]}. The pipeline uses Trim Galore! 0.6.3
(https://www.bioinformatics.babraham.ac.uk/projects/trim\_galore/) for
QC, followed by the trimRRBSdiversityAdaptCustomers.py script from NuGen
for trimming the diversity adapters. This script is available at:
https://github.com/nugentechnologies/NuMetRRBS

All samples had comparable quality levels and no outstanding flags.
Total number of reads was 45-90 million, with an average read length of
about 50 bp. Quality scores were mostly above 30 (including error bars),
with the average above 38. Duplication level was reduced to \(<2\) for
about 95\% of the sequences.

High quality reads were aligned to a custom strain pseudogenomes, using
bowtie2 as implemented in Bismark 0.22 {[}42{]}. The pseudogenomes were
created by incorporating strain-specific SNPs and indels into the
reference genome using g2gtools
(https://github.com/churchill-lab/g2gtools), allowing a more precise
characterization of methylation patterns. Bismark methylation extractor
tool was then used for creating a bed file of estimated methylation
proportions for each animal, which was then translated to the reference
mouse genome (GRCm38) coordinates using g2gtools. Unlike other liftover
tools, g2gtools does not throw away alignments that land on indel
regions. B6 samples were aligned directly to the reference mouse genome.

\hypertarget{analysis-of-histone-modifications}{%
\subsection{Analysis of histone
modifications}\label{analysis-of-histone-modifications}}

\hypertarget{identification-of-chromatin-states}{%
\subsubsection{Identification of chromatin
states}\label{identification-of-chromatin-states}}

We used ChromHMM {[}43{]} to identify chromatin states, which are unique
combinations of the four chromatin modifications, for example, one state
could consist of high levels of both H3K4me3 and H3K4me1, and low levels
of the other two modifications. We conducted all subsequent analyses at
the level of the chromatin state.

To ensure we were analyzing the most biologically meaningful chromatin
states, we calculated chromatin states for all numbers of states between
four and 16, which is the maximum number of states possible with four
binary chromatin modifications (\(2^n\)). We aligned states across the
models by assigning each to one of the sixteen possible binary states
using an emissions probability of 0.3 as the threshold for
presence/absence of the histone mark. We then investigated the stability
of three features across all states: the emissions probabilities (Supp
Fig1), the abundance of each state across transcribed genes (Supp Fig2),
and the effect of each state on transcription (Supp Fig3). Methods for
each of these analyses are described separately below. All measures were
remarkably consistent across all models, but the 14-state model was
characterized by a wide range of relatively abundant states with
relatively strong effects on expression. We used this model for all
subsequent analyses. For more details on how the different models were
compared, see Supplemental Methods.

\hypertarget{genome-distribution-of-chromatin-states}{%
\subsubsection{Genome distribution of chromatin
states}\label{genome-distribution-of-chromatin-states}}

We investigated genomic distributions of chromatin states in two ways.
First, we used the ChromHMM function OverlapEnrichment to calculate
enrichment of each state around known functional elements in the mouse
genome. We analyzed the following features:

\begin{itemize}
\tightlist
\item
  \DIFdelbegin \textbf{\DIFdel{Transcription start sites (TSS)}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Transcription start sites (TSS) }\DIFaddend - Annotations of TSS in the mouse
  genome were provided by RefSeq {[}44{]} and included with the release
  of ChromHMM, which we downloaded on December 9, 2019 {[}43{]}.
\item
  \DIFdelbegin \textbf{\DIFdel{Transcription end sites (TES)}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Transcription end sites (TES) }\DIFaddend - Annotations of TES in the mouse genome
  were provided by RefSeq and included with the release of ChromHMM.
\item
  \DIFdelbegin \textbf{\DIFdel{Transcription factor binding sites (TFBS)}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Transcription factor binding sites (TFBS) }\DIFaddend - We downloaded TFBS
  coordinates from OregAnno {[}45{]} using the UCSC genome browser
  {[}46{]} on May 4, 2021.
\item
  \DIFdelbegin \textbf{\DIFdel{Promoters}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Promoters }\DIFaddend - We downloaded promoter coordinates provided by the
  eukaryotic promoter database {[}47,47{]}, through the UCSC genome
  browser on April 26, 2021.
\item
  \DIFdelbegin \textbf{\DIFdel{Enhancers}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Enhancers }\DIFaddend - We downloaded annotated enhancers provided by ChromHMM
  through the UCSC genome browser on April 26, 2021.
\item
  \DIFdelbegin \textbf{\DIFdel{Candidates of cis regulatory elements in the mouse genome
  (cCREs)}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{Candidates of cis regulatory elements in the mouse genome (cCREs) }\DIFaddend - We
  downloaded cCRE annotations provided by ENCODE {[}48,49{]} through the
  UCSC genome browser on April 26, 2021.
\item
  \DIFdelbegin \textbf{\DIFdel{CpG Islands}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{CpG Islands }\DIFaddend - Annotations of CpG islands in the mouse genome were
  included with the release of ChromHMM.
\end{itemize}

In addition to these enrichments around individual elements, we also
calculated chromatin state abundance relative to the main anatomical
features of a gene. For each transcribed gene, we normalized the base
pair positions to the length of the gene such that the transcription
start site (TSS) was fixed at 0, and the transcription end site (TES)
was fixed at 1. We also included 1000 bp upstream of the TSS and 1000 bp
downstream of the TES, which were converted to values below 0 and above
1 respectively. \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend To map chromatin states to the normalized positions, we
binned the normalized positions into \DIFdelbegin \DIFdel{41 bins defined by the sequence from -2 to 2
}\DIFdelend \DIFaddbegin \DIFadd{bins of relative position
}\DIFaddend incremented by 0.1 \DIFaddbegin \DIFadd{and encompassing all upstream and downstream
positions originally defined as 1kb}\DIFaddend . If a bin encompassed multiple
positions in the gene, we assigned the mean value of the feature of
interest to the bin. To avoid potential contamination from regulatory
regions of nearby genes, we only included genes that were at least 2kb
from their nearest neighbor, for a final set of 14,048 genes.

\hypertarget{chromatin-state-and-gene-expression}{%
\subsubsection{Chromatin state and gene
expression}\label{chromatin-state-and-gene-expression}}

We calculated the effect of each chromatin state on gene expression. We
did this both across genes and across strains. The across-gene analysis
identified states that are associated with high expression and low
expression within the hepatocytes and independent of strain. The
across-strain analysis investigated whether variation in chromatin state
across strains contributed to variation in gene expression across
strains.

For each transcribed gene, we calculated the proportion of the gene body
that was assigned to each chromatin state. We then fit a linear model
separately for each state to calculate the effect of state proportion
with gene expression:

\DIFdelbegin \begin{displaymath}{\DIFdel{\label{eqn:chromatin_effect}}}
\DIFdel{y_{e} = \beta x_{s} + \epsilon
}\end{displaymath}%DIFAUXCMD
\DIFdelend \DIFaddbegin \[\DIFadd{y_{e} = \beta x_{s} + \epsilon}\]\DIFaddend 

where \(y_{e}\) is the rank Z normalized gene expression of the full
transcriptome in a single inbred strain, and \(x_{s}\) is the rank Z
normalized proportion of each gene that was assigned to state \(s\). We
fit this model for each strain and each state to yield one \(\beta\)
coefficient with 95\% confidence interval. The effects were not
different across strains, so we averaged the effects and confidence
intervals across strains to yield one summary effect for each state.

To calculate the effect of each chromatin state across strains, we first
standardized transcript abundance across strains for each transcript. We
also standardized the proportion of each chromatin state for each gene
across strains. We then fit the same linear model, where \(y_{e}\) was a
rank Z normalized vector concatenating all standardized expression
levels across all strains, and \(x_{s}\) was a rank Z normalized vector
concatenating all standardized state proportions across all strains. We
fit the model for each state independently yielding a \(\beta\)
coefficient and 95\% confidence interval for each state.

In addition to calculating the effect of state proportion across the
full gene body, we also performed the same calculations in a
position-based manner. This second analysis yielded an effect of each
state at multiple points along the gene body and a more nuanced view of
the effect of each state.

\hypertarget{analysis-of-dna-methylation}{%
\subsection{Analysis of DNA
methylation}\label{analysis-of-dna-methylation}}

\hypertarget{creation-of-dna-methylome}{%
\subsubsection{Creation of DNA
methylome}\label{creation-of-dna-methylome}}

We combined the DNA methylation data into a single methylome cataloging
the methylated sites across all strains. For each site, we averaged the
percent methylation across the three replicates in each strain. The
final methylome contained 5,311,670 unique sites across the genome.
Because methylated CpG sites can be fully methylated, unmethylated, or
hemi-methylated, we rounded the average percent methylation at each site
to the nearest 0, 50, or 100\%.

\hypertarget{distribution-of-cpg-sites}{%
\subsubsection{Distribution of CpG
sites}\label{distribution-of-cpg-sites}}

We used the enrichment function in ChromHMM described above to identify
enrichment of CpG sites around functional elements in the mouse genome.
We further performed a gene-based analysis of abundance similar to that
in the chromatin states. As a function of relative position on the gene
body, we calculated the density of CpG sites as the average distance to
the next downstream CpG site, as well as the percent methylation at each
site.

\hypertarget{effects-of-dna-methylation-on-gene-expression}{%
\subsubsection{Effects of DNA methylation on gene
expression}\label{effects-of-dna-methylation-on-gene-expression}}

As with chromatin state, we assessed the effect of DNA methylation on
gene expression both across genes and across strains. We used the same
linear model described above, except that \(y_{s}\) became the rank Z
normalized percent methylation either across genes or across strains.
Because the effect of DNA methylation on gene expression is well-known
to be dependent on position, we only calculated a position-dependent
effect on expression. We did not calculate the effect of percent
methylation across the full gene on expression.

\hypertarget{imputation-of-genomic-features-in-diversity-outbred-mice}{%
\subsection{Imputation of genomic features in Diversity Outbred
mice}\label{imputation-of-genomic-features-in-diversity-outbred-mice}}

To assess the extent to which chromatin state and DNA methylation
explained local expression QTLs, we imputed local chromatin state and
DNA methylation into the population of diversity outbred (DO) mice
described above. We compared the effect of the imputed epigenetic
features to imputed SNPs.

All imputations followed the same basic procedure: For each transcript,
we identified the haplotype probabilities in the DO mice at the genetic
marker nearest the gene transcription start site. This matrix held DO
individuals in rows and DO founder haplotypes in columns (Supp. Fig.
\ref{supp_fig:imputation}).

For each transcript, we also generated a three-dimensional array
representing the genomic features (chromatin state, DNA methylation
status, or SNP genotype) derived from the DO founders. This array held
DO founders in rows, feature state in columns, and genomic position in
the third dimension. The feature state for chromatin consisted of states
one through 14, for SNPs feature state consisted of the genotypes A,C,G,
and T.

We then multiplied the haplotype probabilities by each genomic feature
array to obtain the imputed genomic feature for each DO mouse. This
final array held DO individuals in rows, the genomic feature in the
second dimension, and genomic position in the third dimension. This
array is analagous to the genoprobs object in R/qtl2 {[}50{]}. The
genomic position dimension included all positions from 1 kb upstream of
the TSS to 1 kb downstream of the TES. SNP data for the DO founders in
mm10 coordinates were downloaded from the Sanger SNP database
{[}51,52{]}, on July 6, 2021.

To calculate the effect of each imputed genomic feature on gene
expression in the DO population, we fit a linear model. From this linear
model, we calculated the variance explained (\(R^2\)) by each genomic
feature, thereby relating gene expression in the DO to each position of
the imputed feature in and around the gene body.

\hypertarget{results}{%
\section{Results}\label{results}}

Gene expression varied reproducibly across inbred strains of mice. This
is seen as a clustering of individuals from the same strain in a
principal component plot of the hepatocyte transcriptome across strains
(Figure \ref{fig:pc_plots}A). Patterns of DNA methylation (Figure
\ref{fig:pc_plots}B) and individual histone modifications (Figure
\ref{fig:pc_plots}C-F) cluster in a similar pattern. This suggests that
these epigenetic features may relate to gene expression in a manner that
is consistent with genetic background.

\hypertarget{chromatin-state-overview}{%
\subsection{Chromatin state overview}\label{chromatin-state-overview}}

To investigate the association between histone modifications and gene
expression, we used ChromHMM to identify 14 chromatin states composed of
unique combinations of four histone modifications in the hepatocytes of
nine inbred strains of mice. Panel A in Figure \ref{fig:state_overview}
shows the representation of each histone modification across the states.

The states were enriched around known functional elements in the mouse
genome (Figure \ref{fig:state_overview}B). The majority of the states
were enriched around the TSS, and other TSS-related functional elements,
such as promoters and CpG islands. Two states (states 13 and 14) were
primarily found in intergenic regions. Three states (states 6, 2, and 4)
were enriched around known enhancers, and one (\DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 9) was enriched
predominantly near the TES. The majority of these states were also
associated with variation in gene expression \DIFdelbegin \DIFdel{. The colored bars in }\DIFdelend \DIFaddbegin \DIFadd{across strains (}\DIFaddend Figure
\ref{fig:state_overview}C\DIFdelbegin \DIFdel{show the effect of each state on gene
expression across the inbred strains.
For reference, the paired tan bars
show the effect of each chromatin state on gene expression in
hepatocytes. These effects tended to be of the same sign and greater
magnitude than the across-strain effects.
}\DIFdelend \DIFaddbegin \DIFadd{).
}\DIFaddend 

The states in Figure \ref{fig:state_overview} are shown in order of
their effect on expression, which helps illustrate several patterns. The
state with the largest negative effect on gene expression, state 14, is
the absence of all measured modifications. Other states associated with
reduced gene expression contained the repressive mark H3K27me3. The
states with the largest positive effects on expression all had some
combination of the activating marks, H3K4me3, H3K4me1, and H3K27ac. The
repressive mark was less commonly seen in these activating states.

By merging the information from Figure \ref{fig:state_overview}A-C), we
were able to suggest annotations for many of the 14 chromatin states
(Figure \ref{fig:state_overview}D). States with the strongest effects on
expression had the clearest annotations, while states with weaker
effects remained unannotated.

\hypertarget{spatial-distribution-of-epigenetic-modifications-around-gene-bodies}{%
\subsection{Spatial distribution of epigenetic modifications around gene
bodies}\label{spatial-distribution-of-epigenetic-modifications-around-gene-bodies}}

In addition to looking for enrichment of chromatin states near annotated
functional elements, we characterized the fine-grained spatial
distribution of each state around gene bodies (Figure
\ref{fig:state_abundance}A-B). We similarly characterized the
distribution of CpG sites and their percent methylation at this
gene-level scale (Figure \ref{fig:state_abundance}C-D).

The spatial patterns of the individual chromatin states are shown in
(Figure \ref{fig:state_abundance}A), and an overlay of all states
together (Figure \ref{fig:state_abundance}B) emphasizes the difference
in abundance between the most abundant states (states 1, 3, and 14), and
the remaining states, which were relatively rare.

Each chromatin state had a characteristic distribution pattern. For
example, \DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 14, which was characterized by the absence of all
measured histone modifications, was strongly depleted near the TSS,
indicating that this region is commonly subject to the histone
modifications we measured here. In contrast, states 3 and 1 were both
enriched at the TSS. State 3 was very narrowly concentrated right at the
TSS, whereas state 1 was more broadly abundant both upstream and
downstream of the TSS. Both were associated overall with increased
expression across inbred mice (indicated by red shading \DIFaddbegin \DIFadd{in Figure
\ref{fig:state_abundance}}\DIFaddend ), suggesting promoter or enhancer functions.
The third state in this group of expression-enhancing states, state 2,
was depleted nere the TSS, but enriched within the gene body, suggesting
that this state may mark active intragenic enhancers.

States with weaker effects on expression (indicated by grayer shades \DIFaddbegin \DIFadd{in
Figure \ref{fig:state_abundance}}\DIFaddend ) were of lower abundance, but had
distinct distribution patterns around the gene body suggesting the
possibility of distinct functional roles in the regulation of gene
expression.

DNA methylation showed similarly dramatic variation in abundance (Figure
\ref{fig:state_abundance}C-D). The TSS had densely packed CpG sites
relative to the gene body (Figure \ref{fig:state_abundance}C). As
expected, the median CpG site near the TSS was consistently
hypomethylated relative to the median CpG site in intra- and intergenic
regions (Figure \ref{fig:state_abundance}D).

\hypertarget{spatially-resolved-effects-on-gene-expression}{%
\subsection{Spatially resolved effects on gene
expression}\label{spatially-resolved-effects-on-gene-expression}}

The distinct spatial distributions of the chromatin states and
methylated CpG sites around the gene body raised the question as to
whether the effects of these states on gene expression could also be
spatially resolved. To investigate this possibility we tested the
association between both chromatin state and DNA methylation and gene
expression with spatially resolved models (Methods). We tested the
effect of each chromatin state on expression across genes within
hepatocytes (Figure \ref{fig:state_effects}A) and the effect of each
chromatin state on the variation in gene expression across strains
(Figure \ref{fig:state_effects}B).

All chromatin states demonstrated spatially dependent effects on gene
expression within hepatocytes. For many of the states, the effects on
expression were concentrated at or near the TSS, while in the other
states effects were seen across the whole gene. The direction of the
effects matched the overall effects of each state seen previously
(Figure \ref{fig:state_overview}). Remarkably, the spatial effects were
recapitulated for almost every state when we measured across strains.
That is, chromatin states that either enhanced or suppressed gene
expression across hepatocyte genes were similarly related to variation
in \DIFdelbegin \DIFdel{expresion }\DIFdelend \DIFaddbegin \DIFadd{expression }\DIFaddend across strains. \DIFaddbegin \DIFadd{This suggests that the genetic differences
between strains modify chromatin activity in a manner similar to that
used across genes. }\DIFaddend One notable exception was \DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 6, whose presence
upregulated genes within hepatocytes, but did not contribute to
expression variation across strains.

We also examined the effect of percent DNA methylation across genes
within hepatocytes, and across strains (Figure
\ref{fig:DNA_methylation_effect}). As expected, methylation at the TSS
was associated with lower expression in hepatocytes. However, percent
DNA methylation did not contribute at all to expression variation across
strains. This was in part due to an overall lack of variation in percent
DNA methylation at the TSS. These results imply that although percent
DNA methylation is used in gene regulation within a cell type, it is not
heritable and does not contribute to variation in gene expression across
genetically diverse individuals.

\hypertarget{imputed-chromatin-state-explained-expression-variation-in-diversity-outbred-mice}{%
\subsection{Imputed chromatin state explained expression variation in
diversity outbred
mice}\label{imputed-chromatin-state-explained-expression-variation-in-diversity-outbred-mice}}

Thus far, we have used inbred strains of mice to identify correlations
between local chromatin state and gene expression. However, we cannot
establish causality in this population. For that we need a mapping
population in which we can associate genetic or epigenetic variation at
a single locus with changes in gene expression. A mapping population
also allows us to establish the extent to which variation in epigenetic
factors contributes to observed expression quantitative trait loci
(eQTL).

To compare the contribution of genetic and epigenetic features to eQTLs
in a gentically diverse population, we imputed chromatin state, DNA
methylation, and SNPs into DO mice (Methods). Chromatin state is largely
determined by local genotype, especially early in life {[}53{]}, and can
thus be reliably imputed from local genotype. Further, we have shown
here that local chromatin state correlates with variation in gene
expression across inbred strains. DNA methylation, on the other hand, is
known not to be highly heritable {[}54{]}, and thus cannot be reliably
imputed from local genotype. We have also shown here that DNA
methylation is not correlated with variation in gene expression across
inbred strains. The imputation of DNA methylation thus serves as a
negative control--an estimate of a lower bound the ability of a feature
imputed from local haplotype to explain gene expression in a new
population.

\DIFdelbegin \DIFdel{For each transcript in the DO population, we imputed the local chromatin
state across the gene body based on the gene's local founder haplotype
and the chromatin state at the corresponding position in the inbred
mice. We did the same for DNA methylation and SNPs.
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend After imputing each genomic feature into the DO population, we mapped
gene expression to the imputed features and calculated the variance
explained. Examples of each genomic feature and the mapping results for
the gene \textit{Pkd2} are shown in Figure \ref{fig:example_gene}. There
are two particularly interesting regions in this gene. One is at the TSS
and the immediately surrounding area, and the other is just downstream
of the TSS.

These two regions \DIFdelbegin \DIFdel{are colored red,
indicating that they are marked by chromatin states with a positive effect on gene expression. The order of
the rows in this panel helps illustrate that the strains with the most
red in chromatin state space contributed the highest-expressing alleles
to the DO (Figure \ref{fig:example_gene}E). The two haplotypes with the
strongest negative effect on gene expression in the DO have mostly blue
chromatin states in these two regions. These two strains also had the
lowest expression among the inbred mice }\DIFdelend \DIFaddbegin \DIFadd{were both associated with increased gene expression,
and were dominated by activating chromatin states in the strains whose
haplotype effects were positive. They were dominated by repressive
chromatin states for the two strains with negative haplotype effects
}\DIFaddend (Figure \ref{fig:example_gene}\DIFdelbegin \DIFdel{F). The concordance between chromatin state and
gene expression in the DO is seen as the blue pluses in Figure
\ref{fig:example_gene}A that are aligned with the two red regions , which
we suggest are
putative enhancer regions}\DIFdelend \DIFaddbegin \DIFadd{)}\DIFaddend . \DIFaddbegin \DIFadd{We suggest that these two regions are
putative enhancers.
}\DIFaddend 

The spatial patterns in the SNPs only partially mirror those in
chromatin state (Figure Figure \ref{fig:example_gene}C). SNPs underlying
the putative enhancer region at the TSS could potentially influence gene
expression by altering chromatin state. But SNPs downstream of this
region underly invariant chromatin. The downstream enhancer has no
underlying SNPs, suggesting that there is an alternative mechanism for
determining chromatin state at this location.

Percent DNA methylation does not vary across the strains in either of
these putative enhancer regions, and does not contribute to variation in
expression across genetically distinct individuals (Figure
\ref{fig:example_gene}D).

The overall distributions of variance explained by each feature across
all transcripts are shown in Figure \ref{fig:effect_distrubutions}.
These distributions show the haplotype effect for the marker nearest
each transcript compared with the maximum effect across the gene body
for each of the other imputed features. Overall, local haplotype
explained the largest amount of variance of gene expression in the DO
(\DIFdelbegin \DIFdel{\(R^2 =\) 0.17}\DIFdelend \DIFaddbegin \DIFadd{\(R^2 = 0.17\)}\DIFaddend ). The variance explained by local chromatin state was
very highly correlated with that of haplotype (Pearson \(r = 0.96\)) and
explained almost as much variance in gene expression in the DO as local
haplotype (\DIFdelbegin \DIFdel{\(R^2 =\) 0.15). }%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend \DIFaddbegin \DIFadd{\(R^2 = 0.15\)). }\DIFaddend The mean variance explained by SNPs was
lower (\DIFdelbegin \DIFdel{\(R^2 =\) 0.13}\DIFdelend \DIFaddbegin \DIFadd{\(R^2 = 0.13\)}\DIFaddend ) than that explained by haplotype and was not as
highly correlated with local haplotype as chromatin state was (Pearson
\(r = 0.93\)). DNA methylation, the lower bound for variance explained
by a feature imputed from local haplotype, explained the lowest amount
of expression variance in the DO population (\DIFdelbegin \DIFdel{\(R^2 =\) 0.09}\DIFdelend \DIFaddbegin \DIFadd{\(R^2 = 0.09\)}\DIFaddend ), and had a
much lower correlation to haplotype than either chromatin state or SNPs
(Pearson \(r = 0.74\)). \DIFaddbegin \DIFadd{Taken together, these results suggest that the
majority of haplotype-associated variance is encoded by the chromatin
state of the allele. Moreover, this encoding is primarily local and
present in the relevant inbred founder strain.
}\DIFaddend 

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

In this sudy we showed that variation in histone modifications in inbred
mice mirrors genetic variation. We further showed that this variation
was highly related to variation in gene expression across strains. These
observations suggest that cell type-specific patterns of histone
modifications are determined by local genotype, and \DIFdelbegin \DIFdel{may be }\DIFdelend \DIFaddbegin \DIFadd{is likely }\DIFaddend a major
mechanism through which expression QTL (eQTL) are generated. This
hypothesis was supported by the high concordance between chromatin
state, which was imputed from local genotype, and gene expression in an
independent outbred population of mice.

The variation of chromatin state across strains, and its correspondence
with expression variation offers a unique way to identify gene
regulatory regions. Genetic variation serves as a natural perturbation
to the regulatory regions\DIFdelbegin \DIFdel{which }\DIFdelend \DIFaddbegin \DIFadd{, and resulting differences in regulatory
annotations }\DIFaddend can then be linked to variation in gene expression. The
\textit{Pkd2} example illustrates how genetic and epigenetic variation
can be combined to identify two putative enhancer regions for the gene.
The variation in chromatin state further \DIFdelbegin \DIFdel{offers }\DIFdelend \DIFaddbegin \DIFadd{suggests }\DIFaddend a putative mechanism
for the observation of a cis-eQTL at the level of the haplotype.

The discordance between the patterns of chromatin state and SNPs in this
gene are particularly interesting. The variation in chromatin state is
present in the absence of local SNPs, whereas further downstream,
variation in SNPs does not correspond to variation in chromatin state.
This suggests that the presence of the downstream enhancer is determined
by another mechanism, perhaps SNPs acting in \DIFdelbegin \textit{\DIFdel{trans}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \emph{\DIFadd{trans}} \DIFaddend to this
region, or local variation that was not \DIFdelbegin \DIFdel{measured--structural variation, for example}\DIFdelend \DIFaddbegin \DIFadd{measured by SNP genotyping, such
as structural variation}\DIFaddend .

In contrast to chromatin state, percent DNA methylation was not
associated with variation in gene expression across inbred strains or in
the outbred population. At the TSS, this was largely due to a lack of
variation in methylation across strains. An example of this observation
is shown in panel D of Figure \ref{fig:example_gene}. Despite strain
variation in both genotype and chromatin state at the TSS of
\textit{Pkd2}, DNA methylation is invariant -- the CpG island at the TSS
is unmethylated in all strains. Thus, although chromatin state appears
to be highly influenced by local genotype, percent DNA methylation is
not.

Similar observations have been made in human studies {[}54{]}. Multiple
twin studies have estimated the average heritability of individual CpG
sites to be roughly 0.19 {[}55--57{]}, with only about 10\% of CpG sites
having a heritability greater than 0.5 {[}56--58{]}. Trimodal CpG sites,
i.e.~those with methylation percent varying among 0, 50, and 100\%, have
been shown in human brain tissue to be more heritable than unimodal, or
bimodal sites (\(h^2 = 0.8 \pm 0.18\)), and roughly half were associated
with local eQTL {[}59{]}. Here, we did not see an association between
trimodal CpG sites and gene expression across strains (Supplemental
Figure \ref{supp_fig:trimodal}).

The diversity in the effects observed in the 14 chromatin states
highlights the importance of analyzing combinatorial states as opposed
to individual histone modifications. To illustrate this point, consider
the three states with the largest positive effects on transcription.
Each of these three states had a distinct combination of the three
histone marks associated with transcriptional activation: H3K4me1,
H3K4me3, and H3K27ac. State 3 was characterized by high levels of
H3K4me3 and H3K27ac, and low levels of H3K4me1. State 2 was
characterized by high levels of H3K4me1 and H3K27ac, and low levels of
H3K4me3. And state 1 was characterized by high levels of all three
activating marks (Figure \ref{fig:state_overview}). Although all three
states were associated with increased gene expression, each had a
completely distinct spatial distribution. State 3 was distributed in a
very narrow band centered on the TSS, while \DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 1 was distributed
across a much broader r egion centered upstream of the TSS. State 2 had
a completely different distribution -- it was depleted at the TSS, and
most abundant within the gene body and near the TES. This variation in
spatial distribution was mirrored in the spatial effects on
transcription. State 3, which we annotated as an active promoter, was
positively associated with transcription when it was present at the TSS.
In contrast, states 2 and 1, which we annotated as enhancers, were
associated with increased transcription when present anywhere in the
gene body. We would not be able to detect such patterns if analyzing the
histone modifications in isolation. These results highlight the
complexity of the histone code and the importance at analyzing
combinatorial states.

While we were able to annotate several states, particularly those with
the strongest effects on gene expression, other states were more
difficult to annotate. This raises the intruiguing possibility of
identifying new modes of expression regulation through histone
modification. One of these unannotated states, \DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 9, had a weak, but
consistent negative effect on gene transcription centered within the
gene body, downstream of the TSS. This state was characterized by high
levels of H3K4me3 and low levels of the other three modifications.

The modification H3K4me3 is most frequently associated with increased
transcriptional activity {[}60--63{]}, so the association with state 9
with reduced transcription is a deviation from the dominant paradigm.
The physical distribution of this state is also interesting. It was
depleted at the TSS, and enriched just upstream and just downstream of
the TSS. It was also enriched just downstream of the TES, although it
did not appear to influence transcription at this location. The group of
genes marked by \DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 9 were enriched for functions such as stress
response, DNA damage repair, and ncRNA processing suggesting that this
state may be used to regulate subsets of genes involved in responses to
environmental stimuli.

We detected two bivalent states in this survey, which are characterized
by a combination of activating and repressive histone modifications, and
are usually associated with undifferentiated cells {[}64,65{]}. Here we
identified two bivalent states in adult mouse hepatocytes, and annotated
them as a poised enhancer (\DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 12) and a bivalent promoter (\DIFdelbegin \DIFdel{state }\DIFdelend \DIFaddbegin \DIFadd{State }\DIFaddend 11).
Both states were associated with downregulation across inbred strains
when present near the TSS; however this effect was not replicated in the
outbred mice. The lack of replication was perhaps because the effect was
too weak to detect given the number of animals in the population.

Both bivalent promoters and poised enhancers are dynamic states that
change over the course of differentiation and in response to external
stimuli. Bivalent promoters have been studied primarily in the context
of development. They are abundant in undifferentiated cells, and are
typically resolved either to active promoters or to silenced promoters
as the cells differentiate into their final state {[}64{]}. These
promoters have also been shown to be important in the response to
changes in the environment. Their abundance increases in breast cancer
cells in response to hypoxia {[}66{]}. Poised enhancers are also
observed during differentiation and in differentiated cells {[}67{]}. In
concordance with these previous observations, the genes marked by states
12 and 11 were enriched for vascular development and morphogenesis. That
we identified these states in differentiated hepatocytes may indicate
that a subset of developmental genes retain the ability to be activated
under certain circumstances, such as during liver regeneration in
response to damage. It is also possible these states were induced in the
inbred strains in respose to stress, rather than genetically coded. This
could explain why the negative effect on gene expression was not
replicated in the outbred mice. However, given that we detected this
state in all nine inbred strains in relatively equal proportions, this
latter hypothesis seems less likely.

Broadly, local variation in chromatin state was highly correlated with
variation in gene expression across individuals, an observation that was
replicated in an independent population of genetically diverse, outbred
mice. The percent variance explained by chromatin state closely matched
that of haplotype, and exceeded that of individual SNPs. These results
suggest two things: First, a large portion of the effect of local
haplotype on gene expression in mice may be mediated through variation
in chromatin state. Second, the intermediate resolution of chromatin
state between that of individual SNPs and broad haplotypes carries
important imfornation that cannot be resolved at the other levels.
Individual SNPs, although, sometimes causally linked to trait variation,
are highly redundant and cannot be readily used to annotate functional
elements in the genome. Haplotypes aggregate genomic information over
broad regions and are a powerful tool to link genomic variation to trait
variation. However, they are usually too broad to be used to annotate
regions less than a few megabases in length. By combining the mapping
power of haplotypes, the high resolution of SNPs, and the intermediate
resolution of chromatin states, we can begin to build mechanistic
hypotheses that link genetic variation to variation in gene expression
and physiology. Understanding the role that genetic variation plays in
modifying the chromatin state landscape will be critical in making these
links. Through this survey we are providing one of the first rigorous
resources that explores the connection between genetic variation and
epigenetic variation.

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

This work was funded by \DIFdelbegin \DIFdel{XXX}\DIFdelend \DIFaddbegin \DIFadd{the National Institutes of Health R01 GM115518
(to G.W.C), XXX, YYY, and the Jackson Laboratory Directory's Innovation
Funds}\DIFaddend .

\hypertarget{data-and-software-availability}{%
\section{Data and Software
Availability}\label{data-and-software-availability}}

All data used in this study and the code used to analyze it are avalable
as part of a reproducible workflow located at\ldots{} (Figshare?,
Synapse?).

\hypertarget{figure-legends}{%
\section{Figure Legends}\label{figure-legends}}

\begin{figure}[ht]
\centering
\caption{The first two principle components of each genomic feature across
nine inbred strains of mouse. In all panels each point represents
an individual mouse, and strain is indicated by color as shown in
the legend at the bottom of the figure. Each panel is labeled with
the data used to generate the PC plot. (A) Hepatocyte transcriptome - 
all transcripts sequenced in isolated hepatocytes. (B) DNA methylation - 
the percent methylation at all CpG sites shared across all individuals. 
(C-F) Histone modifications - the peak heights of the indicated histone
modification for sites shared across all individuals.}
\label{fig:pc_plots}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Overview of chromatin state composition, genomic distribution,
and effect on expression. The left most panel shows the emission
probabilites for each histone modification in each chromatin state. 
Blue indicates the absence of the histone modification, and red 
indicates the presence of the modification. The panel labeled
genomic enrichment shows the distribution of each state around 
functional elements in the genome. Red indicates that the
state is more likely to be found near the annotated functional 
element than expected by chance. Blue indicates that the state
is less likely to be found near the annotated functional element
than expected by chance. Abbreviations are as follows: TFBS = 
transcription factor binding sites, cCRE = candidate cis-regulatory
element, TSS = transcription start site, 
TES = transcription end site. The panel labeled Expression Effects 
shows the effect of variation in the state on gene expression. 
Bars are colored based on the size and direction the state's effect on
expression. Darker bars show the effects on expression of chromatin 
state variation across strains. Tan bars show the effects on expression
of chromatin state variation across genes. The final column of the figure 
shows plausible annotations for each state based on combining the data in 
the previous three panels. The numbers in parentheses indicate the percent 
of the genome that was assigned to each state.}
\label{fig:state_overview}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Relative abundance of chromatin states and methylated DNA. \DIFaddbeginFL \DIFaddFL{(}\DIFaddendFL A\DIFdelbeginFL \DIFdelFL{. }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{) }\DIFaddendFL Each panel 
shows the abundance of a single chromatin state relative to gene TSS and TES. The 
$y$-axis in each panel is the proportion of genes containing the state. Each
panel has an independent $y$-axis to better show the shape of each curve.
The $x$-axis is the relative gene position. The TSS and TES are marked as vertical
gray dashed lines. \DIFaddbeginFL \DIFaddFL{(}\DIFaddendFL B\DIFdelbeginFL \DIFdelFL{. }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{) }\DIFaddendFL The same data shown in panel A, but with all states overlayed
onto a single set of axes to show the relative abundance of the states. 
\DIFaddbeginFL \DIFaddFL{(}\DIFaddendFL C\DIFdelbeginFL \DIFdelFL{. }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{) }\DIFaddendFL The density of CpG sites relative to the gene body. The $y$-axis shows the 
number of CpG sites per base pair. The density is highest near the TSS. 
CpG sites are less dense within the gene body and in the intergenic space. 
\DIFaddbeginFL \DIFaddFL{(}\DIFaddendFL D\DIFdelbeginFL \DIFdelFL{. }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{) }\DIFaddendFL Percent methylation relative to the gene body. The $y$-axis shows the median 
percent methylation at CpG sites, and the $x$-axis shows relative gene position. 
CpG sites near the TSS are unmethylated relative to intragenic and intergenic
CpG sites.}
\label{fig:state_abundance}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Effects of chromatin states on gene expression. Each column shows 
the effect of each chromatin state on gene expression in a different 
experimental context. The first column shows the effect across genes in 
the inbred mice showing how chromatin states are used within a single 
tissue to increase the expression of some genes and decrease the expression 
of other genes. The second column shows the effect of chromatin state on 
gene expression across strains, showing how variation in chromatin state 
across strains leads to variation in expression of individual genes across 
strains. The third column shows the effect of imputed chromatin state on 
gene expression in a population of diversity outbred mice. These plots show 
the effect on local gene expression of variation in chromatin state across 
genetically diverse individuals. Each column of panels is plotted on a single 
scale for the $y$-axis so the magnitude of the effects in a single column can be 
compared directly to each other. Across a single row, the scale of the $y$-axis 
varies to highlight the similarity of the shape of each curve in each different 
setting. The final column shows the annotation of each state for comparison with
its effects on gene expression. All $y$-axes show the $\beta$ coefficient from 
the linear model shown in \DIFaddbeginFL \DIFaddFL{the }\DIFaddendFL equation\DIFdelbeginFL %DIFDELCMD < [%%%
\DIFdelFL{REF}%DIFDELCMD < ]%%%
\DIFdelendFL . All $x$-axes show the relative 
position along the gene body running from just upstream of the TSS to just downstream 
of the TES. Vertical gray dashed lines mark the TSS and TES in all panels.}
\label{fig:state_effects}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Effect of DNA methylation on gene expression (A) across gene expression
in hepatocytes and (B) across inbred strains. Dark gray line shows estimate
of the effect of percent DNA methylation on gene expression. The $x$-axis is
normalized position along the gene body running from the transcription start
site (TSS) to the transcription end site (TES), marked with vertical gray dashed
lines. The horizontal solid black line indicates an effect of 0. 
The shaded gray area shows 95\% confidence interval arond the model fit.}
\label{fig:DNA_methylation_effect}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Example of epigenetic states and imuptation results for a single 
gene, \textit{Pkd2}. (A) The variance in DO gene expression explained at 
each position along the gene body by each of the imputed genomic 
features: SNPs - red X's, Chromatin State - blue plus signs, and 
Percent Methylation - green circles. The horizontal dashed line shows 
the variance explained by the haplotype. For reference, the arrow 
below this panel runs from the TSS of \textit{Pkd2} to the TES and 
shows the direction of transcription. (B) The chromatin states assigned 
to each 200 bp window in this gene for each inbred mouse strain. States 
are colored by their effect on gene expression in the inbred mice. Red 
indicates a positive effect on gene expression, and blue indicates a 
negative effect. Each row shows the chromatin states for a single inbred 
strain, which is indicated by the label on the left. (C) SNPs along the 
gene body for each inbred strain. The reference genotype is shown in gray. 
SNPs are colored by genotype as shown in the legend. (D) Percent DNA 
methylation for each inbred strain along the \textit{Pkd2} gene body. 
Percentages are binned into 0\% (blue) 50\% (yellow) and 100\% (red). 
(E) Haplotype effects for expression of \textit{Pkd2} in the DO. 
Haplotype effects are colored by from which each allele was derived. 
(F) \textit{Pkd2} expression levels across inbred mouse strains. For 
ease of comparison, all panels B through F are shown in the same order 
as the haplotype effects.}
\label{fig:example_gene}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Chromatin state explains variation in gene expression in an outbred 
population. A. Distributions of gene expression variance explained by different 
genomic features: local haplotype, local imputed chromatin state, local SNP 
genotype, and local imputed DNA methylation status. B. Direct comparisons of 
variance explained by local haplotype, and the three other genomic features: 
imputed chromatin state, SNP genotype, and imputed DNA methylation status. 
Blue lines show $y = x$. Each point is a single transcript.}
\label{fig:effect_distrubutions}
\end{figure}

\pagebreak

\hypertarget{supplemental-figure-legends}{%
\section{Supplemental Figure
Legends}\label{supplemental-figure-legends}}

\begin{figure}[ht]
\centering
\caption{Comparison of emissions probabilities across all ChromHMM models.
Each row contains data for a single ChromHMM model fit to the number of
states indicated on either side of the row. Each set of four columns shows
data for each of the four histone modifications. Each set is separated from
the next by a column of gray for ease of visualization. The bottom row, the
reference row, shows the ideal state that all model states are being compared
to. Blue indicates absence of the histone mark and red indicates presence.
For each ChromHMM model, each state was assigned to one of the reference states
using an emissions probability of 0.3 as a threshold for presence of the histone
modification. If a state was not present in the given model, the corresponding
area is shown in gray. Emissions probabilities near 0 are shown in blue, and 
probabilities near 1 are shown in red. Orange and yellow indicate intermediate
probabilities. Aligning the states across all models shows a remarkable 
stability in the emissions across models, seen as vertical bars of consistent 
color.}
\label{supp_fig:model_emissions_comparison}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Comparison of state abundance across all ChromHMM models. The left-most column
shows the annotation for each state. Unannotated states are marked with a dash. The 
binary heatmap indicates which histone modifications were present in each state: 1 
indicates presence, and 0 indicates absence. The histone modifications are labeled
at the bottom of each column. The continuous heatmap shows the abundance of each state
(in rows) in each ChromHMM model (in columns). The abundance is the proportion of 
transcribed genes with the state present. Less abundant states are shaded blue, and 
more abundant states are shaded yellow, orange, and red. The number of states in the
model is indicated at the bottom of each column. The black box highlights the model
used in this study -- the 14-state model. State abundance was remarkably
stable across the different models.}
\label{supp_fig:model_abundance_comparison}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Comparison of state effect across all ChromHMM models. This figure is 
identical to Figure \ref{supp_fig:model_abundance_comparison}, except that the
cells in the continuous heatmap show the effect of each state on gene expression
across all ChromHMM models. The effect was the $\beta$ coefficient derived from
a linear model. Similar to state abundance, the effects were remarkably stable
across models.}
\label{supp_fig:model_effect_comparison}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Schematic for imputation of histone modifications into the 
DO mice. For a single transcript imputation was made by multiplying 
a three-dimensional array, containing chromatin state by strain by
position, by a two-dimensional array, contatining haplotype probabilities
by DO individual, to create a three-dimensional array, containing 
individual by position by chromatin state probability.}
\label{supp_fig:imputation}
\end{figure}

\begin{figure}[ht]
\centering
\caption{Effects of percent DNA methylation for all (A) CpG sites,
(B) bimodal CpG sites, and (C) trimodal CpG sites. The $y$-axis
in each panel shows the effect of variation in DNA methylation on 
gene expression across hepatocyte genes. The gray polygon shows the
95\% confidence interval of the effect. Although there is a weak
negative effect on transcription of DNA methylation across all 
sites, there was no effect when looking at trimodal sites alone.}
\label{supp_fig:trimodal}
\end{figure}

\pagebreak

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-lawrence2016lateral}{}%
1. Lawrence M, Daujat S, Schneider R. Lateral thinking: How histone
modifications regulate gene expression. Trends in Genetics. Elsevier;
2016;32: 42--56.

\leavevmode\hypertarget{ref-jones2012functions}{}%
2. Jones PA. Functions of dna methylation: Islands, start sites, gene
bodies and beyond. Nature Reviews Genetics. Nature Publishing Group;
2012;13: 484--492.

\leavevmode\hypertarget{ref-moore2013dna}{}%
3. Moore LD, Le T, Fan G. DNA methylation and its basic function.
Neuropsychopharmacology. Nature Publishing Group; 2013;38: 23--38.

\leavevmode\hypertarget{ref-pmid20657582}{}%
4. Ernst J, Kellis M. Discovery and characterization of chromatin states
for systematic annotation of the human genome. Nat Biotechnol. 2010;28:
817--825.

\leavevmode\hypertarget{ref-pmid21441907}{}%
5. Ernst J, Kheradpour P, Mikkelsen TS, Shoresh N, Ward LD, Epstein CB,
et al. Mapping and analysis of chromatin state dynamics in nine human
cell types. Nature. 2011;473: 43--49.

\leavevmode\hypertarget{ref-pmid21701563}{}%
6. Wiench M, John S, Baek S, Johnson TA, Sung MH, Escobar T, et al. DNA
methylation status predicts cell type-specific enhancer activity. EMBO
J. 2011;30: 3028--3039.

\leavevmode\hypertarget{ref-pmid20720541}{}%
7. Ji H, Ehrlich LI, Seita J, Murakami P, Doi A, Lindau P, et al.
Comprehensive methylome map of lineage commitment from haematopoietic
progenitors. Nature. 2010;467: 338--342.

\leavevmode\hypertarget{ref-pmid32671792}{}%
8. Xu R, Li C, Liu X, Gao S. Insights into epigenetic patterns in
mammalian early embryos. Protein Cell. 2021;12: 7--28.

\leavevmode\hypertarget{ref-pmid29625185}{}%
9. Godini R, Lafta HY, Fallahi H. Epigenetic modifications in the
embryonic and induced pluripotent stem cells. Gene Expr Patterns.
2018;29: 1--9.

\leavevmode\hypertarget{ref-pmid33153221}{}%
10. Yi SJ, Kim K. New Insights into the Role of Histone Changes in
Aging. Int J Mol Sci. 2020;21.

\leavevmode\hypertarget{ref-pmid28820059}{}%
11. Wang Y, Yuan Q, Xie L. Histone Modifications in Aging: The
Underlying Mechanisms and Implications. Curr Stem Cell Res Ther.
2018;13: 125--135.

\leavevmode\hypertarget{ref-pmid24459735}{}%
12. Das C, Tyler JK. Histone exchange and histone modifications during
transcription and aging. Biochim Biophys Acta. 2013;1819: 332--342.

\leavevmode\hypertarget{ref-pmid17559965}{}%
13. Fraga MF, Esteller M. Epigenetics and aging: the targets and the
marks. Trends Genet. 2007;23: 413--418.

\leavevmode\hypertarget{ref-pmid30419258}{}%
14. Unnikrishnan A, Freeman WM, Jackson J, Wren JD, Porter H, Richardson
A. The role of DNA methylation in epigenetics of aging. Pharmacol Ther.
2019;195: 172--185.

\leavevmode\hypertarget{ref-pmid25637097}{}%
15. Jung M, Pfeifer GP. Aging and DNA methylation. BMC Biol. 2015;13: 7.

\leavevmode\hypertarget{ref-pmid27259148}{}%
16. Kubben N, Zhang W, Wang L, Voss TC, Yang J, Qu J, et al. Repression
of the Antioxidant NRF2 Pathway in Premature Aging. Cell. 2016;165:
1361--1374.

\leavevmode\hypertarget{ref-pmid16738054}{}%
17. Shumaker DK, Dechat T, Kohlmaier A, Adam SA, Bozovsky MR, Erdos MR,
et al. Mutant nuclear lamin A leads to progressive alterations of
epigenetic control in premature aging. Proc Natl Acad Sci U S A.
2006;103: 8703--8708.

\leavevmode\hypertarget{ref-pmid19896504}{}%
18. Balemans MC, Huibers MM, Eikelenboom NW, Kuipers AJ, Summeren RC
van, Pijpers MM, et al. Reduced exploration, increased anxiety, and
altered social behavior: Autistic-like features of euchromatin histone
methyltransferase 1 heterozygous knockout mice. Behav Brain Res.
2010;208: 47--55.

\leavevmode\hypertarget{ref-pmid19264732}{}%
19. Kleefstra T, Zelst-Stams WA van, Nillesen WM, Cormier-Daire V, Houge
G, Foulds N, et al. Further clinical and molecular delineation of the 9q
subtelomeric deletion syndrome supports a major contribution of EHMT1
haploinsufficiency to the core phenotype. J Med Genet. 2009;46:
598--606.

\leavevmode\hypertarget{ref-pmid20574448}{}%
20. Chi P, Allis CD, Wang GG. Covalent histone
modifications--miswritten, misinterpreted and mis-erased in human
cancers. Nat Rev Cancer. 2010;10: 457--469.

\leavevmode\hypertarget{ref-pmid19892027}{}%
21. Albert M, Helin K. Histone methyltransferases in cancer. Semin Cell
Dev Biol. 2010;21: 209--220.

\leavevmode\hypertarget{ref-pmid22473383}{}%
22. Greer EL, Shi Y. Histone methylation: a dynamic mark in health,
disease and inheritance. Nat Rev Genet. 2012;13: 343--357.

\leavevmode\hypertarget{ref-pmid25693563}{}%
23. Kundaje A, Meuleman W, Ernst J, Bilenky M, Yen A, Heravi-Moussavi A,
et al. Integrative analysis of 111 reference human epigenomes. Nature.
2015;518: 317--330.

\leavevmode\hypertarget{ref-pmid32913075}{}%
24. Kim-Hellmuth S, Aguet F, Oliva M, Muñoz-Aguirre M, Kasela S, Wucher
V, et al. Cell type-specific genetic regulation of gene expression
across human tissues. Science. 2020;369.

\leavevmode\hypertarget{ref-pmid32913073}{}%
25. Ferraro NM, Strober BJ, Einson J, Abell NS, Aguet F, Barbeira AN, et
al. Transcriptomic signatures across human tissues identify functional
rare genetic variation. Science. American Association for the
Advancement of Science; 2020;369: eaaz5900.

\leavevmode\hypertarget{ref-pmid21617055}{}%
26. Pennisi E. The Biology of Genomes. Disease risk links to gene
regulation. Science. 2011;332: 1031.

\leavevmode\hypertarget{ref-pmid19474294}{}%
27. Hindorff LA, Sethupathy P, Junkins HA, Ramos EM, Mehta JP, Collins
FS, et al. Potential etiologic and functional implications of
genome-wide association loci for human diseases and traits. Proc Natl
Acad Sci U S A. 2009;106: 9362--9367.

\leavevmode\hypertarget{ref-pmid23595227}{}%
28. Ernst J, Kellis M. Interplay between chromatin state, regulator
binding, and regulatory motifs in six human cell types. Genome Res.
2013;23: 1142--1154.

\leavevmode\hypertarget{ref-pmid20077036}{}%
29. Collas P. The current state of chromatin immunoprecipitation. Mol
Biotechnol. 2010;45: 87--100.

\leavevmode\hypertarget{ref-Svenson:2012hq}{}%
30. Svenson KL, Gatti DM, Valdar W, Welsh CE, Cheng R, Chesler EJ, et
al. High-resolution genetic mapping using the Mouse Diversity outbred
population. Genetics. 2012;190: 437--447.

\leavevmode\hypertarget{ref-Ashbrook:2019bd}{}%
31. Ashbrook DG, Arends D, Prins P, Mulligan MK, Roy S, Williams EG, et
al. The expanded BXD family of mice: A cohort for experimental systems
genetics and precision medicine. bioRxiv. 2019;139: 387--64.

\leavevmode\hypertarget{ref-Ernst:2012ii}{}%
32. Ernst J, Kellis M. ChromHMM: automating chromatin-state discovery
and characterization. Nature Publishing Group. 2012;9: 215--216.

\leavevmode\hypertarget{ref-Chesler:2008ge}{}%
33. Chesler EJ, Miller DR, Branstetter LR, Galloway LD, Jackson BL,
Philip VM, et al. The Collaborative Cross at Oak Ridge National
Laboratory: developing a powerful resource for systems genetics.
Mammalian Genome. 2008;19: 382--389.

\leavevmode\hypertarget{ref-Gatti:2014ko}{}%
34. Gatti DM, Svenson KL, Shabalin A, Wu L-Y, Valdar W, Simecek P, et
al. Quantitative trait locus mapping methods for diversity outbred mice.
G3 (Bethesda, Md). 2014;4: 1623--1633.

\leavevmode\hypertarget{ref-pmid27309819}{}%
35. Chick JM, Munger SC, Simecek P, Huttlin EL, Choi K, Gatti DM, et al.
Defining the consequences of genetic variation on a proteome-wide scale.
Nature. 2016;534: 500--505.

\leavevmode\hypertarget{ref-pmid28592500}{}%
36. Tyler AL, Ji B, Gatti DM, Munger SC, Churchill GA, Svenson KL, et
al. Epistatic Networks Jointly Influence Phenotypes Related to Metabolic
Disease and Gene Expression in Diversity Outbred Mice. Genetics.
2017;206: 621--639.

\leavevmode\hypertarget{ref-pmid21154709}{}%
37. Langmead B. Aligning short sequencing reads with Bowtie. Curr Protoc
Bioinformatics. 2010;Chapter 11: Unit 11.7.

\leavevmode\hypertarget{ref-pmid19910308}{}%
38. Robinson MD, McCarthy DJ, Smyth GK. edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.
Bioinformatics. 2010;26: 139--140.

\leavevmode\hypertarget{ref-sva}{}%
39. Leek JT, Johnson WE, Parker HS, Fertig EJ, Jaffe AE, Zhang Y, et al.
Sva: Surrogate variable analysis. 2020.

\leavevmode\hypertarget{ref-pmid18798982}{}%
40. Zhang Y, Liu T, Meyer CA, Eeckhoute J, Johnson DS, Bernstein BE, et
al. Model-based analysis of ChIP-Seq (MACS). Genome Biol. 2008;9: R137.

\leavevmode\hypertarget{ref-pmid30348905}{}%
41. Thompson MJ, Chwiałkowska K, Rubbi L, Lusis AJ, Davis RC, Srivastava
A, et al. A multi-tissue full lifespan epigenetic clock for mice. Aging
(Albany NY). 2018;10: 2832--2854.

\leavevmode\hypertarget{ref-pmid21493656}{}%
42. Krueger F, Andrews SR. Bismark: a flexible aligner and methylation
caller for Bisulfite-Seq applications. Bioinformatics. 2011;27:
1571--1572.

\leavevmode\hypertarget{ref-pmid29120462}{}%
43. Ernst J, Kellis M. Chromatin-state discovery and genome annotation
with ChromHMM. Nat Protoc. 2017;12: 2478--2492.

\leavevmode\hypertarget{ref-pmid26553804}{}%
44. O'Leary NA, Wright MW, Brister JR, Ciufo S, Haddad D, McVeigh R, et
al. Reference sequence (RefSeq) database at NCBI: current status,
taxonomic expansion, and functional annotation. Nucleic Acids Res.
2016;44: D733--745.

\leavevmode\hypertarget{ref-pmid26578589}{}%
45. Lesurf R, Cotto KC, Wang G, Griffith M, Kasaian K, Jones SJ, et al.
ORegAnno 3.0: a community-driven resource for curated regulatory
annotation. Nucleic Acids Res. 2016;44: D126--132.

\leavevmode\hypertarget{ref-pmid12045153}{}%
46. Kent WJ, Sugnet CW, Furey TS, Roskin KM, Pringle TH, Zahler AM, et
al. The human genome browser at UCSC. Genome Res. 2002;12: 996--1006.

\leavevmode\hypertarget{ref-pmid27899657}{}%
47. Dreos R, Ambrosini G, Groux R, Cavin Périer R, Bucher P. The
eukaryotic promoter database in its 30th year: focus on non-vertebrate
organisms. Nucleic Acids Res. 2017;45: D51--D55.

\leavevmode\hypertarget{ref-pmid22955616}{}%
48. Dunham I, Kundaje A, Aldred SF, Collins PJ, Davis CA, Doyle F, et
al. An integrated encyclopedia of DNA elements in the human genome.
Nature. 2012;489: 57--74.

\leavevmode\hypertarget{ref-pmid32728249}{}%
49. Moore JE, Purcaro MJ, Pratt HE, Epstein CB, Shoresh N, Adrian J, et
al. Expanded encyclopaedias of DNA elements in the human and mouse
genomes. Nature. 2020;583: 699--710.

\leavevmode\hypertarget{ref-pmid30591514}{}%
50. Broman KW, Gatti DM, Simecek P, Furlotte NA, Prins P, Sen Ś, et al.
R/qtl2: Software for Mapping Quantitative Trait Loci with
High-Dimensional Data and Multiparent Populations. Genetics. 2019;211:
495--502.

\leavevmode\hypertarget{ref-pmid1921910}{}%
51. Jockenhövel F, Grandt D, Weber F, Fritschka E, Philipp T. {[}Plasma
exchange as therapy of recurrent hemolytic-uremic syndrome (HUS) in
adults{]}. Med Klin (Munich). 1991;86: 419--422.

\leavevmode\hypertarget{ref-pmid21921916}{}%
52. Yalcin B, Wong K, Agam A, Goodson M, Keane TM, Gan X, et al.
Sequence-based characterization of structural variation in the mouse
genome. Nature. 2011;477: 326--329.

\leavevmode\hypertarget{ref-pmid16009939}{}%
53. Fraga MF, Ballestar E, Paz MF, Ropero S, Setien F, Ballestar ML, et
al. Epigenetic differences arise during the lifetime of monozygotic
twins. Proc Natl Acad Sci U S A. 2005;102: 10604--10609.

\leavevmode\hypertarget{ref-pmid33931130}{}%
54. Villicaña S, Bell JT. Genetic impacts on DNA methylation: research
findings and future perspectives. Genome Biol. 2021;22: 127.

\leavevmode\hypertarget{ref-pmid27051996}{}%
55. Dongen J van, Nivard MG, Willemsen G, Hottenga JJ, Helmer Q, Dolan
CV, et al. Genetic and environmental influences interact with age and
sex in shaping the human methylome. Nat Commun. 2016;7: 11115.

\leavevmode\hypertarget{ref-pmid24183450}{}%
56. Grundberg E, Meduri E, Sandling JK, Hedman AK, Keildson S, Buil A,
et al. Global analysis of DNA methylation variation in adipose tissue
from twins reveals links to disease-associated variants in distal
regulatory elements. Am J Hum Genet. 2013;93: 876--890.

\leavevmode\hypertarget{ref-pmid22532803}{}%
57. Bell JT, Tsai PC, Yang TP, Pidsley R, Nisbet J, Glass D, et al.
Epigenome-wide scans identify differentially methylated regions for age
and age-related phenotypes in a healthy ageing population. PLoS Genet.
2012;8: e1002629.

\leavevmode\hypertarget{ref-pmid24887635}{}%
58. McRae AF, Powell JE, Henders AK, Bowdler L, Hemani G, Shah S, et al.
Contribution of genetic variation to transgenerational inheritance of
DNA methylation. Genome Biol. 2014;15: R73.

\leavevmode\hypertarget{ref-pmid20485568}{}%
59. Gibbs JR, Brug MP van der, Hernandez DG, Traynor BJ, Nalls MA, Lai
SL, et al. Abundant quantitative trait loci exist for DNA methylation
and gene expression in human brain. PLoS Genet. 2010;6: e1000952.

\leavevmode\hypertarget{ref-pmid15680324}{}%
60. Bernstein BE, Kamal M, Lindblad-Toh K, Bekiranov S, Bailey DK,
Huebert DJ, et al. Genomic maps and comparative analysis of histone
modifications in human and mouse. Cell. 2005;120: 169--181.

\leavevmode\hypertarget{ref-pmid14661024}{}%
61. Schneider R, Bannister AJ, Myers FA, Thorne AW, Crane-Robinson C,
Kouzarides T. Histone H3 lysine 4 methylation patterns in higher
eukaryotic genes. Nat Cell Biol. 2004;6: 73--77.

\leavevmode\hypertarget{ref-pmid12353038}{}%
62. Santos-Rosa H, Schneider R, Bannister AJ, Sherriff J, Bernstein BE,
Emre NC, et al. Active genes are tri-methylated at K4 of histone H3.
Nature. 2002;419: 407--411.

\leavevmode\hypertarget{ref-pmid16728976}{}%
63. Wysocka J, Swigut T, Xiao H, Milne TA, Kwon SY, Landry J, et al. A
PHD finger of NURF couples histone H3 lysine 4 trimethylation with
chromatin remodelling. Nature. 2006;442: 86--90.

\leavevmode\hypertarget{ref-pmid23788621}{}%
64. Voigt P, Tee WW, Reinberg D. A double take on bivalent promoters.
Genes Dev. 2013;27: 1318--1338.

\leavevmode\hypertarget{ref-pmid22513113}{}%
65. Vastenhouw NL, Schier AF. Bivalent histone modifications in early
embryogenesis. Curr Opin Cell Biol. 2012;24: 374--386.

\leavevmode\hypertarget{ref-pmid27800026}{}%
66. Prickaerts P, Adriaens ME, Beucken TVD, Koch E, Dubois L, Dahlmans
VEH, et al. Hypoxia increases genome-wide bivalent epigenetic marking by
specific gain of H3K27me3. Epigenetics Chromatin. 2016;9: 46.

\leavevmode\hypertarget{ref-pmid32432110}{}%
67. Bae S, Lesch BJ. H3K4me1 Distribution Predicts Transcription State
and Poising at Promoters. Front Cell Dev Biol. 2020;8: 289.
\end{cslreferences}

\nolinenumbers


\end{document}
