---
title: Correcting for relatedness in standard mouse mapping populations; and something about epistasis
author:
  - name: Catrina Spruce
    email: Catrina.Spruce@jax.org
    affiliation: The Jackson Laboratory
  - name: Anna L. Tyler
    email: Anna.Tyler@jax.org
    affiliation: The Jackson Laboratory
  - name: Many more people
    email: many more emails
    affiliation: JAX-MG and JAX-GM
  - name: Gregory W. Carter
    email: Gregory.Carter@jax.org
    affiliation: The Jackson Laboratory
    corresponding: Gregory.Carter@jax.org
address:
  - code: The Jackson Laboratory
    address: 600 Main St. Bar Harbor, ME, 04609
abstract: |
  

author_summary: |
  Epigenetics in mice
bibliography: epigenetics.bib
output: rticles::plos_article
csl: plos.csl
---

```{r read_results, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
library(here)
```

# Introduction


# Materials and Methods

## Chromatin Data

## RRBS Data

## Mice

## Data Processing

## Identifying chromatin states
We used ChromHMM to identify chromatin states corresponding to the 
presence and absence of the four chromatin modifications. We calculated
states for all numbers of states between four and 16, which is the maximum
number of states possible with four binary chromatin modifications. 

## Correlating chromatin state with gene expression
To correlate chromatin state with gene expression, we calculated the
proportion of the gene body that was assigned to each chromatin state
across the nine inbred founders. We then correlated the proportion of
each state with the mean gene expression across the founders. We 
calculated these correlations across all states and all state models.

## Assessing abundance of chromatin states across gene bodies
We calculated the relative abundance of each chromatin state
across all gene bodies. We normalized all gene coordinates to
run from 0 at the transcription start site to 1 at the transcription
end site. Upstream regulatory regions were assigned negative coordinates
and downstream regulatory regions were assigned coordinates greater
than 1. We then aligned all genes and grouped the normalized coordinates
into 500 bins. To avoid potential contamination from regulatory
regions of nearby genes, we only included genes that were at least 2kb 
from their nearest neighbor. This included 14048 genes. 
At each position, we then calculated the proportion of genes with the
state present.

## Assessing correlation of chromatin state with expression across gene bodies
We used the normalized gene coordinates calculated above to calculate 
position-based correlations between chromatin state and gene expression.
To do this we used a sliding window across the gene body from normalized
coordinates -1 to +2 and correlated state proportion within each window
with gene expression.


## Imputing chromatin states in DO
To better investigate the effect of chromatin state on local gene expression,
we imputed chromatin state into a diversity outbred (DO) mouse population. 
There is good evidence that, especially early in life, chromatin modifications 
are genetically determined [cite]. For each transcript, we multiplied the 
genotype at the nearest marker by the chromatin state matrix across the full
gene body ($\pm 1kb) to derive an imputed chromatin state matrix identifying
a chromatin state for each DO mouse across the gene body. We used R/qtl2 [cite]
to calculate the effect of each chromatin state on gene expression. We calculated
the total variance explained by both the local genotype as well as the local
chromatin state for each transcript. 

## RRBS processing

# Results

## Nine-state model had highest correlation with gene expression
To identify the ChromHMM model that corresponded best with gene 
expression, we compared the correlation of each state with gene 
expression (Methods) across all ChromHMM models (Supp Fig. XXX). 
Across all models, the correlations between gene expression and 
chromatin state could be binned roughly into five bins: low, 
moderately low, no correlation, moderately high correlation, and 
high correlation. The nine-state model had states in each of these 
categories with the lowest redundancy. Furthermore, state seven in 
the nine-state model had the maximum correlation with gene expression. 
Therefore, we chose the nine-state model for downstream analysis.

## Genomic Position Enrichments Aided Interpretation of States

We were able to annotate some of the states based on their correlation 
with gene expression as well as their enrichments around functionally
annotated positions in the genome. For example, State 5 was characterized
by the presence of both H3K27ac and H3K4me1, the co-occurence of which has 
previously been associated with active enhancers [21106759, 21160473, 29273804].
In line with previous observations, in out study, State 5 was primarily
localized around known enhancers and its presence was positively correlated
with gene expression.

State 7, which was characterized by the presence of three activating
marks, H3k27ac, H3k4me1, and H3K4me3. It was enriched around 
transcription start sites (TSS) and promoters and was also positively 
correlated with gene expression. 

State 3 was characterized by the presence of H3K4me1 and H3k27me3. 
H3K27me3 is associated with downregulation of transcription and
poised enhancers. When paired with H3K4me1

also enriched around 
promoters and TSS, but was most strongly enriched at transcription 
factor binding sites, 

Table?? Figure?? Supp Figure??

## Promoter and enhancer states correlated with increased expression
Chromatin states 5 and 7, which were enriched around enhancers and promoters 
respectively were correlated with increased expression. This was true
both across the liver transcriptome, as well as across strains. That is,
that within a single strain, genes with higher proportions of states 5 
or 7 across the gene body were more highly expressed than genes with 
low proportions of these states. And for individual genes, strains 
with higher proportions of states 5 and 7 had higher expression than 
strains with lower proportions of these states (Figure?).

State 6 was also correlated with increased gene expression to a lesser
degree. This state was also enriched at enhancers, though less so than 
state 5.

## H3K27me3 and the absence of chromatin modification correlated with decreased expression

High abundance of states 1, 2, and 3 were correlated with decreased 
gene expression. State 1 was the absence of chromatin modification,
and states 2 and 3 were the only states with the suppressing 
modification H3K27me3. State 1 was enriched in intergenic regions
and state 3 was enriched around transcription factor binding sites. 
Positional enrichment of state 2 was not clear.

## Differential spatial distribution of states around gene bodies
Chromatin states were each distributed in a specific pattern 
across gene bodies (Methods) (Figure XXX). For example, state 1
was strongly depleted near the TSS, indicating that this region
is commonly subject to chromatin modification. However, its
abundance increased steadily to a peak at the TES. In contrast,
state 7 was present in over 60\% of TSS, but decreased to almost 
0\% near the TES.

The remaining states were relatively low in abundance compared
to states 1 and 7, but also showed gene-body specific distribution
patterns. State 8, was depleted at the TSS, but enriched immediately
downstream of the TSS. State 9 had slight enrichments immediately 
upstream of the TSS and immediately downstream of the TES. The 
enrichment of these states in regulatory regions indicates the 
possibility that these states are used for regulating expression 
levels whereas states 7 and 3 at the transcription start site may
be primarily related to switching gene transcription on and off.

## Correlation of chromatin state and gene expression was differentially distributed across the gene body

We examined whether there was a spatial component to the correlation between
chromatin state and gene expression (Methods). 

Figure XXX shows the Pearson correlation between expression and chromatin 
state across all windows for each chromatin state. The most prominant
position-specific correlations between state and gene expression were
for state 3 and state 9, which were both negatively correlated with 
gene expression exclusively at the TSS. There were no TES-specific
correlations for any state. 

## Results from inbred mice were recapitulated in diversity outbred mice

We investigated the extent to which chromatin state imputed into DO mice 
explained variation in expression across individuals. Although local 
genetic variation explains a large amount of variation in gene 
expression [cite], chromatin state may offer further insight into 
regulation of gene expression at the local level. [more compelling stuff here]

We imputed genome-wide chromatin states in a population of DO mice based 
on their genotype (Methods) and compared the percent variance explained by
local genotype to the maximum percent variance explained by local chromatin 
state for each transcript (Figure XXX.) The two measurements were very 
tightly correlated (Pearson R = 0.95) indicating that chromatin state
determined by genetics is an excellent approximation of the genetic 
effect on gene expression. The imputation further allowed us to observe 
the effects of chromatin state across [500] genetically diverse mice
by measuring chromatin modifications in a handful of inbred mice. Further,
because chromatin modifications are measured at extremely high density,
we can map high-density chromatin effects in the DO mice, which may help 
prioritize functional SNPs within gene bodies and in regulatory regions.

For example, Figure XXX shows chromatin states across the gene Irf5
in the inbred founders along with the LOD score and chromatin state
effects at each position along the gene body as calculated in the DO 
population. The LOD scores and allele effects highlight variation 
at the TSS, and at several internal positions in the gene as potentially
regulating gene expression. 

## DNA methylation varied across the gene body

but did not have any correlation with expression


# Discussion



# Acknowledgements
This work was funded by XXX.

# Data and Software Availability
All data used in this study and the code used to analyze it are avalable as part of 
a reproducible workflow located at... (Figshare?).

CAPE is available at CRAN...

# Supplemental Figure Legends

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first principal component (PC) of
the kinship matrix. Traits with high correlation to the kinship matrix may 
be highly polygenic and thus be susceptible to test statistic inflation due to 
many true positives. To reduce this risk, we selected traits with low correlation
with the first kinship matrix PC. This figure shows the distribution of correlations
between traits and the first kinship PC across populations. 
}
\label{fig:trait_cor}
\end{figure}


\begin{figure}[ht]
\centering
\caption{Reducing $n$ reduces inflation. 
This figure is identical to Fig. \ref{LD} except that we have added a column 
for the F2 that has been subsampled to the same $n$ as the Backcross. This 
subsampling reduces power to detect effects, and thus reduces inflation to 
roughly the same level as that seen in the backcross.
}
\label{subsampled}
\end{figure}

# Supplemental Table Descriptions

\begin{figure}[ht]
\centering
\caption{Correlations between traits and the first PC of the kinship matrix.
}
\label{table:trait_cor}
\end{figure}


# References {#references .unnumbered}
