
## Gene expression
The set of genes with the most highly difergent expression were 
enriched for members of the cytochrome P450 superfamily (32), 
major urinary proteins (15).


The correlation between chromatin state and gene expression, as well as
positional enrichments around functionally annotated positions in the 
genome aided annotation of individual chromatin states. For example, 
chromatin state 5, which was characterized by the co-occurence of H3K27ac 
and H3K4me1, had enriched representation around known enhancers. 
Its presence was also positively correlated with gene expression. This
combination of chromatin modifications has previously been associated with 
active enhancers [21106759, 21160473, 29273804].

State 7, which was characterized by the presence of three activating
marks, H3k27ac, H3k4me1, and H3K4me3. It was enriched around 
transcription start sites (TSS) and promoters and was also positively 
correlated with gene expression. 

State 3 was characterized by the presence of H3K4me1 and H3k27me3. 
H3K27me3 is associated with downregulation of transcription and
poised enhancers. When paired with H3K4me1

also enriched around 
promoters and TSS, but was most strongly enriched at transcription 
factor binding sites, 

Table?? Figure?? Supp Figure??



### Promoter and enhancer states correlated with increased expression
Chromatin states 5 and 7, which were enriched around enhancers and promoters 
respectively, were correlated with increased expression. This was true
both across the liver transcriptome, as well as across strains. That is,
that within a single strain, genes with higher proportions of states 5 
or 7 across the gene body were more highly expressed than genes with 
low proportions of these states. And for individual genes, strains 
with higher proportions of states 5 and 7 had higher expression than 
strains with lower proportions of these states (Figure?).

State 6 was also correlated with increased gene expression to a lesser
degree. This state was also enriched at enhancers, though less so than 
state 5.

### H3K27me3 and the absence of measured marks correlated with decreased expression

High abundance of states 1, 2, and 3 were correlated with decreased 
gene expression. State 1 was the absence of all measured chromatin 
modifications, and states 2 and 3 were the only states with the 
suppressing modification H3K27me3. State 1 was enriched in intergenic 
regions and state 3 was enriched around transcription factor binding 
sites. Positional enrichment of state 2 was not clear.


## Epigenetic features correlated with inbred gene expression

To investigate the relationship between epigenetic modifications and 
gene expression, we calculated the relationship both chromatin state
and DNA methylation and transcript abundance (Methods) (Figure XXX).
Across all genes within individual inbred strains of mice, the percent
methylation at CpG sites was negatively correlated with transcript 
abundance primarily at the TSS. 

The correlation of each chromatin state matched that seen in Figure XXXC,
and the spatial resolution of the relationships added an interesting dimension
to this observation. For example, state 3, which was overall correlated with
lower transcript abundance in Figure XXXC, was shown to have this negative
effect primarily when it was positioned at the TSS. In contrast, state 2,
was also negatively correlated with transcription overall, but this effect
was constant throughout the gene body. State 2 is characterized by H3K27me3,
which has previously been associated with reduced transcription [CITE], but 
was not enriched near any annotated functional genomic elements. 

## Characterizing the landscape of gene expression across inbred strains
Using the variance-stabilized read counts in the inbred strains, we
calculated the standard deviation in expression across strains. We then
binned the genes into 15 bins with increasing standard deviation. We used
the R package gprofiler [CITE] to identify functional enrichments for each
bin. We defined the divergence in expression across strains as the F statistic
from an ANOVA testing the effect of strain on mean expression.

For each group of genes binned by standard deviation,
we performed singular value decomposition on the mean-centered expression
matrix. We then calculated the separation of the strains in the two-dimensional
space defined by the first two principle components. 

## The landscape of gene expression was strain-dependent

Across the inbred strains, the landscape of variation in gene expression 
was highly strain-dependent (Figure XXX). Principal components analysis
Expression in the wild-derived CAST and PWK strains diverged most strongly 
from the overall mean expression, and expression within each strain was 
highly replicable.

There was also a wide range of strain-specificity across individual 
transcripts (Figure XXX). Gene expression ranged from being highly
conserved across strains to highly divergent (Figure XXX). Transcripts
with the lowest variation across strains were enriched for basic cellular 
processes: RNA processing, chromatin modification, and chromosome organization.
Transcripts with the greatest variation across strains were enriched in 
hepatocyte-specific functions: steroid metabolism, small molecule biosynthesis, 
and lipid metabolism (Fig. XXX?)

## Variation in expression in an outbred population maps to imputed chromatin state


A subset of the chromatin states was correlated with gene expression
across genes (Figure XXXC) in a manner that concorded with both their
histone modification profiles and their enrichments near functional genomic 
elements. For example, the histone modification H3K27me3 has been previously
shown to be associated with lower transcript abundance [CITE]. We found this
modification in both states 2 and 3 (Figure XXXA), which were both correlated 
with reduced transcription (Figure XXXC). State 3, furthermore, was enriched
near the TSS, promoters, TFBS, and other regulatory elements supporting a
possible role in transcriptional regulation. 

State 5 was characterized by the presence of two histone modifications 
previously associated with increased transcription: H3K27ac [CITE], and
H3K4me1 [CITE]. The enrichment of this state in enhancers coincides with
previous work showing the presence of these two modifications in active
enhancers [21106759, 21160473, 29273804], and supports the role of this
state in upregulation of gene transcription.

States 6 and 7 were also associated with increased transcription and
the presence of transcriptionally activating histone modifications. 
State 6 was modestly enriched in enhancers, while state 7 was enriched
not only in enhancers, but also stongly near the TSS and other associated
functional elements (Figure XXXB).


## DNA methylation varied across the gene body

In addition to chromatin state, we examined the distribution of DNA
methylation across the gene body, as well as the relationship between
DNA methylation and gene expression in both inbred mice and DO mice.



Within each strain, percent methylation at the gene TSS was slightly 
negatively correlated with gene expression (Pearson r for all strains 
was about -0.2). However, there was very little variation in DNA methylation
across strains, particularly at the TSS, and consequently, there was no
relationship between percent methylation and gene expression across strains.


## Chromatin state but not DNA methylation correlated with gene expression across strains


In stark contrast, DNA methylation was completely uncorrelated with 
variation in gene expression across strains (Figure XXXB). This lack
of correlation is likely due to the low variability of percent methylation
across strains at any given position. Figure XXXC shows the standard
deviation in percent DNA methylation at normalized positions across
the gene body. It is strikingly low everywhere, with the standard deviation
being around 6%, which is likely below any biologically functional threshold.
The variation dips even lower, to around 4% at the TSS, indicating that for
the most part that DNA methylation does not vary across strains and is not 
contributing to strain difference in gene expression.  

## Background

In every mammalian cell nucleus, DNA is wound hierarchically into
dense, and highly regulated structures. At the smallest level, 
DNA is wound into nucleosomic core particles consisting of four
pairs of histone proteins (H2A, H2B, H3, and H4). Each histone
protein has a long tail that projects out of the nucleosome
and which can be post-translationally modified, through 
methylation, acetylation, phosphorylation, and a number of 
other types of modifications [26704082]. Different modifications
at different positions along these tails have markedly different
effects on the transcription of local genes [citations]. Whereas 
trimethylation of the ...