---
title: "Supplemental Methods"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  pdf_document:
    toc: true
---

This document contains detailed methods accompanying the manuscript
Spruce, Tyler, et al. \textit{Variation in epigenetic state correlates with 
gene expression across nine inbred strains of mice.}


## Liver perfusion

To purify hepatocytes from the liver cell population, the mouse livers 
were perfused with collagenase to digest the liver into a single-cell 
suspension, and then isolated using centrifugation. Mice were harvested 
at 9:00 AM and sacrificed by cervical dislocation. Mice were placed over 
a stack of paper towels in preparation to catch excess liquid, and the 
appendages were pinned out to hold the body in place. to keep the fur 
from contaminating the liver sample later, the fur was wiped down with 
70\% ethanol. The mouse skin was then cut open and peeled back to the 
appendages to allow clear access to the abdominal cavity. The fascia was
cut open and back to the ribs, being careful to not nick the liver. 
The intestines and stomach were moved to the right to expose the vena 
cava and hepatic portal vein below the liver. 

For the perfusion, a 23G x $\frac{3}{4}$'' BD Vacutainer Safety-Lok needle 
(REF 367297) was attached to 1.6mm ID BioRad Tygon tubing (R-3603) connected 
to a Pharmacia peristaltic pump that allows a flow of up to 8 ml/min. The 
liver will be processed with three solutions: 5mM EGTA in Leffert's buffer, 
Leffert's buffer wash, and 87 CDU/mL Liberase collagenase with 0.02\% CaCl2 in 
Leffert's buffer. The three solutions were at 37$^{\circ}$C before perfusion. 

The needle was placed into the vena cava for the perfusion superior to the 
kidneys and inferior to the liver. With the peristaltic pump running slowly, 
the vena cava was pierced at shallow 15$^{\circ}$ angle and the needle was 
inserted to a shallow depth (around 2-3mm of the needle tip in the vena cava). 
Once the needle is inserted into the vena cava, the volume on the peristaltic 
pump is increased to 5-7mL/min. The liver will immediately blanch, and the 
hepatic portal vein is immediately severed to allow flushing of the liver. 

The 1x EGTA buffer was used to flush the blood out of the liver and start the 
digestion of the desmosomes connecting the liver cells. To help with the perfusion, 
pressure was applied to the hepatic portal vein for 5 second intervals causing 
more solution to be forced through the liver, which can be seen visually by the 
liver swelling. After 35ml of the 1x EGTA solution is passed through the liver, 
the solution was switched to the 1x Leffert's buffer. The pump was turned off during 
the switch to prevent air from being sucked into the tubing while the tubing is 
transferred to the new solution. To wash, 7-10ml of the Leffert's buffer was passed 
through the liver to flush out the EGTA, which otherwise chelates the calcium ions 
necessary for collagenase activity in the next buffer. The pump was turned off again 
to switch to the Liberase solution. To digest the liver, 25-50mL of Liberase solution 
($\sim4.3$ wunsch units) was passed through the liver. Throughout the perfusion 
process, periodic pressure was applied to the hepatic portal vein to help pump the 
buffers more completely through the liver. As the liver was digested with the Liberase, 
it will swell and look soggy and limp. Over-digestion leads to increased contamination 
with non-hepatocyte cell types, and further reduces cell viability.

After perfusion, which takes around 15-20min to complete, the liver was carefully cut 
out of the abdominal cavity and placed in a petri dish with 35 mL ice-cold 
Leffert's buffer with 0.02\% CaCl$_{2}$. The digested liver was passed through Nitex 
80 $\mu$m nylon mesh (cat #03-80/37) into a 50mL conical, using additional 
ice-cold Leffert's buffer with 0.02\% CaCl2 if necessary, and a rubber policeman. 
After the liver cells from both animals were collected, they were put through two 
wash and spin cycles to purify the hepatocytes and remove other types of cells. To 
isolate the hepatocytes, the much larger size of the hepatocyte cells was exploited 
in very slow 4 min, 50 x g spins that leave smaller other cell types in suspension. 
After each spin, the solution was decanted as waste, and the enriched cell pellet of 
hepatocytes was resuspended in 30ml ice-cold Leffert's buffer with 0.02\% 
CaCl$_{2}$. After the second spin, the solution should be almost clear, 
indicating that other cell types have been removed. The hepatocytes are resuspended 
in room temperature PBS, counted, and volume adjusted to $1 x 10^{6}$ cells/mL. 

We aliquoted $5 x 10^{6}$ cells for each RNA-Seq and bisulfite sequencing, 
and the rest were cross-linked for ChIP assays. Two $5 x 10^6$ aliquots (5mLs) of 
liver cells were removed into two 15mL conicals. These were spun down at 200 rpm for 
5 min, and resuspended in $1200\mu L$ RTL+BME (for RNA-Seq) or frozen as a cell 
pellet in liquid nitrogen (for bisulfite sequencing). Meanwhile, 37\% formaldehyde 
in methanol were added to the remaining cells to a final concentration of 1\%. 
The cells were rotated at room temperature for 5 min to cross-link protein complexes 
to the DNA bound to them. After cross-linking, 10x glycine was added to a 
final concentration of 125 mM and rotated for 5 min to quench the formaldehyde and 
stop cross-linking. The cells were spun down at 2000 rpm for 5 min, decanted, and 
resuspended in PBS to $5 x 10^6$ cells/mL. The cells were divided into $5x10^6$ aliquots 
in 2mL tubes. The tubes were spun down again at 5000 x g for 5 min, decanted, and the 
cell pellets frozen in liquid nitrogen. All cell samples were stored at -80Â°C until used. 

## Histone chromatin immunoprecipitation assays

The histone chromatin immunoprecipitation assays were performed on 
cross-linked hepatocytes using similar protocols. First, the aliquot 
of $5x10^6$ hepatocyte cells was lysed to release the nuclei by rotating 
the sample in hypotonic buffer for 20 min at $4^\circ$C. The cells were 
pelleted by spinning for 10min, 10K x G, at $4^\circ$C. The cells were 
resuspended in 130ul MNase buffer with 1mM PMSF and 1x protease inhibitor 
cocktail (Roche) to prevent histone protein degradation, then 
digested with 15U of MNase. The micrococcal nuclease digests the exposed 
DNA, but leaves the nucleosome-bound DNA intact. After 10 min of incubation 
at $37^\circ$C, the chromatin was digested into primarily mononucleosomes. 
This was confirmed by DNA-purification of the MNase-digested chromatin run 
out on an agarose gel, which yielded mostly 150bp fragments, and few 300bp 
fragments. The MNase digestion was stopped by adding EDTA to 10mM, and 
incubating on ice for 5 min. The digested chromatin was purified by spinning 
out insoluble parts at top speed for 10 min at $4^\circ$C. The chromatin was 
transferred to a new tube and spun again to further remove impurities and 
reduce background in the ChIP assays. The final chromatin was transferred 
to a fresh tube, and used immediately in the ChIP.

To prepare for the ChIP, $20\mu L/1x10^6$ cells Dynabead Protein G beads were 
aliquoted into an Eppendorf tube. A magnetic tube holder was used to attract the 
beads to the wall of the tube, and then the solution was carefully pipetted off, 
leaving only the beads behind. The beads were washed twice with buffer to prepare 
them for binding to the antibody. For this binding step and the chromatin binding 
step, the buffer used was either RIPA buffer for the H3K4me3 and K3K27me3 ChIPs, 
or ChIP buffer for the H3K4me1 ChIP. The ChIP buffer was gentler and less 
stringent than RIPA buffer, which was better for the weaker binding of the 
H3K4me1 antibody that was used. The buffers were supplemented with 50 mg/mL BSA
and 0.5 mg/mL Herring Sperm DNA, both of which are blocking agents that reduce 
background and non-specific binding. The ChIP assays also varied in the amount of 
input chromatin and corresponding size of the reaction that was necessary to yield 
sufficient DNA for sequencing. H3K4me3 ChIP needed only $1.5 x 10^6$ cells, and 
H3K4me1 and K3K27me3 ChIP used $4 x 10^6$ cells. To perform the ChIP, $20\mu$L of 
Dynabeads per $1 x 10^6$ cells is incubated with $5\mu L$ of histone antibody for 
$>20$min in $50\mu L/1x10^6$ cells RIPA (or ChIP) buffer supplemented with 
50 mg/mL BSA, 0.5 mg/mL Herring Sperm DNA, 1xPIC, and 1mM PMSF. The antibodies 
used were (XXX). Once the antibody was bound to the Dynabeads, the beads were 
washed twice with $100\mu L/1x10^6$ cells RIPA buffer with BSA and Herring Sperm 
DNA. 

Next, the MNase-digested chromatin were added, which was at a concentration of 
$1x10^6$ cells/$25\mu L$. The ChIP reaction was incubated overnight with rotation 
at $4^\circ$C, to allow the histone protein to bind to the antibody, which was 
bound to the magnetic beads. In order to calculate enrichment for each ChIP sample, 
a known amount ($10$ or $20\mu L$) of MNase-digested input chromatin was saved.  
The next morning, the ChIPs underwent a series of washes to remove unbound chromatin. 
The H3K4me3 and H3K27me3 ChIPs were washed 3x with $100\mu L/1x10^6$ cells RIPA buffer, 
and the H3K4me1 ChIP was washed with a low salt wash (0.1\% SDS, 1\% Triton X-100, 2mM 
EDTA, 20mM Tris-HCl pH 8, 150 mM NaCl), a high salt wash (0.1\% SDS, 2\% Triton X-100, 
2mM EDTA, 20mM, Tris-HCl, pH 8, 500mM NaCl), and a LiCl wash (0.25 MLiCl, 1\% IGEPAL-CA630, 
1\% deoxycholic acid (sodium salt), 1 mM EDTA, 10 mM Tris-HCl pH 8). After three washes, 
the ChIPs were washed twice with TE buffer and transferred to a new tube during the last 
TE wash to reduce background. At this point, the histone of interest and the histone-bound 
DNA fragment had been purified from the MNase-digested, cross-linked chromatin, and was 
bound by histone-specific antibody to the magnetic Dynabeads. In the next step, a high-salt 
elution buffer is used to degrade the antibody binding interactions to the beads and the 
histone, and concurrently, proteinase K is added to digest the protein away from the 
DNA-protein complexes. The ChIP was incubated with the elution buffer and proteinase K at 
$68^\circ$C for $>6$ hours to liberate the DNA. At the same time, the saved input chromatin 
was also digested in the same buffer. Afterwards, the beads were removed using the magnet, 
and the DNA was purified using the Qiagen PCR purification kit. Quantification was performed 
using the Qubit quantification system, which is accurate to $0.02 ng/\mu L$ and only requires 
a small amount of sample to measure concentration. The ChIP sample was enriched for only DNA 
that was bound to the histone of interest. The goal for each ChIP was to yield 10 ng of ChIP 
DNA for sequencing. Not all samples met this criterion, and the H3K4me1 ChIPs often had a 
total yield of $\sim 2 ng$ of DNA.

To test the efficiency of the ChIPs, quantitative PCR using QuantiFAST was performed. 
Two sets of primers were used, one set in a known region of histone binding (positive control), 
and one set in a region without histone binding (negative control). The qPCR was performed both 
on the ChIP DNA and the input DNA. Then the relative enrichment of positive vs negative assays 
was compared between the ChIP and input DNA.

The ChIP DNA was submitted to The Jackson Lab GES service for library preparation and sequencing. 
Libraries were made using the Kapa Hyper Prep kit with adapters at $0.6\mu M$.  The libraries were 
amplified by 10 cycles of PCR.  These libraries were not size selected, although most fragments 
were $\sim150$ bp due to MNase-digestion. The samples were sequenced with 40 or more million 
reads per sample, which is almost 2x more reads than the ENCODE project, which sequenced using 
20 million reads.

## Selecting the most biologically meaningful ChromHMM model

We compared the chromatin states derived from multiple runs of 
ChromHMM to identify the most biologically meaningful model. 
Across all models, the states were remarkably stable. As 
we increased the number of states detected by the model, new states 
appeared, but previously detected states were not disrupted. This 
stability was apparent in all state measures: emissions probability 
(Supp Fig 1) patterns, overall abundance (Supp. Fig. 2), and effect on 
expresssion (Supp Fig 3). This analysis revealed interesting patterns
in the detected states. For example, one highly abundant state (present 
in 65\% of transcribed genes) detected first in the four-state model 
was split into two distinct states in the 10-state model. These resulting
states were also highly abundant (appearing in 40\% and 41\% of transcribed 
genes), and had distinct emissions probabilities (Supp. Fig. 1). These 
two states remained stable with increasing numbers of clusters through 
to the 16-state model. States arising after the 10-state model were of 
lower abundance, appearing in 2\% or less of transcribed genes. 

All of the higher abundance states were established in the 10-state model. 
However, as we moved toward higher numbers of clusters, the resolution on 
the lower-abundance states improved in terms of the emissions probabilities 
profiles, and strength of the correlation with gene expression. For example, 
the 14-state model better resolved a state that had appeared in the 10-state 
model but was not strongly correlated with gene expression. In the 14-state 
model, the emission patterns were closer to binary, and the strength of the 
correlation with expression was increased. Beyond 14 clusters, the new states 
identified were extremely rare (1\% of transcripts or less), and were not 
strongly correlated with gene expression. We thus selected the 
`r num.states`-state model and the model with the most biologically meaningful 
clusters. 
